Use --warning=cross only if supported by automake

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Constantin Musca <constantinx.musca@intel.com>

--- a/bin/autoreconf.in
+++ b/bin/autoreconf.in
@@ -127,6 +127,8 @@ my $aclocal_supports_warnings = 0;
 my $automake_supports_force_missing = 0;
 # Does automake support -Wfoo?
 my $automake_supports_warnings = 0;
+# Does automake support --warning=cross
+my $automake_supports_cross_warning = 0;
 
 my @prepend_include;
 my @include;
@@ -191,6 +193,7 @@ sub parse_args ()
   $aclocal_supports_warnings = $aclocal_help =~ /--warnings/;
   $automake_supports_force_missing = $automake_help =~ /--force-missing/;
   $automake_supports_warnings = $automake_help =~ /--warnings/;
+  $automake_supports_cross_warning = $automake_help =~ /cross/;
 
   # Dispatch autoreconf's option to the tools.
   # --include;
@@ -244,6 +247,8 @@ sub parse_args ()
       $libtoolize .= ' --debug';
     }
   # --warnings;
+  @warning = grep { $_ ne "cross" } @warning
+    if ! $automake_supports_cross_warning;
   if (@warning)
     {
       my $warn = ' --warnings=' . join (',', @warning);
