From 32f4a4a634c99d4e00cc6149786a9180ad651f76 Mon Sep 17 00:00:00 2001
From: Victor Zverovich <viz@meta.com>
Date: Sat, 22 Mar 2025 06:57:04 -0700
Subject: [PATCH] Workaround an ABI issue in spdlog

Upstream-Status: Backport [https://github.com/fmtlib/fmt/commit/784eac839df77df1852dc1c50b17b76c3aec8f4a]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 include/fmt/format.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/include/fmt/format.h b/include/fmt/format.h
index 4a653007..82f80efd 100644
--- a/include/fmt/format.h
+++ b/include/fmt/format.h
@@ -793,7 +793,9 @@ enum { inline_buffer_size = 500 };
  * converted to `std::string` with `to_string(out)`.
  */
 template <typename T, size_t SIZE = inline_buffer_size,
-          typename Allocator = detail::allocator<T>>
+          // DEPRECATED! std::allocator should be replaced with
+          // detail::allocator
+          typename Allocator = std::allocator<T>>
 class basic_memory_buffer : public detail::buffer<T> {
  private:
   T store_[SIZE];
@@ -2746,7 +2748,7 @@ class bigint {
   }
 
  public:
-  FMT_CONSTEXPR bigint() : exp_(0) {}
+  FMT_CONSTEXPR20 bigint() : exp_(0) {}
   explicit bigint(uint64_t n) { assign(n); }
 
   bigint(const bigint&) = delete;
