Upstream-Status: Pending

Forwarded: not-needed
--- a/system/core/fs_mgr/liblp/reader.cpp
+++ b/system/core/fs_mgr/liblp/reader.cpp
@@ -22,6 +22,7 @@
 #include <unistd.h>
 
 #include <functional>
+#include <cstring>
 
 #include <android-base/file.h>
 #include <android-base/unique_fd.h>
--- a/system/core/fs_mgr/liblp/writer.cpp
+++ b/system/core/fs_mgr/liblp/writer.cpp
@@ -21,6 +21,7 @@
 #include <unistd.h>
 
 #include <string>
+#include <cstring>
 
 #include <android-base/file.h>
 #include <android-base/unique_fd.h>
--- a/system/core/liblog/logger_write.cpp
+++ b/system/core/liblog/logger_write.cpp
@@ -27,6 +27,7 @@
 #include <android/set_abort_message.h>
 #endif
 
+#include <mutex>
 #include <shared_mutex>
 
 #include <android-base/errno_restorer.h>
--- a/system/core/libziparchive/zip_archive_stream_entry.cc
+++ b/system/core/libziparchive/zip_archive_stream_entry.cc
@@ -23,6 +23,7 @@
 #include <sys/types.h>
 #include <unistd.h>
 
+#include <limits>
 #include <memory>
 #include <vector>
 
