Upstream-Status: Pending

Description: Add missing 'std::' scope identifiers.
Forwarded: not-needed
--- a/system/core/libunwindstack/include/unwindstack/DwarfMemory.h
+++ b/system/core/libunwindstack/include/unwindstack/DwarfMemory.h
@@ -29,7 +29,7 @@
   DwarfMemory(Memory* memory) : memory_(memory) {}
   virtual ~DwarfMemory() = default;
 
-  bool ReadBytes(void* dst, size_t num_bytes);
+  bool ReadBytes(void* dst, std::size_t num_bytes);
 
   template <typename SignedType>
   bool ReadSigned(uint64_t* value);
@@ -39,7 +39,7 @@
   bool ReadSLEB128(int64_t* value);
 
   template <typename AddressType>
-  size_t GetEncodedSize(uint8_t encoding);
+  std::size_t GetEncodedSize(uint8_t encoding);
 
   bool AdjustEncodedValue(uint8_t encoding, uint64_t* value);
 
