Upstream-Status: Pending

Description: libusb.h comes from different location
Author: Umang Parmar <umangjparmar@gmail.com>
Forwarded: not-needed
Last-Update: 2018-05-26

--- a/system/core/adb/client/usb_libusb.cpp
+++ b/system/core/adb/client/usb_libusb.cpp
@@ -30,7 +30,7 @@
 #include <thread>
 #include <unordered_map>
 
-#include <libusb/libusb.h>
+#include <libusb-1.0/libusb.h>
 
 #include <android-base/file.h>
 #include <android-base/logging.h>
