Switch to setuptools as distutils is deprecated.

Upstream-Status: Backport [https://pagure.io/sanlock/c/75758fc10db2354dda397d3aba63c7b72a420982]
Signed-off-by: Ross Burton <ross.burton@arm.com>

diff --git a/python/setup.py b/python/setup.py
index b3bfaf1..dfbaf21 100644
--- a/python/setup.py
+++ b/python/setup.py
@@ -4,7 +4,7 @@
 # modify, copy, or redistribute it subject to the terms and conditions
 # of the GNU General Public License v.2.
 
-from distutils.core import setup, Extension
+from setuptools import setup, Extension
 
 sanlocklib = ['sanlock']
 sanlock = Extension(name='sanlock',
