Upstream-Status: Pending

Index: gtk+-2.21.2/gdk/x11/gdkevents-x11.c
===================================================================
--- gtk+-2.21.2.orig/gdk/x11/gdkevents-x11.c	2010-06-22 17:28:04.000000000 +0800
+++ gtk+-2.21.2/gdk/x11/gdkevents-x11.c	2010-06-22 17:28:06.000000000 +0800
@@ -3062,10 +3062,9 @@
 {
   GdkScreenX11 *screen = data;
   
-  if (xsettings_client_process_event (screen->xsettings_client, (XEvent *)xevent))
-    return GDK_FILTER_REMOVE;
-  else
-    return GDK_FILTER_CONTINUE;
+  xsettings_client_process_event (screen->xsettings_client, (XEvent *)xevent);
+
+  return GDK_FILTER_CONTINUE;
 }
 
 static Bool
