Upstream-Status: Pending

=== modified file 'openbox-xdgmenu.c'
--- openbox-xdgmenu.c	2011-09-03 20:13:39 +0000
+++ openbox-xdgmenu.c	2013-12-28 17:41:04 +0000
@@ -235,7 +235,7 @@
     }
     
     g_printf("<item label=\"%s\">\n", g_strjoinv("&amp;", g_strsplit(name,"&",0))),
-    g_printf("<action name=\"Execute\"><execute>%s</execute></action>\n", exec),
+    g_printf("<action name=\"Execute\"><command>%s</command></action>\n", exec),
     g_printf("</item>\n");
     
     g_free(name);

