Upstream-Status: Pending

--- cmake-3.25.2/src/cmake/__init__.py.old	2024-12-12 20:25:48.207389622 +0000
+++ cmake-3.25.2/src/cmake/__init__.py	2024-12-12 20:25:48.207389622 +0000
@@ -44,7 +44,7 @@


 def _program(name: str, args: Iterable[str]) -> int:
-    return subprocess.call([os.path.join(CMAKE_BIN_DIR, name), *args], close_fds=False)
+    return subprocess.call([name] + args, close_fds=False)

 def _program_exit(name: str, *args: str) -> NoReturn:
     if sys.platform.startswith("win"):
