From 7db1b650bb4b01a5194a34cd7573f915656a595b Mon Sep 17 00:00:00 2001
From: Yulong Pei <Yulong.pei@windriver.com>
Date: Thu, 1 Sep 2011 01:03:14 +0800
Subject: [PATCH] replace lynx to curl in apachectl script

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Yulong Pei <Yulong.pei@windriver.com>
---
 support/apachectl.in | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/support/apachectl.in b/support/apachectl.in
index 3281c2e..6ab4ba5 100644
--- a/support/apachectl.in
+++ b/support/apachectl.in
@@ -52,11 +52,11 @@ fi
 # a command that outputs a formatted text version of the HTML at the
 # url given on the command line.  Designed for lynx, however other
 # programs may work.  
-LYNX="@LYNX_PATH@ -dump"
+CURL="/usr/bin/curl"
 #
 # the URL to your server's mod_status status page.  If you do not
 # have one, then status and fullstatus will not work.
-STATUSURL="http://localhost:@PORT@/server-status"
+STATUSURL="http://localhost:@PORT@/"
 #
 # Set this variable to a command that increases the maximum
 # number of file descriptors allowed per child process. This is
@@ -92,10 +92,16 @@ configtest)
     ERROR=$?
     ;;
 status)
-    $LYNX $STATUSURL | awk ' /process$/ { print; exit } { print } '
+    $CURL -s $STATUSURL | grep -o "It works!"
+    if [ $? != 0 ] ; then
+        echo The httpd server does not work!
+    fi
     ;;
 fullstatus)
-    $LYNX $STATUSURL
+    $CURL -s $STATUSURL | grep -o "It works!"
+    if [ $? != 0 ] ; then
+        echo The httpd server does not work!
+    fi
     ;;
 *)
     $HTTPD "$@"
-- 
2.7.4

