From d9b954bd9d0b084d9a1f5159a9f0c45802a51809 Mon Sep 17 00:00:00 2001
From: Paul Eggleton <paul.eggleton@linux.intel.com>
Date: Mon, 22 Apr 2013 16:28:16 +0000

---
Upstream-Status: Pending

 configure.ac | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index e869a6a0..101d6f9f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2514,20 +2514,20 @@ AC_ARG_WITH([libgcrypt],
     if test -f "$withval" && test -x "$withval"; then
       with_libgcrypt_config="$withval"
       with_libgcrypt="yes"
-    else if test -f "$withval/bin/gcrypt-config" && test -x "$withval/bin/gcrypt-config"; then
-      with_libgcrypt_config="$withval/bin/gcrypt-config"
+    else if test -f "$withval/bin/pkg-config" && test -x "$withval/bin/pkg-config"; then
+      with_libgcrypt_config="$withval/bin/pkg-config"
       with_libgcrypt="yes"
     else if test -d "$withval"; then
       GCRYPT_CPPFLAGS="$GCRYPT_CPPFLAGS -I$withval/include"
       GCRYPT_LDFLAGS="$GCRYPT_LDFLAGS -L$withval/lib"
       with_libgcrypt="yes"
     else
-      with_libgcrypt_config="gcrypt-config"
+      with_libgcrypt_config="pkg-config"
       with_libgcrypt="$withval"
     fi; fi; fi
   ],
   [
-    with_libgcrypt_config="libgcrypt-config"
+    with_libgcrypt_config="libpkg-config"
     with_libgcrypt="yes"
   ]
 )
