From dfb83a41aaeae326e9b6f02b233af375bc7b8815 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Fri, 29 Mar 2013 15:17:17 +0100
Subject: [PATCH] setup.py: link in sysroot, not in host directories

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>

Upstream-status: Unknown
---
 setup.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

Index: M2Crypto-0.30.1/setup.py
===================================================================
--- M2Crypto-0.30.1.orig/setup.py
+++ M2Crypto-0.30.1/setup.py
@@ -130,6 +130,7 @@ class _M2CryptoBuildExt(build_ext.build_
         self.set_undefined_options('build', ('bundledlls', 'bundledlls'))
 
         self.libraries = ['ssl', 'crypto']
+        self.openssl = os.environ.get( "STAGING_DIR" )
         if sys.platform == 'win32':
             self.libraries = ['ssleay32', 'libeay32']
             if self.openssl and openssl_version(self.openssl, 0x10100000, True):
@@ -150,8 +151,8 @@ class _M2CryptoBuildExt(build_ext.build_
 
         if self.openssl is not None:
             log.debug('self.openssl = %s', self.openssl)
-            openssl_library_dir = os.path.join(self.openssl, 'lib')
-            openssl_include_dir = os.path.join(self.openssl, 'include')
+            openssl_library_dir = os.environ.get( "STAGING_LIBDIR" )
+            openssl_include_dir = os.environ.get( "STAGING_INCDIR" )
 
             self.library_dirs.append(openssl_library_dir)
             self.include_dirs.append(openssl_include_dir)
