From 16969d42ec6514883bcee87cc89b3e7864481d7e Mon Sep 17 00:00:00 2001
From: Ed Bartosh <ed.bartosh@linux.intel.com>
Date: Tue, 13 Jun 2017 14:55:52 +0300
Subject: [PATCH] Disabled reading host configs.

Upstream-Status: Inappropriate [native]

Signed-off-by: Ed Bartosh <ed.bartosh@linux.intel.com>
---
 config.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/config.c b/config.c
index 2433457..3972150 100644
--- a/config.c
+++ b/config.c
@@ -849,14 +849,6 @@ void read_config(void)
 	memcpy(devices, const_devices,
 	       nr_const_devices*sizeof(struct device));
 
-    (void) ((parse(CONF_FILE,1) |
-	     parse(LOCAL_CONF_FILE,1) |
-	     parse(SYS_CONF_FILE,1)) ||
-	    (parse(OLD_CONF_FILE,1) |
-	     parse(OLD_LOCAL_CONF_FILE,1)));
-    /* the old-name configuration files only get executed if none of the
-     * new-name config files were used */
-
     homedir = get_homedir();
     if ( homedir ){
 	strncpy(conf_file, homedir, MAXPATHLEN );
