From 33a53dc13fd924949a582109b45fedd8d0bed59b Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 27 Jun 2017 07:42:11 -0700
Subject: [PATCH] memory.h: Include stdint.h for uintptr_t

Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 src/libstrongswan/utils/utils/memory.h | 2 ++
 1 file changed, 2 insertions(+)

--- a/src/libstrongswan/utils/utils/memory.h
+++ b/src/libstrongswan/utils/utils/memory.h
@@ -26,6 +26,8 @@
 #include <string.h>
 #endif
 
+#include <stdint.h>
+
 /**
  * Helper function that compares two binary blobs for equality
  */
