From aaec67c133dddda7204d23964732b4b788cbc9ce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Hundeb=C3=B8ll?= <martin@geanix.com>
Date: Wed, 22 May 2024 14:02:55 +0200
Subject: [PATCH] configure: lookup meson exutable from PATH

Upstream-Status: Inappropriate [workaround, would need a real fix for upstream]

Update context for qemu 10.2.0.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 configure | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/configure b/configure
index 82cace1..83bbda9 100755
--- a/configure
+++ b/configure
@@ -993,10 +993,7 @@ get_pwd() {
     fi
 }
 
-meson="$(cd pyvenv/bin; get_pwd)/meson"
-if [ -f "$meson$EXESUF" ]; then
-  meson="$meson$EXESUF"
-fi
+meson=`which meson`
 
 # Conditionally ensure Sphinx is installed.
 
