--- vpnc-0.5.1/makeman.pl.old	2008-03-16 02:17:59.000000000 -0500
+++ vpnc-0.5.1/makeman.pl	2008-03-16 02:29:34.000000000 -0500
@@ -29,7 +29,7 @@ my $vpnc = './vpnc';
 # indenting lists (those originally starting with an asterisk). I hope
 # this pays off when converting the manpage to HTML or such.
 
-open my $LONGHELP, '-|', "$vpnc --long-help";
+open my $LONGHELP, '-|', "cat ../long-help";
 my $vpnc_options    = '';
 my $relative_indent = 0;
 my $indent_needed   = 0;
