Fix for saftest failures.

Upstream-Status: Pending

Signed-off-by: yanjun.zhu <yanjun.zhu@windriver.com>

Index: openhpi-3.6.1/openhpid/safhpi.c
===================================================================
--- openhpi-3.6.1.orig/openhpid/safhpi.c
+++ openhpi-3.6.1/openhpid/safhpi.c
@@ -1933,7 +1933,7 @@ SaErrorT SAHPI_API saHpiSensorThresholds
         oh_release_domain(d); /* Unlock domain */
 
         OH_CALL_ABI(h, set_sensor_thresholds, SA_ERR_HPI_INVALID_CMD, rv,
-                    ResourceId, SensorNum, SensorThresholds);
+                    ResourceId, SensorNum, &tmp);
         oh_release_handler(h);
 
         return rv;
Index: openhpi-3.6.1/utils/sahpi_struct_utils.c
===================================================================
--- openhpi-3.6.1.orig/utils/sahpi_struct_utils.c
+++ openhpi-3.6.1/utils/sahpi_struct_utils.c
@@ -3861,6 +3861,9 @@ SaHpiBoolT oh_valid_textbuffer(SaHpiText
                                         /* found a unpaired surrogate */
                                         return SAHPI_FALSE;
                                 }
+                        } else {
+                            /*the first 2 bytes wrong*/
+                            return SAHPI_FALSE;
                         }
                 }
                 break;
