Upstream-Status: Inappropriate [embedded specific]
  
Signed-off-by: Armin Kuster <akuster808@gmail.com>


Index: git/Makefile
===================================================================
--- git.orig/Makefile
+++ git/Makefile
@@ -7,15 +7,8 @@
 
 P_OUTPUT = output
 P_PWD ?= $(shell pwd)
-P_KVER ?= $(shell uname -r)
 P_BOOTUP_SCRIPT ?= scripts/bootup/lkrg-bootup.sh
 TARGET := p_lkrg
-ifneq ($(KERNELRELEASE),)
-    KERNEL := /lib/modules/$(KERNELRELEASE)/build
-else
-    ## KERNELRELEASE not set.
-    KERNEL := /lib/modules/$(P_KVER)/build
-endif
 
 #
 # Use DEBUG=on for debug build.
@@ -94,14 +87,13 @@ $(TARGET)-objs += src/modules/ksyms/p_re
                   src/p_lkrg_main.o
 
 
-all:
-#	$(MAKE) -C $(KERNEL) M=$(P_PWD) modules CONFIG_DEBUG_SECTION_MISMATCH=y
-	$(MAKE) -C $(KERNEL) M=$(P_PWD) modules
+modules:
+	$(MAKE) -C $(KERNEL_SRC) M=$(P_PWD) modules
 	mkdir -p $(P_OUTPUT)
 	cp $(P_PWD)/$(TARGET).ko $(P_OUTPUT)
 
-install:
-	$(MAKE) -C $(KERNEL) M=$(P_PWD) modules_install
+moduled_install:
+	$(MAKE) -C $(KERNEL_SRC) M=$(P_PWD) modules_install
 	depmod -a
 	$(P_PWD)/$(P_BOOTUP_SCRIPT) install
 
@@ -109,7 +101,7 @@ uninstall:
 	$(P_PWD)/$(P_BOOTUP_SCRIPT) uninstall
 
 clean:
-	$(MAKE) -C $(KERNEL) M=$(P_PWD) clean
+	$(MAKE) -C $(KERNEL_SRC) M=$(P_PWD) clean
 	$(RM) Module.markers modules.order
 	$(RM) $(P_PWD)/src/modules/kmod/client/kmod/Module.markers
 	$(RM) $(P_PWD)/src/modules/kmod/client/kmod/modules.order
