From 1d4e63619fbaf50c60414f7ab46c116fb6ec11dd Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Mon, 20 Feb 2023 11:24:28 +0800
Subject: [PATCH] support type uint64 and uint8 for smbios

Signed-off-by: Joseph Liu <kwliu@nuvoton.com>
---
 types.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/types.hpp b/types.hpp
index cb59510..451ed11 100644
--- a/types.hpp
+++ b/types.hpp
@@ -20,7 +20,7 @@ class Manager;
 
 /** @brief Inventory manager supported property types. */
 using InterfaceVariantType =
-    std::variant<bool, size_t, int64_t, uint16_t, std::string,
+    std::variant<bool, size_t, int64_t, uint8_t, uint16_t, uint64_t, std::string,
                  std::vector<uint8_t>, std::vector<std::string>>;
 
 template <typename T>
-- 
2.34.1

