Upstream Status: Inappropriate [No upstream maintenance]

Signed-off-by: Anne Mulhern <mulhern@yoctoproject.org>

---

Index: Bastille/BastilleBackEnd
===================================================================
--- Bastille.orig/BastilleBackEnd	2013-08-21 12:40:54.000000000 -0400
+++ Bastille/BastilleBackEnd	2013-08-21 12:43:21.895950001 -0400
@@ -52,11 +52,13 @@
 my $force = 0;
 my $debug = 0;
 my $alternate_config=undef;
+my $os_version=undef;
 
 if( Getopt::Long::GetOptions( "n"     => \$nodisclaim,
                               "v"     => \$verbose,
                               "force" => \$force,
 			      "f=s"   => \$alternate_config,
+                              "os=s"  => \$os_version,
 			      "debug" => \$debug) ) {
     $error = 0; # no parse error
 
@@ -66,7 +68,8 @@
 
 &setOptions(
   debug => $debug,
-  verbose => $verbose);
+  verbose => $verbose,
+  os => $os_version);
 &ConfigureForDistro;
 
 if ( $error ) { # GetOptions couldn't parse all of the args
