From 6ff255eff4fea6350b5e0462fee176fadc26fc1c Mon Sep 17 00:00:00 2001
From: Jussi Kukkonen <jussi.kukkonen@intel.com>
Date: Sun, 12 Jun 2016 18:32:49 +0300
Subject: [PATCH] configure.ac: install GtkBuilder interface files for GTK+3
 too

Upstream-Status: Submitted [https://github.com/lathiat/avahi/pull/130]
Signed-off-by: Jussi Kukkonen <jussi.kukkonen@intel.com>
Signed-off-by: Dengke Du <dengke.du@windriver.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 87a9a17..9860dcc 100644
--- a/configure.ac
+++ b/configure.ac
@@ -965,7 +965,7 @@ AC_SUBST(avahi_socket)
 #
 # Avahi interfaces dir
 #
-if test "x$HAVE_PYTHON_DBUS" = "xyes" -o "x$HAVE_GTK" = "xyes"; then
+if test "x$HAVE_PYTHON_DBUS" = "xyes" -o "x$HAVE_GTK" = "xyes" -o "x$HAVE_GTK3" = "xyes"; then
 	interfacesdir="${datadir}/${PACKAGE}/interfaces/"
 	AC_SUBST(interfacesdir)
 fi
-- 
2.8.1

