From 4dddd9e533847f07327cde9a062fc7cbba9d5584 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 10 Jun 2017 00:46:39 -0700
Subject: [PATCH] sigqueue: Rename _sifields to __si_fields on musl

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 memcheck/tests/linux/sigqueue.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/memcheck/tests/linux/sigqueue.c b/memcheck/tests/linux/sigqueue.c
index 4b6fb0f..d37fe93 100644
--- a/memcheck/tests/linux/sigqueue.c
+++ b/memcheck/tests/linux/sigqueue.c
@@ -8,6 +8,11 @@
 #include <syscall.h>
 #include <unistd.h>
 
+/* musl libc defines siginfo_t __si_fields instead of _sifields */
+#if defined(__linux__) && !defined(__GLIBC__)
+#define _sifields __si_fields
+#endif
+
 int main(int argc, char **argv)
 {
   siginfo_t *si;
