From 900b3f9b0765488f0e1f457ff5d5d1e3efe897de Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Wed, 27 Apr 2022 10:01:22 +0200
Subject: [PATCH] Fix drd/tests/shared_timed_mutex.cpp

Upstream-Status: Submitted [https://bugs.kde.org/show_bug.cgi?id=453055]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>
---
 drd/tests/shared_timed_mutex.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drd/tests/shared_timed_mutex.cpp b/drd/tests/shared_timed_mutex.cpp
index 9741169..dfd97a4 100644
--- a/drd/tests/shared_timed_mutex.cpp
+++ b/drd/tests/shared_timed_mutex.cpp
@@ -43,6 +43,7 @@ void g()
         if (test_mutex.try_lock_shared_until(then))
         {
             test_mutex.unlock_shared();
+            break;
         }
     }
     if (i == 3)
-- 
2.30.2

