From fb77fef4f866dac7bcc6d1ae025da60564869f84 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 10 Jun 2017 01:06:11 -0700
Subject: [PATCH 4/6] pth_atfork1.c: Define error() API for musl

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 none/tests/pth_atfork1.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/none/tests/pth_atfork1.c b/none/tests/pth_atfork1.c
index 34201ef..b7f5f2d 100644
--- a/none/tests/pth_atfork1.c
+++ b/none/tests/pth_atfork1.c
@@ -18,7 +18,7 @@
    Boston, MA 02111-1307, USA.  */
 
 #include <errno.h>
-#if !defined(__APPLE__) && !defined(__sun)
+#if !defined(__APPLE__) && !defined(__sun) && defined(__GLIBC__)
 # include <error.h>
 #endif
 #include <stdlib.h>
@@ -27,7 +27,7 @@
 #include <sys/wait.h>
 #include <stdio.h>
 
-#if defined(__APPLE__) || defined(__sun)
+#if defined(__APPLE__) || defined(__sun) || (defined(__linux__) && !defined(__GLIBC__))
 #include <string.h>  /* strerror */
 static void error (int status, int errnum, char* msg)
 {
-- 
2.13.1

