From 206f334d62da17093e784094c9d9e80c853f704f Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 12 May 2017 03:03:46 -0400
Subject: [PATCH] do not support python2

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 configure.ac       | 16 ----------------
 python/Makefile.am | 24 ------------------------
 2 files changed, 40 deletions(-)

diff --git a/configure.ac b/configure.ac
index 200a7cc..8431c51 100644
--- a/configure.ac
+++ b/configure.ac
@@ -22,22 +22,6 @@ AM_MISSING_PROG([AUTOM4TE], [autom4te])
 # Needed by tests/atlocal.in.
 AC_SUBST([O0CFLAGS], [`echo $CFLAGS | sed 's/-O[[0-9]] *//'`])
 
-AM_PATH_PYTHON
-AM_CONDITIONAL([HAVE_PYTHON], test -n "$PYTHON")
-[if test -z "$PYTHON"; then]
-    [echo "The python interpreter was not found in the search path. The Python language bindings"]
-    [echo "will not be built.  If you want to build the Python language bindings, please ensure"]
-    [echo "that python is installed and its directory is included in the search path."]
-    [echo "Then run configure again before attempting to build Satyr."]
-[fi]
-
-# Just PKG_CHECK_MODULES([PYTHON], [python]) works only with python2.7+
-# Below, if python is not found, we set up for python2.6 w/o checking:
-PKG_CHECK_MODULES([PYTHON], [python2],,[
-    PYTHON_LIBS='-L/usr/lib64 -lpython2.6'
-    PYTHON_CFLAGS='-I/usr/include/python2.6'
-])
-
 AC_ARG_ENABLE([python-manpage],
               [AS_HELP_STRING([--disable-python-manpage],
                               [Disable python bindings manpage build])],
diff --git a/python/Makefile.am b/python/Makefile.am
index 36fc2e9..0b73f66 100644
--- a/python/Makefile.am
+++ b/python/Makefile.am
@@ -1,12 +1,5 @@
 SUBDIRS = . doc
 
-py2satyrdir = $(pyexecdir)/satyr
-
-py2satyr_PYTHON = \
-    __init__.py
-
-py2satyr_LTLIBRARIES = _satyr.la
-
 PYEXTFILES = \
     py_cluster.h \
     py_cluster.c \
@@ -75,23 +68,6 @@ PYEXTLDFLAGS = \
     -module \
     -avoid-version
 
-_satyr_la_SOURCES = $(PYEXTFILES)
-_satyr_la_CPPFLAGS = \
-    $(PYEXTCPPFLAGS) \
-    $(PYTHON_CFLAGS)
-_satyr_la_LDFLAGS = \
-    $(PYEXTLDFLAGS) \
-    -export-symbols-regex init_satyr
-
-# Depending on the configure option, we either link libsatyr.so dynamically (as
-# is usual), or directly include its object files (by including the
-# convenience library).
-if ENABLE_STATIC_PYTHON_MODULE
-_satyr_la_LIBADD = ../lib/libsatyr_conv.la
-else
-_satyr_la_LIBADD = ../lib/libsatyr.la
-endif
-
 if WITH_PYTHON3
 py3satyrdir = $(py3execdir)/satyr
 
-- 
2.8.1

