From 1128a98fd1676981e536d8773f363cb832cfa6bb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Tue, 6 Mar 2018 22:28:56 +0100
Subject: [PATCH] Use native tools to build docs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>

---
Upstream-Status: Pending

 docs/Makefile.am | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/docs/Makefile.am b/docs/Makefile.am
index b20918a..47406eb 100644
--- a/docs/Makefile.am
+++ b/docs/Makefile.am
@@ -175,7 +175,7 @@
 	$(AM_V_GEN) touch $@
 
 gnuplot.tex: allterm.h doc2tex$(EXEEXT) $(srcdir)/gnuplot.doc
-	$(AM_V_GEN)./doc2tex$(EXEEXT) -figures $(srcdir)/gnuplot.doc $@
+	$(AM_V_GEN) doc2tex$(EXEEXT) -figures $(srcdir)/gnuplot.doc $@
 
 pdf: pdf_figures
 	$(MAKE) gnuplot.pdf
@@ -255,7 +255,7 @@

 gnuplot-ja.pdf: gnuplot-ja.doc title-ja.tex pdf_figures allterm-ja.h doc2tex$(EXEEXT)
 	iconv -f EUC-JP -t UTF-8 gnuplot-ja.doc > ja.doc
-	./doc2tex$(EXEEXT) -figures -japanese ja.doc | sed 's/titlepag.tex/title-ja.tex/' > gnuplot-ja.tex
+	doc2tex$(EXEEXT) -figures -japanese ja.doc | sed 's/titlepag.tex/title-ja.tex/' > gnuplot-ja.tex
 	lualatex gnuplot-ja
 	lualatex gnuplot-ja
 	makeindex gnuplot-ja

@@ -268,7 +268,7 @@ gnuplot.ps: gnuplot.dvi
 hlp: gnuplot.hlp
 
 gnuplot.hlp: doc2hlp$(EXEEXT) $(srcdir)/gnuplot.doc
-	$(AM_V_GEN)./doc2hlp$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.hlp
+	$(AM_V_GEN)doc2hlp$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.hlp
 
 doc2hlp_SOURCES = doc2hlp.c termdoc.c
 
@@ -276,14 +276,14 @@ doc2hlp_SOURCES = doc2hlp.c termdoc.c
 gih: gnuplot.gih
 
 gnuplot.gih: doc2gih$(EXEEXT) $(srcdir)/gnuplot.doc
-	$(AM_V_GEN)./doc2gih$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.gih
+	$(AM_V_GEN)doc2gih$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.gih
 
 doc2gih_SOURCES = doc2gih.c termdoc.c
 
 # To include all terminals in the .gih file
 allgih: alldoc2gih$(EXEEXT) $(srcdir)/gnuplot.doc
 	@echo "generate gnuplot.gih with all terminals"
-	$(AM_V_at)./alldoc2gih$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.gih
+	$(AM_V_at)alldoc2gih$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.gih
 
 alldoc2gih_SOURCES = doc2gih.c termdoc.c
 alldoc2gih_CPPFLAGS = -DALL_TERM_DOC $(AM_CPPFLAGS)
@@ -298,7 +298,7 @@ $(srcdir)/windows/wgnuplot.hhk
 wxhelp/wgnuplot.hhc wxhelp/wgnuplot.hhk wxhelp/*.html windows/*.png
 
 wxhelp/wgnuplot.html: doc2wxhtml$(EXEEXT) $(srcdir)/gnuplot.doc 
-	$(AM_V_GEN) ./doc2wxhtml$(EXEEXT) $(srcdir)/gnuplot.doc wxhelp/
+	$(AM_V_GEN) doc2wxhtml$(EXEEXT) $(srcdir)/gnuplot.doc wxhelp/
 
 doc2wxhtml_SOURCES = windows/doc2html.c termdoc.c xref.c allterm.h
 doc2wxhtml_CPPFLAGS = -DALL_TERM_DOC -DWXHELP -I../src $(AM_CPPFLAGS)
@@ -318,7 +318,7 @@
 	$(AM_V_at) [ -d html ] || mkdir -p html
 	$(AM_V_at) rm -f html/Overview.html html/Plotting_Styles.html html/Commands.html html/Terminals.html
 	$(AM_V_at) rm -f html/NewFeatures.html
-	$(AM_V_GEN)./doc2web$(EXEEXT) $(srcdir)/gnuplot.doc html
+	$(AM_V_GEN) doc2web$(EXEEXT) $(srcdir)/gnuplot.doc html
 	$(AM_V_GEN) sort -k 3 html/index.hhk > html/index.html
 
 svg_figures: $(GNUPLOT_EXE) $(srcdir)/plotstyles.gnu
@@ -372,7 +372,7 @@ install-info: gnuplot.info
 ipf: gnuplot.ipf
 
 gnuplot.ipf: doc2ipf$(EXEEXT) $(srcdir)/gnuplot.doc
-	$(AM_V_GEN) ./doc2ipf$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.ipf
+	$(AM_V_GEN) doc2ipf$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.ipf
 
 doc2ipf_SOURCES = doc2ipf.c termdoc.c xref.c
 
@@ -380,7 +380,7 @@ doc2ipf_SOURCES = doc2ipf.c termdoc.c xref.c
 rtf: gnuplot.rtf
 
 gnuplot.rtf: doc2rtf$(EXEEXT) $(srcdir)/gnuplot.doc
-	$(AM_V_GEN) ./doc2rtf$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.rtf
+	$(AM_V_GEN) doc2rtf$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.rtf
 
 doc2rtf_SOURCES = doc2rtf.c termdoc.c xref.c
 
@@ -386,7 +386,7 @@
 
 # this is how to check the gnuplot.doc file
 check-local: checkdoc$(EXEEXT)
-	$(AM_V_at)./checkdoc$(EXEEXT) < $(srcdir)/gnuplot.doc; \
+	$(AM_V_at)checkdoc$(EXEEXT) < $(srcdir)/gnuplot.doc; \
 	if test $$? -eq 0; then \
          echo "PASS: gnuplot.doc"; \
 	else \
--- ./docs/Makefile.am.orig	2025-10-06 20:44:25.315270887 +0200
+++ ./docs/Makefile.am	2025-10-06 21:22:31.287527296 +0200
@@ -268,7 +268,7 @@
 hlp: gnuplot.hlp
 
 gnuplot.hlp: doc2hlp$(EXEEXT) $(srcdir)/gnuplot.doc
-	$(AM_V_GEN)./doc2hlp$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.hlp
+	$(AM_V_GEN)doc2hlp$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.hlp
 
 doc2hlp_SOURCES = doc2hlp.c termdoc.c
 
@@ -276,14 +276,14 @@
 gih: gnuplot.gih
 
 gnuplot.gih: doc2gih$(EXEEXT) $(srcdir)/gnuplot.doc
-	$(AM_V_GEN)./doc2gih$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.gih
+	$(AM_V_GEN)doc2gih$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.gih
 
 doc2gih_SOURCES = doc2gih.c termdoc.c
 
 # To include all terminals in the .gih file
 allgih: alldoc2gih$(EXEEXT) $(srcdir)/gnuplot.doc
 	@echo "generate gnuplot.gih with all terminals"
-	$(AM_V_at)./alldoc2gih$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.gih
+	$(AM_V_at)alldoc2gih$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.gih
 
 alldoc2gih_SOURCES = doc2gih.c termdoc.c
 alldoc2gih_CPPFLAGS = -DALL_TERM_DOC $(AM_CPPFLAGS)
@@ -298,7 +298,7 @@
 wxhelp/wgnuplot.hhc wxhelp/wgnuplot.hhk wxhelp/*.html windows/*.png
 
 wxhelp/wgnuplot.html: doc2wxhtml$(EXEEXT) $(srcdir)/gnuplot.doc 
-	$(AM_V_GEN) ./doc2wxhtml$(EXEEXT) $(srcdir)/gnuplot.doc wxhelp/
+	$(AM_V_GEN) doc2wxhtml$(EXEEXT) $(srcdir)/gnuplot.doc wxhelp/
 
 doc2wxhtml_SOURCES = windows/doc2html.c termdoc.c xref.c allterm.h
 doc2wxhtml_CPPFLAGS = -DALL_TERM_DOC -DWXHELP -I../src $(AM_CPPFLAGS)
@@ -318,7 +318,7 @@
 	$(AM_V_at) [ -d html ] || mkdir -p html
 	$(AM_V_at) rm -f html/Overview.html html/Plotting_Styles.html html/Commands.html html/Terminals.html
 	$(AM_V_at) rm -f html/NewFeatures.html
-	$(AM_V_GEN)./doc2web$(EXEEXT) $(srcdir)/gnuplot.doc html
+	$(AM_V_GEN)doc2web$(EXEEXT) $(srcdir)/gnuplot.doc html
 	$(AM_V_GEN) sort -k 3 html/index.hhk > html/index.html
 
 svg_figures: $(GNUPLOT_EXE) $(srcdir)/plotstyles.gnu
@@ -372,7 +372,7 @@
 ipf: gnuplot.ipf
 
 gnuplot.ipf: doc2ipf$(EXEEXT) $(srcdir)/gnuplot.doc
-	$(AM_V_GEN) ./doc2ipf$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.ipf
+	$(AM_V_GEN) doc2ipf$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.ipf
 
 doc2ipf_SOURCES = doc2ipf.c termdoc.c xref.c
 
@@ -380,13 +380,13 @@
 rtf: gnuplot.rtf
 
 gnuplot.rtf: doc2rtf$(EXEEXT) $(srcdir)/gnuplot.doc
-	$(AM_V_GEN) ./doc2rtf$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.rtf
+	$(AM_V_GEN) doc2rtf$(EXEEXT) $(srcdir)/gnuplot.doc gnuplot.rtf
 
 doc2rtf_SOURCES = doc2rtf.c termdoc.c xref.c
 
 # this is how to check the gnuplot.doc file
 check-local: checkdoc$(EXEEXT)
-	$(AM_V_at)./checkdoc$(EXEEXT) < $(srcdir)/gnuplot.doc; \
+	$(AM_V_at)checkdoc$(EXEEXT) < $(srcdir)/gnuplot.doc; \
 	if test $$? -eq 0; then \
 	  echo "PASS: gnuplot.doc"; \
 	else \
