Index: git/plugins.d/charts.d.plugin
===================================================================
--- git.orig/plugins.d/charts.d.plugin
+++ git/plugins.d/charts.d.plugin
@@ -246,16 +246,6 @@ else
     info "configuration file '$myconfig' not found. Using defaults."
 fi
 
-# we check for the timeout command, after we load our
-# configuration, so that the user may overwrite the
-# timeout command we use, providing a function that
-# can emulate the timeout command we need:
-# > timeout SECONDS command ...
-if [ $check_for_timeout -eq 1 ]
-    then
-    require_cmd timeout || exit 1
-fi
-
 # -----------------------------------------------------------------------------
 # internal checks
 
