From 2f54a8cf959ea83c452b1a22acbc47d2738b5fa0 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 23 Nov 2018 15:47:20 +0800
Subject: [PATCH] elfutils: 0.170 -> 0.172

Upstream-Status: Pending [from debian]
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

---
 tests/run-native-test.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/run-native-test.sh b/tests/run-native-test.sh
index 5351e02..70f6681 100755
--- a/tests/run-native-test.sh
+++ b/tests/run-native-test.sh
@@ -83,6 +83,9 @@ native_test()
 # "cannot attach to process: Function not implemented".
 [ "$(uname)" = "GNU/kFreeBSD" ] && exit 77
 
+# hurd's /proc/$PID/maps does not give paths yet.
+[ "$(uname)" = "GNU" ] && exit 77
+
 native_test ${abs_builddir}/allregs
 native_test ${abs_builddir}/funcretval
 
