From 0d863b444c9a54a92dab176b1b656c116923e1ca Mon Sep 17 00:00:00 2001
From: alan-baker <alanbaker@google.com>
Date: Wed, 30 Mar 2022 12:59:28 -0400
Subject: [PATCH] Remove dead variable (#984)

Upstream-Status: Backport [https://github.com/google/amber/commit/627ee453d6047ced0e2dd13cde983b341d0615e3]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/vkscript/command_parser.cc | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/vkscript/command_parser.cc b/src/vkscript/command_parser.cc
index 610c7e6..3429d4d 100644
--- a/src/vkscript/command_parser.cc
+++ b/src/vkscript/command_parser.cc
@@ -486,7 +486,6 @@ Result CommandParser::ParseValues(const std::string& name,
                                   std::vector<Value>* values) {
   assert(values);
 
-  uint32_t row_index = 0;
   auto token = tokenizer_->NextToken();
   size_t seen = 0;
   while (!token->IsEOL() && !token->IsEOS()) {
@@ -515,7 +514,6 @@ Result CommandParser::ParseValues(const std::string& name,
     values->push_back(v);
     token = tokenizer_->NextToken();
 
-    ++row_index;
     ++seen;
   }
 
-- 
2.37.3

