From: rpm-build <rpm-build>
Date: Tue, 26 Nov 2024 11:13:18 +0100
Subject: [PATCH] Disable Werror

Upstream-Status: Pending

Signed-off-by: rpm-build <rpm-build>

diff --git a/src/Makefile b/src/Makefile
index c688005..01b7a80 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -20,7 +20,7 @@ ifeq ($(PLATFORM),Darwin)
 endif
 
 # The base compiler flags. More can be added on command line.
-CFLAGS += -I. -I../inc -O2 -Werror
+CFLAGS += -I. -I../inc
 # Enable a bunch of optional warnings.
 CFLAGS += \
 	-pedantic \
@@ -81,7 +81,7 @@ ifeq ($(PLATFORM),Windows)
 else
 	CC ?= clang
 	CFLAGS += -fPIC
-	CFLAGS += -Wmissing-braces -Wformat=2
+	CFLAGS += -Wmissing-braces
 endif
 
 # Settings for blst.
