From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 6 Nov 2024 16:35:23 +0300
Subject: [PATCH] Let override CC

Upstream-Status: Pending

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/src/Makefile b/src/Makefile
index b733a11..c688005 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -79,7 +79,7 @@ ifeq ($(PLATFORM),Windows)
 	CFLAGS += -D_CRT_SECURE_NO_WARNINGS
 	CFLAGS += -Wno-missing-braces -Wno-format
 else
-	CC = clang
+	CC ?= clang
 	CFLAGS += -fPIC
 	CFLAGS += -Wmissing-braces -Wformat=2
 endif
