From b517158e8ffc6a665506007b20708d6c2589cec1 Mon Sep 17 00:00:00 2001
From: Marcin Juszkiewicz <hrw@openedhand.com>
Date: Fri, 8 Jun 2007 08:35:12 +0000
Subject: [PATCH] mtools: imported from OE

Upstream-Status: Inappropriate [licensing]
---
 Makefile.in  | 11 ++++++-----
 configure.ac | 27 +++++++++++++++++++++++++++
 2 files changed, 33 insertions(+), 5 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 7b305b0..70c8c74 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -26,10 +26,11 @@ USERCFLAGS =
 USERLDFLAGS =
 USERLDLIBS =
 
-MAKEINFO = makeinfo
-TEXI2DVI = texi2dvi
-TEXI2PDF = texi2pdf
-TEXI2HTML = texi2html
+MAKEINFO = @MAKEINFO@
+TEXI2DVI = @TEXI2DVI@
+TEXI2PDF = @TEXI2PDF@
+TEXI2HTML = @TEXI2HTML@
+DVI2PS = @DVI2PS@
 
 
 # do not edit below this line
@@ -185,7 +186,7 @@ dvi: mtools.dvi
 
 ps: mtools.ps
 %.ps: %.dvi
-	dvips -f < $< > $@
+	$(DVI2PS) -f < $< > $@
 
 pdf: mtools.pdf
 %.pdf: %.texi sysconfdir.texi
diff --git a/configure.ac b/configure.ac
index 37f0d00..c93cfb5 100644
--- a/configure.ac
+++ b/configure.ac
@@ -36,6 +36,33 @@ AC_PATH_PROG(INSTALL_INFO, install-info, "")
 AC_C_CONST
 AC_C_INLINE
 
+AC_CHECK_PROG(MAKEINFO, makeinfo, makeinfo, )
+if test "x$MAKEINFO" = "x"; then
+	MAKEINFO="@echo makeinfo missing; true"
+fi
+AC_CHECK_PROG(TEXI2DVI, texi2dvi, texi2dvi, )
+if test "x$TEXI2DVI" = "x"; then
+	TEXI2DVI="@echo texi2dvi missing; true"
+fi
+AC_CHECK_PROG(TEXI2PDF, texi2pdf, texi2pdf, )
+if test "x$TEXI2PDF" = "x"; then
+	TEXI2PDF="@echo texi2pdf missing; true"
+fi
+AC_CHECK_PROG(TEXI2HTML, texi2html, texi2html, )
+if test "x$TEXI2HTML" = "x"; then
+	TEXI2HTML="@echo texi2html missing; true"
+fi    
+AC_CHECK_PROG(DVI2PS, dvi2ps, dvi2ps, )
+if test "x$DVI2PS" = "x"; then
+	DVI2PS="@echo dvi2ps missing; true"
+fi
+
+AC_SUBST(MAKEINFO)
+AC_SUBST(TEXI2DVI)
+AC_SUBST(TEXI2PDF)
+AC_SUBST(TEXI2HTML)
+AC_SUBST(DVI2PS)
+
 dnl Check for configuration options
 dnl Enable OS/2 extended density format disks
 AC_ARG_ENABLE(xdf,
