Rename 'rand' variable to avoid conflict.

Upstream-Status: pending
Signed-off-by: Björn Stenberg <bjst@enea.com>

diff -u glib-2.34.3/tests/refcount/signals.c~ glib-2.34.3/tests/refcount/signals.c
--- glib-2.34.3/tests/refcount/signals.c	2012-11-26 17:52:48.000000000 +0100
+++ glib-2.34.3/tests/refcount/signals.c	2013-02-08 14:24:10.052477546 +0100
@@ -9,7 +9,7 @@
 #define MY_IS_TEST_CLASS(tclass)   (G_TYPE_CHECK_CLASS_TYPE ((tclass), G_TYPE_TEST))
 #define MY_TEST_GET_CLASS(test)    (G_TYPE_INSTANCE_GET_CLASS ((test), G_TYPE_TEST, GTestClass))
 
-static GRand *rand;
+static GRand *grand;
 
 typedef struct _GTest GTest;
 typedef struct _GTestClass GTestClass;
@@ -84,7 +84,7 @@
       NULL
     };
 
-    rand = g_rand_new();
+    grand = g_rand_new();
 
     test_type = g_type_register_static (G_TYPE_OBJECT, "GTest",
         &test_info, 0);
@@ -218,7 +218,7 @@
 static void
 my_test_do_prop (GTest * test)
 {
-  test->value = g_rand_int (rand);
+  test->value = g_rand_int (grand);
   g_object_notify (G_OBJECT (test), "test-prop");
 }
 
