Upstream-Status: Pending

Index: pxaregs-1.14/pxaregs.c
===================================================================
--- pxaregs-1.14.orig/pxaregs.c
+++ pxaregs-1.14/pxaregs.c
@@ -1794,7 +1794,7 @@ static int getmem(u32 addr)
    regaddr = map + (addr & MAP_MASK);
 
    val = *(u32*) regaddr;
-   munmap(0,MAP_SIZE);
+   munmap(map,MAP_SIZE);
 
    return val;
 }
@@ -1829,7 +1829,7 @@ static void putmem(u32 addr, u32 val)
    regaddr = map + (addr & MAP_MASK);
 
    *(u32*) regaddr = val;
-   munmap(0,MAP_SIZE);
+   munmap(map,MAP_SIZE);
 }
 
 static u32 lastaddr = 0;
