From: Al Stone <ahs3@ahs3.net>
Date: Mon, 7 Apr 2014 19:09:37 +0000
Subject: [PATCH 1/2] Fixup aapits build

From http://git.linaro.org/people/al.stone/acpica-tools.git
Upstream-status: Unknown

diff -urN acpica-unix2-20140325/tests/aapits/Makefile acpica-unix2-20140325/tests/aapits/Makefile
--- acpica-unix2-20140325/tests/aapits/Makefile	2014-04-05 14:23:14.683636794 -0600
+++ acpica-unix2-20140325-aapits/tests/aapits/Makefile	2014-04-05 15:10:57.879184598 -0600
@@ -16,6 +16,7 @@
 	atosxfwrap.c \
 	osunixxf.c \
 	../../source/common/ahids.c \
+	../../source/common/ahuuids.c \
 	../../source/common/cmfsize.c \
 	../../source/common/getopt.c \
 	../../source/components/hardware/hwtimer.c \
@@ -174,6 +175,7 @@
 	../../source/components/utilities/utexcep.c \
 	../../source/components/utilities/utfileio.c \
 	../../source/components/utilities/utglobal.c \
+	../../source/components/utilities/uthex.c \
 	../../source/components/utilities/utids.c \
 	../../source/components/utilities/utinit.c \
 	../../source/components/utilities/utlock.c \
@@ -189,6 +191,7 @@
 	../../source/components/utilities/utstate.c \
 	../../source/components/utilities/utstring.c \
 	../../source/components/utilities/uttrack.c \
+	../../source/components/utilities/utuuid.c \
 	../../source/components/utilities/utxface.c \
 	../../source/components/utilities/utxferror.c \
 	../../source/components/utilities/utxfinit.c \
