Upstream-Status: Pending

--- mdadm.d/mdadm-1.3.0/mdadm.h.org	2017-06-20 00:35:55.366978372 -0700
+++ mdadm.d/mdadm-1.3.0/mdadm.h	2017-06-20 00:36:41.244363348 -0700
@@ -29,8 +29,9 @@
 
 #define	__USE_LARGEFILE64
 #include	<unistd.h>
-extern __off64_t lseek64 __P ((int __fd, __off64_t __offset, int __whence));
+//extern __off64_t lseek64 __P ((int __fd, __off64_t __offset, int __whence));
 
+#include	<sys/sysmacros.h>
 #include	<sys/types.h>
 #include	<sys/stat.h>
 #include	<stdlib.h>
