From 4f964821398dff7ab21fec63da15e1e00b2e9277 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Wed, 24 Feb 2021 21:16:14 +0000
Subject: [PATCH] proc/escape.c: add missing include

Upstream-Status: Submitted [https://gitlab.com/procps-ng/procps/-/merge_requests/126]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 proc/escape.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/proc/escape.c b/proc/escape.c
index 2e8fb7d..e1f4612 100644
--- a/proc/escape.c
+++ b/proc/escape.c
@@ -21,6 +21,7 @@
 #include <sys/types.h>
 #include <string.h>
 #include <limits.h>
+#include <langinfo.h>
 #include "procps.h"
 #include "escape.h"
 #include "readproc.h"
