From 7b8d819e012c24df228a313beb86e1942611c904 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Sat, 4 Jun 2022 13:00:12 +0200
Subject: [PATCH] Makefile: correctly list modules when cleaning them

Upstream-Status: Submitted [https://github.com/arsv/perl-cross/pull/133]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6b35fb0..9ef9324 100644
--- a/Makefile
+++ b/Makefile
@@ -462,7 +462,7 @@ clean-subdirs:
 
 # assuming modules w/o Makefiles were never built and need no cleaning
 clean-modules: config.h
-	@for i in $(modules disabled); do \
+	@for i in $(modules) $(disabled); do \
 		test -f $$i/Makefile && \
 		touch $$i/Makefile && \
 		$(MAKE) -C $$i clean \
