From d22f2bb5afcd278b68999f5ce0362328fc8c7723 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Thu, 3 Jun 2021 18:50:56 +0200
Subject: [PATCH] cnf/configure_func_sel.sh: disable thread_safe_nl_langinfo_l

Upstream-Status: Submitted [https://github.com/arsv/perl-cross/pull/115]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 cnf/configure_func_sel.sh | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/cnf/configure_func_sel.sh b/cnf/configure_func_sel.sh
index f48294f..90d350d 100644
--- a/cnf/configure_func_sel.sh
+++ b/cnf/configure_func_sel.sh
@@ -97,5 +97,9 @@ else
 	result "irrelevant"
 fi
 
-# Assume nl_langinfo_l is threadsafe if available
-define d_thread_safe_nl_langinfo_l "$d_nl_langinfo_l"
+# thread_safe_nl_langinfo_l is not enabled by default
+# by upstream, and causes t/Langinfo.t to fail when it is
+# (starting from 5.34.0). This means the configuration is
+# either not well tested, or not at all tested, so we should
+# pick a safer option.
+define d_thread_safe_nl_langinfo_l "undef"
