Use LD and obey LDFLAGS

Signed-off-by: Christopher Larson <chris_larson@mentor.com>
Upstream-Status: Pending

diff --git a/make/linux/Makefile b/make/linux/Makefile
index 391b6e3..a2ac13e 100644
--- a/make/linux/Makefile
+++ b/make/linux/Makefile
@@ -64,6 +64,7 @@ objects_so:=$(patsubst %$(EXTENSION),%.lo,$(sources))
 WARNINGFLAGS=-Wall -Wextra -Wno-unknown-pragmas -ansi -fno-gnu-keywords -Wshadow
 COMPILEFLAGS=$(WARNINGFLAGS) $(CXXFLAGS) $(CPPFLAGS) $(DEBUGFLAGS) $(INCLUDE)
 DEPENDFLAGS  = $(CXXFLAGS) $(INCLUDE)
+LINKFLAGS=$(LDFLAGS)

 ifeq (Darwin,$(shell uname -s))
 all: staticlib
@@ -91,7 +92,7 @@ $(LIBRARY): $(objects)
	$(RANLIB) $@

 $(LIBRARY_SO): $(objects_so)
-	$(CXX) -shared -Wl,-soname,$(LIBRARY_SO_VER) -o $(LIBRARY_SO_VER) $(objects_so)
+	$(LD) $(LINKFLAGS) -shared -Wl,-soname,$(LIBRARY_SO_VER) -o $(LIBRARY_SO_VER) $(objects_so)
	rm -f $(LIBRARY_SO)
	ln -s $(LIBRARY_SO_VER) $(LIBRARY_SO)

diff --git a/make/mingw32/Makefile b/make/mingw32/Makefile
index e5986ef..6ca16aa 100644
--- a/make/mingw32/Makefile
+++ b/make/mingw32/Makefile
@@ -18,6 +18,7 @@ DEBUGFLAGS=-g -DDEBUG
 endif
 CROSS   =
 CXX     = $(CROSS)g++
+LD      = $(CXX)
 CC      = $(CROSS)gcc
 WINDRES = $(CROSS)windres
 RANLIB  = $(CROSS)ranlib
@@ -33,6 +34,7 @@ LIBS = libebml.a
 endif
 INCS = -I"$(shell pwd)/../.."
 COMPILEFLAGS = $(DEBUGFLAGS) $(INCS) $(DLLFLAGS) $(CXXFLAGS)
+LINKFLAGS = $(LDFLAGS)

 .PHONY: all all-before all-after clean clean-custom

@@ -54,7 +56,7 @@ libebml.a: $(OBJ)
	$(RANLIB) $@

 libebml.dll: $(OBJ)
-	$(CXX) -shared -Wl,--export-all -Wl,--out-implib=$@.a -o $@ $(OBJ)
+	$(LD) $(LINKFLAGS) -shared -Wl,--export-all -Wl,--out-implib=$@.a -o $@ $(OBJ)

 depend:
	@echo Calculating dependecies:
--
2.8.0
