From 2e8dc2c28c0938dbbb85ebbac2b9a60be9ccd9f3 Mon Sep 17 00:00:00 2001
From: Max Kellermann <max@musicpd.org>
Date: Wed, 23 Nov 2022 12:25:50 +0100
Subject: [PATCH] SearchPage: use regular integer to fix -Wenum-constexpr-conversion

Upstream-Status: Backport [https://github.com/MusicPlayerDaemon/ncmpc/commit/ddd1757907f0376b5843f707bf182b7827ff6591]
---
 src/SearchPage.cxx | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/SearchPage.cxx b/src/SearchPage.cxx
index 2fa5edbc..3f91c4fe 100644
--- a/src/SearchPage.cxx
+++ b/src/SearchPage.cxx
@@ -81,7 +81,7 @@ search_get_tag_id(const char *name)
 }
 
 struct SearchMode {
-	enum mpd_tag_type table;
+	int table;
 	const char *label;
 };
 
@@ -89,8 +89,8 @@ static constexpr SearchMode mode[] = {
 	{ MPD_TAG_TITLE, N_("Title") },
 	{ MPD_TAG_ARTIST, N_("Artist") },
 	{ MPD_TAG_ALBUM, N_("Album") },
-	{ (enum mpd_tag_type)SEARCH_URI, N_("Filename") },
-	{ (enum mpd_tag_type)SEARCH_ARTIST_TITLE, N_("Artist + Title") },
+	{ SEARCH_URI, N_("Filename") },
+	{ SEARCH_ARTIST_TITLE, N_("Artist + Title") },
 	{ MPD_TAG_COUNT, nullptr }
 };
 
-- 
2.39.0

