From 6b26f3dbf9bf577d71534ab7410de66d06e46ba2 Mon Sep 17 00:00:00 2001
From: Carlos Rafael Giani <dv@pseudoterminal.org>
Date: Sat, 6 Apr 2013 23:52:11 +0200
Subject: [PATCH] Fix crash with gst-inspect Chris Lord <chris@openedhand.com>

Upstream-Status: Pending

Signed-off-by: Carlos Rafael Giani <dv@pseudoterminal.org>
---
 tools/gst-inspect.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/gst-inspect.c b/tools/gst-inspect.c
index 23b7c44..b79b02e 100644
--- a/tools/gst-inspect.c
+++ b/tools/gst-inspect.c
@@ -1556,7 +1556,7 @@ main (int argc, char *argv[])
   g_option_context_add_main_entries (ctx, options, GETTEXT_PACKAGE);
   g_option_context_add_group (ctx, gst_init_get_option_group ());
   if (!g_option_context_parse (ctx, &argc, &argv, &err)) {
-    g_printerr ("Error initializing: %s\n", err->message);
+    g_printerr ("Error initializing: %s\n", err ? err->message : "(null)");
     return -1;
   }
   g_option_context_free (ctx);
-- 
1.8.2

