diff -Naur protobuf-c-0.15.old/src/Makefile.am protobuf-c-0.15/src/Makefile.am
--- protobuf-c-0.15.old/src/Makefile.am	2012-11-28 14:59:57.845251943 +0100
+++ protobuf-c-0.15/src/Makefile.am	2012-11-28 15:00:23.549252632 +0100
@@ -1,5 +1,5 @@
 if BUILD_PROTOC_C
-SUBDIRS = . test
+
 bin_PROGRAMS = protoc-c
 protoc_c_SOURCES = \
 google/protobuf/compiler/c/c_service.cc \
@@ -23,7 +23,7 @@
 lib_LTLIBRARIES = libprotobuf-c.la
 protobufcincludedir = $(includedir)/google/protobuf-c
 
-EXTRA_DIST = CMakeLists.txt test/CMakeLists.txt
+EXTRA_DIST = CMakeLists.txt 
 
 libprotobuf_c_la_SOURCES = \
 google/protobuf-c/protobuf-c-dispatch.c \
