Upstream-Status: Pending

Index: Pyrex-0.9.8.4/Pyrex/Distutils/extension.py
===================================================================
--- Pyrex-0.9.8.4.orig/Pyrex/Distutils/extension.py
+++ Pyrex-0.9.8.4/Pyrex/Distutils/extension.py
@@ -15,7 +15,7 @@ except ImportError:
     warnings = None
 
 class Extension(_Extension.Extension):
-    _Extension.Extension.__doc__ + \
+    _Extension.Extension.__doc__ or "" + \
     """pyrex_include_dirs : [string]
         list of directories to search for Pyrex header files (.pxd) (in
         Unix form for portability)
