From 85b882b4ceb57fe6538f47af58d0a970923fde0e Mon Sep 17 00:00:00 2001
From: Ricardo Ribalda Delgado <ricardo.ribalda@gmail.com>
Date: Thu, 31 Mar 2016 00:20:15 +0200
Subject: [PATCH] 3rdparty/ippicv: Use pre-downloaded ipp

Signed-off-by: Ricardo Ribalda Delgado <ricardo.ribalda@gmail.com>
Signed-off-by: Ismo Puustinen <ismo.puustinen@intel.com>

---
 3rdparty/ippicv/ippicv.cmake | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/3rdparty/ippicv/ippicv.cmake b/3rdparty/ippicv/ippicv.cmake
index ae8748c..305abdb 100644
--- a/3rdparty/ippicv/ippicv.cmake
+++ b/3rdparty/ippicv/ippicv.cmake
@@ -39,18 +39,5 @@ function(download_ippicv root_var)
   endif()
 
   set(THE_ROOT "${OpenCV_BINARY_DIR}/3rdparty/ippicv")
-  ocv_download(FILENAME ${OPENCV_ICV_NAME}
-               HASH ${OPENCV_ICV_HASH}
-               URL
-                 "${OPENCV_IPPICV_URL}"
-                 "$ENV{OPENCV_IPPICV_URL}"
-                 "https://raw.githubusercontent.com/opencv/opencv_3rdparty/${IPPICV_COMMIT}/ippicv/"
-               DESTINATION_DIR "${THE_ROOT}"
-               ID IPPICV
-               STATUS res
-               UNPACK RELATIVE_URL)
-
-  if(res)
-    set(${root_var} "${THE_ROOT}/${OPENCV_ICV_PACKAGE_SUBDIR}" PARENT_SCOPE)
-  endif()
+  set(${root_var} "${THE_ROOT}/${OPENCV_ICV_PACKAGE_SUBDIR}" PARENT_SCOPE)
 endfunction()
