Index: git/cmake/OpenCVGenPkgconfig.cmake
===================================================================
--- git.orig/cmake/OpenCVGenPkgconfig.cmake
+++ git/cmake/OpenCVGenPkgconfig.cmake
@@ -31,7 +31,7 @@ macro(fix_prefix lst isown)
       get_filename_component(libdir "${item}" PATH)
       get_filename_component(_libname "${item}" NAME)
       ocv_get_libname(libname "${_libname}")
-      list(APPEND _lst "-L${libdir}" "-l${libname}")
+      list(APPEND _lst "-l${libname}")
     else()
       list(APPEND _lst "-l${item}")
     endif()
@@ -124,11 +124,14 @@ ocv_list_unique(_extra)
 ocv_list_unique(_3rdparty)
 
 set(OPENCV_PC_LIBS
-  "-L\${exec_prefix}/${OPENCV_LIB_INSTALL_PATH}"
+  "-L\${exec_prefix}/${OPENCV_3P_LIB_INSTALL_PATH}"
   "${_modules}"
 )
 if(BUILD_SHARED_LIBS)
-  set(OPENCV_PC_LIBS_PRIVATE "${_extra}")
+  set(OPENCV_PC_LIBS_PRIVATE
+    "-L\${exec_prefix}/${OPENCV_LIB_INSTALL_PATH}"
+    "${_extra}"
+  )
 else()
   set(OPENCV_PC_LIBS_PRIVATE
     "-L\${exec_prefix}/${OPENCV_3P_LIB_INSTALL_PATH}"
