From e10b9b1f6704057ace39956ae1dc5c7caca07ff1 Mon Sep 17 00:00:00 2001
From: Andrei Dinu <andrei.adrianx.dinu@intel.com>
Date: Mon, 8 Jul 2013 11:53:54 +0300
Subject: [PATCH] Setting the location of nikto on the image

Upstream Status: Inapropriate

Signed-off-by: Andrei Dinu <andrei.adrianx.dinu@intel.com>
---
 nikto.conf |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/nikto.conf b/nikto.conf
index 25b784d..9577033 100644
--- a/nikto.conf
+++ b/nikto.conf
@@ -61,11 +61,11 @@ CIRT=174.142.17.165
 CHECKMETHODS=HEAD GET
 
 # If you want to specify the location of any of the files, specify them here
-# EXECDIR=/opt/nikto				# Location of Nikto
-# PLUGINDIR=/opt/nikto/plugins			# Location of plugin dir
-# DBDIR=/opt/nikto/databases			# Location of plugin dir
-# TEMPLATEDIR=/opt/nikto/templates		# Location of tempmlate dir
-# DOCDIR=/opt/nikto/docs			# Location of docs dir
+EXECDIR=/usr/bin/nikto				# Location of Nikto
+PLUGINDIR=/etc/nikto/plugins			# Location of plugin dir
+DBDIR=/etc/nikto/databases			# Location of plugin dir
+TEMPLATEDIR=/etc/nikto/templates		# Location of tempmlate dir
+DOCDIR=/usr/share/doc/nikto			# Location of docs dir
 
 # Default plugin macros
 @@MUTATE=dictionary;subdomain
-- 
1.7.9.5

