From ae26750c10c049356bb153e3d540573066a36c68 Mon Sep 17 00:00:00 2001
From: Ban Feng <kcfeng0@nuvoton.com>
Date: Tue, 14 Feb 2023 10:13:36 +0800
Subject: [PATCH] entitymanager: test

test

Signed-off-by: Ban Feng <kcfeng0@nuvoton.com>
---
 src/entity_manager.cpp | 1 +
 1 file changed, 1 insertion(+)

Index: git/src/entity_manager.cpp
===================================================================
--- git.orig/src/entity_manager.cpp
+++ git/src/entity_manager.cpp
@@ -622,9 +622,16 @@ void postToDbus(const nlohmann::json& ne
                     createInterface(objServer, boardName, propName,
                                     boardKeyOrig);
 
-                populateInterfaceFromJson(systemConfiguration,
+                if (boardName.substr(38, 11) == "powersupply") {
+                    populateInterfaceFromJson(systemConfiguration,
+                                          jsonPointerPath + propName, iface,
+                                          propValue, objServer,
+                                          sdbusplus::asio::PropertyPermission::readWrite);
+                } else {
+                    populateInterfaceFromJson(systemConfiguration,
                                           jsonPointerPath + propName, iface,
                                           propValue, objServer);
+                }
             }
         }
 
