Upstream-Status: Inappropriate [OE specific]
Signed-off-by: Armin Kuster <akuster808@gmail.com>

Index: git/Cargo.toml
===================================================================
--- git.orig/Cargo.toml
+++ git/Cargo.toml
@@ -91,7 +91,7 @@ hsm-tests-pkcs11 = [ "hsm" ]
 # Make sure that Krill crashes on panics, rather than losing threads and
 # limping on in a bad state.
 [profile.release]
-panic = "abort"
+#panic = "abort"
 
 [dev-dependencies]
 regex     = "1.5.5"
