From 3b02acbb3d5bc93422a6821ce47568633ef4ae5e Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 11 Aug 2018 15:09:24 -0700
Subject: [PATCH] include sys/sysmacros.h for major/minor definition

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 mkfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mkfs.c b/mkfs.c
index e612cbd..b6aa63d 100644
--- a/mkfs.c
+++ b/mkfs.c
@@ -21,6 +21,7 @@
 #include <sys/ioctl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #define __USE_UNIX98
 #include <unistd.h>
 #include <zlib.h>
