From 9be71700eb580c815688584a64621a38867c3fdd Mon Sep 17 00:00:00 2001
From: James Myatt <james.myatt@tessella.com>
Date: Thu, 5 Feb 2015 15:57:14 +0000
Subject: [PATCH] json_tokener requires INF and NAN

---
Upstream-Status: Backport

 json_tokener.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: json-c-0.12/json_tokener.c
===================================================================
--- json-c-0.12.orig/json_tokener.c
+++ json-c-0.12/json_tokener.c
@@ -16,6 +16,7 @@
 #include "config.h"
 
 #include <math.h>
+#include "math_compat.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <stddef.h>
@@ -352,12 +353,10 @@ struct json_object* json_tokener_parse_e
 
     case json_tokener_state_inf: /* aka starts with 'i' */
       {
-	int size;
-	int size_inf;
+	size_t size_inf;
 	int is_negative = 0;
 
 	printbuf_memappend_fast(tok->pb, &c, 1);
-	size = json_min(tok->st_pos+1, json_null_str_len);
 	size_inf = json_min(tok->st_pos+1, json_inf_str_len);
 	char *infbuf = tok->pb->buf;
 	if (*infbuf == '-')
