Upstream-Status: Backport [debian]

--- a/src/Makefile
+++ b/src/Makefile
@@ -37,16 +37,15 @@
 # Use the following only on GNU/Linux and only if you need ps listing like "smsd: MAINPROCESS" and "smsd: GSM1"
 # CFLAGS += -D USE_LINUX_PS_TRICK
 
-all: smsd 
-
-smsd: smsd.c extras.o locking.o cfgfile.o logging.o alarm.o smsd_cfg.o charset.o stats.o blacklist.o whitelist.o modeminit.o pdu.o
-
 ifneq (,$(findstring SOLARIS,$(CFLAGS)))
 ifeq (,$(findstring DISABLE_INET_SOCKET,$(CFLAGS)))
 override LFLAGS += -lsocket -lnsl
 endif
 endif
 
+all: smsd 
+
+smsd: smsd.c extras.o locking.o cfgfile.o logging.o alarm.o smsd_cfg.o charset.o stats.o blacklist.o whitelist.o modeminit.o pdu.o
 ifneq (,$(findstring NOSTATS,$(CFLAGS)))
 	$(CC) $(CFLAGS) -o $@ $^ $(LFLAGS)
 else 
