From a45e086661a3eed29193546c525998d7525ef6d9 Mon Sep 17 00:00:00 2001
From: Tudor Florea <tudor.florea@enea.com>
Date: Tue, 4 Aug 2015 02:46:39 +0200
Subject: [PATCH] Remove hardcoded /usr/local includes from configure.ac

Upstream-Status: Inappropriate [config]

Signed-off-by: Yauhen Kharuzhy <yauhen.kharuzhy@promwad.com>
Signed-off-by: Tudor Florea <tudor.florea@enea.com>

Update for 1.0.49.
Signed-off-by: Zheng Ruoqin <zhengrq.fnst@cn.fujitsu.com>

---
 configure.ac | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/configure.ac b/configure.ac
index 079e6f0..9a1ec06 100644
--- a/configure.ac
+++ b/configure.ac
@@ -96,21 +96,6 @@ AX_CHECK_LINK_FLAG([-Wl,-z,relro], [LDFLAGS="$LDFLAGS -Wl,-z,relro"])
 AX_CHECK_LINK_FLAG([-Wl,-z,now], [LDFLAGS="$LDFLAGS -Wl,-z,now"])
 AX_CHECK_LINK_FLAG([-Wl,-z,noexecstack], [LDFLAGS="$LDFLAGS -Wl,-z,noexecstack"])
 
-if test "x$cross_compiling" != "xyes"; then
-  for path in \
-    /usr/kerberos \
-    /usr/local /opt /usr/local/opt \
-    /usr/openssl@1.1 /opt/openssl@1.1 /usr/local/opt/openssl@1.1 \
-    /usr/openssl /opt/openssl /usr/local/opt/openssl; do
-    if test -d $path/include; then
-      CPPFLAGS="$CPPFLAGS -I${path}/include"
-    fi
-    if test -d $path/lib; then
-      LDFLAGS="$LDFLAGS -L${path}/lib"
-    fi
-  done
-fi
-
 CPPFLAGS="$CPPFLAGS -D_FORTIFY_SOURCE=2"
 
 dnl Checks for header files
-- 
2.7.4

