From 3f288fefbd683d687bde3fc63964da64bc3c8a40 Mon Sep 17 00:00:00 2001
From: Pascal Bach <pascal.bach@siemens.com>
Date: Wed, 11 Aug 2021 15:11:49 +0200
Subject: [PATCH] fix missing expat definition

Upstream-Status: Submitted [https://github.com/pocoproject/poco/pull/3414]
Signed-off-by: Pascal Bach <pascal.bach@siemens.com>
---
 XML/src/ParserEngine.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/XML/src/ParserEngine.cpp b/XML/src/ParserEngine.cpp
index 82d477478..6213e555b 100644
--- a/XML/src/ParserEngine.cpp
+++ b/XML/src/ParserEngine.cpp
@@ -504,7 +504,7 @@ void ParserEngine::init()
 	XML_SetParamEntityParsing(_parser, _externalParameterEntities ? XML_PARAM_ENTITY_PARSING_ALWAYS : XML_PARAM_ENTITY_PARSING_NEVER);
 	XML_SetUnknownEncodingHandler(_parser, handleUnknownEncoding, this);

-#if XML_MAJOR_VERSION > 2 || (XML_MAJOR_VERSION == 2 && XML_MINOR_VERSION >= 4)
+#if XML_DTD && (XML_MAJOR_VERSION > 2 || (XML_MAJOR_VERSION == 2 && XML_MINOR_VERSION >= 4))
 	if (_maximumAmplificationFactor > 1.0)
 	{
 		XML_SetBillionLaughsAttackProtectionMaximumAmplification(_parser, _maximumAmplificationFactor);
--
2.32.0

