From 335b25febd4c864ad0ac08479f5cd43fc21b7d73 Mon Sep 17 00:00:00 2001
From: Nikias Bassen
Date: Mon, 09 Jul 2012 21:17:15 +0000
Subject: Fix building with parallel build makeopts

---
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0bad840..155994e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -57,6 +57,8 @@ ADD_SUBDIRECTORY( plutil )
 ADD_SUBDIRECTORY( include )
 ADD_SUBDIRECTORY( test )
 
+ADD_DEPENDENCIES( plist libcnary )
+
 IF ( SWIG_FOUND AND PYTHONLIBS_FOUND AND PYTHONINTERP_FOUND )
 	ADD_SUBDIRECTORY( swig )
 ENDIF ( SWIG_FOUND AND PYTHONLIBS_FOUND AND PYTHONINTERP_FOUND )
--
cgit v0.9.0.2
