Upstream-Status: Inappropriate [configuration]

Index: gdk-pixbuf-2.22.1/configure.ac
===================================================================
--- gdk-pixbuf-2.22.1.orig/configure.ac	2010-11-26 09:06:34.000000000 +0800
+++ gdk-pixbuf-2.22.1/configure.ac	2010-11-26 09:07:33.000000000 +0800
@@ -287,7 +287,7 @@
 case $enable_explicit_deps in
   auto)
     export SED
-    deplibs_check_method=`(./libtool --config; echo 'eval echo \"$deplibs_check_method\"') | sh`
+    deplibs_check_method=`(./$host_alias-libtool --config; echo 'eval echo \"$deplibs_check_method\"') | sh`
     if test "x$deplibs_check_method" '!=' xpass_all || test "x$enable_static" = xyes ; then
       enable_explicit_deps=yes
     else
@@ -484,7 +484,7 @@
     dnl Now we check to see if our libtool supports shared lib deps
     dnl (in a rather ugly way even)
     if $dynworks; then
-        pixbuf_libtool_config="${CONFIG_SHELL-/bin/sh} ./libtool --config"
+        pixbuf_libtool_config="${CONFIG_SHELL-/bin/sh} ./$host_alias-libtool --config"
         pixbuf_deplibs_check=`$pixbuf_libtool_config | \
             grep '^[[a-z_]]*check[[a-z_]]*_method=[['\''"]]' | \
             sed 's/.*[['\''"]]\(.*\)[['\''"]]$/\1/'`
@@ -957,7 +957,7 @@
 # We are using gmodule-no-export now, but I'm leaving the stripping
 # code in place for now, since pango and atk still require gmodule.
 export SED
-export_dynamic=`(./libtool --config; echo eval echo \\$export_dynamic_flag_spec) | sh`
+export_dynamic=`(./$host_alias-libtool --config; echo eval echo \\$export_dynamic_flag_spec) | sh`
 if test -n "$export_dynamic"; then
   GDK_PIXBUF_DEP_LIBS=`echo $GDK_PIXBUF_DEP_LIBS | sed -e "s/$export_dynamic//"`
 fi
