From e81633a276dd6a9f919e5e5c15481ac50a8e485d Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Fri, 30 Aug 2019 14:59:06 +0800
Subject: [PATCH] support usrmerge

Upstream-Status: Inappropriate [oe-specific]

Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 configure.ac | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4b7e4c8..f1afbd6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -110,8 +110,7 @@ AC_ARG_ENABLE(libicu,
 # If the user specified a libdir ending in lib64 do not append another
 # 64 to the library names.
 #
-base_libdir=`basename "$libdir"`
-case $base_libdir in
+case `basename "$libdir"` in
 lib64)
   enable_lib64=no
 esac
@@ -125,8 +124,8 @@ esac
 #
 case $exec_prefix:$prefix in
 NONE:NONE | NONE:/usr | /usr:*)
-  root_sbindir='/sbin'
-  root_libdir="/${base_libdir}"
+  root_sbindir="${base_sbindir}"
+  root_libdir="${base_libdir}"
   ;;
 *)
   root_sbindir="${sbindir}"
-- 
2.7.4

