From 54bba228ea52d01fd84941d97be23c03f9862b64 Mon Sep 17 00:00:00 2001
From: Carlos Rafael Giani <dv@pseudoterminal.org>
Date: Sat, 6 Apr 2013 01:22:22 +0200
Subject: [PATCH] Disable yasm for libav when --disable-yasm

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Shane Wang <shane.wang@intel.com>
Signed-off-by: Carlos Rafael Giani <dv@pseudoterminal.org>
---
 configure.ac | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure.ac b/configure.ac
index 22ede88..ef3c050 100644
--- a/configure.ac
+++ b/configure.ac
@@ -305,6 +305,12 @@ else
     emblibav_configure_args="$emblibav_configure_args --enable-gpl"
   fi

+  AC_ARG_ENABLE(yasm,
+              [AC_HELP_STRING([--disable-yasm], [disable use of yasm assembler])])
+  if test "x$enable_yasm" = "xno"; then
+    emblibav_configure_args="$emblibav_configure_args --disable-yasm"
+  fi
+
   # if we are cross-compiling, tell libav so
   case $host in
       *android*)
-- 
1.8.2

