From 902b251fd8d6408ee6623e7a92692948cf90187d Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 12 Feb 2019 13:22:21 -0800
Subject: [PATCH] Add CLANG_TIDY check

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7073a0b..2dd2c02 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -60,7 +60,7 @@ if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
         endif()
     endif()
 
-    if(CMAKE_VERSION VERSION_GREATER 3.6)
+    if(CMAKE_VERSION VERSION_GREATER 3.6 AND CLANG_TIDY)
         # Add clang-tidy if available
         option(CLANG_TIDY_FIX "Perform fixes for Clang-Tidy" OFF)
         find_program(
