Upstream-Status: Pending

--- a/external/glslang/src/glslang/Include/PoolAlloc.h
+++ b/external/glslang/src/glslang/Include/PoolAlloc.h
@@ -245,8 +245,9 @@ protected:
 
     int numCalls;           // just an interesting statistic
     size_t totalBytes;      // just an interesting statistic
-private:
+public:
     TPoolAllocator& operator=(const TPoolAllocator&);  // don't allow assignment operator
+private:
     TPoolAllocator(const TPoolAllocator&);  // don't allow default copy constructor
 };
 
