From 9bf1693092385eba9841614613313010221ca01f Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Tue, 29 Jun 2021 20:11:26 +0200
Subject: [PATCH] docs: do not build manpages (requires pandoc)

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 docs/Makefile.am | 2 --
 1 file changed, 2 deletions(-)

diff --git a/docs/Makefile.am b/docs/Makefile.am
index 5a6bd203a..6257767fd 100644
--- a/docs/Makefile.am
+++ b/docs/Makefile.am
@@ -1,7 +1,5 @@
 ## Process this file with automake to produce Makefile.in
 
-SUBDIRS = man
-
 EXTRA_DIST =
 
 EXTRA_DIST += \
-- 
2.32.0

