From e210458d125793915abce30420d866a30305c37a Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Tue, 10 Jan 2017 14:11:30 +0200
Subject: [PATCH] Do not read config files from $HOME

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 lib/rpmrc.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/lib/rpmrc.c b/lib/rpmrc.c
index 9437a0ff1..483585ae4 100644
--- a/lib/rpmrc.c
+++ b/lib/rpmrc.c
@@ -459,8 +459,7 @@ static void setDefaults(void)
     if (!defrcfiles) {
 	defrcfiles = rstrscat(NULL, confdir, "/rpmrc", ":",
 				confdir, "/" RPM_VENDOR "/rpmrc", ":",
-				SYSCONFDIR "/rpmrc", ":",
-			  	"~/.rpmrc", NULL);
+				SYSCONFDIR "/rpmrc", NULL);
     }
 
 #ifndef MACROFILES
@@ -472,8 +471,7 @@ static void setDefaults(void)
   				confdir, "/" RPM_VENDOR "/macros", ":",
 				SYSCONFDIR "/rpm/macros.*", ":",
 				SYSCONFDIR "/rpm/macros", ":",
-				SYSCONFDIR "/rpm/%{_target}/macros", ":",
-				"~/.rpmmacros", NULL);
+				SYSCONFDIR "/rpm/%{_target}/macros", NULL);
     }
 #else
     macrofiles = MACROFILES;
