Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2014-07-17 21:48:51.284570474 +0000
+++ git/configure.ac	2014-07-17 21:53:39.104578318 +0000
@@ -26,7 +26,7 @@
 AC_CANONICAL_HOST
 AC_ISC_POSIX
 
-AM_INIT_AUTOMAKE(1.6 dist-bzip2)
+AM_INIT_AUTOMAKE([dist-bzip2 foreign])
 AC_CONFIG_HEADERS(config.h)
 _XTERM_COLORS
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
