From 5846ba594bedd66c8fd216cad6518f7be11b7a92 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Fri, 20 Dec 2019 16:54:05 +0100
Subject: [PATCH] tests: do not run target executables

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 tests/Makefile | 2 --
 1 file changed, 2 deletions(-)

diff --git a/tests/Makefile b/tests/Makefile
index 3d03721..fbb1d78 100644
--- a/tests/Makefile
+++ b/tests/Makefile
@@ -69,13 +69,11 @@ endif
 
 # unprivileged
 run_psx_test: psx_test
-	./psx_test
 
 psx_test: psx_test.c $(DEPS)
 	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $< -o $@ $(LINKEXTRA) $(LIBPSXLIB)
 
 run_libcap_psx_test: libcap_psx_test
-	./libcap_psx_test
 
 libcap_psx_test: libcap_psx_test.c $(DEPS)
 	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) $< -o $@ $(LINKEXTRA) $(LIBCAPLIB) $(LIBPSXLIB)
