From 5583b48c04d5bf50b56473d88d990c3f0bc45c14 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Fri, 20 Dec 2019 16:54:05 +0100
Subject: [PATCH] tests: do not run target executables

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

---
 tests/Makefile | 2 --
 1 file changed, 2 deletions(-)

diff --git a/tests/Makefile b/tests/Makefile
index bfedbc2..876a8b9 100644
--- a/tests/Makefile
+++ b/tests/Makefile
@@ -19,13 +19,11 @@ sudotest: test run_libcap_launch_test run_libcap_launch_test
 install: all
 
 run_psx_test: psx_test
-	./psx_test
 
 psx_test: psx_test.c $(DEPS)
 	$(CC) $(CFLAGS) $(IPATH) $< -o $@ $(LIBPSXLIB) -Wl,-wrap,pthread_create
 
 run_libcap_psx_test: libcap_psx_test
-	./libcap_psx_test
 
 libcap_psx_test: libcap_psx_test.c $(DEPS)
 	$(CC) $(CFLAGS) $(IPATH) $< -o $@ $(LIBCAPLIB) $(LIBPSXLIB) -Wl,-wrap,pthread_create --static
