From ee47e205255928679701a5b769bd7ae0f815119d Mon Sep 17 00:00:00 2001
From: Gabriel Barbu <gabriel.barbu@enea.com>
Date: Thu, 25 Jul 2013 15:28:33 +0200
Subject: [PATCH] strace: Add ptest

Upstream-Status: Inappropriate

Signed-off-by: Gabriel Barbu <gabriel.barbu@enea.com>
Signed-off-by: Chong Lu <Chong.Lu@windriver.com>
Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>

---
 tests/Makefile.am | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index 8cf4c36..66162eb 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -15,6 +15,7 @@ SIZEOF_LONG = @SIZEOF_LONG@
 MPERS_NAME =
 MPERS_CC_FLAGS =
 ARCH_MFLAGS =
+TEST_SUITE_LOG = test-suite.log
 AM_CFLAGS = $(WARN_CFLAGS)
 AM_CPPFLAGS = $(ARCH_MFLAGS) \
 	      -I$(builddir) \
@@ -538,3 +539,21 @@ BUILT_SOURCES = ksysent.h
 CLEANFILES = ksysent.h
 
 include ../scno.am
+
+buildtest-TESTS: $(check_PROGRAMS) $(TESTS)
+
+install-ptest:
+	install $(BUILDDIR)/strace $(DESTDIR)
+	install $(srcdir)/../strace-log-merge $(DESTDIR)
+	install -d $(DESTDIR)/$(TESTDIR)
+	cp $(BUILDDIR)/$(TESTDIR)/Makefile $(DESTDIR)/$(TESTDIR)
+	sed -i -e 's/^Makefile:/_Makefile:/' $(DESTDIR)/$(TESTDIR)/Makefile
+	sed -i -e 's/bash/sh/' $(DESTDIR)/$(TESTDIR)/Makefile
+	for file in $(check_PROGRAMS); do \
+		install $(BUILDDIR)/$(TESTDIR)/$$file $(DESTDIR)/$(TESTDIR); \
+	done
+	for file in $(EXTRA_DIST); do \
+		install $(srcdir)/$$file $(DESTDIR)/$(TESTDIR); \
+		#sed -i -e 's/$${srcdir=.}/./g' $(DESTDIR)/$(TESTDIR)/$$file; \
+	done
+	for i in net scm_rights-fd rt_sigaction; do sed -i -e 's/$$srcdir/./g' $(DESTDIR)/$(TESTDIR)/$$i.test; done
