Fix error pointed out by gcc10

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
--- a/sigscheme/libgcroots/include/private/gc_priv.h
+++ b/sigscheme/libgcroots/include/private/gc_priv.h
@@ -2026,7 +2026,7 @@ void GC_err_puts(const char *s);
 
 # if defined(NEED_FIND_LIMIT) || \
      defined(USE_PROC_FOR_LIBRARIES) && defined(THREADS)
-JMP_BUF GC_jmp_buf;
+extern JMP_BUF GC_jmp_buf;
 
 /* Set up a handler for address faults which will longjmp to	*/
 /* GC_jmp_buf;							*/
