Disable compiling examples

Upstream-Status: Inappropriate [disable feature]

Signed-off-by: Kevin Tian <kevin.tian@intel.com>
Signed-off-by: Scott Garman <scott.a.garman@intel.com>

Index: dbus-glib-0.108/dbus/Makefile.am
===================================================================
--- dbus-glib-0.108.orig/dbus/Makefile.am
+++ dbus-glib-0.108/dbus/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = . examples
+SUBDIRS = .
 
 AM_CPPFLAGS = \
 	-I$(top_srcdir)						\
