From c96dda04ec024bdef2a15d374e5f8242e041cad4 Mon Sep 17 00:00:00 2001
From: Richard Purdie <richard.purdie@linuxfoundation.org>
Date: Fri, 18 Jul 2014 08:25:16 +0000

---
Upstream-Status: Pending

 configure.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.in b/configure.in
index ac1a7e4..11b9f31 100644
--- a/configure.in
+++ b/configure.in
@@ -1,4 +1,4 @@
-AC_INIT(src/pppoe.c)
-AM_INIT_AUTOMAKE([rp-pppoe], [3.8])
+AC_INIT([rp-pppoe], [3.8])
+AM_INIT_AUTOMAKE([foreign])
 AC_CONFIG_SUBDIRS(src)
 AC_OUTPUT(Makefile)
