diff --git a/glib/Makefile.am b/glib/Makefile.am
index 671cae2..eff55f3 100644
--- a/glib/Makefile.am
+++ b/glib/Makefile.am
@@ -11,14 +11,14 @@ libOpenIPMIglib_la_SOURCES = glib_os_hnd.c
 libOpenIPMIglib_la_CFLAGS = $(GLIB_CFLAGS) $(AM_CFLAGS)
 libOpenIPMIglib_la_LIBADD = $(GDBM_LIB)
 libOpenIPMIglib_la_LDFLAGS = -rdynamic -version-info $(LD_VERSION) \
-	-Wl,-Map -Wl,libOpenIPMIglib.map $(GLIB_LIBS) -L$(libdir) \
+	-Wl,-Map -Wl,libOpenIPMIglib.map $(GLIB_LIBS) \
 	-rpath $(libdir)
 
 libOpenIPMIglib12_la_SOURCES = glib_os_hnd.c
 libOpenIPMIglib12_la_CFLAGS = $(GLIB12_CFLAGS) $(AM_CFLAGS)
 libOpenIPMIglib12_la_LIBADD = $(GDBM_LIB)
 libOpenIPMIglib12_la_LDFLAGS = -rdynamic -version-info $(LD_VERSION) \
-	-Wl,-Map -Wl,libOpenIPMIglib12.map $(GLIB12_LIBS) -L$(libdir) \
+	-Wl,-Map -Wl,libOpenIPMIglib12.map $(GLIB12_LIBS) \
 	-rpath $(libdir)
 
 CLEANFILES = libOpenIPMIglib.map libOpenIPMIglib12.map
