From fc5de1de898fd1a372a2fd2fa493dc57323a029d Mon Sep 17 00:00:00 2001
From: Tudor Florea <tudor.florea@enea.com>
Date: Thu, 3 Dec 2015 04:08:00 +0100
Subject: [PATCH] Display testname for subtest

Signed-off-by: Tudor Florea <tudor.florea@enea.com>
Upstream-Status: Pending

---
 tests/functions.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/functions.sh b/tests/functions.sh
index 5a562a3..098145e 100644
--- a/tests/functions.sh
+++ b/tests/functions.sh
@@ -437,7 +437,7 @@ function ts_init_subtest {
 
 	if [ "$TS_PARSABLE" != "yes" ]; then
 		[ $TS_NSUBTESTS -eq 1 ] && echo
-		printf "%16s: %-27s ..." "" "$TS_SUBNAME"
+		printf "%13s: %-30s ..." "$TS_COMPONENT" "$TS_SUBNAME"
 	fi
 }
 
