From af073c13ef184ca75811df688e0a0a25827b36c3 Mon Sep 17 00:00:00 2001
From: Tudor Florea <tudor.florea@enea.com>
Date: Thu, 3 Dec 2015 04:08:00 +0100
Subject: [PATCH] Define TESTS variable

Signed-off-by: Tudor Florea <tudor.florea@enea.com>
Upstream-Status: Pending

---
 Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile.am b/Makefile.am
index 886598d..1cf4346 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -57,6 +57,7 @@ systemdsystemunit_DATA =
 dist_bashcompletion_DATA =
 check_PROGRAMS =
 dist_check_SCRIPTS =
+TESTS = $(check_PROGRAMS)
 
 PATHFILES =
 ADOCFILES_COMMON =
