From 4b73298b214a5b69ea6edf3c2e21dd82b2b29708 Mon Sep 17 00:00:00 2001
From: Tim Orling <tim.orling@konsulko.com>
Date: Fri, 14 Jan 2022 22:34:59 -0800
Subject: [PATCH 2/2] Cargo.toml: edition 2018 -> 2021

Upstream-Status: Pending

Signed-off-by: Tim Orling <tim.orling@konsulko.com>
---
 src/rust/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rust/Cargo.toml b/src/rust/Cargo.toml
index 174eaa80..7ad053d9 100644
--- a/src/rust/Cargo.toml
+++ b/src/rust/Cargo.toml
@@ -2,7 +2,7 @@
 name = "cryptography-rust"
 version = "0.1.0"
 authors = ["The cryptography developers <cryptography-dev@python.org>"]
-edition = "2018"
+edition = "2021"
 publish = false
 
 [dependencies]
-- 
2.30.2

