From 81ea820b155e887b13ea5986c3407cf93b2737f6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Wed, 2 Jan 2019 18:42:46 +0100
Subject: [PATCH] Do not build gentables helper - we have to use native variant
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>
---
 src/CMakeLists.txt | 23 +----------------------
 1 file changed, 1 insertion(+), 22 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 356bb734..58ff7635 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -399,25 +399,4 @@ else ( MACOSX_FRAMEWORK )
    install ( FILES ${public_main_HEADER} DESTINATION ${INCLUDE_INSTALL_DIR} )
 endif ( MACOSX_FRAMEWORK )
 
-# ******* Auto Generated Lookup Tables ******
-
-include(ExternalProject)
-
-set (GENTAB_SDIR ${CMAKE_CURRENT_SOURCE_DIR}/gentables)
-set (GENTAB_BDIR ${CMAKE_CURRENT_BINARY_DIR}/gentables)
-
-# Use external project to ensure that cmake uses the host compiler when building make_tables.exe
-# To fix cross-compiling fluidsynth from Win32 to ARM (using vcpkg), we need to pass the current generator
-# on to the external project, otherwise (for some unknown reason) the target compiler will be used rather
-# than the host compiler.
-ExternalProject_Add(gentables
-    DOWNLOAD_COMMAND ""
-    SOURCE_DIR ${GENTAB_SDIR}
-    BINARY_DIR ${GENTAB_BDIR}
-    CONFIGURE_COMMAND
-        "${CMAKE_COMMAND}" -DCMAKE_VERBOSE_MAKEFILE=${CMAKE_VERBOSE_MAKEFILE} -G "${CMAKE_GENERATOR}" -B "${GENTAB_BDIR}" "${GENTAB_SDIR}"
-    BUILD_COMMAND
-        "${CMAKE_COMMAND}" --build "${GENTAB_BDIR}"
-    INSTALL_COMMAND ${GENTAB_BDIR}/make_tables.exe "${CMAKE_BINARY_DIR}/"
-)
-add_dependencies(libfluidsynth-OBJ gentables)
+
-- 
2.21.1

