From bc89a1d5329cf8f8a67a5e8c2b7ed53aa6611183 Mon Sep 17 00:00:00 2001
From: Chris Austen <austenc@us.ibm.com>
Date: Thu, 24 Sep 2015 14:49:24 -0500
Subject: [PATCH 1/1] add support for Power8 LE mode

---
 config.guess | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/localedef/config.guess b/localedef/config.guess
index 61f2e4c..c77ec22 100644
--- a/localedef/config.guess
+++ b/localedef/config.guess
@@ -876,6 +876,9 @@ EOF
     ppc64:Linux:*:*)
 	echo powerpc64-unknown-linux-gnu
 	exit 0 ;;
+    ppc64le:Linux:*:*)
+       echo powerpc64le-unknown-linux-gnu
+       exit 0 ;;
     alpha:Linux:*:*)
 	case `sed -n '/^cpu model/s/^.*: \(.*\)/\1/p' < /proc/cpuinfo` in
 	  EV5)   UNAME_MACHINE=alphaev5 ;;
-- 
1.9.1

