Upstream-Status: Pending

--- a/spatialite.pc.in
+++ b/spatialite.pc.in
@@ -8,5 +8,7 @@ includedir=@includedir@
 Name: spatialite
 Description: Spatial SQL database engine based on SQLite
 Version: @VERSION@
-Libs: -L${libdir} -lspatialite @LIBS@ @LIBXML2_LIBS@ @SQLITE3_LIBS@ -lm
+Libs.private: -L${libdir} @LIBS@ @LIBXML2_LIBS@ @SQLITE3_LIBS@ -lm
+Libs: -L${libdir} -lspatialite
+
 Cflags: -I${includedir}
