From 601ef636fc4dfb2af3e7fda88d8ea1c1d92affe4 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Wed, 2 Oct 2019 17:54:15 +0200
Subject: [PATCH] Allow building lua without readline on Linux

Upstream-Status: Pending
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 Makefile      | 2 +-
 src/Makefile  | 3 +++
 src/luaconf.h | 5 +++++
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 119110d..9f6df45 100644
--- a/Makefile
+++ b/Makefile
@@ -36,7 +36,7 @@ RM= rm -f
 # == END OF USER SETTINGS -- NO NEED TO CHANGE ANYTHING BELOW THIS LINE =======
 
 # Convenience platforms targets.
-PLATS= aix bsd c89 freebsd generic linux macosx mingw posix solaris
+PLATS= aix bsd c89 freebsd generic linux linux-no-readline macosx mingw posix solaris
 
 # What to install.
 TO_BIN= lua luac
diff --git a/src/Makefile b/src/Makefile
index 64c78f7..5c0428a 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -109,6 +109,9 @@ generic: $(ALL)
 linux:
 	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline"
 
+linux-no-readline:
+	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX_NO_READLINE" SYSLIBS="-Wl,-E -ldl"
+
 macosx:
 	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS="-lreadline"
 
diff --git a/src/luaconf.h b/src/luaconf.h
index 9eeeea6..d71ca25 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -64,6 +64,11 @@
 #define LUA_USE_READLINE	/* needs some extra libraries */
 #endif
 
+#if defined(LUA_USE_LINUX_NO_READLINE)
+#define LUA_USE_POSIX
+#define LUA_USE_DLOPEN		/* needs an extra library: -ldl */
+#endif
+
 
 #if defined(LUA_USE_MACOSX)
 #define LUA_USE_POSIX
-- 
2.17.1

