Upstream-Status: Pending
diff -ru ruby-1.8.7-p248.orig/ext/extmk.rb ruby-1.8.7-p248/ext/extmk.rb
--- ruby-1.8.7-p248.orig/ext/extmk.rb	2009-12-24 03:01:58.000000000 -0600
+++ ruby-1.8.7-p248/ext/extmk.rb	2010-02-12 15:55:27.370061558 -0600
@@ -354,8 +354,8 @@
   $ruby = '$(topdir)/miniruby' + EXEEXT
 end
 $ruby << " -I'$(topdir)'"
+$ruby << " -I'$(top_srcdir)/lib'"
 unless CROSS_COMPILING
-  $ruby << " -I'$(top_srcdir)/lib'"
   $ruby << " -I'$(extout)/$(arch)' -I'$(extout)/common'" if $extout
   $ruby << " -I./- -I'$(top_srcdir)/ext' -rpurelib.rb"
   ENV["RUBYLIB"] = "-"
