From a8ef81c115a45f05dad145c98e10f3c4940e4e29 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Thu, 26 Jan 2017 16:25:47 +0200
Subject: [PATCH 3/5] Do not hardcode /etc and systemd unit directories

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 650b624a..10199618 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,8 +7,8 @@ if (NOT PYTHON_DESIRED)
     set (PYTHON_DESIRED "2")
 endif()
 
-SET( SYSCONFDIR /etc)
-SET( SYSTEMD_DIR /usr/lib/systemd/system)
+SET( SYSCONFDIR ${CMAKE_INSTALL_SYSCONFDIR})
+SET( SYSTEMD_DIR $ENV{systemd_system_unitdir})
 
 if (${PYTHON_DESIRED} STREQUAL "2")
 	FIND_PACKAGE (PythonInterp REQUIRED)
-- 
2.14.2

