From 887a403e38f590ebf1d04ce600d94cb7a351744c Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Mon, 1 Dec 2014 00:57:42 -0500
Subject: [PATCH] ltrace: allow to enable/disable selinux support

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 configure.ac |   10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6fe5e3b..189885a 100644
--- a/configure.ac
+++ b/configure.ac
@@ -125,8 +125,14 @@ AC_SUBST(libstdcxx_LIBS)
 
 
 dnl Check security_get_boolean_active availability.
-AC_CHECK_HEADERS(selinux/selinux.h)
-AC_CHECK_LIB(selinux, security_get_boolean_active)
+AC_ARG_ENABLE([selinux],
+    AS_HELP_STRING([--enable-selinux],[enable SELinux support [default=auto]]),
+    [], [enable_selinux=auto])
+
+if test x$enable_selinux != xno; then
+    AC_CHECK_HEADERS(selinux/selinux.h)
+    AC_CHECK_LIB(selinux, security_get_boolean_active)
+fi
 
 dnl Whether (and which) elfutils libdw.so to use for unwinding.
 AC_ARG_WITH(elfutils,
-- 
1.7.9.5

