From 02f1f63d3ff58f3473f1689a0322a234ce10b659 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 13 Jan 2023 22:13:45 -0800
Subject: [PATCH] obstack: Fix a clang warning

Upstream-Status: Backport [https://git.savannah.gnu.org/cgit/gnulib.git/commit/?id=0cc39712803ade7b2d4b89c36b143dad72404063]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 gnu/obstack.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnu/obstack.c b/gnu/obstack.c
index 54b675d..4d6a1d5 100644
--- a/gnu/obstack.c
+++ b/gnu/obstack.c
@@ -326,7 +326,7 @@ int obstack_exit_failure = EXIT_FAILURE;
 #   include <libio/iolibio.h>
 #  endif
 
-static _Noreturn void
+static __attribute_noreturn__ void
 print_and_abort (void)
 {
   /* Don't change any of these strings.  Yes, it would be possible to add
-- 
2.39.0

