From 88696037af52d6a7f57bc0fb82598de1a25365cc Mon Sep 17 00:00:00 2001
From: Alistair Francis <alistair.francis@wdc.com>
Date: Fri, 29 Mar 2019 16:09:57 -0700
Subject: [PATCH 2/7] dev_setup.sh: Remove the git dependency

Signed-off-by: Alistair Francis <alistair.francis@wdc.com>
Upstream-Status: Inappropriate [embedded specific]
---
 dev_setup.sh | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/dev_setup.sh b/dev_setup.sh
index 99a44bfd8a7..672b27080c2 100755
--- a/dev_setup.sh
+++ b/dev_setup.sh
@@ -340,10 +340,6 @@ ${YELLOW}Make sure to manually install:$BLUE git python3 python-setuptools pytho
 
 install_deps
 
-# Configure to use the standard commit template for
-# this repo only.
-git config commit.template .gitmessage
-
 # Check whether to build mimic (it takes a really long time!)
 build_mimic='n'
 if [[ $opt_forcemimicbuild == true ]] ; then
@@ -371,18 +367,6 @@ else
     fi
 fi
 
-# Install pep8 pre-commit hook
-HOOK_FILE='./.git/hooks/pre-commit'
-if [[ -n $INSTALL_PRECOMMIT_HOOK ]] || grep -q 'MYCROFT DEV SETUP' $HOOK_FILE; then
-    if [[ ! -f $HOOK_FILE ]] || grep -q 'MYCROFT DEV SETUP' $HOOK_FILE; then
-        echo 'Installing PEP8 check as precommit-hook'
-        echo "#! $(which python)" > $HOOK_FILE
-        echo '# MYCROFT DEV SETUP' >> $HOOK_FILE
-        cat ./scripts/pre-commit >> $HOOK_FILE
-        chmod +x $HOOK_FILE
-    fi
-fi
-
 PYTHON=$(python -c "import sys;print('python{}.{}'.format(sys.version_info[0], sys.version_info[1]))")
 
 # install required python modules
-- 
2.23.0

