From ce24a29cabb8f1774d262e120e5764a2afeb3825 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 7 Apr 2017 16:53:49 -0700
Subject: [PATCH 2/2] fileio: Define DEFFILEMODE if platform is missing

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 fileio.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/fileio.c
+++ b/fileio.c
@@ -36,6 +36,10 @@
 #define DEFFILEMODE 0666
 #endif
 
+#if !defined(DEFFILEMODE)
+# define DEFFILEMODE (S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH)/* 0666*/
+#endif
+
 static char *bkuplocation(const char *);
 static int   bkupleavetmp(const char *);
 
