From c1c980a95d85bcaf8802524d6148783522b300d7 Mon Sep 17 00:00:00 2001
From: Yulong Pei <Yulong.pei@windriver.com>
Date: Wed, 21 Jul 2010 22:33:43 +0800
Subject: [PATCH] change finding path of nss and nspr

Upstream-Status: Pending

Signed-off-by: Yulong Pei <Yulong.pei@windriver.com>
Signed-off-by: Mingli Yu <Mingli.Yu@windriver.com>
Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 configure.ac | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/configure.ac b/configure.ac
index 951b3eb..1fdeb0f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -866,7 +866,7 @@ MOZILLA_MIN_VERSION="1.4"
 NSPR_INCLUDE_MARKER="nspr/nspr.h"
 NSPR_LIB_MARKER="libnspr4$shrext"
 NSPR_LIBS_LIST="-lnspr4 -lplds4 -lplc4"
-NSS_INCLUDE_MARKER="nss/nss.h"
+NSS_INCLUDE_MARKER="nss3/nss.h"
 NSS_LIB_MARKER="libnss3$shrext"
 NSS_LIBS_LIST="-lnss3 -lsmime3"
 
