--- enca-1.9/configure.ac.orig	2006-02-15 00:57:39.000000000 -0800
+++ enca-1.9/configure.ac	2006-02-15 00:57:46.000000000 -0800
@@ -92,9 +92,6 @@
 dnl And for gtk-doc
 gtk_CHECK_GTK_DOC
 
-dnl Check for good random number sources
-AC_CHECK_FILES(/dev/random /dev/urandom /dev/srandom /dev/arandom)
-
 dnl Dirty path hack.  Helps some people with badly set up search paths.
 if test "$prefix" = "NONE"; then
   LDFLAGS="$LDFLAGS -L$ac_default_prefix/lib"
