From 4fd08c0446ca02917014b63f9080c4205958a130 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 20 Mar 2022 01:15:32 -0700
Subject: [PATCH] makefile: Add LIBDIR

Avoid hardcoding /lib

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 1c01f89..2b1df5b 100644
--- a/Makefile
+++ b/Makefile
@@ -11,12 +11,12 @@ all : $(LIB) squish.pc
 
 install : $(LIB) squish.pc
 	install squish.h $(INSTALL_DIR)/include 
-	install libsquish.a $(INSTALL_DIR)/lib
-	install squish.pc $(INSTALL_DIR)/lib/pkgconfig
+	install libsquish.a $(INSTALL_DIR)/$(LIBDIR)
+	install squish.pc $(INSTALL_DIR)/$(LIBDIR)/pkgconfig
 
 uninstall:
 	$(RM) $(INSTALL_DIR)/include/squish.h
-	$(RM) $(INSTALL_DIR)/lib/libsquish.a
+	$(RM) $(INSTALL_DIR)/$(LIBDIR)/libsquish.a
 
 $(LIB) : $(OBJ)
 	$(AR) cr $@ $?
-- 
2.35.1

