Upstream-Status: Pending

from openembedded, added by Qing He <qing.he@intel.com>

Index: libaio-0.3.110/Makefile
===================================================================
--- libaio-0.3.110.orig/Makefile
+++ libaio-0.3.110/Makefile
@@ -15,7 +15,7 @@ all:
 	@$(MAKE) -C src
 
 install:
-	@$(MAKE) -C src install prefix=$(prefix) includedir=$(includedir) libdir=$(libdir)
+	@$(MAKE) -C src install prefix=$(DESTDIR)$(prefix) includedir=$(DESTDIR)$(includedir) libdir=$(DESTDIR)$(libdir)
 
 check:
 	@$(MAKE) -C harness check
