From 3d3ec391038c6ee73897a7fab753b11b589de1ed Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 2 Nov 2016 12:39:20 -0700
Subject: [PATCH 06/10] Disable DVD support

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Makefile.in | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 0d70e9b..050c7ca 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -10,8 +10,7 @@ EC_DIRS= \
 	tools/EventClients
 
 DVDPCODECS_DIRS= \
-	lib \
-	lib/libdvd
+	lib
 
 VideoPlayer_ARCHIVES=xbmc/cores/VideoPlayer/VideoPlayer.a \
                    xbmc/cores/VideoPlayer/DVDCodecs/DVDCodecs.a \
@@ -347,7 +346,6 @@ libaddon: exports
 	$(MAKE) -C lib/addons/library.xbmc.pvr
 	$(MAKE) -C lib/addons/library.kodi.inputstream
 dvdpcodecs: dllloader
-	$(MAKE) -C lib/libdvd
 
 dvdpextcodecs:
 
-- 
2.10.2

