From 9652e6b3c43b4c051f2ff0e000d7ebf5fbab418e Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 29 Aug 2022 10:54:51 -0700
Subject: [PATCH] include string.h for memset prototype

Upstream-Status: Submitted [https://pagure.io/dlm/pull-request/3]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 dlm_controld/lib.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlm_controld/lib.c b/dlm_controld/lib.c
index 8cbdd27f..a7502fcd 100644
--- a/dlm_controld/lib.c
+++ b/dlm_controld/lib.c
@@ -10,6 +10,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <stdint.h>
+#include <string.h>
 #include <errno.h>
 #include <time.h>
 #include <sys/types.h>
