From bb6a26371d15473b380459ac4404bf330634b585 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Fri, 20 Dec 2019 17:21:08 +0100
Subject: [PATCH] kexec: un-break the build on 32 bit x86

Upstream-Status: Pending
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

---
 kexec/arch/i386/Makefile    | 1 +
 kexec/arch/i386/kexec-x86.h | 4 ++++
 2 files changed, 5 insertions(+)

diff --git a/kexec/arch/i386/Makefile b/kexec/arch/i386/Makefile
index f486103..5d560be 100644
--- a/kexec/arch/i386/Makefile
+++ b/kexec/arch/i386/Makefile
@@ -12,6 +12,7 @@ i386_KEXEC_SRCS += kexec/arch/i386/kexec-beoboot-x86.c
 i386_KEXEC_SRCS += kexec/arch/i386/kexec-nbi.c
 i386_KEXEC_SRCS += kexec/arch/i386/x86-linux-setup.c
 i386_KEXEC_SRCS += kexec/arch/i386/crashdump-x86.c
+i386_KEXEC_SRCS += kexec/arch/i386/kexec-mb2-x86.c
 
 dist += kexec/arch/i386/Makefile $(i386_KEXEC_SRCS)			\
 	kexec/arch/i386/crashdump-x86.h					\
diff --git a/kexec/arch/i386/kexec-x86.h b/kexec/arch/i386/kexec-x86.h
index 0f941df..c423171 100644
--- a/kexec/arch/i386/kexec-x86.h
+++ b/kexec/arch/i386/kexec-x86.h
@@ -56,9 +56,13 @@ struct arch_options_t {
 };
 
 int multiboot_x86_probe(const char *buf, off_t len);
+int multiboot2_x86_probe(const char *buf, off_t len);
 int multiboot_x86_load(int argc, char **argv, const char *buf, off_t len,
 	struct kexec_info *info);
+int multiboot2_x86_load(int argc, char **argv, const char *buf, off_t len,
+	struct kexec_info *info);
 void multiboot_x86_usage(void);
+void multiboot2_x86_usage(void);
 
 int multiboot2_x86_load(int argc, char **argv, const char *buf, off_t len,
 			struct kexec_info *info);
