Upstream-Status: Inappropriate [disable feature]

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
--- libidn-1.41.orig/doc/Makefile.am
+++ libidn-1.41/doc/Makefile.am
@@ -73,16 +73,9 @@ BUILT_SOURCES = example.txt example2.txt
 
 # Man pages.
 
-dist_man_MANS = idn.1 $(gdoc_MANS)
+dist_man_MANS = $(gdoc_MANS)
 MAINTAINERCLEANFILES = $(dist_man_MANS)
 
-idn.1: $(top_srcdir)/src/idn.c $(top_srcdir)/src/idn.ggo $(top_srcdir)/.version
-	$(MAKE) -C ../lib
-	$(MAKE) -C ../src
-	$(AM_V_GEN)$(HELP2MAN) \
-		--name="Internationalized Domain Names command line tool" \
-		--output=$@ $(top_builddir)/src/idn$(EXEEXT)
-
 # GDOC
 
 gdoc_TEXINFOS =
