Disable yasm for libav when --disable-yasm

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Shane Wang <shane.wang@intel.com>

diff -r f2f8f74c6e30 configure.ac
--- a/configure.ac	Thu Dec 22 23:56:09 2011 +0800
+++ b/configure.ac	Thu Dec 22 23:57:37 2011 +0800
@@ -325,6 +325,12 @@
         --enable-gpl"
   fi
 
+ AC_ARG_ENABLE(yasm,
+              [AC_HELP_STRING([--disable-yasm], [disable use of yasm assembler])])
+  if test "x$enable_yasm" = "xno"; then
+    embffmpeg_configure_args="$embffmpeg_configure_args --disable-yasm"
+  fi
+
   # if we are cross-compiling, tell ffmpeg so
   target_os=`echo $host_os | sed 's/-gnu//'`
   if test "x$cross_compiling" = xyes; then
