From 28ab6ce1da3d095fdcdd66621329d2a204b4fc2e Mon Sep 17 00:00:00 2001
From: Clayton Casciato <ccasciato@21sw.us>
Date: Thu, 19 Mar 2026 21:24:50 -0600
Subject: [PATCH] Skip pkg Makefile from using its own rust steps

Upstream-Status: Inappropriate [OE Specific]

Signed-off-by: Armin Kuster <akuster808@gmail.com>
Signed-off-by: Clayton Casciato <majortomtosourcecontrol@gmail.com>
---
 Makefile.am | 2 +-
 Makefile.in | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index e99cea8..23e4d08 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -12,7 +12,7 @@ EXTRA_DIST = ChangeLog COPYING LICENSE suricata.yaml.in \
 	     scripts/docs-ubuntu-debian-minimal-build.sh \
        scripts/evedoc.py \
 	     examples/plugins
-SUBDIRS = rust src plugins qa rules doc etc python ebpf \
+SUBDIRS = src plugins qa rules doc etc python ebpf \
           $(SURICATA_UPDATE_DIR)
 DIST_SUBDIRS = $(SUBDIRS) examples/lib/simple examples/lib/custom \
 		examples/lib/live
diff --git a/Makefile.in b/Makefile.in
index a27d8b7..4bd2fc8 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -430,7 +430,7 @@ EXTRA_DIST = ChangeLog COPYING LICENSE suricata.yaml.in \
        scripts/evedoc.py \
 	     examples/plugins
 
-SUBDIRS = rust src plugins qa rules doc etc python ebpf \
+SUBDIRS = src plugins qa rules doc etc python ebpf \
           $(SURICATA_UPDATE_DIR)
 
 DIST_SUBDIRS = $(SUBDIRS) examples/lib/simple examples/lib/custom \
