From 8cfbcb85384c5d4a7bcb12254a5cc0fc543ef7ac Mon Sep 17 00:00:00 2001
From: Allen Kang <jhkang@nuvoton.com>
Date: Tue, 7 Feb 2023 13:51:02 +0800
Subject: [PATCH] Modify power-supply-monitor@.service

Signed-off-by: Allen Kang <jhkang@nuvoton.com>
---
 services/power-supply-monitor@.service | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/services/power-supply-monitor@.service b/services/power-supply-monitor@.service
index 8f794d75..452d50bf 100644
--- a/services/power-supply-monitor@.service
+++ b/services/power-supply-monitor@.service
@@ -1,11 +1,10 @@
 [Unit]
 Description=Power Supply %I Monitor
-Wants=mapper-wait@-xyz-openbmc_project-inventory-system-chassis-motherboard-powersupply%i.service
-After=mapper-wait@-xyz-openbmc_project-inventory-system-chassis-motherboard-powersupply%i.service
-Wants=mapper-wait@-org-openbmc-control-power0.service
-After=mapper-wait@-org-openbmc-control-power0.service
+Wants=mapper-wait@-xyz-openbmc_project-inventory-system-powersupply-powersupply%i.service
+After=mapper-wait@-xyz-openbmc_project-inventory-system-powersupply-powersupply%i.service
 ConditionPathExists=/etc/default/obmc/power-supply-monitor/power-supply-monitor-%i.conf
 
+
 [Service]
 EnvironmentFile=/etc/default/obmc/power-supply-monitor/power-supply-monitor-%i.conf
 ExecStart=/usr/bin/env psu-monitor --path=${DEVPATH} --instance=${INSTANCE} --inventory=${INVENTORY} --num-history-records=${NUM_HISTORY_RECORDS} --sync-gpio-path=${SYNC_GPIO_PATH} --sync-gpio-num=${SYNC_GPIO_NUM}
-- 
2.34.1

