From 264ac99df2370b37d1ae439b4936397c35cac4ab Mon Sep 17 00:00:00 2001
From: Allen Kang <jhkang@nuvoton.com>
Date: Wed, 1 Mar 2023 15:49:03 +0800
Subject: [PATCH 8/8] Get powersupplyName from config setting

Signed-off-by: Allen Kang <jhkang@nuvoton.com>
---
 power-supply/power_supply.cpp | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/power-supply/power_supply.cpp b/power-supply/power_supply.cpp
index 03f7bb3b..45246ca4 100644
--- a/power-supply/power_supply.cpp
+++ b/power-supply/power_supply.cpp
@@ -700,7 +700,6 @@ void PowerSupply::createInventoryObj()
     using namespace phosphor::pmbus;
     using namespace sdbusplus::message;
 
-    log<level::ERR>("Allen power_supply.cpp createInventoryObj");
     // Build the object map and send it to the inventory
     using Properties = std::map<std::string, std::variant<std::string, bool>>;
     using Interfaces = std::map<std::string, Properties>;
@@ -709,11 +708,11 @@ void PowerSupply::createInventoryObj()
     Properties operProps;
     Properties fieldreplaceableProps;
     Properties cachedProps;
-    Properties prettyNameProps;
     Properties presentProps ;
+    Properties itemProps ;
     Interfaces interfaces;
     Object object;
-    std::string powersupplyName = std::string( "powersupply0" ) ;
+    std::string powersupplyName = inventoryPath.substr(strlen("/xyz/openbmc_project/inventory/system/chassis/motherboard/"));
 
     operProps.emplace(FUNCTIONAL_PROP, true);
     interfaces.emplace(OPERATIONAL_STATE_IFACE, std::move(operProps));
@@ -724,11 +723,9 @@ void PowerSupply::createInventoryObj()
     fieldreplaceableProps.emplace(FIELDREPLACEABLE_PROP, true);
     interfaces.emplace(DECOR_REPLACEABLE_IFACE, std::move(fieldreplaceableProps));
 
-    presentProps.emplace(PRESENT_PROP, true);
-    interfaces.emplace(INVENTORY_IFACE, std::move(presentProps));
-
-    prettyNameProps.emplace(PRETTYNAME_PROP, std::move(powersupplyName));
-    interfaces.emplace(INVENTORY_IFACE, std::move(operProps));
+    itemProps.emplace(PRESENT_PROP, true);
+    itemProps.emplace(PRETTYNAME_PROP, std::move(powersupplyName));
+    interfaces.emplace(INVENTORY_IFACE, std::move(itemProps));
 
     presentProps.emplace(PRESENT_PROP, true);
     interfaces.emplace(PSU_INVENTORY_IFACE, std::move(presentProps));
@@ -736,7 +733,6 @@ void PowerSupply::createInventoryObj()
     // For Notify(), just send the relative path of the inventory
     // object so remove the INVENTORY_OBJ_PATH prefix
     auto path = inventoryPath.substr(strlen(INVENTORY_OBJ_PATH));
-    log<level::ERR>((std::string("Allen power_supply.cpp createInventoryObj inventoryPath=") + inventoryPath.c_str()).c_str());
 
     object.emplace(path, std::move(interfaces));
 
-- 
2.34.1

