Upstream-Status: Pending

Description: Added missing headers causing compile errors
Author: Umang Parmar <umangjparmar@gmail.com>
Forwarded: not-needed

--- a/system/core/adb/sysdeps/posix/network.cpp
+++ b/system/core/adb/sysdeps/posix/network.cpp
@@ -22,6 +22,7 @@
 #include <sys/socket.h>
 
 #include <string>
+#include <cstring>
 
 #include <android-base/logging.h>
 #include <android-base/stringprintf.h>
--- a/system/core/base/file.cpp
+++ b/system/core/base/file.cpp
@@ -26,6 +26,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <cstring>
 
 #include <memory>
 #include <mutex>
--- a/system/core/libbacktrace/BacktraceMap.cpp
+++ b/system/core/libbacktrace/BacktraceMap.cpp
@@ -21,6 +21,7 @@
 #include <stdint.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <algorithm>
 
 #include <log/log.h>
 
--- a/system/core/libbacktrace/UnwindStackMap.cpp
+++ b/system/core/libbacktrace/UnwindStackMap.cpp
@@ -20,6 +20,7 @@
 
 #include <string>
 #include <vector>
+#include <algorithm>
 
 #include <backtrace/BacktraceMap.h>
 #include <unwindstack/Elf.h>
