Upstream-Status: Pending

Description: non-x86 arches do not have PAGE_SIZE
Forwarded: not-needed
--- a/system/core/base/cmsg.cpp
+++ b/system/core/base/cmsg.cpp
@@ -33,7 +33,8 @@
                                  const std::vector<int>& fds) {
   size_t cmsg_space = CMSG_SPACE(sizeof(int) * fds.size());
   size_t cmsg_len = CMSG_LEN(sizeof(int) * fds.size());
-  if (cmsg_space >= PAGE_SIZE) {
+  size_t pagesize = static_cast<size_t>(sysconf(_SC_PAGE_SIZE));
+  if (cmsg_space >= pagesize) {
     errno = ENOMEM;
     return -1;
   }
@@ -75,7 +76,8 @@
   fds->clear();
 
   size_t cmsg_space = CMSG_SPACE(sizeof(int) * max_fds);
-  if (cmsg_space >= PAGE_SIZE) {
+  size_t pagesize = static_cast<size_t>(sysconf(_SC_PAGE_SIZE));
+  if (cmsg_space >= pagesize) {
     errno = ENOMEM;
     return -1;
   }
