From d3adfc21c9cc264bd191722f102963cbc4794259 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Wed, 23 May 2018 21:20:35 +0300
Subject: [PATCH] Add a possibility to specify where python modules are
 installed

Upstream-Status: Inappropriate [oe-core specific to solve multilib use case]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1697794c..8ab38818 100644
--- a/Makefile
+++ b/Makefile
@@ -651,7 +651,7 @@ endif
 ifeq ($(PYTHON_BINDINGS),1)
 install_python: libbtrfsutil_python
 	$(Q)cd libbtrfsutil/python; \
-		$(PYTHON) setup.py install --skip-build $(if $(DESTDIR),--root $(DESTDIR)) --prefix $(prefix)
+		$(PYTHON) setup.py install --skip-build $(if $(DESTDIR),--root $(DESTDIR)) --prefix $(prefix) --install-lib=$(PYTHON_SITEPACKAGES_DIR)
 
 .PHONY: install_python
 endif
