diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2b7c4b6..fdccfb2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,8 +2,8 @@ cmake_minimum_required (VERSION 3.5 FATAL_ERROR)
 project (node-manager-proxy CXX)
 set (CMAKE_CXX_STANDARD 17)
 set (CMAKE_CXX_STANDARD_REQUIRED ON)
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti")
-set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-rtti")
+#set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti")
+#set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fno-rtti")
 
 include_directories (${CMAKE_CURRENT_SOURCE_DIR})
 find_package (Boost REQUIRED)
