From a72dc2b1cddcf180ce782860896deae0b12d5000 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Mon, 25 Mar 2019 18:52:48 +0100
Subject: [PATCH] mesonbuild/environment.py: check environment for various
 binaries

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

---
 mesonbuild/environment.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/environment.py b/mesonbuild/environment.py
index 9a029f4..0cfdf9c 100644
--- a/mesonbuild/environment.py
+++ b/mesonbuild/environment.py
@@ -448,7 +448,7 @@ class Environment:
             config = MesonConfigFile.from_config_parser(
                 coredata.load_configs(self.coredata.cross_files))
             properties.host = Properties(config.get('properties', {}), False)
-            binaries.host = BinaryTable(config.get('binaries', {}), False)
+            binaries.host = BinaryTable(config.get('binaries', {}), True)
             if 'host_machine' in config:
                 machines.host = MachineInfo.from_literal(config['host_machine'])
             if 'target_machine' in config:
