From 63b78b7990c5d60f7bc674a26f655caa0bec3c49 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Mon, 25 Mar 2019 17:17:06 +0100
Subject: [PATCH] environment.py: detect windows also if the system string
 contains 'mingw'

Upstream-Status: Pending
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 mesonbuild/envconfig.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/envconfig.py b/mesonbuild/envconfig.py
index f2510c1..5309ef4 100644
--- a/mesonbuild/envconfig.py
+++ b/mesonbuild/envconfig.py
@@ -194,7 +194,7 @@ class MachineInfo:
         """
         Machine is windows?
         """
-        return self.system == 'windows'
+        return self.system == 'windows' or 'mingw' in self.system
 
     def is_cygwin(self):
         """
