From 7bf4c2c02eb7fe1bf24b23e3ba2d7df36495e0aa Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Wed, 13 Feb 2019 17:45:09 +0100
Subject: [PATCH] environment.py: detect windows also if the system string
 contains 'mingw'

Upstream-Status: Pending
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 mesonbuild/environment.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/environment.py b/mesonbuild/environment.py
index b4fc0dc..845077e 100644
--- a/mesonbuild/environment.py
+++ b/mesonbuild/environment.py
@@ -1247,7 +1247,7 @@ class MachineInfo:
         """
         Machine is windows?
         """
-        return self.system == 'windows'
+        return self.system == 'windows' or 'mingw' in self.system
 
     def is_cygwin(self):
         """
