From b38add903c7f701e5b79dd7d93fb7dc61d8252c7 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 20 Feb 2015 09:39:38 +0000
Subject: [PATCH] Avoid using libdir from .la which usually points to a host
 path

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Jonathan Liu <net147@gmail.com>
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 ltmain.sh | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ltmain.sh b/ltmain.sh
index ee938056bef..9ebc7e3d1e0 100644
--- a/ltmain.sh
+++ b/ltmain.sh
@@ -5628,6 +5628,9 @@ func_mode_link ()
 	    absdir="$abs_ladir"
 	    libdir="$abs_ladir"
 	  else
+	    # Instead of using libdir from .la which usually points to a host path,
+	    # use the path the .la is contained in.
+	    libdir="$abs_ladir"
 	    dir="$libdir"
 	    absdir="$libdir"
 	  fi
