From fa3b3b3759986171a85230ba8b53764beafdb37f Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Wed, 19 Jul 2017 14:40:00 +0200
Subject: [PATCH 2/3] fix comparison between pointer and zero character constant

Upstream-Status: Pending [https://github.com/archlinux/svntogit-packages/tree/packages/libmpd/trunk]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/libmpd-database.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libmpd-database.c b/src/libmpd-database.c
index 2480d5e..edafc0a 100644
--- a/src/libmpd-database.c
+++ b/src/libmpd-database.c
@@ -961,7 +961,7 @@ MpdData * mpd_database_get_directory_recursive(MpdObj *mi, const char *path)
 		debug_printf(DEBUG_WARNING,"not connected\n");
 		return NULL;
 	}
-	if(path == '\0' || path[0] == '\0')
+	if(path == NULL || path[0] == '\0')
 	{
 		debug_printf(DEBUG_ERROR, "argumant invalid\n");
 		return NULL;
-- 
2.37.2

