From 062cf74b498c1d12fd76f9d9bca926ac806e0b99 Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Thu, 20 Jun 2019 14:15:18 +0800
Subject: [PATCH] glibc only fix

Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 ctdb/tests/src/test_mutex_raw.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ctdb/tests/src/test_mutex_raw.c b/ctdb/tests/src/test_mutex_raw.c
index 926a525..814343f 100644
--- a/ctdb/tests/src/test_mutex_raw.c
+++ b/ctdb/tests/src/test_mutex_raw.c
@@ -291,8 +291,10 @@ int main(int argc, const char **argv)
 					"failed\n");
 				exit (EXIT_FAILURE);
 			}
+#ifdef __GLIBC__
 		} else if (ret == EBUSY) {
 			printf("INFO: pid=%u\n", mutex->__data.__owner);
+#endif
 		} else if (ret == 0) {
 			pthread_mutex_unlock(mutex);
 		}
-- 
2.7.4

