From 61f40b65637695abe9d9e2f80ca99d3c076c96e8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ga=C3=ABl=20Bonithon?= <gael@xfce.org>
Date: Mon, 16 Mar 2026 16:31:23 +0100
Subject: [PATCH] build: Do not display full path in generated headers

Upstream-Status: Backport [https://gitlab.xfce.org/xfce/xfce4-panel/-/commit/61f40b65637695abe9d9e2f80ca99d3c076c96e8]

Signed-off-by: Zhang Peng <peng.zhang1.cn@windriver.com>
---
 libxfce4panel/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libxfce4panel/meson.build b/libxfce4panel/meson.build
index 38d8ad198..3fee75d58 100644
--- a/libxfce4panel/meson.build
+++ b/libxfce4panel/meson.build
@@ -67,7 +67,7 @@ libpanel_enums += gnome.mkenums(
   install_dir: libpanel_install_dir,
   sources: libpanel_headers,
   fhead: '#if !defined(_LIBXFCE4PANEL_INSIDE_LIBXFCE4PANEL_H) && !defined(LIBXFCE4PANEL_COMPILATION)\n#error "Only <libxfce4panel/libxfce4panel.h> can be included directly, this file may disappear or change contents"\n#endif\n\n#ifndef __LIBXFCE4PANEL_ENUM_TYPES_H__\n#define __LIBXFCE4PANEL_ENUM_TYPES_H__\n#include <glib-object.h>\nG_BEGIN_DECLS\n',
-  fprod: '/* enumerations from "@filename@" */\n',
+  fprod: '/* enumerations from "@basename@" */\n',
   vhead: 'GType @enum_name@_get_type (void) G_GNUC_CONST;\n#define XFCE_TYPE_@ENUMSHORT@ (@enum_name@_get_type())\n',
   ftail: 'G_END_DECLS\n\n#endif /* !__LIBXFCE4PANEL_ENUM_TYPES_H__ */',
 )
@@ -76,7 +76,7 @@ libpanel_enums += gnome.mkenums(
   install_header: false,
   sources: libpanel_headers,
   fhead: '#include "libxfce4panel-enums.h"\n#include "libxfce4panel-enum-types.h"\n\n#include "libxfce4panel-visibility.h"',
-  fprod: '\n/* enumerations from "@filename@" */',
+  fprod: '\n/* enumerations from "@basename@" */',
   vhead: 'GType\n@enum_name@_get_type (void)\n{\n\tstatic GType type = 0;\n\tif (type == 0) {\n\tstatic const G@Type@Value values[] = {',
   vprod: '\t{ @VALUENAME@, "@VALUENAME@", "@valuenick@" },',
   vtail: '\t{ 0, NULL, NULL }\n\t};\n\ttype = g_@type@_register_static ("@EnumName@", values);\n  }\n\treturn type;\n}\n',
--
2.34.1
