From 6ba3f37953e391d5dd293c40e2d8256073a971cf Mon Sep 17 00:00:00 2001
From: Medad CChien <ctcchien@nuvoton.com>
Date: Tue, 15 Feb 2022 10:07:29 +0800
Subject: [PATCH] hwmon temp: add tmp100 support

Signed-off-by: Medad CChien <ctcchien@nuvoton.com>
Signed-off-by: Joseph Liu <kwliu@nuvoton.com>
---
 src/HwmonTempMain.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/HwmonTempMain.cpp b/src/HwmonTempMain.cpp
index 8df8a8c5..75200378 100644
--- a/src/HwmonTempMain.cpp
+++ b/src/HwmonTempMain.cpp
@@ -65,6 +65,7 @@ static auto sensorTypes{
                                 "xyz.openbmc_project.Configuration.NCT7802",
                                 "xyz.openbmc_project.Configuration.SBTSI",
                                 "xyz.openbmc_project.Configuration.SI7020",
+                                "xyz.openbmc_project.Configuration.TMP100",
                                 "xyz.openbmc_project.Configuration.TMP112",
                                 "xyz.openbmc_project.Configuration.TMP175",
                                 "xyz.openbmc_project.Configuration.TMP421",
-- 
2.34.1

