From 28dd66e5b49dff037534de5f0e88624a49ff3adb Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Mon, 15 Aug 2022 14:15:09 +0800
Subject: [PATCH 1/8] add TMP100 support

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 src/HwmonTempMain.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/HwmonTempMain.cpp b/src/HwmonTempMain.cpp
index 9ebdfcdf..0b8cfa69 100644
--- a/src/HwmonTempMain.cpp
+++ b/src/HwmonTempMain.cpp
@@ -67,6 +67,7 @@ static const I2CDeviceTypeMap sensorTypes{
     {"NCT7802", I2CDeviceType{"nct7802", true}},
     {"SBTSI", I2CDeviceType{"sbtsi", true}},
     {"SI7020", I2CDeviceType{"si7020", false}},
+    {"TMP100", I2CDeviceType{"tmp100", true}},
     {"TMP112", I2CDeviceType{"tmp112", true}},
     {"TMP175", I2CDeviceType{"tmp175", true}},
     {"TMP421", I2CDeviceType{"tmp421", true}},
-- 
2.17.1

