From fe08913926c140713c34841f8d14bd00b3867310 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Mon, 15 Aug 2022 14:53:06 +0800
Subject: [PATCH 3/5] change PSU sensor name to meet customer requirement

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 src/PSUSensorMain.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index b5af1769..845af476 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -271,6 +271,7 @@ static void
         name += psuName;
         name += "_";
         name += pwmName;
+	name = psuName + "_PWM_" + pwmName; // change PSU sensor name
 
         std::string objPath = interfacePath;
         objPath += "_";
-- 
2.34.1

