From 2a5c0e85394f335f86912319dd6ef9ace13652d9 Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Fri, 23 Dec 2022 10:07:51 +0800
Subject: [PATCH 5/5] add psu sesnor: p2011 and mx16550

Signed-off-by: Joseph Liu <kwliu@nuvoton.com>
---
 src/PSUSensorMain.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index 845af476..09637fb8 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -49,7 +49,7 @@ static constexpr auto sensorTypes{std::to_array<const char*>(
      "MAX20734",  "MAX20796",    "MAX34451",  "MP2971",    "MP2973",
      "MP5023",    "PLI1209BC",   "pmbus",     "PXE1610",   "RAA228000",
      "RAA228228", "RAA228620",   "RAA229001", "RAA229004", "RAA229126",
-     "TPS53679",  "TPS546D24",   "XDPE11280", "XDPE12284"})};
+     "TPS53679",  "TPS546D24",   "XDPE11280", "XDPE12284", "p2011", "MAX16550"})};
 
 // clang-format off
 static constexpr auto pmbusNames{std::to_array<const char*>({
@@ -96,7 +96,9 @@ static constexpr auto pmbusNames{std::to_array<const char*>({
     "tps53679",
     "tps546d24",
     "xdpe11280",
-    "xdpe12284"
+    "xdpe12284",
+    "p2011",
+    "max16550"
 })};
 //clang-format on
 
-- 
2.34.1

