From 4a1343581b7b9674859d53eac573ebf95017237f Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Wed, 1 Mar 2023 09:43:58 +0800
Subject: [PATCH] add psu sesnor: p2011 and mx16550

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 src/PSUSensorMain.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index 657696b3..96bdee16 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -49,7 +49,7 @@ static constexpr auto sensorTypes{std::to_array<const char*>(
      "MAX20734",  "MAX20796",    "MAX34451",  "MP2971",    "MP2973",
      "MP2975",    "MP5023",      "PLI1209BC", "pmbus",     "PXE1610",
      "RAA228000", "RAA228228",   "RAA228620", "RAA229001", "RAA229004",
-     "RAA229126", "TPS53679",    "TPS546D24", "XDPE11280", "XDPE12284"})};
+     "RAA229126", "TPS53679",    "TPS546D24", "XDPE11280", "XDPE12284", "p2011", "MAX16550"})};
 
 // clang-format off
 static constexpr auto pmbusNames{std::to_array<const char*>({
@@ -97,7 +97,9 @@ static constexpr auto pmbusNames{std::to_array<const char*>({
     "tps53679",
     "tps546d24",
     "xdpe11280",
-    "xdpe12284"
+    "xdpe12284",
+    "p2011",
+    "max16550"
 })};
 //clang-format on
 
-- 
2.17.1

