Upstream-Status: Pending

Signed-off-by: Constantin Musca <constantinx.musca@intel.com>
--- a/extensions/Makefile.am
+++ b/extensions/Makefile.am
@@ -37,8 +37,8 @@ _gen/all.xml: all.xml $(wildcard $(srcdi
 		--xinclude $(tools_dir)/identity.xsl \
 		$< > $@
 
-extensions.html: _gen/all.xml $(tools_dir)/doc-generator.xsl
+extensions.html _gen/svc-gtk-doc.h: _gen/all.xml $(tools_dir)/doc-generator.xsl
 	$(AM_V_GEN)$(XSLTPROC) $(XSLTPROCFLAGS) \
 		$(tools_dir)/doc-generator.xsl \
 		$< > $@
 
