diff --git a/image_verify.cpp b/image_verify.cpp
index b7654aa..c66a4b6 100644
--- a/image_verify.cpp
+++ b/image_verify.cpp
@@ -93,6 +93,7 @@ bool Signature::verifyFullImage()
     bool ret = true;
 #ifdef WANT_SIGNATURE_FULL_VERIFY
     std::vector<std::string> fullImages = {
+        fs::path(imageDirPath) / "image-bios.sig",
         fs::path(imageDirPath) / "image-bmc.sig",
         fs::path(imageDirPath) / "image-hostfw.sig",
         fs::path(imageDirPath) / "image-kernel.sig",
diff --git a/item_updater.cpp b/item_updater.cpp
index b6464f6..46e80cc 100644
--- a/item_updater.cpp
+++ b/item_updater.cpp
@@ -748,6 +748,28 @@ bool ItemUpdater::checkImage(const std::string& filePath,
 }
 
 #ifdef HOST_BIOS_UPGRADE
+std::string restoreBIOSVersion()
+{
+    std::string version = "null";
+    fs::path release = fs::path(PERSIST_DIR) / HOST_RELEASE_FILE;
+    if (fs::exists(release))
+    {
+        try
+        {
+            version = VersionClass::getBMCVersion(release.string());
+        }
+        catch (const std::exception& e)
+        {
+            warning("Failed to parse BIOS version: {ERROR}", "ERROR", e);
+        }
+    }
+    else
+    {
+        info("No bios version file exist");
+    }
+    return version;
+}
+
 void ItemUpdater::createBIOSObject()
 {
     std::string path = BIOS_OBJPATH;
@@ -763,7 +785,7 @@ void ItemUpdater::createBIOSObject()
     createFunctionalAssociation(path);
 
     auto versionId = path.substr(pos + 1);
-    auto version = "null";
+    auto version = restoreBIOSVersion();
     AssociationList assocs = {};
     biosActivation = std::make_unique<Activation>(
         bus, path, *this, versionId, server::Activation::Activations::Active,
diff --git a/meson.build b/meson.build
index e182f46..46cd239 100644
--- a/meson.build
+++ b/meson.build
@@ -58,6 +58,8 @@ conf.set_quoted('ALT_RWFS', '/media/alt/var/persist')
 conf.set_quoted('BMC_ROFS_PREFIX', get_option('media-dir') + '/rofs-')
 # The name of the BMC table of contents file
 conf.set_quoted('OS_RELEASE_FILE', '/etc/os-release')
+# The name of the host firmware version file
+conf.set_quoted('HOST_RELEASE_FILE', 'bios-release')
 # The dir where activation data is stored in files
 conf.set_quoted('PERSIST_DIR', '/var/lib/phosphor-bmc-code-mgmt/')
 
