Upstream-Status: Pending

--- a/linux_syscall_support.h
+++ b/linux_syscall_support.h
@@ -119,14 +119,7 @@ extern "C" {
 
 #ifdef __mips__
 /* Include definitions of the ABI currently in use.                          */
-#ifdef __ANDROID__
-/* Android doesn't have sgidefs.h, but does have asm/sgidefs.h,
- * which has the definitions we need.
- */
 #include <asm/sgidefs.h>
-#else
-#include <sgidefs.h>
-#endif
 #endif
 #endif
 
