Fix defined to match tpm2-tools 4.1.1

Upstream-Status: Submitted https://github.com/tpm2-software/tpm2-tcti-uefi/pull/81
Signed-off-by: Armin Kuster <akuster808@gmail.com>

Index: git/example/tpm2-get-caps-fixed.c
===================================================================
--- git.orig/example/tpm2-get-caps-fixed.c
+++ git/example/tpm2-get-caps-fixed.c
@@ -140,11 +140,11 @@ dump_tpm_properties_fixed (TPMS_TAGGED_P
             Print (L"TPM2_PT_INPUT_BUFFER:\n"
                     "  value: 0x%X\n", value);
             break;
-        case TPM2_PT_HR_TRANSIENT_MIN:
+        case TPM2_PT_TPM2_HR_TRANSIENT_MIN:
             Print (L"TPM2_PT_TPM2_HR_TRANSIENT_MIN:\n"
                     "  value: 0x%X\n", value);
             break;
-        case TPM2_PT_HR_PERSISTENT_MIN:
+        case TPM2_PT_TPM2_HR_PERSISTENT_MIN:
             Print (L"TPM2_PT_TPM2_HR_PERSISTENT_MIN:\n"
                     "  value: 0x%X\n", value);
             break;
