From 037a12301968a56f0c7e492ea4a05d2eecbd4cc6 Mon Sep 17 00:00:00 2001
From: Jate Sujjavanich <jatedev@gmail.com>
Date: Fri, 8 Jan 2021 20:18:00 -0500
Subject: [PATCH 2/3] Fix missing openembedded from ssg/constants.py

---
 ssg/constants.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ssg/constants.py b/ssg/constants.py
index fab7cda5d..2ca289f84 100644
--- a/ssg/constants.py
+++ b/ssg/constants.py
@@ -234,7 +234,8 @@ PRODUCT_TO_CPE_MAPPING = {
 }
 
 MULTI_PLATFORM_LIST = ["rhel", "fedora", "rhosp", "rhv", "debian", "ubuntu",
-                       "wrlinux", "opensuse", "sle", "ol", "ocp", "example"]
+                       "wrlinux", "opensuse", "sle", "ol", "ocp", "example",
+                       "openembedded"]
 
 MULTI_PLATFORM_MAPPING = {
     "multi_platform_debian": ["debian8"],
@@ -249,6 +250,7 @@ MULTI_PLATFORM_MAPPING = {
     "multi_platform_sle": ["sle11", "sle12"],
     "multi_platform_ubuntu": ["ubuntu1404", "ubuntu1604", "ubuntu1804"],
     "multi_platform_wrlinux": ["wrlinux"],
+    "multi_platform_openembedded": ["openembedded"],
 }
 
 RHEL_CENTOS_CPE_MAPPING = {
-- 
2.24.3 (Apple Git-128)

