Upstream-Status: Pending

--- /tmp/configure.in	2010-05-19 13:29:03.000000000 +0200
+++ unix/configure.in	2010-05-19 13:42:05.000000000 +0200
@@ -526,13 +526,9 @@
 	found_xft="yes"
 	dnl make sure package configurator (xft-config or pkg-config
 	dnl says that xft is present.
-	XFT_CFLAGS=`xft-config --cflags 2>/dev/null` || found_xft="no"
-	XFT_LIBS=`xft-config --libs 2>/dev/null` || found_xft="no"
-	if test "$found_xft" = "no" ; then
 	    found_xft=yes
 	    XFT_CFLAGS=`pkg-config --cflags xft fontconfig 2>/dev/null` || found_xft="no"
 	    XFT_LIBS=`pkg-config --libs xft fontconfig 2>/dev/null` || found_xft="no"
-	fi
 	AC_MSG_RESULT([$found_xft])
 	dnl make sure that compiling against Xft header file doesn't bomb
 	if test "$found_xft" = "yes" ; then
