From 6c3ac20477a4bac643088f24df3c042e627fafa9 Mon Sep 17 00:00:00 2001
From: Guillaume Burel <guillaume.burel@stormshield.eu>
Date: Fri, 3 Jan 2020 11:25:54 +0100
Subject: [PATCH] Using native binaries

---
 node.gyp                 |  4 ++--
 tools/v8_gypfiles/v8.gyp | 11 ++++-------
 2 files changed, 6 insertions(+), 9 deletions(-)

--- a/node.gyp
+++ b/node.gyp
@@ -487,6 +487,7 @@
               'action_name': 'run_mkcodecache',
               'process_outputs_as_sources': 1,
               'inputs': [
+                '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
                 '<(mkcodecache_exec)',
               ],
               'outputs': [
@@ -512,6 +513,7 @@
               'action_name': 'node_mksnapshot',
               'process_outputs_as_sources': 1,
               'inputs': [
+                '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
                 '<(node_mksnapshot_exec)',
               ],
               'outputs': [
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -220,6 +220,7 @@
         {
           'action_name': 'run_torque_action',
           'inputs': [  # Order matters.
+            '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
             '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)torque<(EXECUTABLE_SUFFIX)',
             '<@(torque_files)',
           ],
@@ -351,6 +352,7 @@
         {
           'action_name': 'generate_bytecode_builtins_list_action',
           'inputs': [
+            '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
             '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)bytecode_builtins_list_generator<(EXECUTABLE_SUFFIX)',
           ],
           'outputs': [
@@ -533,6 +535,7 @@
             ],
           },
           'inputs': [
+            '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
             '<(mksnapshot_exec)',
           ],
           'outputs': [
@@ -1448,6 +1451,7 @@
         {
           'action_name': 'run_gen-regexp-special-case_action',
           'inputs': [
+            '<(PRODUCT_DIR)/v8-qemu-wrapper.sh',
             '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)gen-regexp-special-case<(EXECUTABLE_SUFFIX)',
           ],
           'outputs': [
