From 4bbee5e3d58bc4911999f3ec2cc5aab8ded6717b Mon Sep 17 00:00:00 2001
From: "Winker Matthias (TT-CA/ENG1)" <Matthias.Winker@de.bosch.com>
Date: Tue, 11 Dec 2018 10:44:29 +0100
Subject: [PATCH] Using native torque

---
 deps/v8/gypfiles/v8.gyp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/deps/v8/gypfiles/v8.gyp b/deps/v8/gypfiles/v8.gyp
index 8c78f02255..434168844e 100644
--- a/deps/v8/gypfiles/v8.gyp
+++ b/deps/v8/gypfiles/v8.gyp
@@ -2831,7 +2831,6 @@
         {
           'action_name': 'run_torque',
           'inputs': [  # Order matters.
-            '<(PRODUCT_DIR)/<(EXECUTABLE_PREFIX)torque<(EXECUTABLE_SUFFIX)',
             '../src/builtins/base.tq',
             '../src/builtins/array.tq',
             '../src/builtins/typed-array.tq',
@@ -2845,7 +2844,10 @@
             '<(SHARED_INTERMEDIATE_DIR)/torque-generated/builtins-typed-array-from-dsl-gen.cc',
             '<(SHARED_INTERMEDIATE_DIR)/torque-generated/builtins-typed-array-from-dsl-gen.h',
           ],
-          'action': ['<@(_inputs)', '-o', '<(SHARED_INTERMEDIATE_DIR)/torque-generated'],
+          'action': [
+            'torque',
+            '<@(_inputs)',
+            '-o', '<(SHARED_INTERMEDIATE_DIR)/torque-generated'],
         },
       ],
     }, # torque
-- 
2.19.2

