From 07ef86e33ed6f7585f0dfaa1732ea17c816655a4 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 14 Jan 2019 11:45:42 -0800
Subject: [PATCH] makehrtf: Disable Wstringop-truncation

Upstream-Status: Inappropriate [Should be fixed in code]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 utils/makemhr/makemhr.cpp | 2 ++
 1 file changed, 2 insertions(+)

--- a/utils/makemhr/makemhr.cpp
+++ b/utils/makemhr/makemhr.cpp
@@ -161,6 +161,7 @@ enum ChannelIndex : uint {
  * pattern string are replaced with the replacement string.  The result is
  * truncated if necessary.
  */
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
 static int StrSubst(const char *in, const char *pat, const char *rep, const size_t maxLen, char *out)
 {
     size_t inLen, patLen, repLen;
@@ -199,6 +200,7 @@ static int StrSubst(const char *in, cons
     return !truncated;
 }
 
+#pragma GCC diagnostic pop
 
 /*********************
  *** Math routines ***
