
Upstream-Status: Inappropriate [configuration]

Signed-off-by: Saul Wold <sgw@linux.intel.com>

Index: Python-2.7.2/Modules/Setup.dist
===================================================================
--- Python-2.7.2.orig/Modules/Setup.dist	2011-06-11 08:46:26.000000000 -0700
+++ Python-2.7.2/Modules/Setup.dist	2011-12-27 15:51:41.244623219 -0800
@@ -248,7 +248,7 @@
 # Message-Digest Algorithm, described in RFC 1321.  The necessary files
 # md5.c and md5.h are included here.
 
-#_md5 md5module.c md5.c
+_md5 md5module.c md5.c
 
 
 # The _sha module implements the SHA checksum algorithms.
