From f7f785c21306010b2367572250b2822df5bc7728 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier at gentoo.org>
Date: Thu, 18 Nov 2010 16:59:14 -0500
Subject: [PATCH] printf-parse: pull in features.h for __GLIBC__

Upstream-Status: Pending

Signed-off-by: Mike Frysinger <vapier at gentoo.org>

---
 lib/printf-parse.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/lib/printf-parse.h b/lib/printf-parse.h
index e7d0f82..d7b4534 100644
--- a/lib/printf-parse.h
+++ b/lib/printf-parse.h
@@ -28,6 +28,9 @@
 
 #include "printf-args.h"
 
+#ifdef HAVE_FEATURES_H
+# include <features.h>	/* for __GLIBC__ */
+#endif
 
 /* Flags */
 #define FLAG_GROUP       1      /* ' flag */
