--- a/src/plugins/Makefile.am	2015-10-25 16:12:29.331415823 +0000
+++ b/src/plugins/Makefile.am	2015-10-25 16:14:37.593415808 +0000
@@ -93,7 +93,7 @@
 id3_id3_la_SOURCES = id3/id3.c id3/id3v1_genres.c
 id3_id3_la_LIBADD = $(PLUGINS_LIBADD)
 
-id3/id3v1_genres.c: $(srcdir)/id3/id3v1_genres.def $(srcdir)/id3/id3v1_genres_gen.awk
+$(srcdir)/id3/id3v1_genres.c: $(srcdir)/id3/id3v1_genres.def $(srcdir)/id3/id3v1_genres_gen.awk
 	$(AWK) -f $(srcdir)/id3/id3v1_genres_gen.awk $(srcdir)/id3/id3v1_genres.def > $@
 
 EXTRA_DIST += id3/id3v1_genres.def id3/id3v1_genres_gen.awk
