From 52360260c0d73a00082d867bc416225fbc2b657e Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Tue, 12 Jul 2011 11:46:37 +0200
Subject: [PATCH] glibmm: add 2.28.2

---
 Makefile.am  | 7 +------
 configure.ac | 1 -
 2 files changed, 1 insertion(+), 7 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 1b366b4..0ff58b5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -23,12 +23,7 @@ src_subdirs = glib/src gio/src
 else
 src_subdirs =
 endif
-if ENABLE_DOCUMENTATION
-doc_subdirs = docs
-else
-doc_subdirs =
-endif
-SUBDIRS = tools $(src_subdirs) glib/glibmm gio/giomm tests examples $(doc_subdirs)
+SUBDIRS = tools $(src_subdirs) glib/glibmm gio/giomm tests
 
 glibmm_includedir = $(includedir)/$(GLIBMM_MODULE_NAME)
 glibmm_include_HEADERS = glib/glibmm.h
diff --git a/configure.ac b/configure.ac
index e706687..198e0d0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -166,7 +166,6 @@ AC_CONFIG_FILES([Makefile
                  gio/src/Makefile
                  gio/giomm/Makefile
                  tests/Makefile
-                 examples/Makefile
                  docs/Makefile
                  docs/reference/Doxyfile
                  MSVC_NMake/glibmm/glibmm.rc
