From: Thilo Cestonaro <thilo.cestonaro@ts.fujitsu.com>
Date: Fri, 18 Nov 2016 10:23:07 +0100
Subject: Build libdbus-c++ tools for target platform rather than enforced for
host platform.

Upstream-Status: Inappropriate [no upstream]
Signed-off-by: Thilo Cestonaro <thilo.cestonaro@ts.fujitsu.com>
---

diff -Naur libdbus-c++-0.9.0.ori/configure.ac libdbus-c++-0.9.0/configure.ac
--- libdbus-c++-0.9.0.ori/configure.ac	2016-11-15 14:25:36.085882774 +0100
+++ libdbus-c++-0.9.0/configure.ac	2016-11-15 14:27:08.814568717 +0100
@@ -64,9 +64,6 @@
 AC_PROG_CC
 AC_PROG_CXX
 
-CXX_FOR_BUILD=${CXX_FOR_BUILD-${CXX}}
-AC_SUBST(CXX_FOR_BUILD)
-
 AM_PROG_LIBTOOL
 
 PKG_PROG_PKG_CONFIG
diff -Naur libdbus-c++-0.9.0.ori/tools/Makefile.am libdbus-c++-0.9.0/tools/Makefile.am
--- libdbus-c++-0.9.0.ori/tools/Makefile.am	2016-11-15 14:25:36.089882803 +0100
+++ libdbus-c++-0.9.0/tools/Makefile.am	2016-11-15 14:26:19.454203583 +0100
@@ -1,7 +1,5 @@
 # hacky, but ...
 
-CXX = $(CXX_FOR_BUILD)
-
 AM_CPPFLAGS = \
 	$(dbus_CFLAGS) \
 	$(xml_CFLAGS) \
