Upstream-Status: Inappropriate

Index: openssl-1.0.2d/crypto/opensslv.h
===================================================================
--- openssl-1.0.2d.orig/crypto/opensslv.h
+++ openssl-1.0.2d/crypto/opensslv.h
@@ -88,7 +88,7 @@ extern "C" {
  * should only keep the versions that are binary compatible with the current.
  */
 # define SHLIB_VERSION_HISTORY ""
-# define SHLIB_VERSION_NUMBER "1.0.0"
+# define SHLIB_VERSION_NUMBER "1.0.2"
 
 
 #ifdef  __cplusplus
