Musl fixes, which should be applied upstream too

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
--- a/src/dbus/common/types.hpp
+++ b/src/dbus/common/types.hpp
@@ -715,7 +715,7 @@ struct TrelInfo
     };
 
     bool               mEnabled;      ///< Whether TREL is enabled.
-    u_int16_t          mNumTrelPeers; ///< The number of TREL peers.
+    uint16_t          mNumTrelPeers; ///< The number of TREL peers.
     TrelPacketCounters mTrelCounters; ///< The TREL counters.
 };
 
--- a/third_party/openthread/repo/src/posix/platform/CMakeLists.txt
+++ b/third_party/openthread/repo/src/posix/platform/CMakeLists.txt
@@ -172,7 +172,7 @@ target_link_libraries(openthread-posix
 )
 
 option(OT_TARGET_OPENWRT "enable openthread posix for OpenWRT" OFF)
-if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux" AND NOT OT_TARGET_OPENWRT)
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux" AND NOT OT_TARGET_OPENWRT AND NOT OT_TARGET_MUSL)
     target_compile_definitions(ot-posix-config
         INTERFACE "OPENTHREAD_POSIX_CONFIG_NAT64_AIL_PREFIX_ENABLE=1"
     )
