From e969ac59335d3fb1cd228f8e7c4f6c2dda4fa536 Mon Sep 17 00:00:00 2001
From: Avinash Sonawane <rootkea@gmail.com>
Date: Mon, 20 Dec 2021 13:33:42 +0530
Subject: [PATCH] Pass correct parameter

Upstream-Status: Backport [https://gitlab.gnome.org/GNOME/libgtop/-/merge_requests/35]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/daemon/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/daemon/main.c b/src/daemon/main.c
index b51addf..eaee3b3 100644
--- a/src/daemon/main.c
+++ b/src/daemon/main.c
@@ -222,8 +222,9 @@ handle_parent_connection (int s)
 		       0, NULL);
 	    break;
     case GLIBTOP_CMND_PROC_IO:
+        memcpy (&pid, parameter, sizeof (pid_t));
 	    glibtop_get_proc_io_l
-		(server, &resp->u.data.proc_io, parameter);
+		(server, &resp->u.data.proc_io, pid);
 	    do_output (s, resp, _offset_data (proc_io),
 		       0, NULL);
 	    break;
-- 
2.37.2

