From b60e364c35d56a2a3164254d98c9931bfa3feed4 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 11 Apr 2012 14:34:43 +0200
Subject: [PATCH] add support for /etc/X11/xserver-system

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

---
Upstream-Status: Pending

 X11/xserver-common | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/X11/xserver-common b/X11/xserver-common
index 52655cb..bb7c2ff 100644
--- a/X11/xserver-common
+++ b/X11/xserver-common
@@ -204,3 +204,8 @@ fi
 if [ "$XSERVER" != "Xorg" ] ; then
     ARGS="$ARGS $MOUSE $KDRIVEARGS"
 fi
+
+# let a local script modify the variables                    
+if [ -e /etc/X11/xserver-system ] ; then                    
+       . /etc/X11/xserver-system                    
+fi                    
