From e2abbe110e71a7b4b56564e6b9d58ffa7f34a9e8 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 11 Apr 2012 14:37:29 +0200
Subject: [PATCH] xserver-common: add dpi and nocursor params for gta01 and
 gta02

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

---
 X11/xserver-common | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/X11/xserver-common b/X11/xserver-common
index d622f2a..a2c0591 100644
--- a/X11/xserver-common
+++ b/X11/xserver-common
@@ -133,15 +133,19 @@ else
             ARGS="$ARGS -screen ${SCREEN_SIZE}"
             DPI="225" ;;
         "gta01" )
-            DPI="285"
+            DPI="280"
             if [ "$XSERVER" != "Xorg" ] ; then
                 ARGS="$ARGS -screen 480x640"
+            else
+                ARGS="$ARGS -dpi ${DPI} -nocursor"
             fi
             ;;
         "gta02")
-            DPI="285"
+            DPI="280"
             if [ "$XSERVER" != "Xorg" ] ; then
                 ARGS="$ARGS -screen ${SCREEN_SIZE}"
+            else
+                ARGS="$ARGS -dpi ${DPI} -nocursor"
             fi
             ;;
         "motorola_ezx_platform")
