From f2d808cea5b9f69158f82e16c06f882a6db10d63 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 11 Apr 2012 14:40:14 +0200
Subject: [PATCH] xserver-common: add support for n900 alias nokia_rx-51_board

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

---
 X11/xserver-common | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/X11/xserver-common b/X11/xserver-common
index a2c0591..b8c37e6 100644
--- a/X11/xserver-common
+++ b/X11/xserver-common
@@ -132,6 +132,13 @@ else
         "nokia_n800" | "nokia_rx-44" | "nokia_n770")
             ARGS="$ARGS -screen ${SCREEN_SIZE}"
             DPI="225" ;;
+        "nokia_rx-51_board" )                                     
+            DPI="267"                                    
+            if [ "$XSERVER" != "Xorg" ] ; then                    
+                ARGS="$ARGS -screen ${SCREEN_SIZE}"               
+            else                                                  
+                ARGS="$ARGS -dpi ${DPI} -nocursor"       
+            fi ;;                                                    
         "gta01" )
             DPI="280"
             if [ "$XSERVER" != "Xorg" ] ; then
