From a194ba2013f9e58cf73cd39b57be68386e501eda Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 11 Apr 2012 14:40:44 +0200
Subject: [PATCH] xserver-common: add support for nexus S alias herring

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

---
 X11/xserver-common | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/X11/xserver-common b/X11/xserver-common
index b8c37e6..9aa6b24 100644
--- a/X11/xserver-common
+++ b/X11/xserver-common
@@ -126,6 +126,9 @@ else
         "htc_tornado")
             ARGS="$ARGS -hide-cursor"
             DPI="100" ;;
+        "herring")
+            DPI="235"
+            ARGS="$ARGS -dpi ${DPI} -nocursor";;
         "generic_omap1510/1610/1710")
             ARGS="$ARGS -screen ${SCREEN_SIZE}"
             DPI="225" ;;
