From dfd832b35b6cb4cf2b6f2b51df7b0999f10c8ca1 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 11 Apr 2012 14:41:06 +0200
Subject: [PATCH] xserver-common: add support for nexus one alias mahimahi

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

---
 X11/xserver-common | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/X11/xserver-common b/X11/xserver-common
index 9aa6b24..8da926e 100644
--- a/X11/xserver-common
+++ b/X11/xserver-common
@@ -129,6 +129,9 @@ else
         "herring")
             DPI="235"
             ARGS="$ARGS -dpi ${DPI} -nocursor";;
+        "mahimahi")
+            DPI="254"
+            ARGS="$ARGS -dpi ${DPI}";;
         "generic_omap1510/1610/1710")
             ARGS="$ARGS -screen ${SCREEN_SIZE}"
             DPI="225" ;;
