From d9e05e53b30c66705852256ff5c603252aa94b2c Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 11 Apr 2012 14:41:52 +0200
Subject: [PATCH] xserver-common: add support for gta04 alias OpenPhoenux

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

---
 X11/xserver-common | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/X11/xserver-common b/X11/xserver-common
index 8da926e..d9325dd 100644
--- a/X11/xserver-common
+++ b/X11/xserver-common
@@ -161,6 +161,14 @@ else
                 ARGS="$ARGS -dpi ${DPI} -nocursor"
             fi
             ;;
+        "gta04")
+            DPI="280"
+            if [ "$XSERVER" != "Xorg" ] ; then
+                ARGS="$ARGS -screen ${SCREEN_SIZE}"
+            else
+                ARGS="$ARGS -dpi ${DPI} -nocursor"
+            fi
+            ;;
         "motorola_ezx_platform")
             ARGS="$ARGS -screen 240x320"
             DPI="170" ;;
