From 57ab521ffc1024f21105ca14be2d041a477942b1 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 17 Aug 2012 13:00:56 +0200
Subject: [PATCH] xserver-common: add support for tuna alias Galaxy Nexus

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>

---
Upstream-Status: Pending

 X11/xserver-common | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/X11/xserver-common b/X11/xserver-common
index d9325dd..017e5ae 100644
--- a/X11/xserver-common
+++ b/X11/xserver-common
@@ -169,6 +169,14 @@ else
                 ARGS="$ARGS -dpi ${DPI} -nocursor"
             fi
             ;;
+        "tuna")
+            DPI="316"
+            if [ "$XSERVER" != "Xorg" ] ; then
+                ARGS="$ARGS -screen ${SCREEN_SIZE}"
+            else
+                ARGS="$ARGS -dpi ${DPI} -nocursor"
+            fi
+            ;;
         "motorola_ezx_platform")
             ARGS="$ARGS -screen 240x320"
             DPI="170" ;;
