From eaf51d6420c63713f62025583681007c3502d387 Mon Sep 17 00:00:00 2001
From: Carlos <carlos.rodriguez@ridgerun.com>
Date: Fri, 15 Dec 2017 10:14:05 -0600
Subject: [PATCH] gstd:yocto compatibility

---
 gstd/Makefile.am                | 2 +-
 3 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/gstd/Makefile.am b/gstd/Makefile.am
index 13c551d..11fe202 100644
--- a/gstd/Makefile.am
+++ b/gstd/Makefile.am
@@ -49,7 +49,7 @@ bin_PROGRAMS = gstd
 
 gstd_SOURCES = gstd.c
 gstd_CFLAGS = $(GST_CFLAGS) $(GIO_CFLAGS) $(GJSON_CFLAGS)
-gstd_LDFLAGS = $(GST_LIBS) $(GIO_LIBS) $(GJSON_LIBS) -Wl,-rpath -Wl,$(libdir)
+gstd_LDFLAGS = $(GST_LIBS) $(GIO_LIBS) $(GJSON_LIBS)
 gstd_LDADD = libgstd-core.la
 
 gstdincludedir = $(includedir)/gstd
-- 
1.9.1

