From eaf51d6420c63713f62025583681007c3502d387 Mon Sep 17 00:00:00 2001
From: Carlos <carlos.rodriguez@ridgerun.com>
Date: Fri, 15 Dec 2017 10:14:05 -0600
Subject: [PATCH] gstd:yocto compatibility

---
 gstd/Makefile.am                | 2 +-
 3 files changed, 9 insertions(+), 3 deletions(-)

--- a/gstd/Makefile.am
+++ b/gstd/Makefile.am
@@ -74,7 +74,7 @@ gstd_CFLAGS = $(GST_CFLAGS)					\
 	      -DGSTD_LOG_STATE_DIR=\"$(gstdlogstatedir)\"	\
 	      -DGSTD_RUN_STATE_DIR=\"$(gstdrunstatedir)\"
 
-gstd_LDFLAGS = $(GST_LIBS) $(GIO_LIBS) $(GIO_UNIX_LIBS) $(GJSON_LIBS) $(LIBD_LIBS) -Wl,-rpath -Wl,$(libdir)
+gstd_LDFLAGS = $(GST_LIBS) $(GIO_LIBS) $(GIO_UNIX_LIBS) $(GJSON_LIBS) $(LIBD_LIBS)
 gstd_LDADD = libgstd-core.la
 
 gstdincludedir = $(includedir)/gstd
