From e671b43a52bf5785fe4aad84712dbabc6c86b39e Mon Sep 17 00:00:00 2001
From: Chris Larson <kergoth@openedhand.com>
Date: Tue, 5 Sep 2006 07:24:58 +0000
Subject: [PATCH] Add dpkg, modified from upstream oe.

Upstream-Status: Inappropriate [disable feature]
---
 Makefile.am | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 74f8df197..35e5eee39 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -11,7 +11,6 @@ SUBDIRS = \
 	$(MAYBE_DSELECT) \
 	scripts \
 	po \
-	man \
 	# EOL
 
 ACLOCAL_AMFLAGS = -I m4
