From ad9024e50a9df4a2b8a1a019e07bb133c3df31bf Mon Sep 17 00:00:00 2001
From: Daniel Watkins <daniel@daniel-watkins.co.uk>
Date: Thu, 28 Aug 2014 18:49:04 +0200
Subject: [PATCH] Switch from using distutils to setuptools.

Upstream-Status: Backport

Signed-off-by: Tim Orling <tim.orling@konsulko.com>
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index c6d57a6..6755ab4 100644
--- a/setup.py
+++ b/setup.py
@@ -12,7 +12,7 @@
 # http://opensource.org/licenses/BSD-3-Clause
 import os
 import sys
-from distutils.core import setup
+from setuptools import setup
 # a simple import wouldn't work if we moved towards a package with __init__
 from _version import __version__
 
-- 
2.30.2

