From 08627a115b498f8b468d3971e0f88b8b46e70925 Mon Sep 17 00:00:00 2001
From: Brian Ma <chma0@nuvoton.com>
Date: Thu, 15 Jul 2021 14:54:27 +0800
Subject: [PATCH] Add more PSU devices support

Signed-off-by: Allen Kang <jhkang@nuvoton.com>
---
 src/PSUSensorMain.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index 38f92a76..9b5caabe 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -96,6 +96,9 @@ static constexpr auto pmbusNames{std::to_array<const char*>({
     "tps546d24",
     "xdpe11280",
     "xdpe12284"
+    "flexpower",
+    "tps53659",
+    "tps53679"
 })};
 //clang-format on
 
-- 
2.34.1

