From 83833a85391da1b16f1f9330d2600034414638e6 Mon Sep 17 00:00:00 2001
From: Brian Ma <chma0@nuvoton.com>
Date: Thu, 15 Jul 2021 14:54:27 +0800
Subject: [PATCH] Add more PSU devices support

---
 src/PSUSensorMain.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index 7ca5d3d3..63d8da2d 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -109,7 +109,10 @@ static constexpr auto pmbusNames{std::to_array<const char*>({
     "raa229004",
     "raa229126",
     "tps546d24",
-    "xdpe12284"
+    "xdpe12284",
+    "flexpower",
+    "tps53659",
+    "tps53679"
 })};
 //clang-format on
 
-- 
2.17.1

