From 57acc3674863538da4f3366bc8376c3a893ddb41 Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Tue, 28 Mar 2023 11:56:33 +0800
Subject: [PATCH] Add more PSU devices support

Signed-off-by: Joseph Liu <kwliu@nuvoton.com>
---
 src/PSUSensorMain.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index 07802cd4..5a1deef1 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -88,6 +88,8 @@ static const I2CDeviceTypeMap sensorTypes{
     {"TPS546D24", I2CDeviceType{"tps546d24", true}},
     {"XDPE11280", I2CDeviceType{"xdpe11280", true}},
     {"XDPE12284", I2CDeviceType{"xdpe12284", true}},
+    {"TPS53659", I2CDeviceType{"tps53659", true}},
+    {"TPS53679", I2CDeviceType{"tps53679", true}},
 };
 
 namespace fs = std::filesystem;
-- 
2.34.1

