From c32fe3424c067baebbf6c263886e04bb99a5af77 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Sat, 24 Aug 2024 15:32:25 +0200
Subject: [PATCH] include/libnetlink.h: add missing include for htobe64
 definitions

Upstream-Status: Submitted [by email to stephen@networkplumber.org netdev@vger.kernel.org]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>
---
 include/libnetlink.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/libnetlink.h b/include/libnetlink.h
index e91505d..938bb40 100644
--- a/include/libnetlink.h
+++ b/include/libnetlink.h
@@ -13,6 +13,7 @@
 #include <linux/neighbour.h>
 #include <linux/netconf.h>
 #include <arpa/inet.h>
+#include <endian.h>
 
 struct rtnl_handle {
 	int			fd;
