From 176c3bd3d86c223fd44b0f15b62d38526bb3b0d2 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 17 Aug 2012 13:00:56 +0200
Subject: [PATCH 13/13] xserver-common: add support for tuna alias Galaxy
 Nexus

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 X11/xserver-common |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/X11/xserver-common b/X11/xserver-common
index d9325dd..017e5ae 100644
--- a/X11/xserver-common
+++ b/X11/xserver-common
@@ -169,6 +169,14 @@ else
                 ARGS="$ARGS -dpi ${DPI} -nocursor"
             fi
             ;;
+        "tuna")
+            DPI="316"
+            if [ "$XSERVER" != "Xorg" ] ; then
+                ARGS="$ARGS -screen ${SCREEN_SIZE}"
+            else
+                ARGS="$ARGS -dpi ${DPI} -nocursor"
+            fi
+            ;;
         "motorola_ezx_platform")
             ARGS="$ARGS -screen 240x320"
             DPI="170" ;;
-- 
1.7.8.6

