From b71eb384522b5ce4629dee6e8be257fb4880fef3 Mon Sep 17 00:00:00 2001
From: Benjamin Gaignard <benjamin.gaignard@linaro.org>
Date: Thu, 20 Apr 2017 14:25:18 +0200

---
 Makefile.am  | 1 -
 configure.ac | 9 ---------
 2 files changed, 10 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index e9113c2..2fe3d54 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,7 +2,6 @@ SUBDIRS = \
 	src \
 	data \
 	etc \
-	docs \
 	m4macros \
 	tests
 
diff --git a/configure.ac b/configure.ac
index 48ee0b2..a1a5fa1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -194,10 +194,6 @@ AC_SUBST(ADDITIONAL_OBJECTS)
 AC_PATH_PROG(XSLTPROC, xsltproc, no)
 AM_CONDITIONAL(HAVE_XSLTPROC, test "x$XSLTPROC" != "xno")
 
-# Check for asciidoc
-AC_PATH_PROG(A2X, a2x, no)
-AM_CONDITIONAL(HAVE_A2X, test "x$A2X" != "xno")
-
 # checking xmllint
 AC_PATH_PROG(XMLLINT, xmllint, no)
 if test "x$XMLLINT" != "xno"; then
@@ -219,11 +215,6 @@ src/Makefile
 data/Makefile
 data/templates/Makefile
 etc/Makefile
-docs/Makefile
-docs/man5/Makefile
-docs/man5/tinyproxy.conf.txt
-docs/man8/Makefile
-docs/man8/tinyproxy.txt
 m4macros/Makefile
 tests/Makefile
 tests/scripts/Makefile
