Origin: Debian packaging
From: Klee Dienes <klee@mit.edu>
Date: Thu, 13 Feb 2014 07:03:26 -0500
Subject: Add CMake support.

---
Upstream-Status: Pending

 CMakeLists.txt | 44 ++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 44 insertions(+)
 create mode 100644 CMakeLists.txt

--- /dev/null
+++ b/CMakeLists.txt
@@ -0,0 +1,50 @@
+cmake_minimum_required (VERSION 2.8.8)
+
+project (iniparser)
+include (GNUInstallDirs)
+
+include_directories (src)
+
+set(INIPARSER_SRCS src/dictionary.c src/iniparser.c)
+set(INIPARSER_HDRS src/dictionary.h src/iniparser.h)
+
+add_library(iniparser-shared SHARED ${INIPARSER_SRCS} ${INIPARSER_HDRS})
+add_library(iniparser-static STATIC ${INIPARSER_SRCS} ${INIPARSER_HDRS})
+
+set_target_properties(iniparser-shared PROPERTIES SOVERSION 1)
+set_target_properties(iniparser-shared PROPERTIES OUTPUT_NAME iniparser)
+set_target_properties(iniparser-static PROPERTIES OUTPUT_NAME iniparser)
+
+install (TARGETS iniparser-shared
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+install (TARGETS iniparser-static
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+
+find_package(Doxygen)
+if (NOT DOXYGEN_FOUND)
+message(FATAL_ERROR "Doxygen is needed to build the documentation. Please install it correctly")
+endif()
+
+file (WRITE ${CMAKE_CURRENT_BINARY_DIR}/iniparser.dox
+  "@INCLUDE = ${CMAKE_CURRENT_SOURCE_DIR}/doc/iniparser.dox\n"
+  "OUTPUT_DIRECTORY = ${CMAKE_CURRENT_BINARY_DIR}\n"
+  )
+
+add_custom_target (doc ALL 
+  COMMAND ${DOXYGEN_EXECUTABLE} ${CMAKE_CURRENT_BINARY_DIR}/iniparser.dox
+  SOURCES doc/iniparser.dox)
+
+enable_testing()
+
+add_test(NAME testsuite
+  COMMAND make
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/test)
+
+install (FILES ${INIPARSER_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/iniparser)
+
+install (DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/html DESTINATION ${CMAKE_INSTALL_DOCDIR})
