From c39d0896930e25c224cc897660fc8511ccae30c8 Mon Sep 17 00:00:00 2001
From: Changqing Li <changqing.li@windriver.com>
Date: Thu, 23 Apr 2020 10:01:12 +0000
Subject: [PATCH] change shebang to python3

Upstream-Status: Pending

Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 pkg_resources/_vendor/appdirs.py   | 2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/pkg_resources/_vendor/appdirs.py b/pkg_resources/_vendor/appdirs.py
index ae67001..933e398 100644
--- a/pkg_resources/_vendor/appdirs.py
+++ b/pkg_resources/_vendor/appdirs.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 # Copyright (c) 2005-2010 ActiveState Software Inc.
 # Copyright (c) 2013 Eddy Petrișor
-- 
2.24.1

