From 607b0bed5e8a5f5c3754cd6ffcf499d8134c37f9 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Mon, 30 Mar 2020 17:11:19 +0200
Subject: [PATCH] gptcurses: correctly include curses.h

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 gptcurses.cc | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/gptcurses.cc b/gptcurses.cc
index 1b18cf2..4ebfde1 100644
--- a/gptcurses.cc
+++ b/gptcurses.cc
@@ -23,11 +23,7 @@
 #include <iostream>
 #include <string>
 #include <sstream>
-#ifdef __APPLE__
 #include <ncurses.h>
-#else
-#include <ncursesw/ncurses.h>
-#endif
 #include "gptcurses.h"
 #include "support.h"
 
