--- a/git/bin/sensor_manager2.py
+++ b/git/bin/sensor_manager2.py
@@ -63,13 +63,13 @@ if __name__ == '__main__':
 	root_sensor.add(obj_path,Sensors.BootProgressSensor(bus,obj_path))
 
 	obj_path = OBJ_PATH+"/host/cpu0/OccStatus"
-	sensor_obj = Sensors.OccStatusSensor(bus,obj_path)
-	sensor_obj.sysfs_attr = "/sys/class/i2c-adapter/i2c-3/3-0050/online"
+	sensor_obj = Sensors.OccStatusSensor(bus,obj_path,
+			"/sys/class/i2c-adapter/i2c-4/4-0050/online")
 	root_sensor.add(obj_path,sensor_obj)
 
 	obj_path = OBJ_PATH+"/host/cpu1/OccStatus"
-	sensor_obj = Sensors.OccStatusSensor(bus,obj_path)
-	sensor_obj.sysfs_attr = "/sys/class/i2c-adapter/i2c-3/3-0051/online"
+	sensor_obj = Sensors.OccStatusSensor(bus,obj_path,
+			"/sys/class/i2c-adapter/i2c-5/5-0050/online")
 	root_sensor.add(obj_path,sensor_obj)
 
 	obj_path = OBJ_PATH+"/host/BootCount"
