Obey LDFLAGS

Signed-off-by: Christopher Larson <chris_larson@mentor.com>
Upstream-Status: Pending

--- hostap-utils-0.4.7.orig/Makefile
+++ hostap-utils-0.4.7/Makefile
@@ -14,17 +14,17 @@ prism2_srec.o: prism2_srec.c util.h
 hostap_crypt_conf: hostap_crypt_conf.c

 hostap_diag: hostap_diag.o util.o
-	$(CC) -o hostap_diag $(CFLAGS) hostap_diag.o util.o
+	$(CC) -o hostap_diag $(CFLAGS) $(LDFLAGS) hostap_diag.o util.o

 hostap_diag.o: hostap_diag.c util.h

 util.o: util.c util.h

 hostap_io_debug: hostap_io_debug.c
-	$(CC) -o hostap_io_debug $(CFLAGS) hostap_io_debug.c
+	$(CC) -o hostap_io_debug $(CFLAGS) $(LDFLAGS) hostap_io_debug.c

 hostap_rid: hostap_rid.o util.o
-	$(CC) -o hostap_rid $(CFLAGS) hostap_rid.o util.o
+	$(CC) -o hostap_rid $(CFLAGS) $(LDFLAGS) hostap_rid.o util.o

 hostap_rid.o: hostap_rid.c util.h
