Upstream-Status: Pending

Subject: gen.c: extend DIRNAMESIZE from 256 to 512

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 lib/dns/gen.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: bind-9.11.3/lib/dns/gen.c
===================================================================
--- bind-9.11.3.orig/lib/dns/gen.c
+++ bind-9.11.3/lib/dns/gen.c
@@ -130,7 +130,7 @@ static const char copyright[] =
 #define TYPECLASSBUF (TYPECLASSLEN + 1)
 #define TYPECLASSFMT "%" STR(TYPECLASSLEN) "[-0-9a-z]_%d"
 #define ATTRIBUTESIZE 256
-#define DIRNAMESIZE 256
+#define DIRNAMESIZE 512
 
 static struct cc {
 	struct cc *next;
