From fd49cbcb7f3402e1a4a99b71b901a564060039b4 Mon Sep 17 00:00:00 2001
From: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
Date: Fri, 24 May 2024 14:16:21 +0300
Subject: [PATCH] Fix building with GCC 14

Signed-off-by: Dmitry Baryshkov <dmitry.baryshkov@linaro.org>
Upstream-Status: Submitted [https://github.com/lm-sensors/lm-sensors/pull/497]
---
 prog/sensord/rrd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/prog/sensord/rrd.c b/prog/sensord/rrd.c
index a531362bddd5..36d685fce514 100644
--- a/prog/sensord/rrd.c
+++ b/prog/sensord/rrd.c
@@ -299,7 +299,7 @@ int rrdInit(void)
 		argv[argc++] = rraBuff;
 		argv[argc] = NULL;
 
-		ret = rrd_create(argc, (char**) argv);
+		ret = rrd_create(argc, (const char**) argv);
 		if (ret == -1) {
 			sensorLog(LOG_ERR, "Error creating RRD file: %s: %s",
 				  sensord_args.rrdFile, rrd_get_error());
@@ -455,7 +455,7 @@ int rrdUpdate(void)
 		const char *argv[] = {
 			"sensord", sensord_args.rrdFile, rrdBuff, NULL
 		};
-		if ((ret = rrd_update(3, (char **) /* WEAK */ argv))) {
+		if ((ret = rrd_update(3, (const char **) /* WEAK */ argv))) {
 			sensorLog(LOG_ERR, "Error updating RRD file: %s: %s",
 				  sensord_args.rrdFile, rrd_get_error());
 		}
-- 
2.39.2

