From bad93f7aec96b1819f4027dbc06b337d2e2227f2 Mon Sep 17 00:00:00 2001
From: Lei Maohui <leimaohui@fujitsu.com>
Date: Thu, 15 Sep 2022 16:35:39 +0900
Subject: [PATCH] Set header file to a fixed path instead of a host path.

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Lei Maohui <leimaohui@fujitsu.com>
---
 .../tracker-miners-enum-types.c.template                        | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libtracker-miners-common/tracker-miners-enum-types.c.template b/src/libtracker-miners-common/tracker-miners-enum-types.c.template
index 4db7ef1..7c44c13 100644
--- a/src/libtracker-miners-common/tracker-miners-enum-types.c.template
+++ b/src/libtracker-miners-common/tracker-miners-enum-types.c.template
@@ -7,7 +7,7 @@
 
 /*** BEGIN file-production ***/
 /* enumerations from "@basename@" */
-#include "@filename@"
+#include "tracker-enums.h"
 /*** END file-production ***/
 
 
