Subject: [PATCH] Search for cython3
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate

Signed-off-by: Markus Volk <f_l_k@gmx.net>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/module/meson.build        2022-02-10 08:49:35.000000000 +0100
+++ b/module/meson.build        2022-03-29 12:58:03.456193737 +0200
@@ -1,4 +1,4 @@
-cython = find_program('cython', required: true)
+cython = find_program('cython3', required: true)
 
 blueman_c = custom_target(
     'blueman_c',

-- 
2.14.3

