From cfee58d5863a535b61aa54690ae205b876f57944 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 2 Feb 2018 22:53:29 -0800
Subject: [PATCH 1/2] libopeniscsiusr: Include limit.h for PATH_MAX

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 libopeniscsiusr/iface.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libopeniscsiusr/iface.c b/libopeniscsiusr/iface.c
index 79898df..a48ef36 100644
--- a/libopeniscsiusr/iface.c
+++ b/libopeniscsiusr/iface.c
@@ -30,6 +30,7 @@
 #include <netdb.h>
 #include <assert.h>
 #include <inttypes.h>
+#include <limits.h>
 
 #include "libopeniscsiusr/libopeniscsiusr.h"
 #include "misc.h"
-- 
2.16.1

