From 2b39f85dcf020647544002cb0b0e734748391dfb Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 2 Feb 2018 23:27:25 -0800
Subject: [PATCH 2/4] idbm.c: Include fcnl.h for O_RDWR and O_CREAT definitions

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 usr/idbm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/usr/idbm.c b/usr/idbm.c
index 5532202..0a51b85 100644
--- a/usr/idbm.c
+++ b/usr/idbm.c
@@ -27,6 +27,7 @@
 #include <errno.h>
 #include <dirent.h>
 #include <limits.h>
+#include <fcntl.h>
 #include <sys/stat.h>
 #include <sys/file.h>
 
-- 
2.16.1

