From 23b068e695881be0e8205ecccadf775fc3d5889d Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 6 Sep 2022 21:25:50 -0700
Subject: [PATCH] utility: Include time.h form time() and strftime() prototypes

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 telnetd/utility.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/telnetd/utility.c b/telnetd/utility.c
index 4811f14..b539777 100644
--- a/telnetd/utility.c
+++ b/telnetd/utility.c
@@ -40,6 +40,7 @@ char util_rcsid[] =
 #define PRINTOPTIONS
 
 #include <stdarg.h>
+#include <time.h> /* for time() anf strftime() */
 #include <sys/utsname.h>
 
 #ifdef AUTHENTICATE
-- 
2.37.3

