gdft.c: remove functions that cause warnings about being unused

Upstream-Status: Backport (Fixed in master branch)

Signed-off-by: Derek Straka <derek@asterius.io>
--- git/src/gdft.c.orig	2016-08-12 10:20:41.418440091 -0400
+++ git/src/gdft.c	2016-08-12 10:22:34.226442987 -0400
@@ -139,16 +139,6 @@
 }
 #else
 
-#ifndef HAVE_LIBFONTCONFIG
-static char * font_pattern(char **fontpath, char *fontpattern)
-{
-	(void)fontpath;
-	(void)fontpattern;
-
-	return "libgd was not built with FontConfig support\n";
-}
-#endif /* HAVE_LIBFONTCONFIG */
-
 #include "gdcache.h"
 /* 2.0.16 Christophe Thomas: starting with FreeType 2.1.6, this is
   mandatory, and it has been supported for a long while. */
@@ -446,16 +436,15 @@
 	return (strcmp (a->fontlist, b->fontlist) == 0 && a->flags == b->flags);
 }
 
+#ifdef HAVE_LIBFONTCONFIG
 static int useFontConfig(int flag)
 {
-#ifdef HAVE_LIBFONTCONFIG
 	if (fontConfigFlag) {
 		return (!(flag & gdFTEX_FONTPATHNAME));
 	}
-#endif
 	return flag & gdFTEX_FONTCONFIG;
 }
-
+#endif
 static void *
 fontFetch (char **error, void *key)
 {
