From 8eca28eddcda4ce8a345ca031f43ff1ed6f37089 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 2 Mar 2015 02:27:55 +0000
Subject: [PATCH 1/9] make man install relative to DESTDIR

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 sim/common/Make-common.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sim/common/Make-common.in b/sim/common/Make-common.in
index 74e5dad3049..9e95c224ba4 100644
--- a/sim/common/Make-common.in
+++ b/sim/common/Make-common.in
@@ -70,7 +70,7 @@ tooldir = $(libdir)/$(target_alias)
 datadir = @datadir@
 datarootdir = @datarootdir@
 mandir = @mandir@
-man1dir = $(mandir)/man1
+man1dir = $(DESTDIR)$(mandir)/man1
 infodir = @infodir@
 includedir = @includedir@
 
-- 
2.36.1

