From a6d3df9d84463ec500d9d01558ab8a17e824e9e5 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 2 Mar 2015 02:27:55 +0000
Subject: [PATCH 01/10] make man install relative to DESTDIR

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 sim/common/Make-common.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sim/common/Make-common.in b/sim/common/Make-common.in
index 8c2cacea1e5..4a52eef2203 100644
--- a/sim/common/Make-common.in
+++ b/sim/common/Make-common.in
@@ -63,7 +63,7 @@ tooldir = $(libdir)/$(target_alias)
 datadir = @datadir@
 datarootdir = @datarootdir@
 mandir = @mandir@
-man1dir = $(mandir)/man1
+man1dir = $(DESTDIR)$(mandir)/man1
 infodir = @infodir@
 includedir = @includedir@
 
-- 
2.33.1

