From 9a85132c4ba7d37a5df146239b3ab1a5854ce478 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 22 Feb 2023 16:24:07 -0800
Subject: [PATCH] gdbserver/linux-low.cc: Fix a typo in ternary operator

Upstream-Status: Submitted [https://sourceware.org/pipermail/gdb-patches/2023-February/197298.html]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 gdbserver/linux-low.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gdbserver/linux-low.cc b/gdbserver/linux-low.cc
index 7e1de397893..95ec871d436 100644
--- a/gdbserver/linux-low.cc
+++ b/gdbserver/linux-low.cc
@@ -5390,7 +5390,7 @@ proc_xfer_memory (CORE_ADDR memaddr, unsigned char *readbuf,
       if (lseek (fd, memaddr, SEEK_SET) != -1)
 	bytes = (readbuf != nullptr
 		 ? read (fd, readbuf, len)
-		 ? write (fd, writebuf, len));
+		 : write (fd, writebuf, len));
 #endif
 
       if (bytes < 0)
