--- a/purgatory/arch/i386/pic.c
+++ b/purgatory/arch/i386/pic.c
@@ -16,7 +16,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
-#include <sys/io.h>
+#include "include/arch/io.h"
 #include <purgatory.h>
 #include "purgatory-x86.h"
 

--- a/purgatory/arch/i386/vga.c
+++ b/purgatory/arch/i386/vga.c
@@ -1,4 +1,4 @@
-#include <sys/io.h>
+#include "include/arch/io.h"
 #include <purgatory.h>
 #include "purgatory-x86.h"
 

