Upstream-Status: Inappropriate [configuration]

remove ms lib options when cross compiling

Signed-Off-By: Armin Kuster <akuster808@gmail.com>

Index: configure.ac
===================================================================
--- a.orig/configure.ac
+++ a/configure.ac
@@ -47,7 +47,6 @@ if test x"${cross_compiling}" = "xno" ;
   AC_MSG_NOTICE([ LDFLAGS = ${LDFLAGS} ])
 else
   AC_MSG_NOTICE([Cross Compiling --- will not update CPPFALGS or LDFLAGS with /usr/local, /opt/local or /sw])
-  LIBS="$LIBS -lws2_32 -lgdi32"
 fi
 
 if test -r /bin/uname.exe ; then
