---
 setup.py |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/setup.py
+++ b/setup.py
@@ -32,8 +32,8 @@
 extens = [Extension('backports/lzma/_lzma',
                     ['backports/lzma/_lzmamodule.c'],
                     libraries = ['lzma'],
-                    include_dirs = [os.path.join(home, 'include'), '/opt/local/include', '/usr/local/include'],
-                    library_dirs = [os.path.join(home, 'lib'), '/opt/local/lib', '/usr/local/lib']
+                    include_dirs = [],
+                    library_dirs = []
                     )]
 
 descr = "Backport of Python 3.3's 'lzma' module for XZ/LZMA compressed files."
