From 2f127ebe425c97b0641fe1bc73247e91e18c2be0 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Thu, 11 Nov 2021 22:06:16 +0100
Subject: [PATCH] libtextstyle: fix builds with automake 1.16.4 and newer

Upstream-Status: Submitted [http://savannah.gnu.org/bugs/?59929]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>
---
 libtextstyle/configure.ac | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/libtextstyle/configure.ac b/libtextstyle/configure.ac
index 63b4c2a..cab9025 100644
--- a/libtextstyle/configure.ac
+++ b/libtextstyle/configure.ac
@@ -17,11 +17,10 @@ dnl along with this program.  If not, see <https://www.gnu.org/licenses/>.
 dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ([2.63])
-AC_INIT
+AC_INIT([libtextstyle],
+    [0.20.1])
 AC_CONFIG_SRCDIR([version.sh])
 AC_CONFIG_AUX_DIR([build-aux])
-. $srcdir/version.sh
-gl_INIT_PACKAGE([libtextstyle], [$VERSION_NUMBER])
 AM_INIT_AUTOMAKE([1.13 silent-rules])
 AM_CONFIG_HEADER([config.h])
 
