From 68b2fb63c8df61d1480121a859f8c955f4910c01 Mon Sep 17 00:00:00 2001
From: Alexander Sosedkin <asosedkin@redhat.com>
Date: Thu, 30 Apr 2026 13:08:01 +0200
Subject: [PATCH] tests/mini-dtls-framents: link to gnulib

Signed-off-by: Alexander Sosedkin <asosedkin@redhat.com>

Upstream-Status: Backport [https://github.com/gnutls/gnutls/commit/68b2fb63c8df61d1480121a859f8c955f4910c01]
Signed-off-by: Peter Marko <peter.marko@siemens.com>
---
 tests/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index f8797964d..1b27df751 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -524,6 +524,7 @@ pathbuf_CPPFLAGS = $(AM_CPPFLAGS) \
 mini_dtls_fragments_CPPFLAGS = $(AM_CPPFLAGS) \
 	-I$(top_srcdir)/gl	\
 	-I$(top_builddir)/gl
+mini_dtls_fragments_LDADD = $(LDADD) ../gl/libgnu.la
 
 if ENABLE_PKCS11
 if !WINDOWS
