Upstream-Status: Pending

diff -ruN ACE_wrappers1/ace/config.h ACE_wrappers/ace/config.h
--- ACE_wrappers1/ace/config.h	1970-01-01 02:00:00.000000000 +0200
+++ ACE_wrappers/ace/config.h	2019-01-10 13:36:52.698012477 +0200
@@ -0,0 +1,1 @@
+#include "ace/config-linux.h"
diff -ruN ACE_wrappers1/include/makeinclude/platform_macros.GNU ACE_wrappers/include/makeinclude/platform_macros.GNU
--- ACE_wrappers1/include/makeinclude/platform_macros.GNU	1970-01-01 02:00:00.000000000 +0200
+++ ACE_wrappers/include/makeinclude/platform_macros.GNU	2018-10-31 15:11:18.265392460 +0200
@@ -0,0 +1,2 @@
+INSTALL_PREFIX = $(D)$(exec_prefix)
+include $(ACE_ROOT)/include/makeinclude/platform_linux.GNU
