diff --git a/arch/arm/boot/dts/nuvoton-npcm750-runbmc-olympus.dts b/arch/arm/boot/dts/nuvoton-npcm750-runbmc-olympus.dts
index b9d6cb0ffb72..0ffcb8015808 100644
--- a/arch/arm/boot/dts/nuvoton-npcm750-runbmc-olympus.dts
+++ b/arch/arm/boot/dts/nuvoton-npcm750-runbmc-olympus.dts
@@ -81,6 +81,13 @@
 			no-map;
 		};
 
+		video_memory: framebuffer@0x1b000000 {
+			compatible = "shared-dma-pool";
+			reg = <0x1b000000 0x02000000>;
+			reusable;
+			linux,cma-default;
+		};
+
 	};
 
 	iio-hwmon {
@@ -989,15 +996,20 @@
 };
 
 &vcd {
-	status = "okay";
+	status = "disabled";
 	memory-region = <&vcd_memory>, <&gfx_memory>;
 };
 
 &ece {
-	status = "okay";
+	status = "disabled";
 	memory-region = <&ece_memory>;
 };
 
+&video {
+	memory-region = <&video_memory>;
+	status = "okay";
+};
+
 &watchdog1 {
 	nuvoton,ext1-reset-type = "wd1";
 	status = "okay";
