Index: rssyl-0.34/configure.ac
===================================================================
--- rssyl-0.34.orig/configure.ac	2014-07-18 07:37:28.981532959 +0000
+++ rssyl-0.34/configure.ac	2014-07-18 07:39:45.881536690 +0000
@@ -1,5 +1,13 @@
+
+dnl plugin version
+m4_define([plugin_major_version],[0])
+m4_define([plugin_minor_version],[34])
+m4_define([plugin_micro_version],[0])
+m4_define([plugin_version], [plugin_major_version.plugin_minor_version.plugin_micro_version])
+
 AC_PREREQ(2.60)
-AC_INIT(src/plugin.c)
+AC_INIT(rssyl, plugin_version)
+AC_CONFIG_SRCDIR(src/plugin.c)
 AC_CONFIG_AUX_DIR(config)
 AM_MAINTAINER_MODE
 AC_CONFIG_HEADERS(config.h)
@@ -7,28 +15,15 @@
 PACKAGE=rssyl
 
 dnl plugin version
-MAJOR_VERSION=0
-MINOR_VERSION=34
-MICRO_VERSION=0
+MAJOR_VERSION=plugin_major_version
+MINOR_VERSION=plugin_minor_version
+MICRO_VERSION=plugin_micro_version
 EXTRA_VERSION=0
-
-if test \( $EXTRA_VERSION -eq 0 \); then
-    if test \( $MICRO_VERSION -eq 0 \); then
-        VERSION=${MAJOR_VERSION}.${MINOR_VERSION} 
-    else
-		VERSION=${MAJOR_VERSION}.${MINOR_VERSION}.${MICRO_VERSION}
-    fi
-else
-    if test \( $MICRO_VERSION -eq 0 \); then
-        VERSION=${MAJOR_VERSION}.${MINOR_VERSION}cvs${EXTRA_VERSION}
-    else
-		VERSION=${MAJOR_VERSION}.${MINOR_VERSION}.${MICRO_VERSION}cvs${EXTRA_VERSION}
-    fi
-fi
+VERSION=plugin_version
 
 AC_CANONICAL_SYSTEM
 
-AM_INIT_AUTOMAKE($PACKAGE, $VERSION, no-define)
+AM_INIT_AUTOMAKE([foreign no-define])
 AC_DEFINE_UNQUOTED(PLUGINVERSION, "$VERSION", [plugin version])
 
 AC_PROG_CC
