From a73bbed89fb049ac424c211629935b26013e2573 Mon Sep 17 00:00:00 2001
From: "Peter A. Bigot" <pab@pabigot.com>
Date: Wed, 14 Nov 2018 09:19:51 -0600
Subject: [PATCH] bthelper: correct path for hciconfig under Yocto

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Peter A. Bigot <pab@pabigot.com>
Signed-off-by: Andrei Gherzan <andrei@gherzan.ro>

---
 usr/bin/bthelper | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/usr/bin/bthelper b/usr/bin/bthelper
index 2133fbc..1fc9245 100755
--- a/usr/bin/bthelper
+++ b/usr/bin/bthelper
@@ -12,8 +12,8 @@ fi
 dev=$1
 
 # Need to bring hci up before looking at MAC as it can be all zeros during init
-/bin/hciconfig $dev up
-if ! /bin/hciconfig $dev | grep -q "Bus: UART"; then
+/usr/bin/hciconfig $dev up
+if ! /usr/bin/hciconfig $dev | grep -q "Bus: UART"; then
     echo Not a UART-attached BT Modem
     exit 0
 fi
@@ -26,7 +26,7 @@ if ( /usr/bin/hcitool -i $dev dev | grep -q -E '\s43:4[35]:' ); then
     BDADDR=`printf '0x%02x 0x%02x 0x%02x 0xeb 0x27 0xb8' $((0x$B3 ^ 0xaa)) $((0x$B2 ^ 0xaa)) $((0x$B1 ^ 0xaa))`
 
     /usr/bin/hcitool -i $dev cmd 0x3f 0x001 $BDADDR
-    /bin/hciconfig $dev reset
+    /usr/bin/hciconfig $dev reset
 else
     echo Raspberry Pi BDADDR already set
 fi
-- 
2.31.1

