From 874679725c443e5d73e3a33d1158ab25442c8a5f Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Tue, 10 Jan 2017 14:11:30 +0200
Subject: [PATCH] Do not read config files from $HOME

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 lib/rpmrc.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/lib/rpmrc.c b/lib/rpmrc.c
index 5d778b8b7..dc8d42aeb 100644
--- a/lib/rpmrc.c
+++ b/lib/rpmrc.c
@@ -485,8 +485,7 @@ static void setDefaults(void)
     if (!defrcfiles) {
 	defrcfiles = rstrscat(NULL, confdir, "/rpmrc", ":",
 				confdir, "/" RPM_VENDOR "/rpmrc", ":",
-				SYSCONFDIR "/rpmrc", ":",
-				userrc, NULL);
+				SYSCONFDIR "/rpmrc", NULL);
     }
 
     /* macrofiles may be pre-set from --macros */
@@ -498,8 +497,7 @@ static void setDefaults(void)
 				confdir, "/" RPM_VENDOR "/macros", ":",
 				SYSCONFDIR "/rpm/macros.*", ":",
 				SYSCONFDIR "/rpm/macros", ":",
-				SYSCONFDIR "/rpm/%{_target}/macros", ":",
-				usermacros, NULL);
+				SYSCONFDIR "/rpm/%{_target}/macros", NULL);
     }
 
     free(usermacros);
