From c5f012750f8102ff54af73ccc2d2b7bfa1f26db4 Mon Sep 17 00:00:00 2001
From: Darren Hart <dvhart@linux.intel.com>
Date: Fri, 29 Apr 2022 13:32:28 +0000
Subject: [PATCH] Add an input group for the /dev/input/* devices

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Darren Hart <dvhart@linux.intel.com>
---
 group.master | 1 +
 1 file changed, 1 insertion(+)

diff --git a/group.master b/group.master
index 1b5e2fb..cea9d60 100644
--- a/group.master
+++ b/group.master
@@ -12,6 +12,7 @@ uucp:*:10:
 man:*:12:
 proxy:*:13:
 kmem:*:15:
+input:*:19:
 dialout:*:20:
 fax:*:21:
 voice:*:22:
