From c6e5c31cd276f69a4db479188efb3ab3e0f35dd8 Mon Sep 17 00:00:00 2001
From: Jacob Kroon <jacob.kroon@gmail.com>
Date: Wed, 30 Jan 2019 04:53:48 +0000
Subject: [PATCH] Add kvm group

Upstream-Status: Pending
Signed-off-by: Jacob Kroon <jacob.kroon@gmail.com>
---
 group.master | 1 +
 1 file changed, 1 insertion(+)

diff --git a/group.master b/group.master
index 7d794b2..7fdd8bf 100644
--- a/group.master
+++ b/group.master
@@ -33,6 +33,7 @@ utmp:*:43:
 video:*:44:
 sasl:*:45:
 plugdev:*:46:
+kvm:*:47:
 staff:*:50:
 games:*:60:
 shutdown:*:70:
