From 6355278b9f744291864c373a32a8da8f84aaaf37 Mon Sep 17 00:00:00 2001
From: Jacob Kroon <jacob.kroon@gmail.com>
Date: Wed, 30 Jan 2019 04:53:48 +0000
Subject: [PATCH] Add kvm group

Upstream-Status: Pending
Signed-off-by: Jacob Kroon <jacob.kroon@gmail.com>
---
 group.master | 1 +
 1 file changed, 1 insertion(+)

diff --git a/group.master b/group.master
index cea9d60..5b62284 100644
--- a/group.master
+++ b/group.master
@@ -34,6 +34,7 @@ utmp:*:43:
 video:*:44:
 sasl:*:45:
 plugdev:*:46:
+kvm:*:47:
 staff:*:50:
 games:*:60:
 shutdown:*:70:
