From 06605158852f9364519391fa11070ba5ec4303e9 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 2 Sep 2017 08:07:17 -0700
Subject: [PATCH] disbale heap checkers and debug allocator on musl

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-Status: Pending

---
 configure.ac | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure.ac b/configure.ac
index 68fd51c..4cb71fe 100644
--- a/configure.ac
+++ b/configure.ac
@@ -53,6 +53,8 @@ case "$host" in
              need_nanosleep=no;;
    *-cygwin*) default_enable_cpu_profiler=no;;
    *-linux*) default_enable_heap_checker=yes; heap_checker_supported=yes;;
+   *-musl*) default_enable_heap_checker=no; default_enable_heap_profiler=no;
+            default_enable_debugalloc=no; default_enable_libunwind=no;
 esac
 
 # Currently only backtrace works on s390 and OSX.
