From 81b4fbb5f52044cb348534c23f10b3884972b09b Mon Sep 17 00:00:00 2001
From: Beat Schaer <beat.schaer@wabtec.com>
Date: Fri, 19 Mar 2021 08:18:58 +0100
Subject: [PATCH] Fixed includes so that it compiles on Ubuntu 20.04

---
 logic.c | 3 +--
 main.c  | 3 ++-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/logic.c b/logic.c
index 97767f5..47ebfaa 100644
--- a/logic.c
+++ b/logic.c
@@ -7,8 +7,7 @@
 #include <string.h>
 #include <errno.h>
 #include <unistd.h>
-#include <attr/xattr.h>
-#include <attr/attributes.h>
+#include <sys/xattr.h>
 #include <fts.h>
 #include <libgen.h>
 #include "logic.h"
diff --git a/main.c b/main.c
index aa11239..f462b98 100644
--- a/main.c
+++ b/main.c
@@ -12,7 +12,8 @@
 #include <linux/limits.h>
 #include <stdbool.h>
 #include <sys/stat.h>
-#include <attr/xattr.h>
+#include <sys/xattr.h>
+#include <errno.h>
 #ifndef _SYS_STAT_H
   #include <linux/stat.h>
 #endif
-- 
2.25.1

