From 3d266a5d77c908566b1a541f3ff8b717f17a9a8a Mon Sep 17 00:00:00 2001
From: Jon Mason <jdmason@kudzu.us>
Date: Mon, 19 Dec 2022 11:36:04 -0500
Subject: [PATCH] configs: vexpress: modify to boot compressed initramfs

Signed-off-by: Jon Mason <jdmason@kudzu.us>
Upstream-Status: Inappropriate
---
 include/configs/vexpress_aemv8.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/configs/vexpress_aemv8.h b/include/configs/vexpress_aemv8.h
index 24d8ca086653..0935461340b7 100644
--- a/include/configs/vexpress_aemv8.h
+++ b/include/configs/vexpress_aemv8.h
@@ -198,8 +198,10 @@
 #define VEXPRESS_RAMDISK_ADDR		0x8fe00000
 
 #define EXTRA_ENV_NAMES							\
+	        "kernel_comp_addr_r=0x90000000\0"			\
+		"kernel_comp_size=0x3000000\0"				\
 		"kernel_name=norkern\0"					\
-		"kernel_alt_name=Image\0"				\
+		"kernel_alt_name=Image.gz\0"				\
 		"ramdisk_name=ramdisk.img\0"				\
 		"fdtfile=board.dtb\0"					\
 		"fdt_alt_name=juno\0"
