From e81feec1ebbc676220cd8759d83f46335602d079 Mon Sep 17 00:00:00 2001
From: Lei Maohui <leimaohui@fujitsu.com>
Date: Fri, 7 Jan 2022 14:37:36 +0900
Subject: [PATCH] Fix build error as following:

*** No rule to make target 'dstat.1', needed by 'docs'.  Stop.

Upstream-Status: Pending

Signed-off-by: Lei Maohui <leimaohui@fujitsu.com>
---
 docs/Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/docs/Makefile b/docs/Makefile
index c031e82..1a85e61 100644
--- a/docs/Makefile
+++ b/docs/Makefile
@@ -11,13 +11,13 @@ all:
 
 dist: docs
 
-docs: dstat.1 $(htmltargets)
+docs: dool.1 $(htmltargets)
 
-install: dstat.1
-	install -Dp -m0644 dstat.1 $(DESTDIR)$(mandir)/man1/dstat.1
+install: dool.1
+	install -Dp -m0644 dool.1 $(DESTDIR)$(mandir)/man1/dool.1
 
 clean:
-	rm -f dstat.1 *.html *.xml
+	rm -f dool.1 *.html *.xml
 
 %.1.html: %.1.adoc
 	asciidoc -d manpage $<
-- 
2.25.1

