Upstream-Status: Pending

Signed-off-by: Scott Garman <scott.a.garman@intel.com>

--- man-1.6e.orig/configure	2006-05-01 14:56:14.000000000 -0300
+++ man-1.6e/configure	2007-05-18 14:35:43.000000000 -0300
@@ -960,7 +960,7 @@
 
 # What sections do we anticipate?
 
-tmpsections="1 1p 8 2 3 3p 4 5 6 7 9 0p tcl n l p o"
+tmpsections="1 1p 8 2 3 3p 3pm 4 5 6 7 9 0p tcl n l p o"
 
 if [ x$default = x ]; then
   echo ""
