Upstream-Status: Pending

Signed-off-by: Scott Garman <scott.a.garman@intel.com>

--- man-1.5j/src/man.conf.in.nocache	Thu Nov 22 15:07:12 2001
+++ man-1.5j/src/man.conf.in	Thu Nov 22 15:07:12 2001
@@ -81,6 +81,9 @@
 # 
 #NOCACHE
 #
+# NOCACHE keeps man from creating cache pages
+NOCACHE
+#
 # Useful paths - note that COL should not be defined when
 # NROFF is defined as "groff -Tascii" or "groff -Tlatin1";
 # not only is it superfluous, but it actually damages the output.
