From 47ede0950740e8919c28774840e77f223d5f3124 Mon Sep 17 00:00:00 2001
From: Derek Straka <derek@asterius.io>
Date: Sun, 11 Mar 2018 19:55:38 -0400
Subject: [PATCH] py-scrypt: remove the hard coded include paths

Upstream-Status: Pending

Signed-off-by: Derek Straka <derek@asterius.io>
---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index 419a982..3f1fe52 100644
--- a/setup.py
+++ b/setup.py
@@ -26,7 +26,6 @@ if sys.platform.startswith('linux'):
         ('_FILE_OFFSET_BITS', '64'),
     ]
     libraries = ['crypto', 'rt']
-    includes = ['/usr/local/include', '/usr/include']
     CFLAGS.append('-O2')
 elif sys.platform.startswith('win32') and os.environ.get('MSYSTEM'):
     msys2_env = os.getenv('MSYSTEM')
