From dd55633e49aa43dede3c8e1770ae8761487f050e Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 6 Aug 2022 21:52:31 -0700
Subject: [PATCH] s390x-se-luks-gencpio: There is no bashism

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/libostree/s390x-se-luks-gencpio | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libostree/s390x-se-luks-gencpio b/src/libostree/s390x-se-luks-gencpio
index e821e2fe..96c1d123 100755
--- a/src/libostree/s390x-se-luks-gencpio
+++ b/src/libostree/s390x-se-luks-gencpio
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # This script creates new initramdisk with LUKS config within
 set -euo pipefail
 
-- 
2.37.1

