From faadbc147394cea9278bc2494f33e3521d88e742 Mon Sep 17 00:00:00 2001
From: Tim Orling <timothy.t.orling@linux.intel.com>
Date: Tue, 9 Jan 2018 18:38:45 -0800
Subject: [PATCH] stress-fcntl: fix build for musl

Signed-off-by: Tim Orling <timothy.t.orling@linux.intel.com>

Upstream-status: Pending

---
 stress-fcntl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/stress-fcntl.c b/stress-fcntl.c
index 2ea867b..e91b345 100644
--- a/stress-fcntl.c
+++ b/stress-fcntl.c
@@ -415,6 +415,7 @@ ofd_lock_abort:	{ /* Nowt */ }
 #if (defined(F_GET_FILE_RW_HINT) && defined(F_SET_FILE_RW_HINT)) | \
     (defined(F_GET_RW_HINT) && defined(F_SET_RW_HINT))
 	{
+		int ret;
 		size_t i;
 		unsigned long hint;
 		static const unsigned long hints[] = {
