From 9b7442069eb68a42d2437181c1b2e710dd077e8b Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 29 Mar 2013 08:48:10 +0400
Subject: [PATCH 08/46] missing-execinfo_h

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-Status: Pending
---
 boehm-gc/include/gc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/boehm-gc/include/gc.h b/boehm-gc/include/gc.h
index 6b38f2d..fca98ff 100644
--- a/boehm-gc/include/gc.h
+++ b/boehm-gc/include/gc.h
@@ -503,7 +503,7 @@ GC_API GC_PTR GC_malloc_atomic_ignore_off_page GC_PROTO((size_t lb));
 #if defined(__linux__) || defined(__GLIBC__)
 # include <features.h>
 # if (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 1 || __GLIBC__ > 2) \
-     && !defined(__ia64__)
+     && !defined(__ia64__) && !defined(__UCLIBC__)
 #   ifndef GC_HAVE_BUILTIN_BACKTRACE
 #     define GC_HAVE_BUILTIN_BACKTRACE
 #   endif
-- 
2.6.3

