From 6ec375eaafc43a2b3c30a0e0e49447d231d81a67 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 9 May 2017 00:57:10 -0700

---
Upstream-Status: Pending

 Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index 8568dd2..5efc91e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,6 +3,8 @@ ACLOCAL_AMFLAGS = -I acinclude
 
 lib_LTLIBRARIES = libSDL2_ttf.la
 
+AUTOMAKE_OPTIONS = foreign
+
 libSDL2_ttfincludedir = $(includedir)/SDL2
 libSDL2_ttfinclude_HEADERS =	\
 	SDL_ttf.h
