From b927b79bff12ebee65b87d3ab1b2eb29519938af Mon Sep 17 00:00:00 2001
From: Adriana Kobylak <anoo@us.ibm.com>
Date: Tue, 8 Aug 2017 12:20:39 -0500
Subject: [PATCH 3/5] Add CONFIG_FLASH_CFI_MTD

---
 include/configs/ast-g5-ncsi.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/ast-g5-ncsi.h b/include/configs/ast-g5-ncsi.h
index 7e40193..31b1de2 100644
--- a/include/configs/ast-g5-ncsi.h
+++ b/include/configs/ast-g5-ncsi.h
@@ -28,6 +28,8 @@
 
 #define CONFIG_HW_WATCHDOG
 
+#define CONFIG_FLASH_CFI_MTD
+
 #define CONFIG_CMD_UBI
 #define CONFIG_CMD_UBIFS
 #define CONFIG_RBTREE
-- 
1.8.2.2

