From c8be0d5e7d17930d819635df6570fa5331a5dedd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Mon, 10 Jan 2022 20:18:00 +0100
Subject: [PATCH] Harcode shebang to avoid pointing to build system's perl
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Pending

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>
---
 src/gdialog.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gdialog.in b/src/gdialog.in
index 2fc4633..f891861 100755
--- a/src/gdialog.in
+++ b/src/gdialog.in
@@ -1,4 +1,4 @@
-#!@PERL@
+#!/usr/bin/env perl
 
 # gdialog -> zenity conversion wrapper
 #
-- 
2.31.1

