From 5c4d6d8538994d5fe9b3b46bfafaf0a605e3bda6 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen.kooi@linaro.org>
Date: Tue, 17 Jun 2014 08:18:17 +0200
Subject: [PATCH] configure: use pkg-config for pcre detection

Signed-off-by: Koen Kooi <koen.kooi@linaro.org>
Upstream-Status: pending
---
 configure.ac | 38 +++++++-------------------------------
 1 file changed, 7 insertions(+), 31 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0c984b7..6edcec1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -70,38 +70,14 @@ AC_MSG_RESULT([$with_pcre])
 
 dnl To make configuring easier, check for a locally built PCRE using the Tools/pcre-build.sh script
 if test x"${with_pcre}" = xyes ; then
-  AC_MSG_CHECKING([whether to use local PCRE])
-  local_pcre_config=no
-  if test -z $PCRE_CONFIG; then
-    if test -f `pwd`/pcre/pcre-swig-install/bin/pcre-config; then
-      PCRE_CONFIG=`pwd`/pcre/pcre-swig-install/bin/pcre-config
-      local_pcre_config=$PCRE_CONFIG
-    fi
-  fi
-  AC_MSG_RESULT([$local_pcre_config])
-fi
-AS_IF([test "x$with_pcre" != xno],
-  [AX_PATH_GENERIC([pcre],
-    [], dnl Minimal version of PCRE we need -- accept any
-    [], dnl custom sed script for version parsing is not needed
-    [AC_DEFINE([HAVE_PCRE], [1], [Define if you have PCRE library])
-     LIBS="$LIBS $PCRE_LIBS"
-     CPPFLAGS="$CPPFLAGS $PCRE_CFLAGS"
-    ],
-    [AC_MSG_FAILURE([
-        Cannot find pcre-config script from PCRE (Perl Compatible Regular Expressions)
-        library package. This dependency is needed for configure to complete,
-        Either:
-        - Install the PCRE developer package on your system (preferred approach).
-        - Download the PCRE source tarball, build and install on your system
-          as you would for any package built from source distribution.
-        - Use the Tools/pcre-build.sh script to build PCRE just for SWIG to statically
-          link against. Run 'Tools/pcre-build.sh --help' for instructions.
-          (quite easy and does not require privileges to install PCRE on your system)
-        - Use configure --without-pcre to disable regular expressions support in SWIG
-          (not recommended).])
-    ])
+  PKG_CHECK_MODULES([PCRE], [libpcre], [
+    AC_DEFINE([HAVE_PCRE], [1], [Define if you have PCRE library])
+    LIBS="$LIBS $PCRE_LIBS"
+    CPPFLAGS="$CPPFLAGS $PCRE_CFLAGS"
+  ], [
+    AC_MSG_WARN([$PCRE_PKG_ERRORS])
   ])
+fi
 
 
 dnl CCache
-- 
1.9.3

