From ebb424eee5599fcc131901c0d82d0bfc0d2f57ab Mon Sep 17 00:00:00 2001
From: "Frank Ch. Eigler" <fche@redhat.com>
Date: Fri, 28 Apr 2023 13:51:27 -0400
Subject: [PATCH] release date fix

Upstream-Status: Backport
---
 NEWS           | 2 +-
 systemtap.spec | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/NEWS b/NEWS
index 695bc990c..46859e304 100644
--- a/NEWS
+++ b/NEWS
@@ -1,4 +1,4 @@
-* What's new in version 4.9, 2023-04-27
+* What's new in version 4.9, 2023-04-28
 
 - ISystemtap: the new interactive systemtap jupyter kernel. This provides
   a simple GUI for writing/running Systemtap scripts in an easy, incremental
diff --git a/systemtap.spec b/systemtap.spec
index 59814f112..fd8023264 100644
--- a/systemtap.spec
+++ b/systemtap.spec
@@ -1307,7 +1307,7 @@ exit 0
 
 # PRERELEASE
 %changelog
-* Thu Apr 27 2023 Frank Ch. Eigler <fche@redhat.com> - 4.9-1
+* Fri Apr 28 2023 Frank Ch. Eigler <fche@redhat.com> - 4.9-1
 - Upstream release, see wiki page below for detailed notes.
   https://sourceware.org/systemtap/wiki/SystemTapReleases
 
-- 
2.40.1

