From aa6cd19ae428769a38fe7d95f98db0a9c19ae90a Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Mon, 25 Feb 2019 15:14:49 +0800
Subject: [PATCH 12/24] fix missing ULONG_LONG_MAX definition in case of musl

Upstream-Status: Inappropriate [musl]

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 src/basic/missing_type.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/basic/missing_type.h b/src/basic/missing_type.h
index 23602eb..7d7c1e4 100644
--- a/src/basic/missing_type.h
+++ b/src/basic/missing_type.h
@@ -39,3 +39,7 @@ typedef int (*__compar_fn_t)(const void *, const void *);
 #ifndef FTW_SKIP_SIBLINGS
 #define FTW_SKIP_SIBLINGS 3
 #endif
+
+#ifndef ULONG_LONG_MAX
+#define ULONG_LONG_MAX ULLONG_MAX
+#endif
-- 
2.7.4

