From 85fd3b59941072e8f90b73b55ef1d2e3fefeb55e Mon Sep 17 00:00:00 2001
From: Tim Lee <timlee660101@gmail.com>
Date: Wed, 4 Jan 2023 13:22:01 +0800
Subject: [PATCH] set HOSTPER bit as 1 for resetting KCS

Signed-off-by: Tim Lee <timlee660101@gmail.com>
---
 common/board_r.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/common/board_r.c b/common/board_r.c
index c835ff8e26..c807cc21f8 100644
--- a/common/board_r.c
+++ b/common/board_r.c
@@ -822,6 +822,23 @@ void board_init_r(gd_t *new_gd, ulong dest_addr)
 #endif
 	gd->flags &= ~GD_FLG_LOG_READY;
 
+	/* Set HOSTPER (BIT8) as 1 for resetting KCS */
+	/* SWRSTC1B/SWRSTC2B/SWRSTC3B register */
+	setbits_le32(0xf080108C, BIT(8));
+	setbits_le32(0xf0801090, BIT(8));
+	setbits_le32(0xf0801094, BIT(8));
+
+	/* WD0RCRB/WD1RCRB/WD2RCRB register */
+	setbits_le32(0xf0801080, BIT(8));
+	setbits_le32(0xf0801084, BIT(8));
+	setbits_le32(0xf0801088, BIT(8));
+
+	/* TIPRSTCB register */
+	setbits_le32(0xf0801098, BIT(8));
+
+	/* CORSTCB register */
+	setbits_le32(0xf080109C, BIT(8));
+
 #ifdef CONFIG_NEEDS_MANUAL_RELOC
 	for (i = 0; i < ARRAY_SIZE(init_sequence_r); i++)
 		init_sequence_r[i] += gd->reloc_off;
-- 
2.17.1

