From d23dd33e373340f6fddf11904839d1a118824401 Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Mon, 1 Dec 2014 02:08:27 -0500
Subject: [PATCH] apache2: allow to disable selinux support

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 configure.in |   14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/configure.in b/configure.in
index df94ee5..8c3ab21 100644
--- a/configure.in
+++ b/configure.in
@@ -466,10 +466,16 @@ getloadavg
 dnl confirm that a void pointer is large enough to store a long integer
 APACHE_CHECK_VOID_PTR_LEN
 
-AC_CHECK_LIB(selinux, is_selinux_enabled, [
-   AC_DEFINE(HAVE_SELINUX, 1, [Defined if SELinux is supported])
-   APR_ADDTO(AP_LIBS, [-lselinux])
-])
+# SELinux support
+AC_ARG_ENABLE(selinux,APACHE_HELP_STRING(--enable-selinux,Enable SELinux support [default=auto]),
+    [],[enable_selinux=auto])
+
+if test x$enable_selinux != xno; then
+    AC_CHECK_LIB(selinux, is_selinux_enabled, [
+        AC_DEFINE(HAVE_SELINUX, 1, [Defined if SELinux is supported])
+        APR_ADDTO(AP_LIBS, [-lselinux])
+    ])
+fi
 
 AC_CACHE_CHECK([for gettid()], ac_cv_gettid,
 [AC_TRY_RUN(#define _GNU_SOURCE
-- 
1.7.9.5

