From 9130d340c37861a59e812e334e2b13577e705a6c Mon Sep 17 00:00:00 2001
From: Wang Mingyu <wangmy@cn.fujitsu.com>
Date: Fri, 25 Dec 2020 15:10:58 +0900
Subject: [PATCH] avoid attr unless wanted

Upstream-Status: Inappropriate [embedded specific]
---
 lib/replace/wscript | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/lib/replace/wscript b/lib/replace/wscript
index 3593eaf..504a1cb 100644
--- a/lib/replace/wscript
+++ b/lib/replace/wscript
@@ -968,8 +968,6 @@ def build(bld):
     if not bld.CONFIG_SET('HAVE_INET_ATON'):     REPLACE_SOURCE += ' inet_aton.c'
     if not bld.CONFIG_SET('HAVE_INET_NTOP'):     REPLACE_SOURCE += ' inet_ntop.c'
     if not bld.CONFIG_SET('HAVE_INET_PTON'):     REPLACE_SOURCE += ' inet_pton.c'
-    if not bld.CONFIG_SET('HAVE_GETXATTR') or bld.CONFIG_SET('XATTR_ADDITIONAL_OPTIONS'):
-                                                 REPLACE_SOURCE += ' xattr.c'
 
     if not bld.CONFIG_SET('HAVE_CLOSEFROM'):
         REPLACE_SOURCE += ' closefrom.c'
@@ -984,7 +982,7 @@ def build(bld):
                       # hide_symbols=bld.BUILTIN_LIBRARY('replace'),
                       private_library=True,
                       provide_builtin_linking=True,
-                      deps='dl attr' + extra_libs)
+                      deps='dl' + extra_libs)
 
     replace_test_cflags = ''
     if bld.CONFIG_SET('HAVE_WNO_FORMAT_TRUNCATION'):
-- 
2.25.1

