--- a/lib/replace/wscript	2015-11-13 16:04:04.000000000 +0100
+++ b/lib/replace/wscript	2015-11-13 16:23:20.000000000 +0100
@@ -837,8 +837,6 @@ def build(bld):
     if not bld.CONFIG_SET('HAVE_INET_ATON'):     REPLACE_SOURCE += ' inet_aton.c'
     if not bld.CONFIG_SET('HAVE_INET_NTOP'):     REPLACE_SOURCE += ' inet_ntop.c'
     if not bld.CONFIG_SET('HAVE_INET_PTON'):     REPLACE_SOURCE += ' inet_pton.c'
-    if not bld.CONFIG_SET('HAVE_GETXATTR') or bld.CONFIG_SET('XATTR_ADDITIONAL_OPTIONS'):
-                                                 REPLACE_SOURCE += ' xattr.c'
 
     if not bld.CONFIG_SET('HAVE_CLOSEFROM'):
         REPLACE_SOURCE += ' closefrom.c'
@@ -852,7 +850,7 @@ def build(bld):
                       # at the moment:
                       # hide_symbols=bld.BUILTIN_LIBRARY('replace'),
                       private_library=True,
-                      deps='crypt dl nsl socket rt attr' + extra_libs)
+                      deps='crypt dl nsl socket rt ' + extra_libs)
 
     replace_test_cflags="-Wno-format-zero-length"
     if bld.CONFIG_SET('HAVE_WNO_FORMAT_TRUNCATION'):
