Upstream-Status: Pending

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@intel.com>
Signed-off-by: Maxin B. John <maxin.john@intel.com>
---
diff -Naur pycurl-7.19.5.2-orig/setup.py pycurl-7.19.5.2/setup.py
--- pycurl-7.19.5.2-orig/setup.py	2015-11-02 15:42:24.000000000 +0200
+++ pycurl-7.19.5.2/setup.py	2015-11-02 17:59:36.121527273 +0200
@@ -154,7 +154,7 @@
         optbuf = ''
         sslhintbuf = ''
         errtext = ''
-        for option in ["--libs", "--static-libs"]:
+        for option in ["--libs"]:
             p = subprocess.Popen((CURL_CONFIG, option),
                 stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             stdout, stderr = p.communicate()
