From 72dfeec0e49478b0bfb471c4155044391bad8e6c Mon Sep 17 00:00:00 2001
From: Ross Burton <ross.burton@arm.com>
Date: Fri, 8 Dec 2023 10:35:25 +0000
Subject: [PATCH] Don't use docdir from environment

Upstream-Status: Submitted [https://gitlab.gnome.org/GNOME/gtk-doc/-/merge_requests/73]
Signed-off-by: Ross Burton <ross.burton@arm.com>
---
 buildsystems/autotools/gtkdocize.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/buildsystems/autotools/gtkdocize.in b/buildsystems/autotools/gtkdocize.in
index 83127bf..76dcbfd 100755
--- a/buildsystems/autotools/gtkdocize.in
+++ b/buildsystems/autotools/gtkdocize.in
@@ -39,7 +39,7 @@ set - $args
 
 # assume working directory if srcdir is not set
 test "$srcdir" || srcdir=.
-test "$docdir" || docdir="$srcdir"
+docdir="$srcdir"
 
 # detect configure script
 no_configure_found=0
