From e30d4c4c557568b01d5adce9df2714de8d3921c2 Mon Sep 17 00:00:00 2001
From: sewenew <sewe.code@gmail.com>
Date: Tue, 15 Apr 2025 22:57:08 +0800
Subject: [PATCH] update cmake_minimum_required to 3.5

Signed-off-by: Moritz Haase <Moritz.Haase@bmw.de>
Upstream-Status: Backport [e30d4c4c557568b01d5adce9df2714de8d3921c2]
---
 CMakeLists.txt      | 2 +-
 test/CMakeLists.txt | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e0a8cf4..9c742f9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 function(GET_VERSION VERSION_PART VERSION_NUM)
     set(VERSION_REGEX "^const int VERSION_${VERSION_PART} = (.+);$")
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 3423956..88aa079 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -1,6 +1,6 @@
 project(test_redis++)
 
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 set(REDIS_PLUS_PLUS_TEST_SOURCES src/sw/redis++/test_main.cpp)
 
