From 2f200ceed289f935b5e7ec230454a22dd76e42b0 Mon Sep 17 00:00:00 2001
From: leimaohui <leimaohui@cn.fujitsu.com>
Date: Mon, 12 Jan 2015 11:52:35 +0800
Subject: [PATCH] crash: add new recipe

---
Upstream-Status: Pending

 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 79aef176..7266e305 100644
--- a/Makefile
+++ b/Makefile
@@ -295,6 +295,7 @@ gdb_patch:
 
 library: ${OBJECT_FILES}
 	ar -rs ${PROGRAM}lib.a ${OBJECT_FILES}
+	${RANLIB} ${PROGRAM}lib.a 
 
 gdb: force
 	rm -f ${GDB_OFILES}
