Add alises for arm64 which is same as aarch64

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Pending

Index: git/SConstruct
===================================================================
--- git.orig/SConstruct
+++ git/SConstruct
@@ -990,6 +990,7 @@ elif endian == "big":
 processor_macros = {
     'arm'     : { 'endian': 'little', 'defines': ('__arm__',) },
     'aarch64' : { 'endian': 'little', 'defines': ('__arm64__', '__aarch64__')},
+    'arm64'   : { 'endian': 'little', 'defines': ('__arm64__', '__aarch64__')},
     'i386'    : { 'endian': 'little', 'defines': ('__i386', '_M_IX86')},
     'ppc64le' : { 'endian': 'little', 'defines': ('__powerpc64__',)},
     's390x'   : { 'endian': 'big',    'defines': ('__s390x__',)},
Index: git/src/third_party/IntelRDFPMathLib20U1/SConscript
===================================================================
--- git.orig/src/third_party/IntelRDFPMathLib20U1/SConscript
+++ git/src/third_party/IntelRDFPMathLib20U1/SConscript
@@ -301,7 +301,7 @@ if processor == 'i386':
 elif processor == 'arm':
     cpp_defines['IA32'] = '1'
     cpp_defines['ia32'] = '1'
-elif processor == "aarch64":
+elif processor == "aarch64" or processor == 'arm64':
     cpp_defines['efi2'] = '1'
     cpp_defines['EFI2'] = '1'
 # Using 64 bit little endian
Index: git/src/third_party/wiredtiger/SConscript
===================================================================
--- git.orig/src/third_party/wiredtiger/SConscript
+++ git/src/third_party/wiredtiger/SConscript
@@ -139,7 +139,7 @@ condition_map = {
     'POSIX_HOST'   : not env.TargetOSIs('windows'),
     'WINDOWS_HOST' : env.TargetOSIs('windows'),
 
-    'ARM64_HOST'   : env['TARGET_ARCH'] == 'aarch64',
+    'ARM64_HOST'   : env['TARGET_ARCH'] in ('aarch64', 'arm64'),
     'POWERPC_HOST' : env['TARGET_ARCH'] == 'ppc64le',
     'X86_HOST'     : env['TARGET_ARCH'] == 'x86_64',
     'ZSERIES_HOST' : env['TARGET_ARCH'] == 's390x',
