musl has valid d_ino implementation

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Pending

Index: tar-1.28/m4/d-ino.m4
===================================================================
--- tar-1.28.orig/m4/d-ino.m4
+++ tar-1.28/m4/d-ino.m4
@@ -40,7 +40,8 @@ AC_DEFUN([gl_CHECK_TYPE_STRUCT_DIRENT_D_
            [gl_cv_struct_dirent_d_ino=no],
            [case "$host_os" in
                            # Guess yes on glibc systems with Linux kernel.
-              linux*-gnu*) gl_cv_struct_dirent_d_ino="guessing yes" ;;
+              linux*-gnu*|linux*-musl*)
+                           gl_cv_struct_dirent_d_ino="guessing yes" ;;
                            # Guess no on native Windows.
               mingw*)      gl_cv_struct_dirent_d_ino="guessing no" ;;
                            # If we don't know, assume the worst.
