Signed-off-by: Diego Rondini <diego.ml@zoho.com>
Upstream-Status: Inappropriate

Index: watchdog-5.15/watchdog.conf
===================================================================
--- watchdog-5.15.orig/watchdog.conf
+++ watchdog-5.15/watchdog.conf
@@ -28,7 +28,7 @@
 #retry-timeout		= 60
 #repair-maximum		= 1
 
-#watchdog-device	= /dev/watchdog
+watchdog-device	= /dev/watchdog
 
 # Defaults compiled into the binary
 #temperature-sensor	=
