From caf59c46ea5dc49e4dbf756c642b03e52c1b5468 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 20 Dec 2019 07:02:24 -0800
Subject: [PATCH] Respect CMAKE_INSTALL_LIBDIR in installed CMake files

Upstream-Status: Submitted [https://github.com/google/effcee/pull/36]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 CMakeLists.txt        | 1 +
 effcee/CMakeLists.txt | 4 ++--
 2 files changed, 3 insertions(+), 2 deletions(-)

--- a/external/effcee/CMakeLists.txt
+++ b/external/effcee/CMakeLists.txt
@@ -27,6 +27,7 @@ endif()
 
 include(cmake/setup_build.cmake)
 include(cmake/utils.cmake)
+include(GNUInstallDirs)
 
 add_subdirectory(third_party)
 add_subdirectory(effcee)
--- a/external/effcee/effcee/CMakeLists.txt
+++ b/external/effcee/effcee/CMakeLists.txt
@@ -14,8 +14,8 @@ install(
   DESTINATION
     include/effcee)
 install(TARGETS effcee
-  LIBRARY DESTINATION lib
-  ARCHIVE DESTINATION lib)
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 if(EFFCEE_BUILD_TESTING)
   add_executable(effcee-test
