Upstream-Status: Inappropriate [embedded specific]

--- a/configure.ac	2011-10-20 21:56:02.230663987 +0200
+++ b/configure.ac	2011-08-17 15:01:30.000000000 +0200
@@ -227,8 +227,8 @@
   LIBTOOL="$sh_libtool"
   SVN_LIBTOOL="$sh_libtool"
 else
-  sh_libtool="$abs_builddir/libtool"
-  SVN_LIBTOOL="\$(SHELL) $sh_libtool"
+  sh_libtool="$abs_builddir/$host_alias-libtool"
+  SVN_LIBTOOL="\$(SHELL) \$(abs_builddir)/$host_alias-libtool"
 fi
 AC_SUBST(SVN_LIBTOOL)
 
