From a04fd2c8ac9e0f7a6fc17f02b2a95227b3d0aae4 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Tue, 31 Mar 2020 16:06:39 +0200
Subject: [PATCH] modulemd: generate the manpage only if the feature is enabled

Upstream-Status: Pending
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 modulemd/meson.build | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modulemd/meson.build b/modulemd/meson.build
index 9ef1902..9f71e72 100644
--- a/modulemd/meson.build
+++ b/modulemd/meson.build
@@ -494,6 +494,7 @@ if with_docs
     )
 endif
 
+if with_manpages == 'enabled'
 help2man_opts = [
   '--no-info',
   '--section=1',
@@ -506,3 +507,4 @@ custom_target(
     ],
     install: true,
     install_dir: join_paths(get_option('mandir'), 'man1'))
+endif
