From 4b79f603b948709906c108796087de7b5643f61a Mon Sep 17 00:00:00 2001
From: Yang Lyu <yangl@axis.com>
Date: Mon, 26 Feb 2018 12:48:34 +0100
Subject: [PATCH] Correct including directory for conf

---
 src/launch/config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/launch/config.c b/src/launch/config.c
index 6a68699..579471b 100644
--- a/src/launch/config.c
+++ b/src/launch/config.c
@@ -1022,7 +1022,7 @@ static void config_parser_end_fn(void *userdata, const XML_Char *name) {
                                     state->file,
                                     state->current->include.selinux_root_relative ?
                                         bus_selinux_policy_root() :
-                                        state->file->path,
+                                        NULL,
                                     state->current->cdata);
                 if (r) {
                         state->error = error_trace(r);
-- 
2.1.4

