From e4d3a7470b307693660d0412732e7266d1738d8c Mon Sep 17 00:00:00 2001
From: Christopher Larson <chris_larson@mentor.com>
Date: Tue, 13 Dec 2016 10:29:17 -0700
Subject: [PATCH 6/6] Don't set up -m32/-m64, we do that ourselves

Upstream-Status: Inappropriate
Signed-off-by: Christopher Larson <chris_larson@mentor.com>

---
 tools/build/src/tools/gcc.jam | 39 ---------------------------------------
 1 file changed, 39 deletions(-)

diff --git a/tools/build/src/tools/gcc.jam b/tools/build/src/tools/gcc.jam
index e4fc6c32..37914fd0 100644
--- a/tools/build/src/tools/gcc.jam
+++ b/tools/build/src/tools/gcc.jam
@@ -337,45 +337,6 @@ rule set-address-model-options ( targets * : sources * : properties * )
     {
         local option ;
         local target-os = [ feature.get-values target-os : $(properties) ] ;
-        if $(target-os) = aix
-        {
-            if $(model) = 32
-            {
-                option = -maix32 ;
-            }
-            else
-            {
-                option = -maix64 ;
-            }
-        }
-        else if $(target-os) = hpux
-        {
-            if $(model) = 32
-            {
-                option = -milp32 ;
-            }
-            else
-            {
-                option = -mlp64 ;
-            }
-        }
-        else
-        {
-            local arch = [ feature.get-values architecture : $(properties) ] ;
-            if $(arch) = power || $(arch) = sparc || $(arch) = x86
-            {
-                if $(model) = 32
-                {
-                    option = -m32 ;
-                }
-                else if $(model) = 64
-                {
-                    option = -m64 ;
-                }
-            }
-            # For darwin, the model can be 32_64. darwin.jam will handle that
-            # on its own.
-        }
         OPTIONS on $(targets) += $(option) ;
     }
 }
-- 
2.15.1

