Upstream-Status: Pending

Dont strip during install let the packaging handle it

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Index: tools/jailtest/Makefile
===================================================================
--- tools.orig/jailtest/Makefile	2003-02-07 20:10:27.000000000 -0800
+++ tools/jailtest/Makefile	2012-09-05 12:13:56.061615854 -0700
@@ -14,4 +14,4 @@
 
 install : $(BIN)
 	install -d $(DESTDIR)$(BIN_DIR)
-	install -s $(BIN) $(DESTDIR)$(BIN_DIR)
+	install $(BIN) $(DESTDIR)$(BIN_DIR)
Index: tools/mconsole/Makefile
===================================================================
--- tools.orig/mconsole/Makefile	2012-09-05 12:05:09.869607043 -0700
+++ tools/mconsole/Makefile	2012-09-05 12:14:07.725607469 -0700
@@ -14,4 +14,4 @@
 
 install : $(BIN)
 	install -d $(DESTDIR)$(BIN_DIR)
-	install -s $(BIN) $(DESTDIR)$(BIN_DIR)
+	install $(BIN) $(DESTDIR)$(BIN_DIR)
Index: tools/moo/Makefile
===================================================================
--- tools.orig/moo/Makefile	2003-06-26 09:27:00.000000000 -0700
+++ tools/moo/Makefile	2012-09-05 12:14:20.981604580 -0700
@@ -22,4 +22,4 @@
 
 install : $(BIN)
 	install -d $(DESTDIR)$(BIN_DIR)
-	install -s $(BIN) $(DESTDIR)$(BIN_DIR)
+	install $(BIN) $(DESTDIR)$(BIN_DIR)
Index: tools/port-helper/Makefile
===================================================================
--- tools.orig/port-helper/Makefile	2003-02-07 20:05:31.000000000 -0800
+++ tools/port-helper/Makefile	2012-09-05 12:14:39.429609678 -0700
@@ -14,4 +14,4 @@
 
 install : $(BIN)
 	install -d $(DESTDIR)$(LIB_DIR)
-	install -s $(BIN) $(DESTDIR)$(LIB_DIR)
+	install $(BIN) $(DESTDIR)$(LIB_DIR)
Index: tools/tunctl/Makefile
===================================================================
--- tools.orig/tunctl/Makefile	2003-02-07 20:05:04.000000000 -0800
+++ tools/tunctl/Makefile	2012-09-05 12:14:49.985606470 -0700
@@ -14,4 +14,4 @@
 
 install : $(BIN)
 	install -d $(DESTDIR)$(BIN_DIR)
-	install -s $(BIN) $(DESTDIR)$(BIN_DIR)
+	install $(BIN) $(DESTDIR)$(BIN_DIR)
Index: tools/uml_net/Makefile
===================================================================
--- tools.orig/uml_net/Makefile	2003-02-07 20:04:55.000000000 -0800
+++ tools/uml_net/Makefile	2012-09-05 12:15:00.961607290 -0700
@@ -21,4 +21,4 @@
 
 install : $(BIN)
 	install -d $(DESTDIR)$(BIN_DIR)
-	install -s -m 04755 $(BIN) $(DESTDIR)$(BIN_DIR)
+	install -m 04755 $(BIN) $(DESTDIR)$(BIN_DIR)
Index: tools/uml_router/Makefile
===================================================================
--- tools.orig/uml_router/Makefile	2003-02-07 20:04:39.000000000 -0800
+++ tools/uml_router/Makefile	2012-09-05 12:15:10.649607571 -0700
@@ -20,4 +20,4 @@
 
 install : $(BIN)
 	install -d $(DESTDIR)$(BIN_DIR)
-	install -s $(BIN) $(DESTDIR)$(BIN_DIR)
+	install $(BIN) $(DESTDIR)$(BIN_DIR)
Index: tools/watchdog/Makefile
===================================================================
--- tools.orig/watchdog/Makefile	2003-02-07 20:04:25.000000000 -0800
+++ tools/watchdog/Makefile	2012-09-05 12:15:35.561612203 -0700
@@ -14,4 +14,4 @@
 
 install : $(BIN)
 	install -d $(DESTDIR)$(BIN_DIR)
-	install -s $(BIN) $(DESTDIR)$(BIN_DIR)
+	install $(BIN) $(DESTDIR)$(BIN_DIR)
