From db9e6a4b7a48dee62fccb662347dd638a2d286d5 Mon Sep 17 00:00:00 2001
From: Lei Maohui <leimaohui@fujitsu.com>
Date: Thu, 2 Dec 2021 14:29:06 +0900
Subject: [PATCH] Remove unsuitble part for cross-compile.

Upstream-Status: Inappropriate [OE specific]

Signed-off-by: Lei Maohui <leimaohui@fujitsu.com>
---
 configure.ac               | 6 ------
 src/libpcp/src/GNUmakefile | 1 -
 2 files changed, 7 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -607,12 +607,6 @@ fi
 which=$WHICH
 AC_SUBST(which)
 
-if test -z "$AR"; then
-    AC_PATH_PROGS(AR, [gcc-ar ar], /usr/bin/ar)
-fi
-if test ! -x "$AR"; then
-    AC_MSG_ERROR([cannot find a valid 'ar' command.])
-fi
 ar=$AR
 AC_SUBST(ar)
 
--- a/src/libpcp/src/GNUmakefile
+++ b/src/libpcp/src/GNUmakefile
@@ -179,8 +179,6 @@ exports:	exports.in mk.exports
 # consistent in all placess.
 #
 check.done:	$(OBJECTS)
-	./check-statics
-	./check-errorcodes
 	touch check.done
 
 ifneq ($(LIBTARGET),)
