From 3863d8bd71d6a5638cf984f8d8f9cccac0c7f2a4 Mon Sep 17 00:00:00 2001
From: Alistair Francis <alistair.francis@wdc.com>
Date: Mon, 23 Jul 2018 11:31:17 -0700
Subject: [PATCH] Makefile: Don't install the docs

Signed-off-by: Alistair Francis <alistair.francis@wdc.com>
Upstream-Status: Inappropriate [disable feature]
---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index 47a447f12..7c0f9848f 100644
--- a/Makefile
+++ b/Makefile
@@ -148,7 +148,6 @@ rpm dpkg: pkg-clean
 
 install uninstall:
 	$(MAKE) -C src $@
-	$(MAKE) -C doc $@
 
 .PHONY: all clean clobber test check cstyle check-license install uninstall\
 	source rpm dpkg pkg-clean pcheck check-remote format doc require-rpmem\
-- 
2.14.4

