From 03404e6bd1ebfb363c971016a40056d17b12ded0 Mon Sep 17 00:00:00 2001
From: CH Li <chli30@nuvoton.com>
Date: Thu, 18 Jul 2019 14:02:48 +0800
Subject: [PATCH] meta-olympus-nuvoton: op-pwrctl: enable server power control
 feature

---
 power_control_obj.c | 26 ++++++++++++++++++++------
 1 file changed, 20 insertions(+), 6 deletions(-)

diff --git a/power_control_obj.c b/power_control_obj.c
index dedcbe2..cde80a0 100644
--- a/power_control_obj.c
+++ b/power_control_obj.c
@@ -273,12 +273,26 @@ on_set_power_state(ControlPower *pwr,
 							power_gpio->power_up_outs[i].name, error);
 					continue;
 				}
-				power_up_out = state ^ !power_gpio->power_up_pols[i];
-				g_print("PowerControl: setting power up %s to %d\n",
-						power_gpio->power_up_outs[i].name, (int)power_up_out);
-				error = gpio_write(power_pin, power_up_out);
-				if(error != GPIO_OK) {
-					continue;
+				if (state){
+					g_print("Power requested state: %d\n",state);
+					power_up_out = state ^ !power_gpio->power_up_pols[i];
+					g_print("PowerControl: setting power up %s to %d\n",
+							power_gpio->power_up_outs[i].name, (int)power_up_out);
+					error = gpio_write(power_pin, power_up_out);
+					if(error != GPIO_OK) {
+						continue;
+					}
+					sleep(1);
+				} else{
+					g_print("Power requested state: %d\n",state);
+					power_up_out = state ^ power_gpio->power_up_pols[i];
+					g_print("PowerControl: setting power up %s to %d\n",
+							power_gpio->power_up_outs[i].name, (int)power_up_out);
+					error = gpio_write(power_pin, power_up_out);
+					if(error != GPIO_OK) {
+						continue;
+					}
+					sleep(1);
 				}
 				gpio_close(power_pin);
 			}
-- 
2.17.1

