From 08f44b0ffe7bccf0a17fe3d595024853a86a8d0c Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Fri, 27 Dec 2019 12:34:29 +0800
Subject: [PATCH] skip Listen for BootProgress signal

---
power_control_obj.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/power_control_obj.c b/power_control_obj.c
index dedcbe2..9170cfb 100644
--- a/power_control_obj.c
+++ b/power_control_obj.c
@@ -442,7 +442,7 @@ on_bus_acquired(GDBusConnection *connection,
 			"handle-init",
 			G_CALLBACK(on_init),
 			object); /* user_data */
-
+#if 0
 	/* Listen for BootProgress signal from BootProgress sensor */
 	g_dbus_connection_signal_subscribe(connection,
 			NULL, /* service */
@@ -454,7 +454,7 @@ on_bus_acquired(GDBusConnection *connection,
 			(GDBusSignalCallback) on_boot_progress,
 			object, /* user data */
 			NULL );
-
+#endif
 	/* Export the object (@manager takes its own reference to @object) */
 	g_dbus_object_manager_server_set_connection(manager, connection);
 	g_dbus_object_manager_server_export(manager, G_DBUS_OBJECT_SKELETON(object));
-- 
2.17.1

