From 43e9db656431ffb22b429d5fca4ce3b4af21bc9e Mon Sep 17 00:00:00 2001
From: Herrie <github.com@herrie.org>
Date: Mon, 24 Jul 2017 21:30:16 +0200

---
 libpurple/protocols/irc/irc.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libpurple/protocols/irc/irc.h b/libpurple/protocols/irc/irc.h
index 596ddaf..b70f3a9 100644
--- a/libpurple/protocols/irc/irc.h
+++ b/libpurple/protocols/irc/irc.h
@@ -40,9 +40,9 @@
 
 #define IRC_DEFAULT_CHARSET "UTF-8"
 #define IRC_DEFAULT_AUTODETECT FALSE
-#define IRC_DEFAULT_ALIAS "purple"
+#define IRC_DEFAULT_ALIAS "OE-user"
 
-#define IRC_DEFAULT_QUIT "Leaving."
+#define IRC_DEFAULT_QUIT "Powered by OE: www.openembedded.org"
 
 #define IRC_BUFSIZE_INCREMENT 1024
 #define IRC_MAX_BUFSIZE 16384
