--- a/configure.ac
+++ b/configure.ac
@@ -1488,7 +1488,7 @@ AC_ARG_WITH([python],
 
 if test "x$enable_dbus" = "xyes" || test "x$enable_consoleui" = "xyes" ; then
 	if test -z "$PYTHON" -o "x$PYTHON" = "xyes"; then
-		AC_PATH_PROG([PYTHON], [python], [no])
+		AC_PATH_PROG([PYTHON], [python3], [no])
 	fi
 
 	if test x"$PYTHON" = x"no" ; then
@@ -1568,7 +1568,7 @@ dnl Check for Python headers (currently
 dnl (Thanks to XChat)
 if test "x$enable_consoleui" = "xyes" -a ! -z "$PYTHON" -a x"$PYTHON" != x"no" ; then
         AC_MSG_CHECKING(for Python compile flags)
-        PKG_CHECK_MODULES(PY, python, [
+        PKG_CHECK_MODULES(PY, python3, [
                 AC_SUBST(PY_CFLAGS)
                 AC_SUBST(PY_LIBS)
         ], [
