Index: fbida-2.10/GNUmakefile
===================================================================
--- fbida-2.10.orig/GNUmakefile
+++ fbida-2.10/GNUmakefile
@@ -51,7 +51,7 @@ HAVE_LIBSANE	:= $(call ac_lib,sane_init,
 HAVE_LIBCURL	:= $(call ac_lib,curl_easy_init,curl)
 HAVE_LIBLIRC	:= $(call ac_lib,lirc_init,lirc_client)
 HAVE_MOTIF	:= $(call ac_lib,XmStringGenerate,Xm,-L/usr/X11R6/$(LIB) -lXpm -lXt -lXext -lX11)
-JPEG_VER        := $(call ac_jpeg_ver)
+JPEG_VER        := 62
 endef
 
 # transposing
