From 46f9f2d2f573f77ebed735db16f7dc9936b9c8c3 Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Sat, 28 May 2022 00:33:41 +0800
Subject: [PATCH] scm npcm845 board setting

---
 board/nuvoton/arbel/arbel.c | 38 +++++++++++++++++++++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/board/nuvoton/arbel/arbel.c b/board/nuvoton/arbel/arbel.c
index a674757064..dbb7ecbfaf 100644
--- a/board/nuvoton/arbel/arbel.c
+++ b/board/nuvoton/arbel/arbel.c
@@ -77,6 +77,44 @@ int board_init(void)
 	arbel_clk_init();
 	arbel_eth_init();
 
+	/* Do not reset FLM moudule in watchdog reset */
+	clrbits_le32(0xF0801080, BIT(6));
+#if 0
+	/* UART0 to Pin 41/42, UART Switch to SI2 */
+	clrbits_le32(0xf080026c, BIT(1)); // Clear MFSEL4.BSPASEL
+
+	/* Set Pin 41/42 as UART0 function */
+	setbits_le32(0xf0800260, BIT(9)); // Set MFSEL1.BSPSEL
+
+	/* Set SI2 as Serial Interface function */
+	setbits_le32(0xf0800260, BIT(11)); // Set MFSEL1.HSI2ASEL
+	/* Set UART switch mode 1 */
+	clrbits_le32(0xf0800038, 0x7);
+#endif
+	/* clear VGAIOEN */
+	clrbits_le32(0xF080003C, BIT(6));
+
+	/*set PCIRREL*/
+	setbits_le32(0xf08000a0, BIT(0));
+
+	clrbits_le32(0xf0801080, BIT(7) | BIT(8) | BIT(9) | BIT(5));
+	clrbits_le32(0xf0801084, BIT(7) | BIT(8) | BIT(9) | BIT(5));
+	clrbits_le32(0xf0801088, BIT(7) | BIT(8) | BIT(9) | BIT(5));
+	clrbits_le32(0xf080108c, BIT(7) | BIT(8) | BIT(9) | BIT(5));
+	clrbits_le32(0xf0801090, BIT(7) | BIT(8) | BIT(9) | BIT(5));
+	clrbits_le32(0xf0801094, BIT(7) | BIT(8) | BIT(9) | BIT(5));
+	clrbits_le32(0xf0801098, BIT(7) | BIT(8) | BIT(9) | BIT(5));
+	clrbits_le32(0xf080109c, BIT(7) | BIT(8) | BIT(9) | BIT(5));
+
+	clrbits_le32(0xf0801038, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf080103c, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf0801040, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf0801044, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf0801048, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf080104c, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf0801050, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf080105c, BIT(31) | BIT(24) | BIT(25));
+
 	return 0;
 }
 
-- 
2.34.1

