From d7041359ea0beca3b2f27b3ee4699103685872c2 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Tue, 14 Feb 2023 11:42:49 +0800
Subject: [PATCH] uboot scm dts

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 arch/arm/dts/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index e64d162b20..3aba574f67 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -1033,7 +1033,8 @@ dtb-$(CONFIG_TARGET_DURIAN) += phytium-durian.dtb
 dtb-$(CONFIG_TARGET_PRESIDIO_ASIC) += ca-presidio-engboard.dtb
 
 dtb-$(CONFIG_TARGET_ARBEL_EVB) += \
-	nuvoton-npcm845-evb.dtb
+	nuvoton-npcm845-evb.dtb \
+	nuvoton-npcm845-scm.dtb
 
 dtb-$(CONFIG_ARCH_NPCM7xx) += \
 		nuvoton-npcm750-buv.dtb \
-- 
2.17.1

