From 5e632c5e2722be4d887ada6256f4715d745b3e7c Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Tue, 14 Feb 2023 13:15:33 +0800
Subject: [PATCH] scm-npcm845 board setting

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 board/nuvoton/arbel/arbel.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/board/nuvoton/arbel/arbel.c b/board/nuvoton/arbel/arbel.c
index a674757064..06e5a12e4e 100644
--- a/board/nuvoton/arbel/arbel.c
+++ b/board/nuvoton/arbel/arbel.c
@@ -77,6 +77,25 @@ int board_init(void)
 	arbel_clk_init();
 	arbel_eth_init();
 
+	/* do not reset SGPIO/eSPI/HGPIO/GFX/TMPS during soft reset */
+	clrbits_le32(0xf0801080, BIT(7) | BIT(9) | BIT(5));
+	clrbits_le32(0xf0801084, BIT(7) | BIT(9) | BIT(5));
+	clrbits_le32(0xf0801088, BIT(7) | BIT(9) | BIT(5));
+	clrbits_le32(0xf080108c, BIT(7) | BIT(9) | BIT(5));
+	clrbits_le32(0xf0801090, BIT(7) | BIT(9) | BIT(5));
+	clrbits_le32(0xf0801094, BIT(7) | BIT(9) | BIT(5));
+	clrbits_le32(0xf0801098, BIT(7) | BIT(9) | BIT(5));
+	clrbits_le32(0xf080109c, BIT(7) | BIT(9) | BIT(5));
+
+	clrbits_le32(0xf0801038, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf080103c, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf0801040, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf0801044, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf0801048, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf080104c, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf0801050, BIT(31) | BIT(24) | BIT(25));
+	clrbits_le32(0xf080105c, BIT(31) | BIT(24) | BIT(25));
+
 	return 0;
 }
 
-- 
2.17.1

