From e27d0f06c37d57e1fe12abde18b58f15ee71b3e7 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Thu, 19 May 2022 14:25:14 +0800
Subject: [PATCH] i2c mw zero len

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 cmd/i2c.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/cmd/i2c.c b/cmd/i2c.c
index 5d0e207874..5c0e501a7e 100644
--- a/cmd/i2c.c
+++ b/cmd/i2c.c
@@ -681,6 +681,11 @@ static int do_i2c_mw(struct cmd_tbl *cmdtp, int flag, int argc,
 	else
 		count = 1;
 
+	if (count == 0) {
+		printf("write zero len\n");
+		dm_i2c_write(dev, addr, &byte, 0);
+		return 0;
+	}
 	while (count-- > 0) {
 #if CONFIG_IS_ENABLED(DM_I2C)
 		ret = dm_i2c_write(dev, addr++, &byte, 1);
-- 
2.17.1
