From 261f135a8f84ead592d6861a37a3f6a33e7e3fb6 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Mon, 15 Aug 2022 11:40:22 +0800
Subject: [PATCH] set fiu0 drd_cfg 4-byte addr

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 board/nuvoton/arbel/arbel.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/board/nuvoton/arbel/arbel.c b/board/nuvoton/arbel/arbel.c
index 29cc8b7bdc..b1a567e033 100644
--- a/board/nuvoton/arbel/arbel.c
+++ b/board/nuvoton/arbel/arbel.c
@@ -87,6 +87,9 @@ int board_init(void)
 	arbel_clk_init();
 	arbel_eth_init();
 
+	/* set FIU0 drd config, 4-byte addr */
+	writel(0x0301100b, 0xfb000000);
+
 	/* Do not reset FLM moudule in watchdog reset */
 	clrbits_le32(0xF0801080, BIT(6));
 #if 0
-- 
2.17.1
