
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- Numeric-23.7/setup.py~nolapack
+++ Numeric-23.7/setup.py
@@ -32,7 +32,7 @@
     mathlibs = []
 
 # delete all but the first one in this list if using your own LAPACK/BLAS
-sourcelist = [os.path.join('Src', 'lapack_litemodule.c'),
+sourcelist = [
               #os.path.join('Src', 'blas_lite.c'), 
               #os.path.join('Src', 'f2c_lite.c'), 
               #os.path.join('Src', 'zlapack_lite.c'),
@@ -40,12 +40,12 @@
              ]
 # set these to use your own BLAS;
 
-library_dirs_list = ['/usr/lib/atlas']
-libraries_list = ['lapack', 'cblas', 'f77blas', 'atlas', 'g2c'] 
+library_dirs_list = []
+libraries_list = [] 
 
 # set to true (1), if you also want BLAS optimized matrixmultiply/dot/innerproduct
-use_dotblas = 1 
-include_dirs = ['/usr/include/atlas']  
+use_dotblas = 0
+include_dirs = []  
                    # You may need to set this to find cblas.h
                    #  e.g. on UNIX using ATLAS this should be ['/usr/include/atlas']
 extra_link_args = []
