From be5f1439a1f36c0bc714411d20186045bd3e9539 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Tue, 12 Jul 2011 12:48:14 +0200
Subject: [PATCH] remove bogus check for host-side jack installs, we are crosscompiling and don't care about these misguided checks

---
 configure.ac |   45 ---------------------------------------------
 1 files changed, 0 insertions(+), 45 deletions(-)

diff --git a/configure.ac b/configure.ac
index 0a7e6bc..4033c21 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,51 +3,6 @@ dnl $Id: configure.ac 4467 2011-06-23 15:10:03Z paul $
 
 AC_INIT(jackd/jackd.c)
 
-
-dnl
-dnl Check for existing JACK installs
-dnl 
-
-AC_MSG_CHECKING([existing, conflicting JACK installs])
-not_overwriting=0
-installs=
-for dir in /usr/lib /usr/local/lib /opt/lib ; do
-    if test -d $dir ; then
-	if test $(find $dir/ -name 'libjack.so.*' 2>/dev/null | wc -l) -gt 0 ; then
-	    if echo $prefix/lib | grep -vs $dir >/dev/null 2>&1 ; then
-		not_overwriting=$(expr $not_overwriting + 1)
-	    fi
-	    installs="$installs $dir"
-	fi	    
-    fi
-done
-
-if test $not_overwriting -gt 0 ; then
-    echo 
-    echo
-    echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
-    echo "You appear to have at least one existing installation of JACK."
-    echo
-    echo "Complete or partial JACK installs exist in:$installs"
-    echo 
-    echo "Installing this version will leave at least one of these"
-    echo "existing installations installed and this will probably break"
-    echo "JACK on your machine. "
-    echo
-    echo "Before building, you should first remove the existing JACK"
-    echo "installation(s). "
-    echo
-    echo "Alternatively use ./configure --prefix=... to force overwriting"
-    echo "the existing install."
-    echo
-    echo "WARNING: ON ANY DEBIAN-DERIVED DISTRIBUTION (Debian, Ubuntu etc)"
-    echo "CHANGING THE INSTALLATION PREFIX WILL NOT PRODUCE A WORKING JACK"
-    echo "INSTALL. Please contact the distribution packager for JACK and"
-    echo "ask them to fix their packaging."
-    echo "!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!"
-    exit 1
-fi
-
 AC_CONFIG_AUX_DIR(config)
 AC_CANONICAL_TARGET
 
-- 
1.6.6.1

