From 564901ab78da2b2b1bed92351dc3c102dc0a8154 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 6 Aug 2019 17:51:39 -0700
Subject: [PATCH] Set use_tcl to be empty string if tcl is disabled

Upstream-Status: Inappropriate [Cross-compile specific]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/configure.ac
+++ b/configure.ac
@@ -1225,7 +1225,7 @@ AC_ARG_ENABLE(tcl,
   [], [enable_tcl=yes])
 
 if test "x$enable_tcl" != "xyes"; then
-  use_tcl="No (disabled)"
+  use_tcl=""
 fi
 
 if test "x$use_tcl" = "x"; then
--- a/Makefile.am
+++ b/Makefile.am
@@ -11,7 +11,7 @@ pkginclude_HEADERS = $(top_builddir)/gra
 dist_man_MANS = graphviz.7
 
 # $(subdirs) contains the list from: AC_CONFIG_SUBDIRS
-SUBDIRS = $(subdirs) lib plugin cmd tclpkg doc contrib share graphs tests
+SUBDIRS = $(subdirs) lib plugin cmd doc contrib share graphs tests
 
 .PHONY: doxygen
 doxygen:
