From 5e53b643615fd53f3ff960521bbe9d95641e5f68 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Sun, 20 Jan 2019 23:58:29 +0100
Subject: [PATCH] Use native mkdefs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>
---
 lib/gvpr/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/gvpr/Makefile.am
+++ b/lib/gvpr/Makefile.am
@@ -59,7 +59,7 @@ mkdefs$(EXEEXT):
 	$(HOSTCC) $(srcdir)/mkdefs.c -o mkdefs$(EXEEXT)
 
 gdefs.h: $(top_srcdir)/lib/gvpr/gprdata mkdefs$(EXEEXT)
-	./mkdefs$(EXEEXT) gdefs.h < $(top_srcdir)/lib/gvpr/gprdata
+	mkdefs$(EXEEXT) gdefs.h < $(top_srcdir)/lib/gvpr/gprdata
 
 EXTRA_DIST = $(man_MANS) $(pdf_DATA) gprdata mkdefs.c gdefs.h trie.c gvprlib.vcxproj*
 
