Add config option --with-dev-dsp.

Upstream-Status: Pending

Signed-off-by: Zhang Xiao <xiao.zhang@windriver.com>
---
--- a/configure.ac
+++ b/configure.ac
@@ -106,6 +106,16 @@ AC_ARG_WITH(python,[  --with-python[[=AR
   fi
   ])
 
+AC_ARG_WITH(dev-dsp,[  --with-dev-dsp          enables dev/dsp for entropy producing. auto for auto-detecting dev/dep on host. [[default=no]]],[
+  if test "$withval" = yes; then
+    AC_DEFINE([HAVE_DEV_DSP], 1)
+  else
+    if test "$withval" = auto;  then
+      ac_detect_dev_dsp=yes
+    fi
+  fi
+  ],[ac_detect_dev_dsp=no])
+
 # Check for expert mode
 if test "$ac_enable_expert_mode" = yes; then
   BEE_EXPERT_MODE
@@ -464,7 +474,7 @@ linux*)
       ac_cv_have_dev_dsp=no
     fi
     ])
-  if test "$ac_cv_have_dev_dsp" = yes; then
+  if test "$ac_cv_have_dev_dsp" = yes && test "$ac_detect_dev_dsp" = yes; then
     AC_DEFINE([HAVE_DEV_DSP], 1)
   fi
   ;;
