From 7727dbfef4dd7ced3958ae9ada59a8899976b12d Mon Sep 17 00:00:00 2001
From: Markus Volk <f_l_k@t-online.de>
Date: Mon, 12 Dec 2022 18:40:31 +0100
Subject: [PATCH] xdg-desktop-portal.pc.in: add pc_sysrootdir dir

---
 xdg-desktop-portal.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xdg-desktop-portal.pc.in b/xdg-desktop-portal.pc.in
index 585c0c8..15b5817 100644
--- a/xdg-desktop-portal.pc.in
+++ b/xdg-desktop-portal.pc.in
@@ -2,7 +2,7 @@ prefix=@prefix@
 datarootdir=@datarootdir@
 datadir=@datadir@
 
-interfaces_dir=${datadir}/dbus-1/interfaces/
+interfaces_dir=${pc_sysrootdir}${datadir}/dbus-1/interfaces
 
 Name: xdg-desktop-portal
 Description: Desktop integration portal
-- 
2.34.1

