Upstream-Status: Backport

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
diff -Nurd bluez-4.101/configure.ac bluez-4.101/configure.ac
--- bluez-4.101/configure.ac	2012-06-22 19:36:49.000000000 +0300
+++ bluez-4.101/configure.ac	2013-01-07 06:13:18.385888966 +0200
@@ -2,7 +2,7 @@
 AC_INIT(bluez, 4.101)

 AM_INIT_AUTOMAKE([foreign subdir-objects color-tests])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)

 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
