From 7e8839731ae3fdf7502b91643aa6b252f6c5eaa6 Mon Sep 17 00:00:00 2001
From: Adrian Bunk <bunk@stusta.de>
Date: Thu, 14 May 2020 00:08:21 +0300
Subject: The C.UTF-8 locale is not portable, use std::locale::classic()

Upstream-Status: Inappropriate [Similar fix already in apt >= 1.3]
Signed-off-by: Adrian Bunk <bunk@stusta.de>
---
 ftparchive/writer.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ftparchive/writer.cc b/ftparchive/writer.cc
index 25963820a..6ad571d2a 100644
--- a/ftparchive/writer.cc
+++ b/ftparchive/writer.cc
@@ -984,7 +984,7 @@ ReleaseWriter::ReleaseWriter(FileFd * const GivenOutput, string const &/*DB*/) :
    AddPatterns(_config->FindVector("APT::FTPArchive::Release::Patterns"));
 
    time_t const now = time(NULL);
-   auto const posix = std::locale("C.UTF-8");
+   auto const posix = std::locale::classic();
 
    // FIXME: use TimeRFC1123 here? But that uses GMT to satisfy HTTP/1.1
    std::ostringstream datestr;
-- 
2.20.1

