From 9da1b0dbdcc90455bc9de49f73a96e7d18f83493 Mon Sep 17 00:00:00 2001
From: Julian Andres Klode <julian.klode@canonical.com>
Date: Tue, 18 Feb 2025 10:29:40 +0100
Subject: [PATCH] strutl: Add missing #include <cstdint> [gcc 15]

Closes: #1096322

Upstream-Status: Backport [https://salsa.debian.org/apt-team/apt/-/commit/f82dcd7e4ebb3f70d28e9feb3621676f8c0cc024]

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 apt-pkg/contrib/strutl.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/apt-pkg/contrib/strutl.cc b/apt-pkg/contrib/strutl.cc
index 67100f1..c0a1cbc 100644
--- a/apt-pkg/contrib/strutl.cc
+++ b/apt-pkg/contrib/strutl.cc
@@ -26,6 +26,7 @@
 
 #include <algorithm>
 #include <array>
+#include <cstdint>
 #include <iomanip>
 #include <limits>
 #include <locale>
