Disable documentation

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Constantin Musca <constantinx.musca@intel.com>

Index: apt-1.2.24/Makefile
===================================================================
--- apt-1.2.24.orig/Makefile
+++ apt-1.2.24/Makefile
@@ -19,7 +19,7 @@ all headers library clean veryclean bina
 	$(MAKE) -C cmdline $@
 	$(MAKE) -C ftparchive $@
 	$(MAKE) -C dselect $@
-	$(MAKE) -C doc $@
+#	$(MAKE) -C doc $@
 	$(MAKE) -C po $@
 	# FIXME: -C test has issue swith parallel builds, investigate!
 	-$(MAKE) -C test $@
