Add test-install rule to support ptest execution

Signed-off-by: Radu Patriu <radu.patriu@enea.com>

Index: openvswitch-2.1.2/Makefile.am
===================================================================
--- openvswitch-2.1.2.orig/Makefile.am
+++ openvswitch-2.1.2/Makefile.am
@@ -300,3 +300,5 @@
 include python/compat/automake.mk
 include tutorial/automake.mk
 include vtep/automake.mk
+include test.mk
+
Index: openvswitch-2.1.2/test.mk
===================================================================
--- /dev/null
+++ openvswitch-2.1.2/test.mk
@@ -0,0 +1,49 @@
+TEST_DEST ?= ${prefix}/lib/openvswitch
+TEST_ROOT ?= ${prefix}/lib/openvswitch
+TEST_DEPEND =
+
+if HAVE_OPENSSL
+TEST_DEPEND += $(TESTPKI_FILES)
+endif
+
+test-install: $(TEST_DEPEND)
+	@list='$(noinst_PROGRAMS) $(EXTRA_DIST) $(dist_check_SCRIPTS) $(TEST_DEPEND) tests/atlocal tests/atconfig' ;\
+	install -d $(TEST_DEST)/tests ;\
+	install -d $(TEST_DEST)/python ;\
+	install -d $(TEST_DEST)/python/ovs ;\
+	install -d $(TEST_DEST)/python/ovs/db ;\
+	install -d $(TEST_DEST)/python/ovs/unixctl ;\
+	install -d $(TEST_DEST)/vswitchd ;\
+	install vswitchd/vswitch.ovsschema $(TEST_DEST)/vswitchd ;\
+	install -d $(TEST_DEST)/debian ;\
+	install debian/ovs-monitor-ipsec $(TEST_DEST)/debian ;\
+	install -d $(TEST_DEST)/build-aux ;\
+	install build-aux/check-structs $(TEST_DEST)/build-aux ;\
+	install -d $(TEST_DEST)/xenserver ;\
+	install xenserver/usr_share_openvswitch_scripts_ovs-xapi-sync $(TEST_DEST)/xenserver ;\
+	install xenserver/opt_xensource_libexec_interface-reconfigure $(TEST_DEST)/xenserver ;\
+	install xenserver/opt_xensource_libexec_InterfaceReconfigure.py $(TEST_DEST)/xenserver ;\
+	install xenserver/opt_xensource_libexec_InterfaceReconfigureBridge.py $(TEST_DEST)/xenserver ;\
+	install xenserver/opt_xensource_libexec_InterfaceReconfigureVswitch.py $(TEST_DEST)/xenserver ;\
+	install -d $(TEST_DEST)/vtep ;\
+	install vtep/vtep.ovsschema $(TEST_DEST)/vtep ;\
+	for p in $$list ; do \
+	  echo $$p ;\
+	  p=$${p#./} ;\
+	  pre=$${p#tests\/} ;\
+	  if test $$pre != $$p ; then \
+	    echo installing $$p to $(TEST_DEST)/tests/$$pre ;\
+	    install $$p $(TEST_DEST)/tests/$$pre ;\
+	    continue ;\
+	  fi ;\
+	  pre=$${p#python\/ovs\/} ;\
+	  if test $$pre != $$p ; then \
+	    echo installing $$p to $(TEST_DEST)/python/ovs/$$pre ;\
+	    install $$p $(TEST_DEST)/python/ovs/$$pre ;\
+	    continue ;\
+	  fi; \
+	done ;\
+	sed -i 's|abs_builddir=.*|abs_builddir='"'"'$(TEST_ROOT)/tests'"'"'|g' $(TEST_DEST)/tests/atconfig
+	sed -i 's|abs_srcdir=.*|abs_srcdir='"'"'$(TEST_ROOT)/tests'"'"'|g' $(TEST_DEST)/tests/atconfig
+	sed -i 's|abs_top_srcdir=.*|abs_top_srcdir='"'"'$(TEST_ROOT)'"'"'|g' $(TEST_DEST)/tests/atconfig
+	sed -i 's|abs_top_builddir=.*|abs_top_builddir='"'"'$(TEST_ROOT)'"'"'|g' $(TEST_DEST)/tests/atconfig
