Patch obtained from OpenEmbedded.

Upstream-Status: Pending

Signed-off-by: Scott Garman <scott.a.garman@intel.com>

--- sgml-common-0.6.3/bin/Makefile.am~autohell.patch	2001-01-30 15:42:21.000000000 +0100
+++ sgml-common-0.6.3/bin/Makefile.am	2004-06-17 11:51:11.000000000 +0200
@@ -1,6 +1,5 @@
 primitive_scripts = sgmlwhich
 
-bindir = $(prefix)/bin
 bin_SCRIPTS = install-catalog $(primitive_scripts)
 
 EXTRA_DIST =  install-catalog.in $(primitive_scripts)
--- /dev/null	2003-09-23 19:59:22.000000000 +0200
+++ sgml-common-0.6.3/bin/sgmlwhich.in	2004-06-17 11:51:11.000000000 +0200
@@ -0,0 +1,7 @@
+#!/bin/sh
+# Location of the main configuration file for consistency with non-LSB platforms
+# Send any comments to Eric Bischoff <eric@caldera.de>
+# This program is under GPL license. See LICENSE file for details.
+
+echo "@sysconfdir@/sgml/sgml.conf"
+# Simple, isn't it? ;-)
--- /dev/null	2003-09-23 19:59:22.000000000 +0200
+++ sgml-common-0.6.3/config/sgml.conf.in	2004-06-17 11:51:11.000000000 +0200
@@ -0,0 +1,2 @@
+SGML_BASE_DIR=@datadir@/sgml
+SGML_CATALOGS_DIR=@sysconfdir@/sgml
--- sgml-common-0.6.3/configure.in~autohell.patch	2001-05-25 18:40:54.000000000 +0200
+++ sgml-common-0.6.3/configure.in	2004-06-17 11:53:22.000000000 +0200
@@ -3,11 +3,6 @@
 
 AM_INIT_AUTOMAKE(sgml-common, 0.6.3)
 
-docdir='$(prefix)/doc'
-AC_SUBST(docdir)
-
-dnl Checks for programs.
-
 dnl Checks for libraries.
 
 dnl Checks for header files.
@@ -18,6 +13,5 @@
 
 AC_OUTPUT(Makefile \
           bin/Makefile bin/install-catalog \
-          config/Makefile \
-          doc/Makefile doc/HTML/Makefile doc/man/Makefile doc/refentry/Makefile \
+          config/Makefile config/sgml.conf bin/sgmlwhich \
           sgml-iso-entities/Makefile xml-iso-entities/Makefile)

--- sgml-common-0.6.3/Makefile.am~no-docs	2001-05-25 18:22:01.000000000 +0200
+++ sgml-common-0.6.3/Makefile.am	2004-06-17 11:49:09.000000000 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = bin config doc sgml-iso-entities xml-iso-entities
+SUBDIRS = bin config sgml-iso-entities xml-iso-entities
 
 sgmlcodir = $(prefix)/share/sgml
 sgmlco_DATA = xml.dcl
