From 699420d306cadbb5285942e09f4383d84b01985d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Fri, 13 Jan 2017 13:21:23 +0100
Subject: [PATCH] replace freetype-config by pkg-config
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Pending

Signed-off-by: Andreas Müller <schnitzeltony@googlemail.com>
---
 configure.ac | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index f0efd48..3af5ecb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1006,12 +1006,12 @@ case $host_os_gui in
 
 	xft_found=no
 	if test x$enable_xft != xno; then
-	    AC_PATH_PROG(FTCONFIG, freetype-config)
+	    AC_PATH_PROG(PKGCONFIG,pkg-config)
 
-	    if test "x$FTCONFIG" != x; then
-		CPPFLAGS="`$FTCONFIG --cflags` $CPPFLAGS"
-		CFLAGS="`$FTCONFIG --cflags` $CFLAGS"
-		CXXFLAGS="`$FTCONFIG --cflags` $CXXFLAGS"
+	    if test "x$PKGCONFIG" != x; then
+	        CPPFLAGS="`$PKGCONFIG --cflags xft` $CPPFLAGS"
+	        CFLAGS="`$PKGCONFIG --cflags xft` $CFLAGS"
+	        CXXFLAGS="`$PKGCONFIG --cflags xft` $CXXFLAGS"
 
 		AC_CHECK_LIB(fontconfig, FcPatternCreate)
 		AC_CHECK_HEADER(X11/Xft/Xft.h,
-- 
2.7.4

