From 6c75656b68cb6e38b039ae532bd39437cd6daec5 Mon Sep 17 00:00:00 2001
From: Saul Wold <sgw@linux.intel.com>
Date: Wed, 16 Aug 2017 11:18:01 +0800
Subject: [PATCH] dirmngr uses libgpg error

Upstream-Status: Pending
Signed-off-by: Saul Wold <sgw@linux.intel.com>

Rebase to 2.1.23

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

---
 dirmngr/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dirmngr/Makefile.am b/dirmngr/Makefile.am
index 00d3c42..450d873 100644
--- a/dirmngr/Makefile.am
+++ b/dirmngr/Makefile.am
@@ -101,6 +101,7 @@ dirmngr_LDADD = $(libcommonpth) \
         $(DNSLIBS) $(LIBASSUAN_LIBS) \
 	$(LIBGCRYPT_LIBS) $(KSBA_LIBS) $(NPTH_LIBS) \
 	$(NTBTLS_LIBS) $(LIBGNUTLS_LIBS) $(LIBINTL) $(LIBICONV) $(NETLIBS) \
+	$(GPG_ERROR_LIBS) \
         $(dirmngr_robj)
 if USE_LDAP
 dirmngr_LDADD += $(ldaplibs)
