From ddfe6ed6a89226985e8c9f0751c026aabc0927a0 Mon Sep 17 00:00:00 2001
From: Saul Wold <sgw@linux.intel.com>
Date: Thu, 13 Dec 2012 19:03:52 -0800
Subject: [PATCH] use echo only in init

Upstream-Status: Inappropriate [embedded specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 scheduler/cups.sh.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scheduler/cups.sh.in b/scheduler/cups.sh.in
index 74cce18..c57f0db 100644
--- a/scheduler/cups.sh.in
+++ b/scheduler/cups.sh.in
@@ -51,7 +51,7 @@ case "`uname`" in
 		ECHO_ERROR=:
 		;;
 
-	Linux*)
+	DisableLinux*)
 		IS_ON=/bin/true
 		if test -f /etc/init.d/functions; then
 			. /etc/init.d/functions
