From 1fb07162a9ed187cccf06e34c9bf841d15c6e64e Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Sun, 30 Jan 2011 16:37:27 +0100
Subject: [PATCH] don't try to run generated binaries

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>

---
 ppdc/Makefile | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/ppdc/Makefile b/ppdc/Makefile
index e563988..973dd3f 100644
--- a/ppdc/Makefile
+++ b/ppdc/Makefile
@@ -189,8 +189,8 @@ genstrings:		genstrings.o libcupsppdc.a ../cups/$(LIBCUPSSTATIC) \
 	$(LD_CXX) $(ARCHFLAGS) $(LDFLAGS) -o genstrings genstrings.o \
 		libcupsppdc.a ../cups/$(LIBCUPSSTATIC) $(LIBGSSAPI) $(SSLLIBS) \
 		$(DNSSDLIBS) $(COMMONLIBS) $(LIBZ)
-	echo Generating localization strings...
-	./genstrings >sample.c
+#	echo Generating localization strings...
+#	./genstrings >sample.c
 
 
 #
@@ -209,9 +209,9 @@ ppdc-static:		ppdc.o libcupsppdc.a ../cups/$(LIBCUPSSTATIC) foo.drv foo-fr.po
 		../cups/$(LIBCUPSSTATIC) $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
 		$(COMMONLIBS) $(LIBZ)
 	$(CODE_SIGN) -s "$(CODE_SIGN_IDENTITY)" $@
-	echo Testing PPD compiler...
-	./ppdc-static -l en,fr -I ../data foo.drv
-	./ppdc-static -l en,fr -z -I ../data foo.drv
+#	echo Testing PPD compiler...
+#	./ppdc-static -l en,fr -I ../data foo.drv
+#	./ppdc-static -l en,fr -z -I ../data foo.drv
 
 
 #
@@ -240,17 +240,17 @@ ppdi-static:		ppdc-static ppdi.o libcupsppdc.a  ../cups/$(LIBCUPSSTATIC)
 		../cups/$(LIBCUPSSTATIC) $(LIBGSSAPI) $(SSLLIBS) $(DNSSDLIBS) \
 		$(COMMONLIBS) $(LIBZ)
 	$(CODE_SIGN) -s "$(CODE_SIGN_IDENTITY)" $@
-	echo Testing PPD importer...
-	$(RM) -r ppd ppd2 sample-import.drv
-	./ppdc-static -l en -I ../data sample.drv
-	./ppdi-static -I ../data -o sample-import.drv ppd/*
-	./ppdc-static -l en -I ../data -d ppd2 sample-import.drv
-	if diff -r ppd ppd2 >/dev/null; then \
-		echo PPD import OK; \
-	else \
-		echo PPD import FAILED; \
-		exit 1; \
-	fi
+#	echo Testing PPD importer...
+#	$(RM) -r ppd ppd2 sample-import.drv
+#	./ppdc-static -l en -I ../data sample.drv
+#	./ppdi-static -I ../data -o sample-import.drv ppd/*
+#	./ppdc-static -l en -I ../data -d ppd2 sample-import.drv
+#	if diff -r ppd ppd2 >/dev/null; then \
+#		echo PPD import OK; \
+#	else \
+#		echo PPD import FAILED; \
+#		exit 1; \
+#	fi
 
 
 #
