From 3e9a965dcd65ab2d40b753b6f792a1a4559182aa Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Sun, 30 Jan 2011 16:37:27 +0100
Subject: [PATCH 2/4] don't try to run generated binaries

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>

---
 ppdc/Makefile | 32 ++++++++++++++++----------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/ppdc/Makefile b/ppdc/Makefile
index 32e2e0b..f1478d4 100644
--- a/ppdc/Makefile
+++ b/ppdc/Makefile
@@ -186,8 +186,8 @@ genstrings:		genstrings.o libcupsppdc.a ../cups/$(LIBCUPSSTATIC) \
 	$(LD_CXX) $(ARCHFLAGS) $(ALL_LDFLAGS) -o genstrings genstrings.o \
 		libcupsppdc.a $(LINKCUPSSTATIC)
 	$(CODE_SIGN) -s "$(CODE_SIGN_IDENTITY)" $@
-	echo Generating localization strings...
-	./genstrings >sample.c
+#	echo Generating localization strings...
+#	./genstrings >sample.c
 
 
 #
-- 
2.17.1

