From ff6c7168c3f26094b3a18298208a28831d1c1fd5 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Sun, 30 Jan 2011 16:37:27 +0100
Subject: [PATCH] don't try to run generated binaries

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 ppdc/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ppdc/Makefile b/ppdc/Makefile
index e36ed11..3fe97e1 100644
--- a/ppdc/Makefile
+++ b/ppdc/Makefile
@@ -187,8 +187,8 @@ genstrings:		genstrings.o libcupsppdc.a ../cups/$(LIBCUPSSTATIC) \
 	$(LD_CXX) $(ARCHFLAGS) $(ALL_LDFLAGS) -o genstrings genstrings.o \
 		libcupsppdc.a $(LINKCUPSSTATIC)
 	$(CODE_SIGN) -s "$(CODE_SIGN_IDENTITY)" $@
-	echo Generating localization strings...
-	./genstrings >sample.c
+#	echo Generating localization strings...
+#	./genstrings >sample.c
 
 
 #
