From 122963c764b06a4b487b32d4d1da330bd83da4d8 Mon Sep 17 00:00:00 2001
From: fundawang <fundawang@yeah.net>
Date: Mon, 17 Mar 2025 19:05:08 +0800
Subject: [PATCH] Fix libname of Libs.private

Upstream-Status: Backport [1c712194e604f6dd4f8a881e09d8236d4f770b67]
Signed-off-by: Moritz Haase <Moritz.Haase@bmw.de>
---
 src/createrepo_c.pc.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/createrepo_c.pc.cmake b/src/createrepo_c.pc.cmake
index 466487e2f490db28576611924e3d9ac94a99cb51..b7eb66318781b01f2b1348ab2c9dbd24d9a5c32f 100644
--- a/src/createrepo_c.pc.cmake
+++ b/src/createrepo_c.pc.cmake
@@ -8,5 +8,5 @@ Version: @VERSION@
 Requires: glib-2.0 rpm libcurl sqlite3
 Requires.private: zlib libxml-2.0
 Libs: -L${libdir} -lcreaterepo_c
-Libs.private: -lbz2 -lzma
+Libs.private: -lbz2 -llzma
 Cflags: -I${includedir}
