Index: libol-0.3.18/configure.in
===================================================================
--- libol-0.3.18.orig/configure.in	2006-03-27 14:44:52.000000000 +0000
+++ libol-0.3.18/configure.in	2014-07-18 07:05:56.029481372 +0000
@@ -1,7 +1,7 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(src/abstract_io.c)
+AC_INIT(libol, "0.3.18")
 
-AM_INIT_AUTOMAKE(libol, "0.3.18", 1)
+AM_INIT_AUTOMAKE([foreign])
 if test -n "$SNAPSHOT_VERSION"; then
         VERSION=$VERSION+$SNAPSHOT_VERSION
 fi
