From d88aae2230683517b6d5b62cced51da81317a722 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 29 Aug 2022 22:37:14 -0700
Subject: [PATCH] clif: Include string.h for mem* function prototypes

Upstream-Status: Submitted [https://github.com/intel/openlldp/pull/86]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 clif.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/clif.c b/clif.c
index cad6f75..0758a7e 100644
--- a/clif.c
+++ b/clif.c
@@ -32,6 +32,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/socket.h>
