From 01641d146e4e6bea954e4a4ee1f6230b822665b4 Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Tue, 15 Aug 2017 15:37:49 +0800
Subject: [PATCH 06/11] site.h: enable gentle shutdown

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Chen Qi <Qi.Chen@windriver.com>

Rebase to 4.3.6
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 includes/site.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: dhcp-4.4.1/includes/site.h
===================================================================
--- dhcp-4.4.1.orig/includes/site.h
+++ dhcp-4.4.1/includes/site.h
@@ -295,7 +295,7 @@
    situations.  We plan to revisit this feature and may
    make non-backwards compatible changes including the
    removal of this define.  Use at your own risk.  */
-/* #define ENABLE_GENTLE_SHUTDOWN */
+#define ENABLE_GENTLE_SHUTDOWN
 
 /* Include old error codes.  This is provided in case you
    are building an external program similar to omshell for
