From 37e227ae1766568d8de55e268bb69bc15eee3529 Mon Sep 17 00:00:00 2001
From: Jussi Kukkonen <jussi.kukkonen@intel.com>
Date: Thu, 26 Nov 2015 16:01:47 +0200
Subject: [PATCH] Enable xdg-terminal

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Jussi Kukkonen <jussi.kukkonen@intel.com>
---
 scripts/Makefile.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scripts/Makefile.in b/scripts/Makefile.in
index d167393..430032d 100644
--- a/scripts/Makefile.in
+++ b/scripts/Makefile.in
@@ -20,7 +20,8 @@ SCRIPTS		= \
     xdg-open \
     xdg-email \
     xdg-screensaver \
-    xdg-settings
+    xdg-settings \
+    xdg-terminal
 #    xdg-su
 #    xdg-copy \
 #    xdg-file-dialog
-- 
2.1.4

