Use G_IR_SCANNER without passing additional environment

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
--- a/cmake/modules/GObjectIntrospection.cmake
+++ b/cmake/modules/GObjectIntrospection.cmake
@@ -122,9 +122,7 @@ macro(gir_add_introspection gir)
 		file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${_gir_name}_files "${_gir_files}")
 
 		add_custom_command(
-			COMMAND ${CMAKE_COMMAND} -E env "CC='${CMAKE_C_COMPILER}'" LDFLAGS=
-				${INTROSPECTION_SCANNER_ENV}
-				${G_IR_SCANNER}
+			COMMAND ${G_IR_SCANNER}
 				${INTROSPECTION_SCANNER_ARGS}
 				--namespace=${_gir_namespace}
 				--nsversion=${_gir_version}
@@ -240,7 +238,6 @@ macro(gir_add_introspection_simple gir_l
 		${_gir_identifies_prefixes}
 		${_gir_deps}
 		--add-include-path=${SHARE_INSTALL_PREFIX}/gir-1.0
-		--library-path=${LIB_INSTALL_DIR}
 		${_extra_library_path}
 		--pkg-export ${pkg_export_prefix}-${gir_library_version}
 		--c-include=${c_include}
