From b7b3caa9b5c8abcd3c371d0683001cd681e104b3 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 15 Jul 2017 10:19:22 -0700
Subject: [PATCH 1/2] Add another variable LIBS to provides libraries from env

Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index cc7e823..ae30a07 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ ifneq ($(VERBOSE_TEST),)
 	VERBOSE_TEST="--verbose"
 endif
 
-grubby_LIBS = -lblkid -lpopt
+grubby_LIBS = -lblkid -lpopt ${LIBS}
 
 all: grubby
 
