From 8f844ac7f44b8dc428d06cd6958c5f32d383d01c Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 15 Jul 2017 10:19:50 -0700
Subject: [PATCH 2/2] include paths.h for _PATH_MOUNTED

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 grubby.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/grubby.c b/grubby.c
index 47a1a15..59f74a9 100644
--- a/grubby.c
+++ b/grubby.c
@@ -34,6 +34,7 @@
 #include <libgen.h>
 #include <execinfo.h>
 #include <signal.h>
+#include <paths.h>
 #include <blkid/blkid.h>
 
 #include "log.h"
-- 
2.13.3

