Author: Marcin Juszkiewicz <marcin.juszkiewicz@linaro.org>

Upstream-Status: unsuitable
---
 build/linux.gcc.inc |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

Index: tbb2017_20170118oss/build/linux.gcc.inc
===================================================================
--- tbb2017_20170118oss.orig/build/linux.gcc.inc
+++ tbb2017_20170118oss/build/linux.gcc.inc
@@ -32,8 +32,9 @@ DYLIB_KEY = -shared
 EXPORT_KEY = -Wl,--version-script,
 LIBDL = -ldl
 
-CPLUS = g++
-CONLY = gcc
+CPLUS = $(CXX)
+CONLY = $(CC)
+CPLUS_FLAGS = $(CXXFLAGS)
 LIB_LINK_FLAGS = $(DYLIB_KEY) -Wl,-soname=$(BUILDING_LIBRARY)
 LIBS += -lpthread -lrt
 LINK_FLAGS = -Wl,-rpath-link=. -rdynamic
Index: tbb2017_20170118oss/build/linux.clang.inc
===================================================================
--- tbb2017_20170118oss.orig/build/linux.clang.inc
+++ tbb2017_20170118oss/build/linux.clang.inc
@@ -31,8 +31,9 @@ DYLIB_KEY = -shared
 EXPORT_KEY = -Wl,--version-script,
 LIBDL = -ldl
 
-CPLUS = clang++
-CONLY = clang
+CPLUS = $(CXX)
+CONLY = $(CC)
+CPLUS_FLAGS = $(CXXFLAGS)
 LIB_LINK_FLAGS = $(DYLIB_KEY) -Wl,-soname=$(BUILDING_LIBRARY)
 LIBS += -lpthread -lrt
 LINK_FLAGS = -Wl,-rpath-link=. -rdynamic
