From 2d61587f85d1bf57287d12b470afb94c0bf9249d Mon Sep 17 00:00:00 2001
From: Davide Cavalca <dcavalca@fedoraproject.org>
Date: Mon, 27 Jun 2022 16:54:35 -0700
Subject: [PATCH] cli: Fix format security warning

Upstream-Status: Submitted [https://github.com/Microsemi/switchtec-user/pull/316]

---
 cli/graph.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cli/graph.c b/cli/graph.c
index 5748914..95609de 100644
--- a/cli/graph.c
+++ b/cli/graph.c
@@ -134,7 +134,7 @@ static void draw_status(WINDOW *win, int x_off, int x_cnt, const char *status)
 	mvwprintw(win, 0, 0, "     ");
 	for (i = 0; i < x_cnt * 2 + x_off; i++)
 		mvwaddch(win, 1, i, ' ');
-	mvwprintw(win, 1, x_off, status);
+	mvwprintw(win, 1, x_off, "%s", status);
 	wrefresh(win);
 }
 
-- 
2.39.2

