From 8eeaee82dcfdf47f16ad880e416b722827f41bdb Mon Sep 17 00:00:00 2001
From: Vyacheslav Yurkov <Vyacheslav.Yurkov@bruker.com>
Date: Mon, 23 May 2022 19:37:32 +0200
Subject: [PATCH] Makefile: proper location of LDFLAGS

Signed-off-by: Vyacheslav Yurkov <Vyacheslav.Yurkov@bruker.com>
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index e3c5207..14b155e 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 CFLAGS = -Wall -g
-LFLAGS = -lm
+LDFLAGS = -lm
 CC = gcc
 
 all: overlay
@@ -7,7 +7,7 @@ all: overlay
 objects = fsck.o common.o lib.o check.o mount.o path.o overlayfs.o
 
 overlay: $(objects)
-	$(CC) $(LFLAGS) $(objects) -o fsck.overlay
+	$(CC) $(objects) -o fsck.overlay $(LDFLAGS) 
 
 .c.o:
 	$(CC) $(CFLAGS) -c $<
