Upstream-Status: Submitted [https://github.com/probonopd/zsync-curl/pull/41]
Signed-off-by: Ross Burton <ross.burton@arm.com>

diff --git a/src/configure.ac b/src/configure.ac
index 10de8c4..6488140 100644
--- a/src/configure.ac
+++ b/src/configure.ac
@@ -9,7 +9,6 @@ AC_CONFIG_MACRO_DIR([autotools])
 AC_CANONICAL_HOST
 
 AM_INIT_AUTOMAKE([dist-bzip2 no-dist-gzip foreign check-news -Woverride -Wobsolete -Wportability -Wsyntax -Wunsupported])
-AM_MAINTAINER_MODE
 
 dnl --- Check for programs
 AC_PROG_CC
@@ -23,6 +22,7 @@ AS_IF([test "x$enable_profile" = "xyes"], [
    ZS_CFLAGS="${ZS_CFLAGS} -pg" ])
 
 dnl --- Header files, typedefs, structures, libraries
+AC_USE_SYSTEM_EXTENSIONS
 AC_C_CONST
 AC_HEADER_STDC
 # string.h, memory.h, stdlib.h both included in the default header checks
@@ -55,10 +55,6 @@ case $host_os in
        host_os_mingw32=yes
        LIBS="${LIBS} -lwsock32"
        ;;
-   *)
-       AC_DEFINE([_XOPEN_SOURCE], 600, [Enable POSIX extensions if present])
-       AC_DEFINE([_BSD_SOURCE],1, [Enable BSD extensions if present])
-       ;;
 esac
 AM_CONDITIONAL([MINGW32], test "x$host_os_mingw32" = "xyes")
 
diff --git a/src/librcksum/Makefile.am b/src/librcksum/Makefile.am
index f4fce72..47754eb 100644
--- a/src/librcksum/Makefile.am
+++ b/src/librcksum/Makefile.am
@@ -1,4 +1,4 @@
-
+AM_CPPFLAGS = "-I$(top_srcdir)"
 
 noinst_LIBRARIES = librcksum.a
 
diff --git a/src/librcksum/md4.h b/src/librcksum/md4.h
index e90603a..0ed7485 100644
--- a/src/librcksum/md4.h
+++ b/src/librcksum/md4.h
@@ -18,11 +18,8 @@
 
 #include "zsglobal.h"
 
-#ifdef HAVE_INTTYPES_H
-#include <inttypes.h>
-#else
+#include <stdint.h>
 #include <sys/types.h>
-#endif
 
 #define	MD4_BLOCK_LENGTH		64
 #define	MD4_DIGEST_LENGTH		16
diff --git a/src/libzsync/Makefile.am b/src/libzsync/Makefile.am
index 0a6ae85..33a5951 100644
--- a/src/libzsync/Makefile.am
+++ b/src/libzsync/Makefile.am
@@ -1,3 +1,4 @@
+AM_CPPFLAGS = "-I$(top_srcdir)"
 
 noinst_LIBRARIES = libzsync.a
 
diff --git a/src/libzsync/sha1.h b/src/libzsync/sha1.h
index 3946505..87f3794 100644
--- a/src/libzsync/sha1.h
+++ b/src/libzsync/sha1.h
@@ -11,11 +11,8 @@
 
 #include "config.h"
 
-#ifdef HAVE_INTTYPES_H
-#include <inttypes.h>
-#else
+#include <stdint.h>
 #include <sys/types.h>
-#endif
 
 #define	SHA1_BLOCK_LENGTH		64
 #define	SHA1_DIGEST_LENGTH		20
diff --git a/src/libzsync/zmap.c b/src/libzsync/zmap.c
index f052dba..030b2a5 100644
--- a/src/libzsync/zmap.c
+++ b/src/libzsync/zmap.c
@@ -28,9 +28,6 @@
 #include <string.h>
 #include <sys/types.h>
 #include <arpa/inet.h>
-#ifdef HAVE_INTTYPES_H
-#include <inttypes.h>
-#endif
 
 #ifdef WITH_DMALLOC
 # include <dmalloc.h>
diff --git a/src/make.c b/src/make.c
index 191b527..ba7f3a9 100644
--- a/src/make.c
+++ b/src/make.c
@@ -28,9 +28,6 @@
 #include <time.h>
 
 #include <arpa/inet.h>
-#ifdef HAVE_INTTYPES_H
-#include <inttypes.h>
-#endif
 
 #include <sys/types.h>
 #include <sys/stat.h>
