Upstream-Status: Inappropriate [configuration]

---
 docs/reference/Makefile.am |    2 ++
 1 file changed, 2 insertions(+)

--- gypsy.orig/docs/reference/Makefile.am
+++ gypsy/docs/reference/Makefile.am
@@ -81,10 +81,12 @@ expand_content_files=
 # e.g. GTKDOC_LIBS=$(top_builddir)/gtk/$(gtktargetlib)
 
 INCLUDES=-I$(top_srcdir) $(GYPSY_CFLAGS)
 GTKDOC_LIBS=$(top_builddir)/gypsy/libgypsy.la $(GYPSY_LIBS)
 
+EXTRA_DIST =
+CLEANFILES =
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
 include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
