From d332b480257aa98b63d39c3c94896a111536f937 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Wed, 23 Aug 2017 18:38:26 +0300
Subject: [PATCH 2/2] Do not build anything in help/ as it requires itstool.

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

---
 meson.build | 1 -
 1 file changed, 1 deletion(-)

diff --git a/meson.build b/meson.build
index 3986273..ae2f65e 100644
--- a/meson.build
+++ b/meson.build
@@ -164,7 +164,6 @@ subdir('tools')
 subdir('tests')
 
 subdir('data')
-subdir('help')
 subdir('po')
 
 meson.add_install_script('build-aux/meson/post_install.sh')
-- 
2.7.4

