From 26778716cc866abb33859917b840669693c180a5 Mon Sep 17 00:00:00 2001
From: Tim Lee <timlee660101@gmail.com>
Date: Mon, 20 Jun 2022 15:06:19 +0800
Subject: [PATCH 2/5] Redfish Add power metrics support

Signed-off-by: Tim Lee <timlee660101@gmail.com>
---
 redfish-core/lib/power.hpp | 75 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 75 insertions(+)

diff --git a/redfish-core/lib/power.hpp b/redfish-core/lib/power.hpp
index 7b2a47ac7..03d3f83af 100644
--- a/redfish-core/lib/power.hpp
+++ b/redfish-core/lib/power.hpp
@@ -283,6 +283,81 @@ inline void requestRoutesPower(App& app)
                     // be null if the limit is not enabled.
                     value = powerCap * std::pow(10, scale);
                 }
+
+                auto PowerMetricHandler =
+                [sensorAsyncResp](
+                    const boost::system::error_code ec2,
+                    const std::vector<std::pair<std::string, dbus::utility::DbusVariantType>>&
+                        properties2) {
+                    if (ec2)
+                    {
+                        messages::internalError(sensorAsyncResp->asyncResp->res);
+                        BMCWEB_LOG_ERROR
+                            << "Power Metric GetAll handler: Dbus error " << ec2;
+                        return;
+                    }
+                    nlohmann::json& tempArray2 =
+                        sensorAsyncResp->asyncResp->res.jsonValue["PowerControl"];
+                    nlohmann::json& sensorJson2 = tempArray2.back();
+                    for (const std::pair<std::string, dbus::utility::DbusVariantType>& property :
+                        properties2)
+                    {
+                        if (!property.first.compare("IntervalInMin"))
+                        {
+                            const uint64_t* i =
+                            std::get_if<uint64_t>(
+                                &property.second);
+
+                            if (i)
+                            {
+                                nlohmann::json& value2 =
+                                    sensorJson2["PowerMetrics"]["IntervalInMin"];
+                                value2 = *i;
+                            }
+                        }
+                        else if (!property.first.compare("MinConsumedWatts"))
+                        {
+                            const uint16_t* i =
+                            std::get_if<uint16_t>(
+                                &property.second);
+                            if (i)
+                            {
+                                nlohmann::json& value2 =
+                                    sensorJson2["PowerMetrics"]["MinConsumedWatts"];
+                                value2 = *i;
+                            }
+                        }
+                        else if (!property.first.compare("MaxConsumedWatts"))
+                        {
+                            const uint16_t* i =
+                            std::get_if<uint16_t>(
+                                &property.second);
+                            if (i)
+                            {
+                                nlohmann::json& value2 =
+                                    sensorJson2["PowerMetrics"]["MaxConsumedWatts"];
+                                value2 = *i;
+                            }
+                        }
+                        else if (!property.first.compare("AverageConsumedWatts"))
+                        {
+                            const uint16_t* i =
+                            std::get_if<uint16_t>(
+                                &property.second);
+                            if (i)
+                            {
+                                nlohmann::json& value2 =
+                                    sensorJson2["PowerMetrics"]["AverageConsumedWatts"];
+                                value2 = *i;
+                            }
+                        }
+                    }
+                };
+                crow::connections::systemBus->async_method_call(
+                    std::move(PowerMetricHandler), "xyz.openbmc_project.NodeManagerProxy",
+                    "/xyz/openbmc_project/Power/PowerMetric",
+                    "org.freedesktop.DBus.Properties", "GetAll",
+                    "xyz.openbmc_project.Power.PowerMetric");
             };
 
             crow::connections::systemBus->async_method_call(
-- 
2.17.1

