Upstream-Status: Pending

Subject: functions: avoid exit 1 which causes init scripts to fail

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 rc.d/init.d/functions | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: initscripts-9.72/rc.d/init.d/functions
===================================================================
--- initscripts-9.72.orig/rc.d/init.d/functions
+++ initscripts-9.72/rc.d/init.d/functions
@@ -686,6 +686,6 @@ if [ "$_use_systemctl" = "1" ]; then
     fi
 fi
 
-strstr "$(cat /proc/cmdline)" "rc.debug" && set -x
+strstr "$(cat /proc/cmdline)" "rc.debug" && set -x || true
 return 0
 
