From e46b056282c8420f096d5c34d78c00f816788784 Mon Sep 17 00:00:00 2001
From: Fan Xin <fan.xin@jp.fujitsu.com>
Date: Mon, 5 Jun 2017 16:26:47 +0900
Subject: [PATCH 1/2] Upstream-Status: Inappropriate [configuration]

Signed-off-by: Xiaofeng Yan <xiaofeng.yan@windriver.com>
Signed-off-by: Saul Wold <sgw@linux.intel.com>

Rebase on 9.72

Signed-off-by: Fan Xin <fan.xin@jp.fujitsu.com>
Upstream-Status: Pending

---
 rc.d/init.d/functions | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/rc.d/init.d/functions b/rc.d/init.d/functions
index 2e3da964..1a204dec 100644
--- a/rc.d/init.d/functions
+++ b/rc.d/init.d/functions
@@ -59,7 +59,7 @@ systemctl_redirect () {
 [ -z "${COLUMNS:-}" ] && COLUMNS=80
 
 if [ -z "${CONSOLETYPE:-}" ]; then
-    if [ -c "/dev/stderr" ] && [ -r "/dev/stderr" ]; then
+    if [ -c "/dev/stderr" ] && [ -r "/dev/stderr" ] && [ -e /sbin/consoletype ]; then
         CONSOLETYPE="$(/sbin/consoletype < /dev/stderr 2>/dev/null)"
     else
         CONSOLETYPE="serial"
-- 
2.15.1

