From aa19cb131c953e4520be5284d86964f371ee059f Mon Sep 17 00:00:00 2001
From: Yat Ho <lagoho7@gmail.com>
Date: Mon, 13 May 2024 16:21:28 +0800
Subject: [PATCH] miniupnpc: bump CMake version to 3.14

Upstream-Status: Backport [https://github.com/transmission/miniupnpc/commit/e7b0fa83c313c50a66da76ad6fb446e5bfdb482f]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0a53e27..2c713b5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required (VERSION 2.6)
+cmake_minimum_required (VERSION 3.14)
 
 project (miniupnpc C)
 set (MINIUPNPC_VERSION 2.0)
