From 9b66394c712ec0d0fcb2052baa7f590621a53461 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Tue, 8 Nov 2016 13:15:46 -0500
Subject: [PATCH] selinux: drop selinux support

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 src/systemdhook.c |   12 ------------
 1 file changed, 12 deletions(-)

--- a/src/systemdhook.c
+++ b/src/systemdhook.c
@@ -16,7 +16,6 @@
 #include <errno.h>
 #include <inttypes.h>
 #include <linux/limits.h>
-#include <selinux/selinux.h>
 #include <yajl/yajl_tree.h>
 #include <stdbool.h>
 
@@ -129,9 +128,6 @@ static int chperm(const char *path, cons
 				closedir(dir);
 				return -1;
 			}
-			if (setfilecon (full_path, label) < 0) {
-				pr_perror("Failed to set context %s on %s", label, full_path);
-			}
 
 			if (doChown) {
 				/* Change uid and gid to something the container can handle */
@@ -496,14 +492,6 @@ static int prestart(const char *rootfs,
 				return -1;
 			}
 		}
-
-		if (strcmp("", mount_label)) {
-			rc = setfilecon(journal_dir, (security_context_t)mount_label);
-			if (rc < 0) {
-				pr_perror("Failed to set journal dir selinux context");
-				return -1;
-			}
-		}
 
 		/* Attempt to creare /var/log/journal inside of rootfs,
 		   if successful, or directory exists, mount tmpfs on top of
