Add configure check for gtk2+ and objc++

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Pending

Index: boinc-client_release-7.6-7.6.33/configure.ac
===================================================================
--- boinc-client_release-7.6-7.6.33.orig/configure.ac
+++ boinc-client_release-7.6-7.6.33/configure.ac
@@ -64,6 +64,7 @@ m4_ifdef([AC_PROG_OBJCXX],[AC_PROG_OBJCX
 dnl ------
 AC_PROG_F77
 AC_PROG_CPP
+AC_PROG_OBJCXX
 AC_PROG_MAKE_SET
 SAH_LINKS
 AC_LANG_PUSH(C)
@@ -1081,6 +1082,8 @@ AM_CONDITIONAL(ENABLE_BOINCCRYPT, [test
 AM_CONDITIONAL(INSTALL_HEADERS, [test "${enable_install_headers}" = yes])
 AM_CONDITIONAL(HAVE_CUDA_LIB, [test "${enable_client}" = yes -a -f ./coprocs/CUDA/posix/${boinc_platform}/libcudart.so])
 
+PKG_CHECK_MODULES([GTK2], [gtk+-2.0])
+
 dnl ======================================================================
 dnl some more vodoo required for building portable client-binary (client, clientgui)
 dnl ======================================================================
