From daabb4110661c4358ec73293d5c0b2106f567c1f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Sun, 23 Oct 2016 23:54:14 +0200
Subject: [PATCH] Makefile.am: remove useless and breaking code
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Pending

Signed-off-by: Andreas Müller <schnitzeltony@googlemail.com>
---
 Makefile.am | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 89a8a7f..e582a34 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -24,15 +24,6 @@ EXTRA_DIST = \
 	m4 \
 	$(NULL)
 
-# Print out an informative summary.
-all-local:
-	@$(ECHO) "Done."
-	@$(ECHO)
-	@if test "x$(MAKECMDGOALS)" = "xall-am" -o "x$(.TARGETS)" = "xall-am" -o "x$(MAKECMDGOALS)" = "x" -o "x$(.TARGETS)" = "x" ; then \
-	  $(ECHO) "---" ;\
-	  $(ECHO) "Run 'make install' to begin installation into $(prefix)" ;\
-	fi
-	@$(ECHO)
 
 # Upload documentation
 DOC = docs/html docs/latex/ftgl.pdf
-- 
2.5.5

