From 8ce50bf569ec2d931735180079bbe507730626c8 Mon Sep 17 00:00:00 2001
From: Yann Dirson <yann@blade-group.com>
Date: Mon, 7 Sep 2020 18:17:21 +0200
Subject: [PATCH] Remove buggy test confusing host and target

---
Upstream-Status: Pending

 src/libcec/cmake/CheckPlatformSupport.cmake | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/src/libcec/cmake/CheckPlatformSupport.cmake b/src/libcec/cmake/CheckPlatformSupport.cmake
index 1d99bd8..31dac4f 100644
--- a/src/libcec/cmake/CheckPlatformSupport.cmake
+++ b/src/libcec/cmake/CheckPlatformSupport.cmake
@@ -266,14 +266,6 @@ else()
                 DESTINATION python/cec)
       endif()
     else()
-      if(EXISTS "/etc/os-release")
-        file(READ "/etc/os-release" OS_RELEASE)
-        string(REGEX MATCH "ID(_LIKE)?=debian" IS_DEBIAN ${OS_RELEASE})
-        if (IS_DEBIAN)
-          SET(PYTHON_PKG_DIR "dist-packages")
-        endif()
-      endif()
-
       if (NOT PYTHON_PKG_DIR)
         SET(PYTHON_PKG_DIR "site-packages")
       endif()
-- 
2.28.0

