From da235923d178ffbaa72ebed9959468a8fedb72a3 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Tue, 16 Aug 2022 16:02:14 +0800
Subject: [PATCH 11/21] dbus-sdr: do not replace '_' for sensor name

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 dbus-sdr/sensorcommands.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dbus-sdr/sensorcommands.cpp b/dbus-sdr/sensorcommands.cpp
index 281a7891..ca22c5c0 100644
--- a/dbus-sdr/sensorcommands.cpp
+++ b/dbus-sdr/sensorcommands.cpp
@@ -474,7 +474,7 @@ std::string parseSdrIdFromPath(const std::string& path)
             name.resize(FULL_RECORD_ID_STR_MAX_LENGTH);
         }
     }
-    std::replace(name.begin(), name.end(), '_', ' ');
+    //std::replace(name.begin(), name.end(), '_', ' ');
     return name;
 }
 
-- 
2.34.1

