From c37e9853c38dc7b22fa0417ad11ca34023e2563f Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Tue, 16 Aug 2022 16:22:58 +0800
Subject: [PATCH 14/21] fix percentage type show

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 dbus-sdr/sensorcommands.cpp | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/dbus-sdr/sensorcommands.cpp b/dbus-sdr/sensorcommands.cpp
index 3967d5f1..5039a948 100644
--- a/dbus-sdr/sensorcommands.cpp
+++ b/dbus-sdr/sensorcommands.cpp
@@ -1591,6 +1591,18 @@ void constructSensorSdrHeaderKey(uint16_t sensorNum, uint16_t recordID,
     record.key.owner_lun = lun;
     record.key.sensor_number = sensornumber;
 }
+
+bool isPercentage(DbusInterfaceMap& sensorMap)
+{
+    const static std::string PERCENT =
+        "xyz.openbmc_project.Sensor.Value.Unit.Percent";
+    auto sensorObject = sensorMap.find(sensor::sensorInterface);
+    std::string unit;
+    unit = ipmi::mappedVariant<std::string>(sensorObject->second, "Unit",
+                                            std::string());
+    return unit == PERCENT;
+}
+
 bool constructSensorSdr(ipmi::Context::ptr ctx, uint16_t sensorNum,
                         uint16_t recordID, const std::string& service,
                         const std::string& path,
@@ -1700,7 +1712,14 @@ bool constructSensorSdr(ipmi::Context::ptr ctx, uint16_t sensorNum,
         (rExpSign << 7) | (rExpBits << 4) | (bExpSign << 3) | bExpBits;
 
     // Set the analog reading byte interpretation accordingly
-    record.body.sensor_units_1 = (bSigned ? 1 : 0) << 7;
+    if (isPercentage(sensorMap))
+    {
+        get_sdr::body::set_percentage(&record.body);
+    }
+    else
+    {
+        record.body.sensor_units_1 = (bSigned ? 1 : 0) << 7;
+    }
 
     // TODO(): Perhaps care about Tolerance, Accuracy, and so on
     // These seem redundant, but derivable from the above 5 attributes
-- 
2.17.1

