From bebbd79e8b3d81030bfa19a97d682583c6967105 Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Fri, 11 Nov 2022 15:55:51 +0800
Subject: [PATCH 14/21] fix percentage type show

Signed-off-by: Joseph Liu <kwliu@nuvoton.com>
---
 dbus-sdr/sensorcommands.cpp | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/dbus-sdr/sensorcommands.cpp b/dbus-sdr/sensorcommands.cpp
index ca22c5c0..d54873af 100644
--- a/dbus-sdr/sensorcommands.cpp
+++ b/dbus-sdr/sensorcommands.cpp
@@ -1605,6 +1605,18 @@ void constructSensorSdrHeaderKey(uint16_t sensorNum, uint16_t recordID,
     record.key.owner_lun = lun;
     record.key.sensor_number = sensornumber;
 }
+
+bool isPercentage(DbusInterfaceMap& sensorMap)
+{
+    const static std::string PERCENT =
+        "xyz.openbmc_project.Sensor.Value.Unit.Percent";
+    auto sensorObject = sensorMap.find(sensor::sensorInterface);
+    std::string unit;
+    unit = ipmi::mappedVariant<std::string>(sensorObject->second, "Unit",
+                                            std::string());
+    return unit == PERCENT;
+}
+
 bool constructSensorSdr(
     ipmi::Context::ptr ctx,
     const std::unordered_set<std::string>& ipmiDecoratorPaths,
@@ -1716,7 +1728,14 @@ bool constructSensorSdr(
         (rExpSign << 7) | (rExpBits << 4) | (bExpSign << 3) | bExpBits;
 
     // Set the analog reading byte interpretation accordingly
-    record.body.sensor_units_1 = (bSigned ? 1 : 0) << 7;
+    if (isPercentage(sensorMap))
+    {
+        get_sdr::body::set_percentage(&record.body);
+    }
+    else
+    {
+        record.body.sensor_units_1 = (bSigned ? 1 : 0) << 7;
+    }
 
     // TODO(): Perhaps care about Tolerance, Accuracy, and so on
     // These seem redundant, but derivable from the above 5 attributes
-- 
2.34.1

