From e00e62cd0c56576a8840b8deeeebe61203ffd4a0 Mon Sep 17 00:00:00 2001
From: Tim Lee <timlee660101@gmail.com>
Date: Tue, 26 Jul 2022 16:08:48 +0800
Subject: [PATCH 2/2] hwmon temp: add tmp100 support

Signed-off-by: Tim Lee <timlee660101@gmail.com>
---
 src/HwmonTempMain.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/HwmonTempMain.cpp b/src/HwmonTempMain.cpp
index 565ab7cb..cc8b3f91 100644
--- a/src/HwmonTempMain.cpp
+++ b/src/HwmonTempMain.cpp
@@ -72,7 +72,8 @@ static auto sensorTypes{
                                 "xyz.openbmc_project.Configuration.TMP441",
                                 "xyz.openbmc_project.Configuration.TMP75",
                                 "xyz.openbmc_project.Configuration.W83773G",
-                                "xyz.openbmc_project.Configuration.BMCOnDieThermalSensor"})};
+                                "xyz.openbmc_project.Configuration.BMCOnDieThermalSensor",
+                                "xyz.openbmc_project.Configuration.TMP100"})};
 
 static struct SensorParams
     getSensorParameters(const std::filesystem::path& path)
-- 
2.17.1

