From da433dbe79f2d4d5d7d79869c669594c99c5de9c Mon Sep 17 00:00:00 2001
From: Jasper Orschulko <jasper@fancydomain.eu>
Date: Wed, 16 Jun 2021 19:00:30 +0200
Subject: [PATCH] Add CC tag to build

Upstream-Status: Pending
Signed-off-by: Jasper Orschulko <jasper@fancydomain.eu>
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 5e1d37dd..f7a6dece 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -36,7 +36,7 @@ AUTOMAKE_OPTIONS = \
     subdir-objects
 
 ACLOCAL_AMFLAGS = -I m4
-LIBTOOLFLAGS = --verbose
+LIBTOOLFLAGS = --verbose --tag=CC
 
 SUBDIRS = lib # lib goes first to build first
 if WITH_EXAMPLES
-- 
2.32.0

