diff --git a/mesonbuild/dependencies.py b/mesonbuild/dependencies.py
index da73a57b7ecd..22100f69c580 100644
--- a/mesonbuild/dependencies.py
+++ b/mesonbuild/dependencies.py
@@ -65,7 +65,7 @@ class Dependency():
     def need_threads(self):
         return False
 
-    def get_pkgconfig_variable(self, variable_name):
+    def get_pkgconfig_variable(self, variable_name, use_native=False):
         raise MesonException('Tried to get a pkg-config variable from a non-pkgconfig dependency.')
 
 class InternalDependency(Dependency):
@@ -177,8 +177,12 @@ class PkgConfigDependency(Dependency):
         return s.format(self.__class__.__name__, self.name, self.is_found,
                         self.version_reqs)
 
-    def _call_pkgbin(self, args):
-        p, out = Popen_safe([self.pkgbin] + args, env=os.environ)[0:2]
+    def _call_pkgbin(self, args, use_native=False):
+        if use_native:
+            pkgbin = [self.pkgbin + "-native"]
+        else:
+            pkgbin = [self.pkgbin]
+        p, out = Popen_safe(pkgbin + args, env=os.environ)[0:2]
         return (p.returncode, out.strip())
 
     def _set_cargs(self):
@@ -212,8 +216,8 @@ class PkgConfigDependency(Dependency):
                 self.is_libtool = True
             self.libs.append(lib)
 
-    def get_pkgconfig_variable(self, variable_name):
-        ret, out = self._call_pkgbin(['--variable=' + variable_name, self.name])
+    def get_pkgconfig_variable(self, variable_name, use_native=False):
+        ret, out = self._call_pkgbin(['--variable=' + variable_name, self.name], use_native=use_native)
         variable = ''
         if ret != 0:
             if self.required:
@@ -933,10 +937,10 @@ class QtBaseDependency(Dependency):
             corekwargs = {'required': 'false', 'silent': 'true'}
             core = PkgConfigDependency(self.qtpkgname + 'Core', env, corekwargs)
         # Used by self.compilers_detect()
-        self.bindir = core.get_pkgconfig_variable('host_bins')
+        self.bindir = core.get_pkgconfig_variable('host_bins', use_native=True)
         if not self.bindir:
             # If exec_prefix is not defined, the pkg-config file is broken
-            prefix = core.get_pkgconfig_variable('exec_prefix')
+            prefix = core.get_pkgconfig_variable('exec_prefix', use_native=True)
             if prefix:
                 self.bindir = os.path.join(prefix, 'bin')
 
