From 84bb692d6b047c09266de154f404af9817fa04aa Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Thu, 29 Mar 2018 15:59:05 +0800
Subject: [PATCH 01/10] contrib.mak: fix for parallel build

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>

Rebase to 9.23
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 contrib/contrib.mak | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/contrib/contrib.mak b/contrib/contrib.mak
index f5a2fa4..4999752 100644
--- a/contrib/contrib.mak
+++ b/contrib/contrib.mak
@@ -1067,6 +1067,7 @@ $(DEVOBJ)dviprlib.$(OBJ) : $(JAPSRC)dviprlib.c $(JAPSRC)dviprlib.h \
 	$(DEVCC) $(O_)$@ $(C_) $(JAPSRC)dviprlib.c
 
 extra-dmprt-install: install-libdata
+	mkdir -p $(DESTDIR)$(gsdatadir)$(D)lib
 	$(INSTALL_DATA) $(JAPSRC)dmp_init.ps $(DESTDIR)$(gsdatadir)$(D)lib || exit 1
 	$(INSTALL_DATA) $(JAPSRC)dmp_site.ps $(DESTDIR)$(gsdatadir)$(D)lib || exit 1
 	$(INSTALL_DATA) $(JAPSRC)escp_24.src $(DESTDIR)$(gsdatadir)$(D)lib || exit 1
@@ -1235,6 +1236,7 @@ $(DEVOBJ)gdevalps.$(OBJ) : $(JAPSRC)gdevalps.c $(PDEVH) \
 ### ----------------- Additional .upp files ---------------- ###
 
 extra-upp-install: install-libdata
+	mkdir -p $(DESTDIR)$(gsdatadir)$(D)lib
 	for f in $(CONTRIBSRC)uniprint$(D)*.upp; do \
 	    $(INSTALL_DATA) $$f $(DESTDIR)$(gsdatadir)$(D)lib || exit 1; \
 	done
-- 
1.8.3.1

