From 6c5cf9f87a9047bc56ccd8b6096b8ec4a6af96ab Mon Sep 17 00:00:00 2001
From: Patrick Williams <patrick@stwcx.xyz>
Date: Fri, 5 Aug 2016 16:53:21 -0500
Subject: [PATCH 2/2] external: Utilize DESTDIR in shared makefile

Signed-off-by: Patrick Williams <patrick@stwcx.xyz>
---
 external/shared/Makefile | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/external/shared/Makefile b/external/shared/Makefile
index e0b3ff3..4baa6b4 100644
--- a/external/shared/Makefile
+++ b/external/shared/Makefile
@@ -39,15 +39,15 @@ all: links arch_links $(OBJS)
 	$(CC) -shared -Wl,-soname,libflash.so -o $(SHARED_NAME) $(OBJS)
 
 install-lib: all
-	install -D -m 0755 $(SHARED_NAME) $(LIBDIR)/$(SHARED_NAME)
-	ln -sf $(SHARED_NAME) $(LIBDIR)/libflash.so
+	install -D -m 0755 $(SHARED_NAME) $(DESTDIR)$(LIBDIR)/$(SHARED_NAME)
+	ln -sf $(SHARED_NAME) $(DESTDIR)$(LIBDIR)/libflash.so
 
 install-dev: links arch_links
-	mkdir -p $(INCDIR)
-	install -m 0644 $(LIBFLASH_H) $(ARCHFLASH_H) $(INCDIR)
+	mkdir -p $(DESTDIR)$(INCDIR)
+	install -m 0644 $(LIBFLASH_H) $(ARCHFLASH_H) $(DESTDIR)$(INCDIR)
 
 install: install-lib install-dev
 
 uninstall:
-	rm -f $(LIBDIR)/libflash*
-	rm -rf $(INCDIR)
+	rm -f $(DESTDIR)$(LIBDIR)/libflash*
+	rm -rf $(DESTDIR)$(INCDIR)
-- 
2.9.0

