From 4e8eab047a43e48bd541f2887104299fcfb99b0d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Tue, 27 Nov 2018 12:03:20 +0100
Subject: [PATCH] Use native mtp-hotplug
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate [OE-specific]

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>
---
 Makefile.am | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 7b7e06b..083e1c6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -16,16 +16,16 @@ hwdb_DATA=69-libmtp.hwdb
 noinst_DATA=libmtp.usermap libmtp.fdi
 
 libmtp.usermap: util/mtp-hotplug
-	util/mtp-hotplug > libmtp.usermap
+	mtp-hotplug > libmtp.usermap
 
 @UDEV_RULES@: util/mtp-hotplug
-	util/mtp-hotplug -u -p"@UDEV@" @UDEV_GROUP@ @UDEV_MODE@ > @UDEV_RULES@
+	mtp-hotplug -u -p"@UDEV@" @UDEV_GROUP@ @UDEV_MODE@ > @UDEV_RULES@
 
 libmtp.fdi: util/mtp-hotplug
-	util/mtp-hotplug -H > libmtp.fdi
+	mtp-hotplug -H > libmtp.fdi
 
 $(hwdb_DATA): util/mtp-hotplug
-	util/mtp-hotplug -w > $(hwdb_DATA)
+	mtp-hotplug -w > $(hwdb_DATA)
 
 CLEANFILES = libmtp.usermap @UDEV_RULES@ libmtp.fdi libmtp.hwdb
 endif
-- 
2.14.5

