From e117ccc2b4ebb116ab63a3aa6097104964ee223b Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Mon, 19 Apr 2021 23:02:45 -0700
Subject: [PATCH] do not build tests sub directory

Upstream-Status: Inappropriate [OE Specific]

Signed-off-by: Hongxu Jia <Hongxu.Jia@windriver.com>
Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 468f4b54..c47c9c0e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -351,9 +351,6 @@ if(BUILD_VIEWER)
   add_subdirectory(media)
 endif()
 
-add_subdirectory(tests)
-
-
 if(BUILD_VIEWER)
   add_subdirectory(release)
 endif()
-- 
2.25.1

