From 076d0e12a7be6cd2108e4ca0dcde1cb658918fa5 Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Mon, 19 Apr 2021 23:02:45 -0700
Subject: [PATCH] do not build tests sub directory

Upstream-Status: Inappropriate [OE Specific]

Signed-off-by: Hongxu Jia <Hongxu.Jia@windriver.com>
Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 CMakeLists.txt | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7bf99441..bda80598 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -304,9 +304,6 @@ if(BUILD_VIEWER)
   add_subdirectory(media)
 endif()
 
-add_subdirectory(tests)
-
-
 add_subdirectory(release)
 
 # uninstall
-- 
2.30.2

