From 23f827ea94284656972652f61f2c6aad96092cc8 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 20 Jul 2017 23:00:29 -0400
Subject: [PATCH] fix compile failure against musl C library

Upstream-Status: Pending

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 lib/rpm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/lib/rpm.c
+++ b/lib/rpm.c
@@ -34,6 +34,11 @@
 #include <assert.h>
 #include <string.h>
 
+#if !defined(__GLIBC__)
+typedef int (*__compar_fn_t) (const void*, const void*);
+typedef __compar_fn_t comparison_fn_t;
+#endif
+
 struct sr_rpm_package *
 sr_rpm_package_new()
 {
