Upstream-Status: Pending

upstream: not applicable
source: stolen from debian

Index: anthy-9100h/src-util/Makefile.am
===================================================================
--- anthy-9100h.orig/src-util/Makefile.am
+++ anthy-9100h/src-util/Makefile.am
@@ -8,7 +8,8 @@ ELISP_FILES = anthy.el anthy-dic.el anth
  anthy-isearch.el anthy-azik.el anthy-kyuri.el
 EXTRA_DIST = $(ELISP_FILES) typetab dic-tool-usage.txt anthy.i
 bin_PROGRAMS = anthy-dic-tool anthy-agent anthy-morphological-analyzer
-ELCFILES = anthy.elc anthy-dic.elc anthy-azik.elc anthy-conf.elc anthy-isearch.elc anthy-kyuri.elc leim-list.elc
+#ELCFILES = anthy.elc anthy-dic.elc anthy-azik.elc anthy-conf.elc anthy-isearch.elc anthy-kyuri.elc leim-list.elc
+ELCFILES = 
 if ELISP
 lisp_LISP = $(ELISP_FILES)
 endif
