diff -Nurp anthy-9100h.org/calctrans/Makefile.am anthy-9100h/calctrans/Makefile.am
--- anthy-9100h.org/calctrans/Makefile.am	2007-10-27 23:02:59.000000000 +0900
+++ anthy-9100h/calctrans/Makefile.am	2014-11-11 17:14:42.152999991 +0900
@@ -15,31 +15,31 @@ dict_source_files = anthy.cand_info anth
 
 update_params:
 	rm -f parsed_data2
-	./proccorpus $(srcdir)/corpus.?.txt > parsed_data
-	./calctrans parsed_data -o $(srcdir)/corpus_info
-	./calctrans parsed_data -e -o $(srcdir)/weak_words
+	proccorpus $(srcdir)/corpus.?.txt > parsed_data
+	calctrans parsed_data -o $(srcdir)/corpus_info
+	calctrans parsed_data -e -o $(srcdir)/weak_words
 	make do_update_params
 
 update_params0:
 	rm -f parsed_data2
-	./proccorpus $(srcdir)/corpus.0.txt > parsed_data
-	./calctrans parsed_data -o $(srcdir)/corpus_info
-	./calctrans parsed_data -e -o $(srcdir)/weak_words
+	proccorpus $(srcdir)/corpus.0.txt > parsed_data
+	calctrans parsed_data -o $(srcdir)/corpus_info
+	calctrans parsed_data -e -o $(srcdir)/weak_words
 	make do_update_params
 
 update_params2:
-	./proccorpus $(srcdir)/corpus.?.txt >> parsed_data2
-	./calctrans parsed_data parsed_data2 -o $(srcdir)/corpus_info
-	./calctrans parsed_data parsed_data2 -e -o $(srcdir)/weak_words
+	proccorpus $(srcdir)/corpus.?.txt >> parsed_data2
+	calctrans parsed_data parsed_data2 -o $(srcdir)/corpus_info
+	calctrans parsed_data parsed_data2 -e -o $(srcdir)/weak_words
 	make do_update_params
 
 do_update_params:
-	./calctrans -c $(srcdir)/corpus_info
+	calctrans -c $(srcdir)/corpus_info
 	rm -f $(dict_source_files)
 	make $(dict_source_files)
 
-$(dict_source_files): $(srcdir)/corpus_info $(srcdir)/weak_words calctrans
-	./calctrans -c $(srcdir)/corpus_info $(srcdir)/weak_words
+$(dict_source_files): $(srcdir)/corpus_info $(srcdir)/weak_words
+	calctrans -c $(srcdir)/corpus_info $(srcdir)/weak_words
 
 noinst_DATA = $(dict_source_files)
 CLEANFILES = $(dict_source_files) parsed_data parsed_data2 anthy.feature_info
diff -Nurp anthy-9100h.org/depgraph/Makefile.am anthy-9100h/depgraph/Makefile.am
--- anthy-9100h.org/depgraph/Makefile.am	2008-11-29 21:46:14.000000000 +0900
+++ anthy-9100h/depgraph/Makefile.am	2014-11-11 17:12:03.813999991 +0900
@@ -9,11 +9,11 @@ CLEANFILES = anthy.dep
 EXTRA_DIST = indepword.txt $(DEPWORDS)
 
 # Generate the dictionary
-noinst_PROGRAMS = mkdepgraph
-mkdepgraph_SOURCES = mkdepgraph.c
-mkdepgraph_LDADD =  ../src-main/libanthy.la ../src-worddic/libanthydic.la
+#noinst_PROGRAMS = mkdepgraph
+#mkdepgraph_SOURCES = mkdepgraph.c
+#mkdepgraph_LDADD =  ../src-main/libanthy.la ../src-worddic/libanthydic.la
 
-anthy.dep : mkdepgraph $(DEPWORDS)
-	./mkdepgraph
+anthy.dep : $(DEPWORDS)
+	mkdepgraph
 
 noinst_DATA = anthy.dep
diff -Nurp anthy-9100h.org/mkanthydic/Makefile.am anthy-9100h/mkanthydic/Makefile.am
--- anthy-9100h.org/mkanthydic/Makefile.am	2007-04-14 18:01:10.000000000 +0900
+++ anthy-9100h/mkanthydic/Makefile.am	2014-11-11 17:10:34.792999991 +0900
@@ -3,12 +3,12 @@ noinst_SCRIPTS =
 CLEANFILES = anthy.dic
 INCLUDES = -I$(top_srcdir)/ -DSRCDIR=\"$(srcdir)\"
 
-noinst_PROGRAMS = mkfiledic
-mkfiledic_SOURCES = mkfiledic.c
-mkfiledic_LDADD = ../src-diclib/libdiclib.la
+#noinst_PROGRAMS = mkfiledic
+#mkfiledic_SOURCES = mkfiledic.c
+#mkfiledic_LDADD = ../src-diclib/libdiclib.la
 
-anthy.dic : mkfiledic ../mkworddic/anthy.wdic ../depgraph/anthy.dep ../calctrans/anthy.cand_info ../calctrans/anthy.trans_info ../calctrans/anthy.corpus_array ../calctrans/anthy.corpus_bucket
-	./mkfiledic
+anthy.dic : ../mkworddic/anthy.wdic ../depgraph/anthy.dep ../calctrans/anthy.cand_info ../calctrans/anthy.trans_info ../calctrans/anthy.corpus_array ../calctrans/anthy.corpus_bucket
+	mkfiledic
 
 
 # To install 
diff -Nurp anthy-9100h.org/mkworddic/Makefile.am anthy-9100h/mkworddic/Makefile.am
--- anthy-9100h.org/mkworddic/Makefile.am	2009-01-23 12:31:31.000000000 +0900
+++ anthy-9100h/mkworddic/Makefile.am	2014-11-11 17:13:31.946999992 +0900
@@ -26,14 +26,14 @@ DIC_FILES = @top_srcdir@/alt-cannadic/gc
 	    @top_srcdir@/mkworddic/udict
 
 # Generate the dictionary
-noinst_PROGRAMS = mkworddic
-mkworddic_SOURCES = mkdic.c writewords.c mkudic.c calcfreq.c mkdic.h
-mkworddic_LDADD = ../src-worddic/libanthydic.la
+#noinst_PROGRAMS = mkworddic
+#mkworddic_SOURCES = mkdic.c writewords.c mkudic.c calcfreq.c mkdic.h
+#mkworddic_LDADD = ../src-worddic/libanthydic.la
 
 noinst_DATA = anthy.wdic
 
-anthy.wdic : mkworddic $(DIC_FILES)
-	   ./mkworddic -f ./dict.args
+anthy.wdic : $(DIC_FILES)
+	   mkworddic -f ./dict.args
 
 
 # To install
