Disable PAM

Upstream-Status: Inappropriate [config]

Signed-off-by: Roy.Li <rongqing.li@windriver.com>
diff -ur vsftpd-2.0.1_org/builddefs.h vsftpd-2.0.1_patch/builddefs.h
--- vsftpd-2.0.1_org/builddefs.h	2004-07-02 16:36:59.000000000 +0200
+++ vsftpd-2.0.1_patch/builddefs.h	2004-07-21 09:34:49.044900488 +0200
@@ -2,7 +2,7 @@
 #define VSF_BUILDDEFS_H
 
 #define VSF_BUILD_TCPWRAPPERS
-#define VSF_BUILD_PAM
+#undef VSF_BUILD_PAM
 #undef VSF_BUILD_SSL
 
 #endif /* VSF_BUILDDEFS_H */
