From 6120cee594c543d17ae015bc601012debcf57177 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 23 Oct 2023 10:43:42 +0100
Subject: [PATCH] include string.h for strncpy()

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Backport [master: 6120cee594c543d17ae015bc601012debcf57177]
---
 interface.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/interface.c b/interface.c
index 29229ad6f53d..9a83c36933e7 100644
--- a/interface.c
+++ b/interface.c
@@ -5,6 +5,7 @@
  * @note SPDX-License-Identifier: GPL-2.0+
  */
 #include <stdlib.h>
+#include <string.h>
 #include "interface.h"
 
 struct interface {
-- 
2.34.1

