From ecfc8e287eb6811e42c4e2800a913822f554d7c8 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 31 Aug 2022 19:09:18 -0700
Subject: [PATCH] include string.h for strncpy()

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 interface.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/interface.c b/interface.c
index 29229ad..9a83c36 100644
--- a/interface.c
+++ b/interface.c
@@ -5,6 +5,7 @@
  * @note SPDX-License-Identifier: GPL-2.0+
  */
 #include <stdlib.h>
+#include <string.h>
 #include "interface.h"
 
 struct interface {
