From c73d19ffa909a865e5ff5a16921724a33765cfd1 Mon Sep 17 00:00:00 2001
From: Stanley Chu <yschu@nuvoton.com>
Date: Fri, 13 May 2022 14:07:17 +0800
Subject: [PATCH] skip POWER_BUTTON and POST_COMPLETE

Signed-off-by: Stanley Chu <yschu@nuvoton.com>
---
 src/power_control.cpp | 14 +++++++++-----
 1 file changed, 9 insertions(+), 5 deletions(-)

diff --git a/src/power_control.cpp b/src/power_control.cpp
index 05aa45c..7fbeb26 100644
--- a/src/power_control.cpp
+++ b/src/power_control.cpp
@@ -2621,7 +2621,7 @@ int main(int argc, char* argv[])
             return -1;
         }
     }
-
+#if 0
     // Request POWER_BUTTON GPIO events
     if (powerButtonConfig.type == ConfigType::GPIO)
     {
@@ -2637,7 +2637,7 @@ int main(int argc, char* argv[])
             power_control::dbusGPIOMatcher(powerButtonConfig,
                                            powerButtonHandler);
     }
-
+#endif
     // Request RESET_BUTTON GPIO events
     if (resetButtonConfig.type == ConfigType::GPIO)
     {
@@ -2691,7 +2691,7 @@ int main(int argc, char* argv[])
         "PropertiesChanged',arg0='xyz.openbmc_project.State.Host.Misc'",
         hostMiscHandler);
 #endif
-
+#if 0
     // Request POST_COMPLETE GPIO events
     if (postCompleteConfig.type == ConfigType::GPIO)
     {
@@ -2713,6 +2713,7 @@ int main(int argc, char* argv[])
             "postComplete name should be configured from json config file");
         return -1;
     }
+#endif
 
     // initialize NMI_OUT GPIO.
     if (!nmiOutConfig.lineName.empty())
@@ -3062,6 +3063,7 @@ int main(int argc, char* argv[])
     sdbusplus::asio::object_server buttonsServer =
         sdbusplus::asio::object_server(conn);
 
+#if 0
     if (!powerButtonConfig.lineName.empty())
     {
         // Power Button Interface
@@ -3116,7 +3118,7 @@ int main(int argc, char* argv[])
 
         powerButtonIface->initialize();
     }
-
+#endif
     if (!resetButtonConfig.lineName.empty())
     {
         // Reset Button Interface
@@ -3267,7 +3269,8 @@ int main(int argc, char* argv[])
     // Get the initial OS state based on POST complete
     //      0: Asserted, OS state is "Standby" (ready to boot)
     //      1: De-Asserted, OS state is "Inactive"
-    OperatingSystemStateStage osState;
+    OperatingSystemStateStage osState = OperatingSystemStateStage::Standby;
+    #if 0
     if (postCompleteConfig.type == ConfigType::GPIO)
     {
         osState = postCompleteLine.get_value() > 0
@@ -3280,6 +3283,7 @@ int main(int argc, char* argv[])
                       ? OperatingSystemStateStage::Inactive
                       : OperatingSystemStateStage::Standby;
     }
+    #endif
 
     osIface->register_property(
         "OperatingSystemState",
-- 
2.17.1

