Upstream-Status:Inappropriate [embedded specific]

Index: perl-5.24.1/installperl
===================================================================
--- perl-5.24.1.orig/installperl
+++ perl-5.24.1/installperl
@@ -2,8 +2,8 @@
 
 BEGIN {
     chdir '..' if !-d 'lib' and -d '../lib';
-    @INC = 'lib';
-    $ENV{PERL5LIB} = 'lib';
+#    @INC = 'lib';
+#    $ENV{PERL5LIB} = 'lib';
 
     # This needs to be at BEGIN time, before any use of Config
     # install_lib itself loads and imports Config into main::
