Upstream-Status: Pending

Dont assume errno type

--- ez-ipupdate-3.0.11b7/cache_file.c.orig	2014-07-02 14:01:07.126029412 -0600
+++ ez-ipupdate-3.0.11b7/cache_file.c	2014-07-02 14:08:27.422026332 -0600
@@ -43,11 +43,11 @@
 #include <cache_file.h>
 
 #if HAVE_STRERROR
-extern int errno;
+#  include <errno.h>
 #  define error_string strerror(errno)
 #elif HAVE_SYS_ERRLIST
 extern const char *const sys_errlist[];
-extern int errno;
+#  include <errno.h>
 #  define error_string (sys_errlist[errno])
 #else
 #  define error_string "error message not found"
@@ -63,11 +63,11 @@
 #  define dprintf(x)
 #endif
 #if HAVE_STRERROR
-extern int errno;
+#  include <errno.h>
 #  define error_string strerror(errno)
 #elif HAVE_SYS_ERRLIST
 extern const char *const sys_errlist[];
-extern int errno;
+#  include <errno.h>
 #  define error_string (sys_errlist[errno])
 #else
 #  define error_string "error message not found"
