Fix build with gettext 0.20.x

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Michael Opdenacker <michael.opdenacker@bootlin.com>
--- a/configure.ac
+++ b/configure.ac
@@ -34,7 +34,7 @@
 AC_PROG_LIBTOOL
 
 ALL_LINGUAS="be cs da en_GB eo es fr hr hu nl pl ro ru sk sv uk vi"
-AM_GNU_GETTEXT
+AM_GNU_GETTEXT([external])
 
 dnl --------------------------------------------------
 dnl System checks
@@ -397,7 +397,6 @@
 Makefile
 m4/Makefile
 po/Makefile.in
-intl/Makefile
 include/Makefile
 share/Makefile
 win32/Makefile
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,8 +2,8 @@
 
 AUTOMAKE_OPTIONS = foreign dist-zip
 
-SUBDIRS = po intl include share win32 @OPT_SUBDIRS@
-DIST_SUBDIRS = po intl include share win32 ogg123 oggenc oggdec ogginfo \
+SUBDIRS = po include share win32 @OPT_SUBDIRS@
+DIST_SUBDIRS = po include share win32 ogg123 oggenc oggdec ogginfo \
 	vcut vorbiscomment m4
 
 EXTRA_DIST = config.rpath README AUTHORS COPYING CHANGES
