From 71628ddc91b6efb9b922a3fcf8cc18522f5387be Mon Sep 17 00:00:00 2001
From: "Roy.Li" <rongqing.li@windriver.com>
Date: Mon, 20 Feb 2012 13:51:49 +0000
Subject: [PATCH] Disable PAM

Upstream-Status: Inappropriate [config]

Signed-off-by: Roy.Li <rongqing.li@windriver.com>

---
 builddefs.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/builddefs.h b/builddefs.h
index 0106d1a..f48a568 100644
--- a/builddefs.h
+++ b/builddefs.h
@@ -2,7 +2,7 @@
 #define VSF_BUILDDEFS_H
 
 #define VSF_BUILD_TCPWRAPPERS
-#define VSF_BUILD_PAM
+#undef VSF_BUILD_PAM
 #undef VSF_BUILD_SSL
 
 #endif /* VSF_BUILDDEFS_H */
