From d7c7cb223ddb8b816332532c9d8f79f3ba16c911 Mon Sep 17 00:00:00 2001
From: Scott Murray <scott.murray@konsulko.com>
Date: Sun, 26 Apr 2026 05:15:36 +0300
Subject: [PATCH] Fixes build issues

Upstream-Status: Inappropriate [next version has many changes to configure.ac]
Signed-off-by: Armin Kuster <akuster@mvista.com>
(Updated for 0.19.3)
Signed-off-by: Scott Murray <scott.murray@konsulko.com>
---
 configure.ac | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index eb39aaa..c538c58 100644
--- a/configure.ac
+++ b/configure.ac
@@ -14,6 +14,7 @@ dnl The name of the configure h-file.
 AC_CONFIG_HEADERS(include/config.h)
 
 dnl Checks for programs.
+m4_pattern_allow([AC_MSG_ERROR])
 AC_PROG_CC
 if test "x$ac_cv_prog_cc_c99" = xno; then
     AC_MSG_ERROR([AIDE needs a C99 compatible compiler])
@@ -282,8 +283,6 @@ AC_CHECK_HEADERS(syslog.h inttypes.h fcntl.h ctype.h)
 
 AIDE_PKG_CHECK_MANDATORY(pcre2, PCRE2, libpcre2-8)
 
-AX_PTHREAD(compoptionstring="${compoptionstring}use pthread: mandatory\\n", [AC_MSG_ERROR([AIDE requires pthread])])
-
 AC_ARG_WITH([fstype],
             AS_HELP_STRING([--without-fstype],
             [Disable file system type support for restricted rules (Linux only)]),
-- 
2.47.3

