From 1d8ae4b32bd76c19ec238f30eb9b1ee582cbe990 Mon Sep 17 00:00:00 2001
From: Jackie Huang <jackie.huang@windriver.com>
Date: Fri, 2 Mar 2018 01:10:58 -0800
Subject: [PATCH] xmlsec1: fix a typo in examples/verify3.c

Upstream-Status: Submitted [https://github.com/lsh123/xmlsec/pull/153]

Signed-off-by: Jackie Huang <jackie.huang@windriver.com>

---
 examples/verify3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/examples/verify3.c b/examples/verify3.c
index 2d26ae7..68f52ab 100644
--- a/examples/verify3.c
+++ b/examples/verify3.c
@@ -1,4 +1,4 @@
-4/** 
+/**
  * XML Security Library example: Verifying a file signed with X509 certificate
  *
  * Verifies a file signed with X509 certificate. 
