From c212d228c25cb583f52a6d31e9f0ec7bc1f9c506 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 26 Mar 2022 01:10:02 -0700
Subject: [PATCH] util: Correct path to iniparser.h

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 util/parse-configs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/parse-configs.c b/util/parse-configs.c
index c834a07..1b7ffa6 100644
--- a/util/parse-configs.c
+++ b/util/parse-configs.c
@@ -4,7 +4,7 @@
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
-#include <iniparser.h>
+#include <iniparser/iniparser.h>
 #include <sys/stat.h>
 #include <util/parse-configs.h>
 #include <util/strbuf.h>
-- 
2.35.1

