From 391a2bdea7adea3c6d7abde82f2f1289e36f3aae Mon Sep 17 00:00:00 2001
From: Sergey Solomin <sergey.solomin@us.ibm.com>
Date: Wed, 31 Aug 2016 17:16:30 -0500
Subject: [PATCH 1/1] 4 byte read support #466

---
 tools/i2cdump.c | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)

diff --git a/tools/i2cdump.c b/tools/i2cdump.c
index a7bba72..70af678 100644
--- a/tools/i2cdump.c
+++ b/tools/i2cdump.c
@@ -33,6 +33,9 @@
 #include "util.h"
 #include "../version.h"
 
+#define I2C_SMBUS_DWORD 7
+#define N_BYTES_TO_READ 4
+
 static void help(void)
 {
 	fprintf(stderr,
@@ -46,6 +49,7 @@ static void help(void)
 		"    s (SMBus block)\n"
 		"    i (I2C block)\n"
 		"    c (consecutive byte)\n"
+		"    d (double word)\n"
 		"    Append p for SMBus PEC\n");
 }
 
@@ -184,6 +188,9 @@ int main(int argc, char *argv[])
 	} else if (!strncmp(argv[flags+3], "c", 1)) {
 		size = I2C_SMBUS_BYTE;
 		pec = argv[flags+3][1] == 'p';
+	} else if (!strncmp(argv[flags+3], "d", 1)) {
+		size = I2C_SMBUS_DWORD;
+		pec = argv[flags+3][1] == 'p';
 	} else if (!strcmp(argv[flags+3], "i"))
 		size = I2C_SMBUS_I2C_BLOCK_DATA;
 	else {
@@ -285,6 +292,7 @@ int main(int argc, char *argv[])
 			size == I2C_SMBUS_BLOCK_DATA ? "smbus block" :
 			size == I2C_SMBUS_I2C_BLOCK_DATA ? "i2c block" :
 			size == I2C_SMBUS_BYTE ? "byte consecutive read" :
+			size == I2C_SMBUS_DWORD ? "double word" :
 			size == I2C_SMBUS_BYTE_DATA ? "byte" : "word");
 		if (pec)
 			fprintf(stderr, "PEC checking enabled.\n");
@@ -313,6 +321,32 @@ int main(int argc, char *argv[])
 		}
 	}
 
+	/* handle mode 'd' (double word read) */
+	if (size == I2C_SMBUS_DWORD) {
+		unsigned char buff[N_BYTES_TO_READ];
+		struct i2c_rdwr_ioctl_data msgset;
+		struct i2c_msg msg[1];
+
+		msg[0].addr = address;
+		msg[0].flags = I2C_M_RD;
+		msg[0].len = N_BYTES_TO_READ;
+		msg[0].buf = buff;
+
+		msgset.msgs = msg;
+		msgset.nmsgs = 1;
+
+		if (ioctl( file, I2C_RDWR, &msgset ) < 0) {
+			fprintf(stderr, "Error: Could not read "
+				"double word. %s\n", strerror(errno));
+			exit(1);
+		}
+		for (i = 0; i < N_BYTES_TO_READ; i++) {
+			printf ("%02x ", buff[i]);
+		}
+		printf ("\n");
+		exit(0);
+	}
+
 	/* See Winbond w83781d data sheet for bank details */
 	if (bank && size != I2C_SMBUS_BLOCK_DATA) {
 		res = i2c_smbus_read_byte_data(file, bankreg);
-- 
1.8.2.2

