From dd65685c24bb25a800ed42388ed0a1e8792b2ed2 Mon Sep 17 00:00:00 2001
From: falkTX <falktx@gmail.com>
Date: Sat, 24 Nov 2018 14:29:58 +0100
Subject: [PATCH] Fix OSX build

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Backport [https://github.com/jackaudio/jack2/commit/d851fada460d42508a6f82b19867f63853062583]
---
 posix/JackNetUnixSocket.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/posix/JackNetUnixSocket.cpp b/posix/JackNetUnixSocket.cpp
index 7a5585ef..a4cb5036 100644
--- a/posix/JackNetUnixSocket.cpp
+++ b/posix/JackNetUnixSocket.cpp
@@ -169,7 +169,7 @@ namespace Jack
 
     int JackNetUnixSocket::Bind()
     {
-        return bind(fSockfd, reinterpret_cast<socket_address_t*>(&fRecvAddr), sizeof(socket_address_t));
+        return ::bind(fSockfd, reinterpret_cast<socket_address_t*>(&fRecvAddr), sizeof(socket_address_t));
     }
 
     int JackNetUnixSocket::BindWith(const char* ip)
-- 
2.20.1

