From d155ffd33630f80b23c31b3a6e01da221dd9bd13 Mon Sep 17 00:00:00 2001
From: Luca Boccassi <luca.boccassi@gmail.com>
Date: Fri, 30 Mar 2018 16:36:22 +0100
Subject: [PATCH] Problem: out of date with zproject

Upstream-Status: backport
Signed-off-by: Adrian Bunk <bunk@stusta.de>

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a3b81722..4f203461 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,7 +38,7 @@ if (NOT CMAKE_BUILD_TYPE)
             OUTPUT_STRIP_TRAILING_WHITESPACE
         )
         message(STATUS "git workspace root [${git_result}]: ${git_root}")
-        if ( "${git_result}" == "0" )
+        if ( "${git_result}" STREQUAL "0" )
             set (CMAKE_BUILD_TYPE Debug)
         else ()
             set (CMAKE_BUILD_TYPE Release)
@@ -608,7 +608,7 @@ add_custom_command(
     TARGET  distclean
 )
 
-include(ClangFormat)
+include(ClangFormat OPTIONAL)
 
 ########################################################################
 # summary

