From 945df2da87a412758ceab4ac5711abbb269a47c1 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 8 Jun 2017 23:09:51 -0700
Subject: [PATCH] audio_alsa09.c: alsa drain fix

Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 audio_alsa09.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/audio_alsa09.c b/audio_alsa09.c
index 534e3db..d9450e6 100644
--- a/audio_alsa09.c
+++ b/audio_alsa09.c
@@ -506,7 +506,7 @@ void esd_audio_flush(void)
 	}
 
 	if (alsa_playback_handle != NULL)
-		snd_pcm_drain( alsa_playback_handle );
+		snd_pcm_drop( alsa_playback_handle );
   
 	if (alsadbg) 
 		print_state();
