Upstream-Status: Inappropriate [configuration]

Signed-off-by: Mei Lei <lei.mei@intel.com>

diff --git a/configure.ac b/configure.ac
index c1fe224..f5ac628 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1374,7 +1374,8 @@ if test -n "$WITH_DIET_LIBC" ; then
 	INCLUDES="$INCLUDES -D_REENTRANT"
 fi
 AC_SUBST(INCLUDES)
-AM_MKINSTALLDIRS
+MKINSTALLDIRS="mkdir -p"
+AC_SUBST(MKINSTALLDIRS)
 dnl
 dnl Build CFLAGS
 dnl
