From 7b259580800e259d232229dc89f97058b56e2fe8 Mon Sep 17 00:00:00 2001
From: "Hongjun.Yang" <hongjun.yang@windriver.com>
Date: Wed, 22 Oct 2014 10:02:48 +0800
Subject: [PATCH] Add ptest for tcpdump

Upstream-Status: Pending

Signed-off-by: Hongjun.Yang <hongjun.yang@windriver.com>
---
 Makefile.in | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

Index: tcpdump-4.9.0/Makefile.in
===================================================================
--- tcpdump-4.9.0.orig/Makefile.in
+++ tcpdump-4.9.0/Makefile.in
@@ -436,9 +436,17 @@ distclean:
 	    tests/failure-outputs.txt
 	rm -rf autom4te.cache tests/DIFF tests/NEW
 
-check: tcpdump
+buildtest-TESTS: tcpdump
+
+runtest-PTEST:
 	(cd tests && ./TESTrun.sh)
 
+install-ptest:
+	cp -r tests                     $(DESTDIR)
+	cp -r config.h                  $(DESTDIR)
+	install -m 0755 Makefile        $(DESTDIR)
+	ln -sf /usr/sbin/tcpdump        $(DESTDIR)/tcpdump
+
 extags: $(TAGFILES)
 	ctags $(TAGFILES)
 
