From 88077ed156a714f32bb408f422278d61611ffd41 Mon Sep 17 00:00:00 2001
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Sat, 26 Jul 2014 16:57:46 -0300
Subject: [PATCH] configure.in: Fix string argument syntax

Upstream-Status: Submitted

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
---
 configure.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index bfe1b82..e939bbd 100644
--- a/configure.in
+++ b/configure.in
@@ -65,7 +65,7 @@ fi)
 
 DIRECTFB_REQUIRED_VERSION=1.4.14
 
-PKG_CHECK_MODULES(DIRECTFB, directfb >= $DIRECTFB_REQUIRED_VERSION,[ WATER=`$PKG_CONFIG --variable=water directfb` ],
+PKG_CHECK_MODULES([DIRECTFB], directfb >= $DIRECTFB_REQUIRED_VERSION,[ WATER=`$PKG_CONFIG --variable=water directfb` ],
 	AC_MSG_ERROR([$DIRECTFB_PKG_ERRORS
 *** DirectFB $DIRECTFB_REQUIRED_VERSION or newer is required. The latest version
 *** of DirectFB is available from http://www.directfb.org/.]))
-- 
1.7.10.4

