Author: Marcin Juszkiewicz <marcin.juszkiewicz@linaro.org>
  
Upstream-Status: unsuitable
---

diff --git a/build/linux.clang.inc b/build/linux.clang.inc
index fe9b5c98..b0dcd68b 100644
--- a/build/linux.clang.inc
+++ b/build/linux.clang.inc
@@ -12,8 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-CPLUS ?= clang++
-CONLY ?= clang
+CPLUS ?= $(CXX)
+CONLY ?= $(CC)
 COMPILE_ONLY = -c -MMD
 PREPROC_ONLY = -E -x c++
 INCLUDE_KEY = -I
diff --git a/build/linux.gcc.inc b/build/linux.gcc.inc
index d820c15d..62c76afd 100644
--- a/build/linux.gcc.inc
+++ b/build/linux.gcc.inc
@@ -12,8 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-CPLUS ?= g++
-CONLY ?= gcc
+CPLUS ?= $(CXX)
+CONLY ?= $(CC)
 COMPILE_ONLY = -c -MMD
 PREPROC_ONLY = -E -x c++
 INCLUDE_KEY = -I
