From 2a43027f7ecf2bb3ce73f95bcf575c56bc495d07 Mon Sep 17 00:00:00 2001
From: Alejandro del Castillo <alejandro.delcastillo@ni.com>
Date: Wed, 9 Dec 2015 17:38:05 -0600
Subject: [PATCH] configure.ac: use pkg-config for libsolv

Signed-off-by: Alejandro del Castillo <alejandro.delcastillo@ni.com>
---
 configure.ac | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/configure.ac b/configure.ac
index 52e1025..d17dc5c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -159,17 +159,7 @@ if test "x$want_solver" != "xno"; then
                  AC_MSG_ERROR(Specify which solver with --enable-solver=<SOLVER>)],
             [libsolv],
                 [AC_MSG_RESULT(libsolv)
-
-                SOLVER_CFLAGS="-I/usr/local/include/"
-                SOLVER_LIBS="-lsolv -lsolvext"
-                AC_CHECK_LIB([solv],
-                             [solver_solve],
-                             [],
-                             [AC_MSG_ERROR(libsolv not found)])
-                dnl TODO: remove previous 6 lines and uncomment line below to use
-                dnl pkg-config once there is a release of libsolv with pkg-config:
-                dnl PKG_CHECK_MODULES(SOLVER, libsolv)
-
+                PKG_CHECK_MODULES(SOLVER, libsolv)
                 AC_DEFINE(HAVE_SOLVER_LIBSOLV,1,[Define if you want to use libsolv])],
             # default
             [AC_MSG_RESULT(no)
-- 
1.9.1

