Upstream-Status: Pending

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@intel.com>

Index: pycurl-7.19.3.1/setup.py
===================================================================
--- pycurl-7.19.3.1.orig/setup.py	2014-07-16 12:57:24.065346887 +0000
+++ pycurl-7.19.3.1/setup.py	2014-07-16 12:57:53.057347678 +0000
@@ -147,7 +147,7 @@
         optbuf = ''
         sslhintbuf = ''
         errtext = ''
-        for option in ["--libs", "--static-libs"]:
+        for option in ["--libs"]:
             p = subprocess.Popen((CURL_CONFIG, option),
                 stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             stdout, stderr = p.communicate()
