From 8ec1b100afda5b3e31f9d44af04b04b16dfff2e4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Tue, 18 May 2021 00:15:46 +0200
Subject: [PATCH] Find jack with pkgconfig
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Pending

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 122fe93..ab7d7f5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -274,7 +274,8 @@ ELSE()
 
   ELSEIF(UNIX)
 
-    FIND_PACKAGE(Jack)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules (JACK jack)
     IF(JACK_FOUND)
       OPTION(PA_USE_JACK "Enable support for Jack" ON)
     ELSE()
-- 
2.31.1

