rpmdb: more verbose error logging in rpmTempFile()

Upstream-Status: Inappropriate [debugging]

Signed-off-by: Markus Lehtonen <markus.lehtonen@linux.intel.com>

diff --git a/rpmdb/signature.c b/rpmdb/signature.c
index 100204b..e56ab3d 100644
--- a/rpmdb/signature.c
+++ b/rpmdb/signature.c
@@ -72,12 +72,16 @@ int rpmTempFile(const char * prefix, const char ** fnptr, void * fdptr)
 #endif
 
 	temput = urlPath(tempfn, &tfn);
-	if (*tfn == '\0') goto errxit;
+	if (*tfn == '\0') {
+            rpmlog(RPMLOG_ERR, _("error creating tempfile: empty filename\n"));
+            goto errxit;
+        }
 
 	switch (temput) {
 	case URL_IS_DASH:
 	case URL_IS_HKP:
 	case URL_IS_MONGO:	/* XXX FIXME */
+            rpmlog(RPMLOG_ERR, _("error creating temporary file, invalid uri type: %d\n"), temput);
 	    goto errxit;
 	    /*@notreached@*/ /*@switchbreak@*/ break;
 	case URL_IS_HTTPS:
-- 
2.6.6

