diff -Nur a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts
--- a/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts	2022-03-20 10:43:23.588685705 +0200
+++ b/arch/arm64/boot/dts/nuvoton/nuvoton-npcm845-evb.dts	2022-03-20 10:45:04.256074792 +0200
@@ -130,6 +130,13 @@
 		};
 	};
 
+	firmware {
+		optee {
+			compatible = "linaro,optee-tz";
+			method = "smc";
+		};
+	};
+
 	ahb {
 		/*rstc: rstc@f0801000 {
 			nuvoton,sw-reset-number = <2>;
