---
 configure.ac |    1 +
 1 file changed, 1 insertion(+)

Index: git/configure.ac
===================================================================
--- git.orig/configure.ac
+++ git/configure.ac
@@ -56,6 +56,7 @@ AM_CONDITIONAL([ENABLE_EXAMPLES], [test
 # Checks for programs.
 AC_PROG_CC
 AC_PROG_CC_C99
+AM_PROG_AR
 
 # Enable automake silent build rules.
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
