From 788aace494f79e8201b18ebcdf1592b5030c5295 Mon Sep 17 00:00:00 2001
From: Adrian Freihofer <adrian.freihofer@siemens.com>
Date: Wed, 4 Dec 2019 17:23:46 +0100
Subject: [PATCH] remove tests dep on gobject-intro

---
 src/ws/Makefile-ws.am | 54 ---------------------------------------------------
 1 file changed, 54 deletions(-)

--- a/src/ws/Makefile-ws.am
+++ b/src/ws/Makefile-ws.am
@@ -58,46 +58,6 @@ pam_cockpit_cert_so_LDFLAGS = -shared
 pam_cockpit_cert_so_SOURCES = src/ws/pam_cockpit_cert.c
 
 # -----------------------------------------------------------------------------
-# test-server: server for running the html/browser unit tests against
-
-check_PROGRAMS += test-server
-
-test_server_CPPFLAGS = $(libcockpit_ws_a_CPPFLAGS) $(TEST_CPP)
-test_server_LDADD = $(libcockpit_ws_a_LIBS) $(TEST_LIBS)
-
-test_server_SOURCES = \
-	src/ws/mock-service.c \
-	src/ws/mock-service.h \
-	src/ws/test-server.c \
-	$(NULL)
-
-test_server_CPPFLAGS += -I$(top_builddir)/src/ws
-nodist_test_server_SOURCES = $(GDBUS_CODEGEN_GENERATED)
-
-BUILT_SOURCES += $(GDBUS_CODEGEN_GENERATED)
-CLEANFILES += $(GDBUS_CODEGEN_GENERATED)
-GDBUS_CODEGEN_GENERATED = \
-	src/ws/mock-dbus-tests.h \
-	src/ws/mock-dbus-tests.c \
-	$(NULL)
-
-EXTRA_DIST += $(GDBUS_CODEGEN_XML)
-GDBUS_CODEGEN_XML = $(srcdir)/src/ws/com.redhat.Cockpit.DBusTests.xml
-
-GDBUS_CODEGEN_INVOCATION = \
-	$(AM_V_GEN) gdbus-codegen \
-	--interface-prefix com.redhat.Cockpit.DBusTests \
-	--c-namespace Test \
-	--c-generate-object-manager \
-	$(NULL)
-
-src/ws/mock-dbus-tests.h: $(GDBUS_CODEGEN_XML)
-	$(GDBUS_CODEGEN_INVOCATION) --header --output $@ $<
-
-src/ws/mock-dbus-tests.c: $(GDBUS_CODEGEN_XML)
-	$(GDBUS_CODEGEN_INVOCATION) --body --output $@ $<
-
-# -----------------------------------------------------------------------------
 # Unit tests
 
 check_SCRIPTS += src/ws/mock-cat-with-init
