From 788aace494f79e8201b18ebcdf1592b5030c5295 Mon Sep 17 00:00:00 2001
From: Adrian Freihofer <adrian.freihofer@siemens.com>
Date: Wed, 4 Dec 2019 17:23:46 +0100
Subject: [PATCH] remove tests dep on gobject-intro

---
 src/ws/Makefile-ws.am | 54 ---------------------------------------------------
 1 file changed, 54 deletions(-)

diff --git a/src/ws/Makefile-ws.am b/src/ws/Makefile-ws.am
index 009130941..34e13d7fe 100644
--- a/src/ws/Makefile-ws.am
+++ b/src/ws/Makefile-ws.am
@@ -246,60 +246,6 @@ EXTRA_DIST += \
 
 # ----------------------------------------------------------------------------------------------------
 
-noinst_PROGRAMS += test-server
-check_PROGRAMS += test-server
-
-GDBUS_CODEGEN_XML = $(srcdir)/src/ws/com.redhat.Cockpit.DBusTests.xml
-
-GDBUS_CODEGEN_GENERATED = \
-	src/ws/mock-dbus-tests.h \
-	src/ws/mock-dbus-tests.c \
-	$(NULL)
-
-# FIXME: --header/--body and --output are only available from GLib 2.56.
-# just use --generate-c-code and a bit of dependency ugliness for now
-GDBUS_CODEGEN_INVOCATION = \
-	$(AM_V_GEN) gdbus-codegen \
-	--interface-prefix com.redhat.Cockpit.DBusTests \
-	--c-namespace Test \
-	--c-generate-object-manager \
-	--generate-c-code src/ws/mock-dbus-tests \
-	$(GDBUS_CODEGEN_XML)
-
-BUILT_SOURCES += $(GDBUS_CODEGEN_GENERATED)
-CLEANFILES += $(GDBUS_CODEGEN_GENERATED)
-EXTRA_DIST += $(GDBUS_CODEGEN_XML)
-
-src/ws/mock-dbus-tests.h: $(GDBUS_CODEGEN_XML)
-	$(GDBUS_CODEGEN_INVOCATION)
-
-src/ws/mock-dbus-tests.c: $(GDBUS_CODEGEN_XML) src/ws/mock-dbus-tests.h
-	$(GDBUS_CODEGEN_INVOCATION)
-
-test_server_SOURCES = \
-	src/ws/mock-service.c \
-	src/ws/mock-service.h \
-	src/ws/test-server.c \
-	$(NULL)
-
-nodist_test_server_SOURCES = \
-	$(GDBUS_CODEGEN_GENERATED) \
-	$(NULL)
-
-test_server_CFLAGS = 					\
-	-I$(builddir)/src/ws \
-	-I$(top_srcdir)/src/ws \
-	-DG_LOG_DOMAIN=\"test-server\"			\
-	$(GIO_CFLAGS)					\
-	$(COCKPIT_WS_CFLAGS) \
-	$(NULL)
-
-test_server_LDADD = 					\
-	$(libcockpit_ws_LIBS) \
-	$(GIO_LIBS)					\
-	-lpam 						\
-	$(NULL)
-
 WS_CHECKS = \
 	test-base64 \
 	test-creds \
-- 
2.11.0

