Subject: [PATCH 1/6] urandom-xauth-changes-to-options.h

Upstream-Status: Inappropriate [configuration]
---
 default_options.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/default_options.h b/default_options.h
index 349338c..5ffac25 100644
--- a/default_options.h
+++ b/default_options.h
@@ -289,7 +289,7 @@ group1 in Dropbear server too */
 
 /* The command to invoke for xauth when using X11 forwarding.
  * "-q" for quiet */
-#define XAUTH_COMMAND "/usr/bin/xauth -q"
+#define XAUTH_COMMAND "xauth -q"
 
 
 /* If you want to enable running an sftp server (such as the one included with
-- 
2.25.1

