Subject: [PATCH 4/6] fix 2kb keys

Upstream-Status: Inappropriate [configuration]
---
 kex.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: dropbear-2017.75/kex.h
===================================================================
--- dropbear-2017.75.orig/kex.h
+++ dropbear-2017.75/kex.h
@@ -106,6 +106,6 @@ int curve25519_donna(unsigned char *out,
 #endif
 
 
-#define MAX_KEXHASHBUF 2000
+#define MAX_KEXHASHBUF 3000
 
 #endif /* DROPBEAR_KEX_H_ */
