Upstream-Status: Inappropriate [configuration]

---
 configure.in     |    6 ------
 methods/makefile |    7 -------
 2 files changed, 13 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -86,12 +86,6 @@ AC_CHECK_HEADER(db.h,
 
 LIBS="$saveLIBS"
 
-AC_CHECK_LIB(curl, curl_easy_init,
-     [AC_CHECK_HEADER(curl/curl.h,
-        curl_ok=yes,
-        curl_ok=no)],
-     AC_MSG_ERROR([failed: I need CURL due https support]),
-)
 
 AC_SUBST(BDBLIB)
 
--- a/methods/makefile
+++ b/methods/makefile
@@ -51,13 +51,6 @@ LIB_MAKES = apt-pkg/makefile
 SOURCE = http.cc http_main.cc rfc2553emu.cc connect.cc server.cc
 include $(PROGRAM_H)
 
-# The https method
-PROGRAM=https
-SLIBS = -lapt-pkg -lcurl $(INTLLIBS)
-LIB_MAKES = apt-pkg/makefile
-SOURCE = https.cc server.cc
-include $(PROGRAM_H)
-
 # The ftp method
 PROGRAM=ftp
 SLIBS = -lapt-pkg $(SOCKETLIBS) $(INTLLIBS)
