include sys/select.h for FD_* macros

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Pending

Index: gnome-keyring-2.32.1/pkcs11/rpc-layer/gkm-rpc-daemon-standalone.c
===================================================================
--- gnome-keyring-2.32.1.orig/pkcs11/rpc-layer/gkm-rpc-daemon-standalone.c
+++ gnome-keyring-2.32.1/pkcs11/rpc-layer/gkm-rpc-daemon-standalone.c
@@ -32,6 +32,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/select.h>
 
 #include <dlfcn.h>
 #include <pthread.h>
