From 9a5e655cf57301008cd61d53c8a410a7f397e650 Mon Sep 17 00:00:00 2001
From: Andrei Gherzan <andrei.gherzan@windriver.com>
Date: Tue, 18 Nov 2014 15:51:30 +0100
Subject: [PATCH 4/4] Modify systemd config directory

Upstream-Status: Inappropriate [Configuration Specific]
Signed-off-by: Andrei Gherzan <andrei.gherzan@windriver.com>
---
 systemd/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e6b44a2..0e885bf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -201,7 +201,7 @@ if(WITH_SYSTEMD OR WITH_SYSTEMD_WATCHDOG OR WITH_SYSTEMD_JOURNAL)
 
     set(systemd_SRCS ${PROJECT_SOURCE_DIR}/systemd/3rdparty/sd-daemon.c)
 
-    set(SYSTEMD_UNITDIR "${CMAKE_INSTALL_PREFIX}/lib/systemd/system" CACHE PATH
+    set(SYSTEMD_UNITDIR "/lib/systemd/system" CACHE PATH
         "Set directory to install systemd unit files")
 
     add_subdirectory(systemd)
