From 7a524d49b3d4459280f18942df2980603400ec52 Mon Sep 17 00:00:00 2001
From: Bian Naimeng <biannm@cn.fujitsu.com>
Date: Fri, 19 Jun 2015 11:54:44 +0900
Subject: [PATCH] Using PKG_CHECK_MODULES to found headers and libraries of
 smbclient

Signed-off-by: Bian Naimeng <biannm@cn.fujitsu.com>
---
 configure.in    | 5 +++++
 src/Makefile.am | 3 +++
 2 files changed, 8 insertions(+)

diff --git a/configure.in b/configure.in
index ce21aef..ecd7119 100644
--- a/configure.in
+++ b/configure.in
@@ -125,6 +125,11 @@ AC_CHECK_LIB(smbclient, smbc_setOptionUserData,
     )]
 )
 
+dnl *****************************************************************
+dnl *** Check libsmbclient by pkgconfig to get cflags and ldflags ***
+dnl *****************************************************************
+PKG_CHECK_MODULES(SMBCLIENT, smbclient)
+
 dnl ******************
 dnl *** Final step ***
 dnl ******************
diff --git a/src/Makefile.am b/src/Makefile.am
index 41519d8..9fc97e9 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -16,3 +16,6 @@ smbnetfs_SOURCES = \
 	event.c			event.h			\
 	reconfigure.c		reconfigure.h		\
 	main.c
+
+smbnetfs_CFLAGS=${SMBCLIENT_CFLAGS}
+smbnetfs_LDFLAGS=${SMBCLIENT_LDFLAGS}
-- 
1.8.4.2

