From dac72d809be9faf9380b181df0c19a2c6d744c54 Mon Sep 17 00:00:00 2001
From: "Edgar E. Iglesias" <edgar.iglesias@gmail.com>
Date: Mon, 28 Aug 2017 19:53:54 -0700
Subject: [PATCH] Disable the warning message for eh_frame_hdr

Signed-off-by: Edgar E. Iglesias <edgar.iglesias@gmail.com>
Signed-off-by: Manjukumar Matha <manjukumar.harthikote-matha@xilinx.com>
Upstream-Status: Inappropriate [workaround]

---
 bfd/elf-eh-frame.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/bfd/elf-eh-frame.c b/bfd/elf-eh-frame.c
index 95697c4..704121d 100644
--- a/bfd/elf-eh-frame.c
+++ b/bfd/elf-eh-frame.c
@@ -1042,10 +1042,13 @@ _bfd_elf_parse_eh_frame (bfd *abfd, struct bfd_link_info *info,
   goto success;
 
  free_no_table:
-  (*info->callbacks->einfo)
+  /* FIXME: Remove the microblaze specifics when relaxing gets fixed.  */
+  if (bfd_get_arch(abfd) != bfd_arch_microblaze) {
     /* xgettext:c-format */
-    (_("%P: error in %B(%A); no .eh_frame_hdr table will be created.\n"),
-     abfd, sec);
+    (*info->callbacks->einfo)
+      (_("%P: error in %B(%A); no .eh_frame_hdr table will be created.\n"),
+       abfd, sec);
+  }
   hdr_info->u.dwarf.table = FALSE;
   if (sec_info)
     free (sec_info);
