--- faac-1.28/common/mp4v2/atom_standard.cpp.orig	2016-06-16 07:59:21.657040761 -0400
+++ faac-1.28/common/mp4v2/atom_standard.cpp	2016-06-16 08:01:50.613036724 -0400
@@ -22,15 +22,15 @@
 #include "mp4common.h"
 #include "atoms.h"
 
-static const char name[5]={0251,'n', 'a', 'm', '\0'};
-static const char art[5]={0251,'A', 'R', 'T', '\0'};
-static const char wrt[5]={0251,'w', 'r', 't', '\0'};
-static const char alb[5]={0251,'a', 'l', 'b', '\0'};
-static const char day[5]={0251,'d', 'a', 'y', '\0'};
-static const char too[5]={0251,'t', 'o', 'o', '\0'};
-static const char cmt[5]={0251,'c', 'm', 't', '\0'};
-static const char gen[5]={0251,'g', 'e', 'n', '\0'};
-static const char grp[5]={0251,'g', 'r', 'p', '\0'};
+static const char name[5]={(char)0251,'n', 'a', 'm', '\0'};
+static const char art[5]={(char)0251,'A', 'R', 'T', '\0'};
+static const char wrt[5]={(char)0251,'w', 'r', 't', '\0'};
+static const char alb[5]={(char)0251,'a', 'l', 'b', '\0'};
+static const char day[5]={(char)0251,'d', 'a', 'y', '\0'};
+static const char too[5]={(char)0251,'t', 'o', 'o', '\0'};
+static const char cmt[5]={(char)0251,'c', 'm', 't', '\0'};
+static const char gen[5]={(char)0251,'g', 'e', 'n', '\0'};
+static const char grp[5]={(char)0251,'g', 'r', 'p', '\0'};
 
 MP4StandardAtom::MP4StandardAtom (const char *type) : MP4Atom(type)
 {
--- faac-1.28/common/mp4v2/mp4atom.cpp.orig	2016-06-16 08:03:07.777034633 -0400
+++ faac-1.28/common/mp4v2/mp4atom.cpp	2016-06-16 08:03:54.629033363 -0400
@@ -264,11 +264,11 @@
       }
       break;
     case 0251:
-      static const char name[5]={0251,'n', 'a', 'm', '\0'};
-      static const char cmt[5]={0251,'c', 'm', 't', '\0'};
-      static const char cpy[5]={0251,'c', 'p', 'y', '\0'};
-      static const char des[5]={0251,'d', 'e', 's','\0'};
-      static const char prd[5]={0251, 'p', 'r', 'd', '\0'};
+      static const char name[5]={(char)0251,'n', 'a', 'm', '\0'};
+      static const char cmt[5]={(char)0251,'c', 'm', 't', '\0'};
+      static const char cpy[5]={(char)0251,'c', 'p', 'y', '\0'};
+      static const char des[5]={(char)0251,'d', 'e', 's','\0'};
+      static const char prd[5]={(char)0251, 'p', 'r', 'd', '\0'};
       if (ATOMID(type) == ATOMID(name) ||
 	  ATOMID(type) == ATOMID(cmt) ||
 	  ATOMID(type) == ATOMID(cpy) ||

