From edd575a546d56bb5683aff19782b16963d61fd0b Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 2 Nov 2011 16:54:57 +0100
Subject: [PATCH] Upstream-Status: Pending

---
 pear/Makefile.frag | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pear/Makefile.frag b/pear/Makefile.frag
index bbe8ec3..16f43e2 100644
--- a/pear/Makefile.frag
+++ b/pear/Makefile.frag
@@ -12,7 +12,7 @@ PEAR_SUFFIX = -ds a$(program_suffix)
 PEAR_INSTALLER_URL = https://pear.php.net/install-pear-nozlib.phar
 
 install-pear-installer: $(SAPI_CLI_PATH)
-	@$(top_builddir)/sapi/cli/php $(PEAR_INSTALL_FLAGS) pear/install-pear-nozlib.phar -d "$(peardir)" -b "$(bindir)" ${PEAR_PREFIX} ${PEAR_SUFFIX}
+	@$(PHP_NATIVE_DIR)/php $(PEAR_INSTALL_FLAGS) $(builddir)/install-pear-nozlib.phar -d "$(peardir)" -b "$(bindir)" ${PEAR_PREFIX} ${PEAR_SUFFIX}
 
 install-pear:
 	@echo "Installing PEAR environment:      $(INSTALL_ROOT)$(peardir)/"
