From 179102310e9a3729fb5b00f2a7bf24b12f4a7a83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@gmail.com>
Date: Sat, 16 Apr 2022 20:04:14 +0200
Subject: [PATCH] Allow building gir in cross environments
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

For us this works perfectly fine

Upstream-Status: Inappropriate [OE-specific]

Signed-off-by: Andreas Müller <schnitzeltony@gmail.com>
Signed-off-by: Markus Volk <f_l_k@t-online.de>

--- a/meson.build	2022-09-18 14:48:53.000000000 +0200
+++ b/meson.build	2022-11-26 14:05:13.302487232 +0100
@@ -51,7 +51,7 @@
 gen_locations_variant = find_program('build-aux/meson/gen_locations_variant.py')
 
 g_ir_scanner = find_program('g-ir-scanner', required: get_option('introspection'))
-build_gir = get_option('introspection') and g_ir_scanner.found() and not meson.is_cross_build()
+build_gir = get_option('introspection') and g_ir_scanner.found()
 build_docs = get_option('gtk_doc') and build_gir
 
 schemas_srcdir = meson.project_source_root() / 'schemas'
