From 8f3ed8df4721991958a5becf75a69493d67e7514 Mon Sep 17 00:00:00 2001
From: "Roy.Li" <rongqing.li@windriver.com>
Date: Tue, 5 Sep 2017 10:24:10 +0800
Subject: [PATCH] [PATCH] do not use the hardcode libdir

Upstream-Status: Pending

Signed-off-by: Roy.Li <rongqing.li@windriver.com>
Signed-off-by: Jackie Huang <jackie.huang@windriver.com>

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 00b359e..ba50e6f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -114,7 +114,7 @@ if test "$with_libpq" != yes; then
 	if test -d "$with_libpq"; then
 		PATH="$with_libpq/bin:$PATH"
 		CPPFLAGS="$CPPFLAGS -I$with_libpq/include -I$with_libpq/include/postgresql/internal"
-		LDFLAGS="$LDFLAGS -L$with_libpq/lib"
+		LDFLAGS="$LDFLAGS -L$with_libpq/${base_libdir}"
 	else
 		if test -x "$with_libpq"; then
 			PG_CONFIG=$with_libpq
