Install the examples to $libdir/clutter/.

Signed-off-by: Ross Burton <ross.burton@intel.com>
Upstream-Status: Inappropriate

Index: clutter-1.26.2/examples/Makefile.am
===================================================================
--- clutter-1.26.2.orig/examples/Makefile.am
+++ clutter-1.26.2/examples/Makefile.am
@@ -33,7 +33,8 @@ AM_CPPFLAGS = \
 	-I$(top_srcdir)/clutter \
 	-I$(top_builddir)/clutter
 
-noinst_PROGRAMS = $(all_examples)
+exampledir = $(pkglibdir)/examples
+example_PROGRAMS = $(all_examples)
 
 EXTRA_DIST = redhand.png
 
