Description: install cmake files into /usr/lib/cmake/cpprestsdk
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Upstream-Status: https://github.com/Microsoft/cpprestsdk/pull/737
Forwarded: https://github.com/Microsoft/cpprestsdk/pull/737
Last-Update: 2018-04-19

--- cpprest-2.10.2.orig/CMakeLists.txt
+++ cpprest-2.10.2/CMakeLists.txt
@@ -18,7 +18,7 @@ enable_testing()
 set(WERROR ON CACHE BOOL "Treat Warnings as Errors.")
 set(CPPREST_EXCLUDE_WEBSOCKETS OFF CACHE BOOL "Exclude websockets functionality.")
 set(CPPREST_EXCLUDE_COMPRESSION OFF CACHE BOOL "Exclude compression functionality.")
-set(CPPREST_EXPORT_DIR lib/cpprestsdk CACHE STRING "Directory to install CMake config files.")
+set(CPPREST_EXPORT_DIR lib/cmake/cpprestsdk CACHE STRING "Directory to install CMake config files.")
 set(CPPREST_INSTALL_HEADERS ON CACHE BOOL "Install header files.")
 set(CPPREST_INSTALL ON CACHE BOOL "Add install commands.")
 
