From a7914d5ffee6ffdfb3f2b8ebcc22c8367d078301 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 27 Mar 2021 22:08:43 -0700
Subject: [PATCH] make overriding CFLAGS possible

Author: Jakub Wilk <jwilk@debian.org>

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 base64/Makefile | 2 +-
 src/Makefile    | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/base64/Makefile b/base64/Makefile
index 30a2c5c..783a248 100644
--- a/base64/Makefile
+++ b/base64/Makefile
@@ -3,7 +3,7 @@ BINARIES = base64
 # Build flags (uncomment one)
 #############################
 # Release build flags
-CFLAGS += -O3
+CFLAGS ?= -O3
 #############################
 # Debug build flags
 #CFLAGS += -g
diff --git a/src/Makefile b/src/Makefile
index 28b2382..48801fc 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -3,7 +3,7 @@ LIBRARIES = libb64.a
 # Build flags (uncomment one)
 #############################
 # Release build flags
-CFLAGS += -O3
+CFLAGS ?= -O3
 #############################
 # Debug build flags
 #CFLAGS += -g
