From 9de4f5fa81891e14e205fe61e1408f1218369e55 Mon Sep 17 00:00:00 2001
From: Leon Anavi <leon.anavi@konsulko.com>
Date: Tue, 18 May 2021 15:03:22 +0000
Subject: [PATCH] config.mk: Fix compiler and linker

Fix LIBS

Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>
---
Upstream-Status: Pending

 config.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.mk b/config.mk
index 2eb9fb0..032bf9f 100644
--- a/config.mk
+++ b/config.mk
@@ -19,7 +19,7 @@ WEBEXTLIBS = `pkg-config --libs webkit2gtk-4.0 webkit2gtk-web-extension-4.0 gio-
 
 # includes and libs
 INCS = $(X11INC) $(GTKINC)
-LIBS = $(X11LIB) $(GTKLIB) -lgthread-2.0
+LIBS = -lc -lX11 ${GTKLIB} -lgthread-2.0
 
 # flags
 CPPFLAGS = -DVERSION=\"$(VERSION)\" -DGCR_API_SUBJECT_TO_CHANGE \
-- 
2.17.1

