From c00668541f68c17a5e453a4dcc2a86022b20af87 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Mon, 30 Jul 2012 23:52:49 +0200

---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index 7c15b6b..9896631 100644
--- a/configure.in
+++ b/configure.in
@@ -23,7 +23,7 @@ AM_CONFIG_HEADER(config.h)
 dnl Checks for programs.
 AC_PROG_CC
 
-dnl AC_PROG_CXX
+AC_PROG_CXX
 dnl AC_PROG_AWK
 dnl AM_PROG_LEX
 dnl AC_PROG_YACC
