From be4032079c931704f52e29f5da5c01cde24ac842 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 16 Jan 2020 10:44:58 -0800
Subject: [PATCH] ppc: Fix signature for co_create API

Upstream-Status: Submitted [https://github.com/fluent/fluent-bit/pull/1886]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 lib/flb_libco/ppc.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/flb_libco/ppc.c b/lib/flb_libco/ppc.c
index e6536d56..533256b3 100644
--- a/lib/flb_libco/ppc.c
+++ b/lib/flb_libco/ppc.c
@@ -279,7 +279,9 @@ static uint32_t* co_create_(unsigned size, uintptr_t entry) {
   return t;
 }
 
-cothread_t co_create(unsigned int size, void (*entry_)(void)) {
+cothread_t co_create(unsigned int size, void (*entry_)(void),
+                     size_t *out_size) {
+
   uintptr_t entry = (uintptr_t)entry_;
   uint32_t* t = 0;
 
@@ -325,7 +327,7 @@ cothread_t co_create(unsigned int size, void (*entry_)(void)) {
     t[10] = (uint32_t)(sp >> shift >> shift);
     t[11] = (uint32_t)sp;
   }
-
+  *out_size = size;
   return t;
 }
 
-- 
2.25.0

