From 2d9687840b97186b80053dd262209e39455ac876 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Wed, 8 May 2019 15:31:23 +0200
Subject: [PATCH] gstreamer: add a missing format string

Upstream-Status: Pending
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 .../platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp b/Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp
index e1e497ee..483fd65c 100644
--- a/Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp
+++ b/Source/WebCore/platform/graphics/gstreamer/WebKitWebSourceGStreamer.cpp
@@ -738,7 +738,7 @@ static GstStateChangeReturn webKitWebSrcChangeState(GstElement* element, GstStat
     WebKitWebSrc* src = WEBKIT_WEB_SRC(element);
 
 #if GST_CHECK_VERSION(1, 14, 0)
-    GST_DEBUG_OBJECT(src, gst_state_change_get_name(transition));
+    GST_DEBUG_OBJECT(src, "%s", gst_state_change_get_name(transition));
 #endif
     switch (transition) {
     case GST_STATE_CHANGE_READY_TO_NULL:
