Signed-off-by: Christopher Larson <chris_larson@mentor.com>
Upstream-Status: Pending

From 897563a0397266d8ceb058f172e16b06419b2593 Mon Sep 17 00:00:00 2001
From: Daniel Schepler <dschepler@gmail.com>
Date: Mon, 26 Mar 2018 17:48:34 +0300
Subject: [PATCH] Fix FTBFS in x32

===================================================================

---
 Source/WTF/wtf/Platform.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Source/WTF/wtf/Platform.h b/Source/WTF/wtf/Platform.h
index 850e298..551c047 100644
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -132,7 +132,11 @@
 /* CPU(X86_64) - AMD64 / Intel64 / x86_64 64-bit */
 #if   defined(__x86_64__) \
     || defined(_M_X64)
+#ifdef __ILP32__
+#define WTF_CPU_X86_64_32 1
+#else
 #define WTF_CPU_X86_64 1
+#endif
 #define WTF_CPU_X86_SSE2 1
 #define WTF_CPU_KNOWN 1
 #endif
