From e12a93174f9b652604dda8d8464b9559b62b29d5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Hundeb=C3=B8ll?= <martin@geanix.com>
Date: Wed, 22 May 2024 14:02:55 +0200
Subject: [PATCH 10/11] configure: lookup meson exutable from PATH

Upstream-Status: Inappropriate [workaround, would need a real fix for upstream]
---
 configure | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/configure b/configure
index 3cd736b139..482a1f8ef3 100755
--- a/configure
+++ b/configure
@@ -956,12 +956,7 @@ fi
 $mkvenv ensuregroup --dir "${source_path}/python/wheels" \
      ${source_path}/pythondeps.toml meson || exit 1
 
-# At this point, we expect Meson to be installed and available.
-# We expect mkvenv or pip to have created pyvenv/bin/meson for us.
-# We ignore PATH completely here: we want to use the venv's Meson
-# *exclusively*.
-
-meson="$(cd pyvenv/bin; pwd)/meson"
+meson=`which meson`
 
 # Conditionally ensure Sphinx is installed.
 
-- 
2.44.0

