Imported from Fedora

--- M2Crypto/SWIG/_m2crypto.i	2017-09-26 11:26:33.000000000 +0200
+++ M2Crypto-0.26.4/SWIG/_m2crypto.i	2017-09-26 21:04:14.080330741 +0200
@@ -7,6 +7,7 @@
  * Copyright (c) 2009-2010 Heikki Toivonen. All rights reserved.
  *
  */
+%import "gcc_macros.h"
 
 %module(threads=1) m2crypto
 /* We really don't need threadblock (PyGILState_Ensure() etc.) anywhere.
@@ -15,11 +16,6 @@
 %nothreadblock;
 %nothreadallow;
 
-#if SWIG_VERSION >= 0x030000
-#define __WCHAR_MAX__ __WCHAR_MAX
-#define __WCHAR_MIN__ __WCHAR_MIN
-#endif
-
 %{
 #if __GNUC__ < 5
 #pragma GCC diagnostic ignored "-Wunused-label"
