From 458e9450cecf703f55536e609365162719585900 Mon Sep 17 00:00:00 2001
From: "Roy.Li" <rongqing.li@windriver.com>
Date: Wed, 10 Sep 2014 17:03:29 +0800
Subject: [PATCH] do not compile host tools, since we have native

Upstream-Status: Pending

Signed-off-by: Roy.Li <rongqing.li@windriver.com>
---
 configure.ac | 23 +----------------------
 1 file changed, 1 insertion(+), 22 deletions(-)

diff --git a/configure.ac b/configure.ac
index b46130d..cf58654 100644
--- a/configure.ac
+++ b/configure.ac
@@ -90,28 +90,7 @@ AC_CANONICAL_BUILD
 # $native_tools is also added to SUBDIRS in the main Makefile.am,
 # so that make compiles the native tools first.
 #
-if test "$cross_compiling" = no; then
-  native_tools=
-else
-  native_tools=tools
-  test -d "$native_tools" || mkdir "$native_tools"
-  confdir=`(cd "$srcdir";pwd)`
-  # Make sure the secondary configure won't fail with
-  # "error: source directory already configured".
-  rm -f config.status
-  AC_MSG_NOTICE([[Doing configure of native tools (${build}).]])
-  cd "$native_tools" || exit 1
-  # Run secondary configure in alternate environment or
-  # it gets the wrong CC etc.  env -i gives this build host configure
-  # a clean environment.
-  env -i CC="${BUILD_CC}" AR="${BUILD_AR}" RANLIB="${BUILD_RANLIB}" \
-      PATH="${PATH}" \
-      tools_only=1 \
-    ${confdir}/configure --build=${build} --host=${build} \
-                         --disable-rpath --disable-nls
-  cd .. || exit 1
-  AC_MSG_NOTICE([[Continuing with main configure (${host}).]])
-fi
+native_tools=
 AC_SUBST(native_tools)
 AM_CONDITIONAL(TOOLS_ONLY, [[test "x$tools_only" = x1]])
 
-- 
1.9.1

