From a4e03fbb4d82f91069d96005405f89c45b8fd157 Mon Sep 17 00:00:00 2001
From: Chen Qi <Qi.Chen@windriver.com>
Date: Wed, 17 Apr 2019 09:24:37 +0800
Subject: [PATCH] dc.tests: fix two test case to also depend on DC_BIG

Upstream-Status: Submitted [http://lists.busybox.net/pipermail/busybox/2019-April/087205.html]

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 testsuite/dc.tests | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/testsuite/dc.tests b/testsuite/dc.tests
index 1708a48..8c3af41 100755
--- a/testsuite/dc.tests
+++ b/testsuite/dc.tests
@@ -41,6 +41,9 @@ testing "dc complex without spaces (multiple args)" \
 	"16\n" \
 	"" ""
 
+optional FEATURE_DC_BIG
+# All tests below depend on FEATURE_DC_BIG
+
 testing "dc read" \
 	"dc -finput" \
 	"2\n9\n1\n" \
@@ -51,9 +54,6 @@ testing "dc read string" \
 	"2\nstr\n1\n" \
 	"1?2\nf" "[str]\n"
 
-optional FEATURE_DC_BIG
-# All tests below depend on FEATURE_DC_BIG
-
 testing "dc '>a' (conditional execute string) 1" \
 	"dc" \
 	"1\n9\n" \
-- 
2.7.4

