From a06c77557ed951249d5b344441ad6ec57410e63f Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex@linutronix.de>
Date: Sun, 3 Oct 2021 21:52:16 +0200
Subject: [PATCH] Makefile: do not use -Werror

Upstream-Status: Inappropriate [oe-core specific]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>
---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index 97973ce..78273ff 100644
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,6 @@ BUILDFLAGS = \
 	-DSBINDIR=\"$(SBINDIR)\" \
 	-I${CURDIR}/include \
 	-Wall \
-	-Werror \
 	$(NULL)
 
 TESTFLAGS =	\
-- 
2.34.1

