From e0df1f07d1707d5daf0358cc60b30f06121f7e60 Mon Sep 17 00:00:00 2001
From: Zang Ruochen <zangrc.fnst@cn.fujitsu.com>
Date: Fri, 25 Dec 2020 11:41:43 +0900
Subject: [PATCH] don't fail if GLOB_BRACE is not defined

Signed-off-by: Zang Ruochen <zangrc.fnst@cn.fujitsu.com>
---
Upstream-Status: Pending

 src/util.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/util.c b/src/util.c
index 841ec12..59595da 100644
--- a/src/util.c
+++ b/src/util.c
@@ -32,6 +32,12 @@
 #include "names.h"
 #include "yaml-helpers.h"
 
+/* Don't fail if the standard library
+ * doesn't provide brace expansion */
+#ifndef GLOB_BRACE
+#define GLOB_BRACE 0
+#endif
+
 NETPLAN_ABI GHashTable*
 wifi_frequency_24;
 
-- 
2.25.1

