From 05194070775b597829abfa0d9a5f96d19cc064b8 Mon Sep 17 00:00:00 2001
From: Kumar Gala <galak@kernel.crashing.org>
Date: Thu, 11 Aug 2011 01:26:33 -0500
Subject: [PATCH 2/6] automake: Update for python.m4 to respect libdir

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Kumar Gala <galak@kernel.crashing.org>

Updated for automake-1.12.6

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 m4/python.m4 | 32 ++++++--------------------------
 1 file changed, 6 insertions(+), 26 deletions(-)

diff --git a/m4/python.m4 b/m4/python.m4
index 0b1db26..07db0eb 100644
--- a/m4/python.m4
+++ b/m4/python.m4
@@ -99,6 +99,8 @@ AC_DEFUN([AM_PATH_PYTHON],
     [am_cv_python_version=`$PYTHON -c "import sys; print ('%u.%u' % sys.version_info[[:2]])"`])
   AC_SUBST([PYTHON_VERSION], [$am_cv_python_version])
 
+  AC_SUBST([PYTHON_LIB_PREFIX], ['${libdir}'])
+
   dnl At times, e.g., when building shared libraries, you may want
   dnl to know which OS platform Python thinks this is.
   dnl
@@ -268,18 +270,7 @@ except ImportError:
    else
      am_py_prefix=$am_cv_python_prefix
    fi
-   am_cv_python_pythondir=`$PYTHON -c "
-$am_python_setup_sysconfig
-if can_use_sysconfig:
-  try:
-    $am_python_setup_scheme
-    sitedir = sysconfig.get_path('purelib', scheme, vars={'base':'$am_py_prefix'})
-  except:
-    sitedir = sysconfig.get_path('purelib', vars={'base':'$am_py_prefix'})
-else:
-  from distutils import sysconfig
-  sitedir = sysconfig.get_python_lib(0, 0, prefix='$am_py_prefix')
-sys.stdout.write(sitedir)"`
+   am_cv_python_pythondir=`echo "$PYTHON_LIB_PREFIX/python$PYTHON_VERSION/site-packages"`
    #
    case $am_cv_python_pythondir in
    $am_py_prefix*)
@@ -289,7 +280,7 @@ sys.stdout.write(sitedir)"`
    *)
      case $am_py_prefix in
        /usr|/System*) ;;
-       *) am_cv_python_pythondir="\${PYTHON_PREFIX}/lib/python$PYTHON_VERSION/site-packages"
+       *) am_cv_python_pythondir=$PYTHON_LIB_PREFIX/python$PYTHON_VERSION/site-packages
           ;;
      esac
      ;;
@@ -315,18 +306,7 @@ sys.stdout.write(sitedir)"`
    else
      am_py_exec_prefix=$am_cv_python_exec_prefix
    fi
-   am_cv_python_pyexecdir=`$PYTHON -c "
-$am_python_setup_sysconfig
-if can_use_sysconfig:
-  try:
-    $am_python_setup_scheme
-    sitedir = sysconfig.get_path('platlib', scheme, vars={'platbase':'$am_py_exec_prefix'})
-  except:
-    sitedir = sysconfig.get_path('platlib', vars={'platbase':'$am_py_exec_prefix'})
-else:
-  from distutils import sysconfig
-  sitedir = sysconfig.get_python_lib(1, 0, prefix='$am_py_exec_prefix')
-sys.stdout.write(sitedir)"`
+   am_cv_python_pyexecdir=`echo "$PYTHON_LIB_PREFIX/python$PYTHON_VERSION/site-packages"`
    #
    case $am_cv_python_pyexecdir in
    $am_py_exec_prefix*)
@@ -336,7 +316,7 @@ sys.stdout.write(sitedir)"`
    *)
      case $am_py_exec_prefix in
        /usr|/System*) ;;
-       *) am_cv_python_pyexecdir="\${PYTHON_EXEC_PREFIX}/lib/python$PYTHON_VERSION/site-packages"
+       *) am_cv_python_pyexecdir=$PYTHON_LIB_PREFIX/python$PYTHON_VERSION/site-packages
           ;;
      esac
      ;;
-- 
2.39.2

