From a4290bf1887f9203288858ca76bdd20b2edf337a Mon Sep 17 00:00:00 2001
From: Baptiste Durand <baptiste.durand@gmail.com>
Date: Sun, 9 Sep 2018 20:50:41 +0200
Subject: [PATCH] Update Gettext version

Upstream-Status: Inappropriate [oe-core specific]

Signed-off-by: Baptiste Durand <baptiste.durand@gmail.com>
---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -14,7 +14,7 @@
 # License along with MiniDLNA; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
 # USA.
-AC_INIT(MiniDLNA,1.1.3,,minidlna)
+AC_INIT(MiniDLNA,1.2.1,,minidlna)
 #LT_INIT
 
 AC_CANONICAL_TARGET
@@ -28,7 +28,7 @@ m4_ifdef([AC_USE_SYSTEM_EXTENSIONS], [AC
 
 AM_ICONV
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION(0.18)
+AM_GNU_GETTEXT_VERSION(0.20)
 
 # Checks for programs.
 AC_PROG_AWK
