From 7c00d75d16da18a9998fc4cca28d3c953dd54ceb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20P=C3=A9ron?= <peron.clem@gmail.com>
Date: Mon, 14 Feb 2022 18:37:22 +0100
Subject: [PATCH] Cmds: automatically create folder

---
Upstream-Status: Pending

 Cmds | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Cmds b/Cmds
index e590b38..3df23b8 100644
--- a/Cmds
+++ b/Cmds
@@ -44,7 +44,8 @@ ifeq ("$(origin PREFIX)", "$(filter $(origin PREFIX), file command line)")
   INSTX = echo -e "  INST\t$(1)" && install -d $(2) && \
 	install -C $(1) $(2)/$(shell basename $(1))
 else
-  INSTX = echo -e "  INST\t$(1)" && install -C $(1) $(2)/$(shell basename $(1))
+  INSTX = echo -e "  INST\t$(1)" && install -d $(2) && \
+	install -C $(1) $(2)/$(shell basename $(1))
 endif
 
 MKDIR = echo -e  "  MKDIR\t$(1)" && mkdir -p $(1)
-- 
2.32.0

