From 3667cced504bc40b176d4570d99a83fe67599f3a Mon Sep 17 00:00:00 2001
From: Drew DeVault <sir@cmpwn.com>
Date: Wed, 4 Sep 2024 10:07:46 +0200
Subject: [PATCH] Makefile: drop -static

Upstream-Status: Backport [https://git.sr.ht/~sircmpwn/scdoc/commit/3667cced504bc40b176d4570d99a83fe67599f3a]
Signed-off-by: Alex Kiernan <alex.kiernan@gmail.com>
---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index 96ad48cd04a6..49f076aadd39 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,6 @@
 VERSION=1.11.3
 CFLAGS?=-g
 MAINFLAGS:=-DVERSION='"$(VERSION)"' -Wall -Wextra -Werror -Wno-unused-parameter
-LDFLAGS+=-static
 INCLUDE+=-Iinclude
 PREFIX?=/usr/local
 BINDIR?=$(PREFIX)/bin
