Index: git/configure.ac
===================================================================
--- git.orig/configure.ac
+++ git/configure.ac
@@ -1109,11 +1109,7 @@ AC_ARG_WITH([crypto],
      [],
      [crypto=gcrypt])
 
-if test "x${libexecdir}" = xNONE; then
-	probe_dir="/usr/local/libexec/openscap"
-else
-	EXPAND_DIR(probe_dir,"${libexecdir}/openscap")
-fi
+probe_dir="/usr/local/libexec/openscap"
 
 AC_SUBST(probe_dir)
 
