Upstream-Status:Inappropriate [embedded specific]

Index: perl-5.24.1/Makefile.SH
===================================================================
--- perl-5.24.1.orig/Makefile.SH
+++ perl-5.24.1/Makefile.SH
@@ -710,7 +710,7 @@ bitcount.h: generate_uudmap\$(HOST_EXE_E
 
 $spitshell >>$Makefile <<'!NO!SUBS!'
 generate_uudmap$(HOST_EXE_EXT): generate_uudmap$(OBJ_EXT)
-	$(CC) -o generate_uudmap$(EXE_EXT) $(LDFLAGS) generate_uudmap$(OBJ_EXT) $(libs)
+	$(BUILD_CC) -o generate_uudmap$(EXE_EXT) generate_uudmap.c
 
 !NO!SUBS!
 ;;
