From ea6829c152c18e55cbe2a05317bc12a2be65ed65 Mon Sep 17 00:00:00 2001
From: Liu Yiding <liuyd.fnst@fujitsu.com>
Date: Mon, 9 Jun 2025 03:22:06 +0000
Subject: [PATCH] remove man page

Upstream-Status: Inappropriate [OE specific]
Signed-off-by: Liu Yiding <liuyd.fnst@fujitsu.com>
---
 configure.ac      |  1 -
 utils/Makefile.am | 23 +----------------------
 2 files changed, 1 insertion(+), 23 deletions(-)

diff --git a/configure.ac b/configure.ac
index b843037..224311f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -65,7 +65,6 @@ IMSETTINGS_MODULEDIR="${libdir}/imsettings"
 AC_SUBST(IMSETTINGS_MODULEDIR)
 
 AM_MISSING_PROG([GIT], [git])
-AM_MISSING_PROG([DB2MAN], [db2x_docbook2man])
 
 dnl ======================================================================
 dnl functions testing
diff --git a/utils/Makefile.am b/utils/Makefile.am
index b492532..2236257 100644
--- a/utils/Makefile.am
+++ b/utils/Makefile.am
@@ -20,19 +20,11 @@ DEPS =							\
 	$(NULL)
 EXTRA_DIST =						\
 	$(autostart_in_files)				\
-	$(man_xml_files)				\
-	$(man1_MANS)					\
 	$(NULL)
 DISTCLEANFILES =					\
 	$(autostart_DATA)				\
 	$(NULL)
-BUILT_SOURCES =						\
-	$(man1_MANS)					\
-	$(NULL)
-MAINTAINERCLEANFILES =					\
-	$(man1_MANS)					\
-	$(NULL)
-SUFFIXES = .1.xml
+
 
 ##
 # Local Rules
@@ -50,12 +42,6 @@ bin_PROGRAMS =						\
 libexec_PROGRAMS =					\
 	imsettings-check				\
 	$(NULL)
-man1_MANS =						\
-	imsettings-info.1				\
-	imsettings-list.1				\
-	imsettings-reload.1				\
-	imsettings-switch.1				\
-	$(NULL)
 #
 autostartdir = $(sysconfdir)/xdg/autostart
 #
@@ -64,13 +50,6 @@ autostart_in_files =					\
 	$(NULL)
 autostart_DATA = $(autostart_in_files:.desktop.in=.desktop)
 #
-man_xml_files = $(man1_MANS:.1=.1.xml)
-#
-.xml.1:
-	$(AM_V_GEN) $(DB2MAN) $<
-.1.xml.1:
-	$(AM_V_GEN) $(DB2MAN) $<
-#
 imsettings_check_SOURCES =				\
 	imsettings-check.c				\
 	$(NULL)
-- 
2.43.0

