--- /tmp/Makefile.old	2006-01-16 22:51:34.000000000 +0100
+++ minilite-050/Makefile	2009-05-01 16:36:32.000000000 +0200
@@ -45,7 +45,7 @@
 
 install-program: all
 	mkdir -p $(DESTDIR)$(PREFIX)/bin
-	$(INSTALL) -s $(PACKAGE) $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
+	$(INSTALL) $(PACKAGE) $(DESTDIR)$(PREFIX)/bin/$(PACKAGE)
 	mkdir -p $(DESTDIR)$(PREFIX)/share/applications
 	$(INSTALL) -m 644 $(PACKAGE).desktop $(DESTDIR)$(PREFIX)/share/applications/$(PACKAGE).desktop
 	mkdir -p $(DESTDIR)$(PREFIX)/share/pixmaps/
