From 22afc5d9aaa215c3c87ba21c77d47da44ab3b113 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Fri, 26 Aug 2016 18:20:32 +0300
Subject: [PATCH] Use pkg-config for pcre dependency instead of -config script.

RP 2014/5/22
Upstream-Status: Pending
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 configure.ac | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 5383cec..c29a902 100644
--- a/configure.ac
+++ b/configure.ac
@@ -651,10 +651,18 @@ AC_ARG_WITH([pcre],
 )
 AC_MSG_RESULT([$WITH_PCRE])
 
-if test "$WITH_PCRE" != no; then
-  if test "$WITH_PCRE" != yes; then
-    PCRE_LIB="-L$WITH_PCRE/lib -lpcre"
-    CPPFLAGS="$CPPFLAGS -I$WITH_PCRE/include"
+if test "$WITH_PCRE" != "no"; then
+  PKG_CHECK_MODULES(PCREPKG, [libpcre], [
+				PCRE_LIB=${PCREPKG_LIBS}
+				CPPFLAGS="$CPPFLAGS ${PCREPKG_CFLAGS}"
+  ], [
+				AC_MSG_ERROR([pcre pkgconfig not found, install the pcre-devel package or build with --without-pcre])
+  ])
+
+  if test x"$PCRE_LIB" != x; then
+    AC_DEFINE([HAVE_LIBPCRE], [1], [libpcre])
+    AC_DEFINE([HAVE_PCRE_H], [1], [pcre.h])
+    AC_SUBST(PCRE_LIB)
   else
     AC_PATH_PROG([PCRECONFIG], [pcre-config])
     if test -n "$PCRECONFIG"; then
-- 
2.15.0

