From 6ede7d01b18a4d9eeaccd25f5c1ab9985cb65307 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 12 Feb 2023 21:25:04 -0800
Subject: [PATCH] zlib: Include unistd.h for open/close C APIs

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 third_party/zlib/gzguts.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/third_party/zlib/gzguts.h b/third_party/zlib/gzguts.h
index 57faf37..3c700c2 100644
--- a/third_party/zlib/gzguts.h
+++ b/third_party/zlib/gzguts.h
@@ -19,6 +19,7 @@
 #endif
 
 #include <stdio.h>
+#include <unistd.h>
 #include "zlib.h"
 #ifdef STDC
 #  include <string.h>
