From b37554e0bc3cf383e6547c5c6a69c6f6849c09e3 Mon Sep 17 00:00:00 2001
From: Eric Leblond <eric@regit.org>
Date: Wed, 17 Jul 2019 12:35:12 +0200
Subject: [PATCH] af-packet: fix build on recent Linux kernels

Upstream-Status: Backport
Signed-off-by: Armin kuster <akuster808@gmail.com>
---
 src/source-af-packet.c | 4 ++++
 1 file changed, 4 insertions(+)

Index: suricata-4.1.5/src/source-af-packet.c
===================================================================
--- suricata-4.1.5.orig/src/source-af-packet.c
+++ suricata-4.1.5/src/source-af-packet.c
@@ -68,6 +68,10 @@
 #include <linux/sockios.h>
 #endif
 
+#if HAVE_LINUX_SOCKIOS_H
+#include <linux/sockios.h>
+#endif
+
 #ifdef HAVE_PACKET_EBPF
 #include "util-ebpf.h"
 #include <bpf/libbpf.h>
