Skip pkg Makefile from using its own rust steps

Upstream-Status: Inappropriate [OE Specific]

Signed-off-by: Armin Kuster <akuster808@gmail.com>

Index: suricata-7.0.0/Makefile.in
===================================================================
--- suricata-7.0.0.orig/Makefile.in
+++ suricata-7.0.0/Makefile.in
@@ -424,7 +424,7 @@ EXTRA_DIST = ChangeLog COPYING LICENSE s
 	     acsite.m4 \
 	     scripts/generate-images.sh
 
-SUBDIRS = $(HTP_DIR) rust src qa rules doc contrib etc python ebpf \
+SUBDIRS = $(HTP_DIR) src qa rules doc contrib etc python ebpf \
           $(SURICATA_UPDATE_DIR)
 
 CLEANFILES = stamp-h[0-9]*
Index: suricata-7.0.0/Makefile.am
===================================================================
--- suricata-7.0.0.orig/Makefile.am
+++ suricata-7.0.0/Makefile.am
@@ -8,7 +8,7 @@ EXTRA_DIST = ChangeLog COPYING LICENSE s
 	     lua \
 	     acsite.m4 \
 	     scripts/generate-images.sh
-SUBDIRS = $(HTP_DIR) rust src qa rules doc contrib etc python ebpf \
+SUBDIRS = $(HTP_DIR) src qa rules doc contrib etc python ebpf \
           $(SURICATA_UPDATE_DIR)
 
 CLEANFILES = stamp-h[0-9]*
