Upstream-Status: Pending

Description: Bring Clang's _Nonnull keyword to GCC
Author: Kai-Chung Yan
Forwarded: not-needed
--- a/system/core/adb/sysdeps.h
+++ b/system/core/adb/sysdeps.h
@@ -40,11 +40,12 @@
 #include "sysdeps/network.h"
 #include "sysdeps/stat.h"
 
+#define _Nonnull
+#define _Nullable
+
 #ifdef _WIN32
 
 // Clang-only nullability specifiers
-#define _Nonnull
-#define _Nullable
 
 #include <ctype.h>
 #include <direct.h>
