From 2f7382b35d59fe08034603497e82ffb943fedef1 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Wed, 30 Jun 2021 15:31:16 +0200
Subject: [PATCH] libdnf/dnf-context.cpp: do not try to access BDB database

Upstream-Status: Inappropriate [upstream needs to rework this to support
sqlite]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 libdnf/dnf-context.cpp | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/libdnf/dnf-context.cpp b/libdnf/dnf-context.cpp
index 86f71a79..9cdcf769 100644
--- a/libdnf/dnf-context.cpp
+++ b/libdnf/dnf-context.cpp
@@ -2264,20 +2264,6 @@ dnf_context_setup(DnfContext *context,
         !dnf_context_set_os_release(context, error))
         return FALSE;
 
-    /* setup a file monitor on the rpmdb, if we're operating on the native / */
-    if (g_strcmp0(priv->install_root, "/") == 0) {
-        rpmdb_path = g_build_filename(priv->install_root, "var/lib/rpm/Packages", NULL);
-        file_rpmdb = g_file_new_for_path(rpmdb_path);
-        priv->monitor_rpmdb = g_file_monitor_file(file_rpmdb,
-                               G_FILE_MONITOR_NONE,
-                               NULL,
-                               error);
-        if (priv->monitor_rpmdb == NULL)
-            return FALSE;
-        g_signal_connect(priv->monitor_rpmdb, "changed",
-                         G_CALLBACK(dnf_context_rpmdb_changed_cb), context);
-    }
-
     /* copy any vendor distributed cached metadata */
     if (!dnf_context_copy_vendor_cache(context, error))
         return FALSE;
