From 320b8378ee30eb5e0fe83a8b397f822f2f88a4c1 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 1 Mar 2015 22:25:07 +0000
Subject: [PATCH] %% original patch: mediactl-pkgconfig.patch

---
Upstream-Status: Pending

 utils/media-ctl/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/utils/media-ctl/Makefile.am b/utils/media-ctl/Makefile.am
index e255e16..ff7b417 100644
--- a/utils/media-ctl/Makefile.am
+++ b/utils/media-ctl/Makefile.am
@@ -20,6 +20,7 @@ libv4l2subdev_la_LIBADD = libmediactl.la
 mediactl_includedir=$(includedir)/mediactl
 noinst_HEADERS = mediactl.h v4l2subdev.h
 
+pkgconfig_DATA = libmediactl.pc
 bin_PROGRAMS = media-ctl
 media_ctl_SOURCES = media-ctl.c options.c options.h tools.h
 media_ctl_LDADD = libmediactl.la libv4l2subdev.la
