From f7109d6b2fcb291824d795071c04a492d9fbc45b Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 1 Mar 2015 22:25:07 +0000
Subject: [PATCH] %% original patch: export-mediactl-headers.patch

---
Upstream-Status: Pending

 utils/media-ctl/Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/utils/media-ctl/Makefile.am b/utils/media-ctl/Makefile.am
index ff7b417..6ce656f 100644
--- a/utils/media-ctl/Makefile.am
+++ b/utils/media-ctl/Makefile.am
@@ -17,8 +17,8 @@ CLEANFILES = $(BUILT_SOURCES)
 nodist_libv4l2subdev_la_SOURCES = $(BUILT_SOURCES)
 libv4l2subdev_la_SOURCES = libv4l2subdev.c
 libv4l2subdev_la_LIBADD = libmediactl.la
-mediactl_includedir=$(includedir)/mediactl
-noinst_HEADERS = mediactl.h v4l2subdev.h
+otherincludedir = $(includedir)/mediactl
+otherinclude_HEADERS = mediactl.h v4l2subdev.h
 
 pkgconfig_DATA = libmediactl.pc
 bin_PROGRAMS = media-ctl
