From fa40b874f6470ec11a8fd7b0c9909d0cdd2d6feb Mon Sep 17 00:00:00 2001
From: Dan Callaghan <dan.callaghan@opengear.com>
Date: Fri, 5 Feb 2021 08:56:34 +1000
Subject: [PATCH] enable LTO

Upstream-Status: Pending
---
 Cargo.toml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Cargo.toml b/Cargo.toml
index 7a2f6c8..cdb6b5d 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -3,3 +3,6 @@
 name = "rust-hello-world"
 version = "0.0.1"
 authors = ["Cody P Schafer <dev@codyps.com>"]
+
+[profile.release]
+lto = true
-- 
2.28.0

