From ba9015386cbc044e111d7c266f13e2be045e4bf1 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen.kooi@linaro.org>
Date: Tue, 17 Jun 2014 09:10:57 +0200
Subject: [PATCH] configure: use pkg-config for PCRE detection

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Koen Kooi <koen.kooi@linaro.org>
---
 configure.in | 26 +++++---------------------
 1 file changed, 5 insertions(+), 21 deletions(-)

diff --git a/configure.in b/configure.in
index 38c1d0a..c799aec 100644
--- a/configure.in
+++ b/configure.in
@@ -221,27 +221,11 @@ else if which $with_pcre 2>/dev/null; then :; else
 fi
 fi
 
-AC_CHECK_TARGET_TOOLS(PCRE_CONFIG, [pcre2-config pcre-config],
-                      [`which $with_pcre 2>/dev/null`], $with_pcre)
-
-if test "x$PCRE_CONFIG" != "x"; then
-  if $PCRE_CONFIG --version >/dev/null 2>&1; then :; else
-    AC_MSG_ERROR([Did not find working script at $PCRE_CONFIG])
-  fi
-  case `$PCRE_CONFIG --version` in
-  [1[0-9].*])
-    AC_DEFINE(HAVE_PCRE2, 1, [Detected PCRE2]) 
-    ;;
-  [[1-5].*])
-    AC_MSG_ERROR([Need at least pcre version 6.0])
-    ;;
-  esac
-  AC_MSG_NOTICE([Using external PCRE library from $PCRE_CONFIG])
-  APR_ADDTO(PCRE_INCLUDES, [`$PCRE_CONFIG --cflags`])
-  APR_ADDTO(PCRE_LIBS, [`$PCRE_CONFIG --libs8 2>/dev/null || $PCRE_CONFIG --libs`])
-else
-  AC_MSG_ERROR([pcre(2)-config for libpcre not found. PCRE is required and available from http://pcre.org/])
-fi
+PKG_CHECK_MODULES([PCRE], [libpcre], [
+  AC_DEFINE([HAVE_PCRE], [1], [Define if you have PCRE library])
+], [
+  AC_MSG_ERROR([$PCRE_PKG_ERRORS])
+])
 APACHE_SUBST(PCRE_LIBS)
 
 AC_MSG_NOTICE([])
-- 
2.25.1

