From d7160d647c8f8847f3ea8d7b0eb222936962c1af Mon Sep 17 00:00:00 2001
From: Sinan Kaya <sinan.kaya@microsoft.com>
Date: Tue, 11 Feb 2020 11:58:02 -0500
Subject: [PATCH] Add fPIC option

Signed-off-by: Sinan Kaya <sinan.kaya@microsoft.com>
---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 74a3a4c..4f837b3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,6 +79,9 @@ option(ABSL_USE_GOOGLETEST_HEAD
 
 option(ABSL_RUN_TESTS "If ON, Abseil tests will be run." OFF)
 
+# link fails on arm64 and x86-64 without this
+add_compile_options(-fPIC)
+
 if(${ABSL_RUN_TESTS})
   # enable CTest.  This will set BUILD_TESTING to ON unless otherwise specified
   # on the command line
-- 
2.20.1.windows.1

