From e600d3c09d2f97a197f86e6987d44d142e7b7cdf Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 27 Jun 2017 21:12:17 -0700
Subject: [PATCH] orrery: Fix sprintf format

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 orrery.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/orrery.c b/orrery.c
index e3a1f6d..1b31a9f 100644
--- a/orrery.c
+++ b/orrery.c
@@ -3767,7 +3767,7 @@ static void drawOptsScreens(void)
 		      displayWidth/2 - gdk_string_width(smallFont, scratchString)/2,
 		      12, scratchString);
       for (day = 0; day < 7; day++) {
-	sprintf(scratchString, dayName[day]);
+	sprintf(scratchString, "%s", dayName[day]);
 	gdk_draw_string(pixmap, smallFont, blueGC,
 			CAL_DAY_WIDTH/2 + day*CAL_DAY_WIDTH + CAL_LEFT_OFFSET -
 			gdk_string_width(smallFont, scratchString)/2,
-- 
2.13.2

