From f98b390a2d323f7f92fb0492b0943d201afe5b8f Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 23 Aug 2022 10:40:53 -0700
Subject: [PATCH] config: Include sys/types.h for mode_t defintion

Upstream-Status: Submitted [https://github.com/ccache/ccache/pull/1145]
Signed-off-by: Khem Raj <raj.khem@gmail.com>

---
 src/Config.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/Config.hpp b/src/Config.hpp
index a9e08ec..9e7af40 100644
--- a/src/Config.hpp
+++ b/src/Config.hpp
@@ -25,6 +25,8 @@
 
 #include "third_party/nonstd/optional.hpp"
 
+#include <sys/types.h>
+
 #include <cstdint>
 #include <functional>
 #include <limits>
