From 5f3911737c63a098caf73f1365320fe1b3a7638b Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 21 Apr 2016 03:05:57 -0400
Subject: [PATCH] support cross-complication

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
Signed-off-by: Leon Anavi <leon.anavi@konsulko.com>
---
 setup.py | 35 -----------------------------------
 1 file changed, 35 deletions(-)

diff --git a/setup.py b/setup.py
index eb83cfe..3f9567b 100755
--- a/setup.py
+++ b/setup.py
@@ -13,41 +13,6 @@ import sys
 # - Need gcc
 # - Need C headers
 # - Need libcap headers
-if not sys.platform.startswith('linux'):
-    sys.stderr.write("This module only works on linux\n")
-    sys.exit(1)
-
-kvers = os.uname()[2]
-if kvers < '2.6.18' and not os.environ.get("PRCTL_SKIP_KERNEL_CHECK",False):
-    sys.stderr.write("This module requires linux 2.6.18 or newer\n")
-    sys.exit(1)
-
-if sys.version_info[:2] < (2,4):
-    sys.stderr.write("This module requires python 2.4 or newer\n")
-    sys.exit(1)
-
-exit = False
-try:
-    subprocess.call(['gcc','-v'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-except:
-    sys.stderr.write("You need to install gcc to build this module\n")
-    sys.exit(1)
-
-sp = subprocess.Popen(['cpp'], stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=os.environ)
-sp.communicate('#include <sys/prctl.h>\n'.encode())
-if sp.returncode:
-    sys.stderr.write("You need to install libc development headers to build this module\n")
-    exit = True
-
-sp = subprocess.Popen(['cpp'], stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, env=os.environ)
-sp.communicate('#include <sys/capability.h>\n'.encode())
-if sp.returncode:
-    sys.stderr.write("You need to install libcap development headers to build this module\n")
-    exit = True
-
-if exit:
-    sys.exit(1)
-
 _prctl = Extension("_prctl",
                    sources = ['_prctlmodule.c'],
                    depends = ['securebits.h'],
-- 
2.17.1

