Description: Fix kernel 5.8 forbidding use of vermagic.h header file
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Origin: https://www.virtualbox.org/ticket/19644
Bug-Ubuntu: https://launchpad.net/bugs/1884652
Last-Update: 2020-08-10

--- virtualbox-6.1.12-dfsg.orig/src/VBox/Additions/linux/sharedfolders/vfsmod.c
+++ virtualbox-6.1.12-dfsg/src/VBox/Additions/linux/sharedfolders/vfsmod.c
@@ -53,7 +53,9 @@
 #include <linux/seq_file.h>
 #include <linux/vfs.h>
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 5, 62)
-# include <linux/vermagic.h>
+# if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
+#  include <linux/vermagic.h>
+# endif
 #endif
 #include <VBox/err.h>
 #include <iprt/path.h>
