Index: dejagnu-1.4.4/configure.in
===================================================================
--- dejagnu-1.4.4.orig/configure.in	2014-07-18 07:05:49.997481207 +0000
+++ dejagnu-1.4.4/configure.in	2014-07-18 07:05:50.085481210 +0000
@@ -1,10 +1,10 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_PREREQ(2.13)
-AC_INIT(runtest.exp)
+AC_INIT(dejagnu, 1.4.4)
 dnl AC_CONFIG_AUX_DIR(..)
 
 dnl These are required by automake
-AM_INIT_AUTOMAKE(dejagnu, 1.4.4)
+AM_INIT_AUTOMAKE([foreign])
 AM_MAINTAINER_MODE
 AC_PROG_MAKE_SET
 
Index: dejagnu-1.4.4/example/calc/configure.in
===================================================================
--- dejagnu-1.4.4.orig/example/calc/configure.in	2002-04-26 03:32:40.000000000 +0000
+++ dejagnu-1.4.4/example/calc/configure.in	2014-07-18 07:11:59.085491266 +0000
@@ -1,8 +1,8 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_PREREQ(2.5)
-AC_INIT(calc.c)
+AC_INIT(calc, 1.1)
 AM_CONFIG_HEADER(calc.h)
-AM_INIT_AUTOMAKE(calc, 1.1)
+AM_INIT_AUTOMAKE([foreign])
 
 AC_PROG_CC
 AC_PROG_INSTALL
Index: dejagnu-1.4.4/example/hello/configure.in
===================================================================
--- dejagnu-1.4.4.orig/example/hello/configure.in	2002-08-31 05:46:16.000000000 +0000
+++ dejagnu-1.4.4/example/hello/configure.in	2014-07-18 07:12:23.721491937 +0000
@@ -25,7 +25,7 @@
 # ------------------------------------------------------------------------
 
 AC_INIT(helloworld, demo-version, philip.wilsey@ieee.org)
-AM_INIT_AUTOMAKE(helloworld, demo-version)
+AM_INIT_AUTOMAKE([foreign])
 
 #AC_CONFIG_SRCDIR([hello.cc])
 #AC_CONFIG_HEADER([config.h])
