Upstream-Status: Pending

--- a/src/meson.build	2023-01-24 09:22:59.874074134 +0100
+++ b/src/meson.build	2023-01-24 09:27:00.494627305 +0100
@@ -73,12 +73,12 @@
 
 
 # Translate and install policy file
-i18n.merge_file (
-  input: 'org.opensuse.cupspkhelper.mechanism.policy.in',
-  output: 'org.opensuse.cupspkhelper.mechanism.policy',
-  type: 'xml',
-  data_dirs: join_paths (meson.source_root (), 'src'),
-  po_dir: join_paths (meson.source_root (), 'po'),
-  install: true,
-  install_dir: join_paths (prefix, datadir, 'polkit-1', 'actions')
-)
+#i18n.merge_file (
+#  input: 'org.opensuse.cupspkhelper.mechanism.policy.in',
+#  output: 'org.opensuse.cupspkhelper.mechanism.policy',
+#  type: 'xml',
+#  data_dirs: join_paths (meson.source_root (), 'src'),
+#  po_dir: join_paths (meson.source_root (), 'po'),
+#  install: true,
+#  install_dir: join_paths (prefix, datadir, 'polkit-1', 'actions')
+#)
