Upstream Status: Iinappropriate

Issues do to build env.

Signed-off-by: Armin Kuster <akuster808@gmail.com>

Index: aircrack-ng-1.2-rc2/src/Makefile
===================================================================
--- aircrack-ng-1.2-rc2.orig/src/Makefile
+++ aircrack-ng-1.2-rc2/src/Makefile
@@ -3,8 +3,6 @@ include $(AC_ROOT)/common.mak
 
 TEST_DIR	= $(AC_ROOT)/test
 
-CFLAGS		+= -Iinclude
-
 iCC             = $(shell find /opt/intel/cc/*/bin/icc)
 iCFLAGS         = -w -mcpu=pentiumpro -march=pentiumpro $(COMMON_CFLAGS)
 iOPTFLAGS       = -O3 -ip -ipo -D_FILE_OFFSET_BITS=64
@@ -102,7 +100,7 @@ endif
 
 
 ifeq ($(subst TRUE,true,$(filter TRUE true,$(sqlite) $(SQLITE))),true)
-	LIBSQL		= -L/usr/local/lib -lsqlite3
+	LIBSQL		= -lsqlite3
 else
 	LIBSQL		=
 endif
