Upstream-Status: Inappropriate [configuration]
---
 appearance/mb-appearance.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/appearance/mb-appearance.c b/appearance/mb-appearance.c
index 6af8376..71fd57b 100644
--- a/appearance/mb-appearance.c
+++ b/appearance/mb-appearance.c
@@ -317,11 +317,13 @@ on_gconf_value_changed (GConfClient* client, const gchar* key, GConfValue* value
     gtk_font_button_set_font_name (GTK_FONT_BUTTON (font_button),
                                    gconf_value_get_string (value));
   } else if (strcmp (key, HANDED_KEY) == 0) {
+#if 0
     GEnumValue *e;
     e = g_enum_get_value_by_name (g_type_class_ref (GTK_TYPE_CORNER_TYPE),
                                   gconf_value_get_string (value));
     gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (handed_check),
                                   e->value == GTK_CORNER_TOP_RIGHT);
+#endif
   }
 }
 
@@ -408,10 +410,12 @@ main (int argc, char **argv) {
 
   /* Left/Right Handed */
   
+#if 0
   handed_check = gtk_check_button_new_with_mnemonic (_("_Left-handed"));
   frame = new_frame (_("Orientation"),handed_check);
   gtk_box_pack_start (GTK_BOX (box), frame, TRUE, TRUE, 0);
   g_signal_connect (handed_check, "toggled", G_CALLBACK (on_handed_set), NULL);
+#endif
 
   gconf_client_add_dir (gconf, INTERFACE_DIR, GCONF_CLIENT_PRELOAD_ONELEVEL, NULL);
   g_signal_connect (gconf, "value-changed", G_CALLBACK (on_gconf_value_changed), NULL);
-- 
2.8.1

