From a490e82dca5a669b0af27a13d74759d8f77e2333 Mon Sep 17 00:00:00 2001
From: Vyacheslav Yurkov <v.yurkov@precitec.de>
Date: Mon, 3 Oct 2022 18:25:15 +0200
Subject: [PATCH] fix(build): do not install git files

Signed-off-by: Vyacheslav Yurkov <v.yurkov@precitec.de>
---
Upstream-Status: Backport [https://github.com/open62541/open62541/commit/a0328d4cb527f9778a262fc4a6b42bfbdffc38e9]

 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1934374e..bc5c8bad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1703,6 +1703,7 @@ install(DIRECTORY ${UA_install_tools_dirs}
     FILES_MATCHING
     PATTERN "*"
     PATTERN "*.pyc" EXCLUDE
+    PATTERN ".git*" EXCLUDE
     )
 
 install(FILES ${UA_install_tools_files} DESTINATION ${open62541_install_tools_dir})
