From d59e03b5a82b3e0debc3a3c77270bd160f4309f9 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 8 Jun 2021 20:42:19 -0700
Subject: [PATCH 4/4] syscalls: Add quotactl_path

Upstream-Status: Submitted [https://github.com/seccomp/libseccomp/pull/327]

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 include/seccomp-syscalls.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/seccomp-syscalls.h b/include/seccomp-syscalls.h
index 4ff814c..dd347d3 100644
--- a/include/seccomp-syscalls.h
+++ b/include/seccomp-syscalls.h
@@ -284,6 +284,7 @@
 #define __PNR_clock_nanosleep			-10250
 #define __PNR_gettimeofday			-10251
 #define __PNR_fcntl				-10252
+#define __PNR_quotactl_path			-10253
 
 /*
  * libseccomp syscall definitions
@@ -1557,6 +1558,12 @@
 #define __SNR_riscv_flush_icache	__PNR_riscv_flush_icache
 #endif
 
+#ifdef __NR_quotactl_path
+#define __SNR_quotactl_path       __NR_quotactl_path
+#else
+#define __SNR_quotactl_path       __PNR_quotactl_path
+#endif
+
 #ifdef __NR_rmdir
 #define __SNR_rmdir			__NR_rmdir
 #else
-- 
2.33.0

