--- /tmp/configure.in	2009-04-13 11:26:43.000000000 +0200
+++ nautilus-2.26.1/configure.in	2009-04-13 11:28:37.000000000 +0200
@@ -375,25 +375,6 @@
 
 dnl ===========================================================================
 
-dnl strftime checks
-
-AC_TRY_RUN([#include <time.h>
-                int main ()
-                {
-                  char buf[100];
-                  struct tm tm = {0};
-                  tm.tm_year = 99;
-                  if (strftime(buf, 100, "%EY", &tm) == 4 &&
-		      strcmp (buf, "1999")==0)
-                    return 0;
-                  return 1;
-                }
-            ],
-	    AC_DEFINE(HAVE_STRFTIME_EXTENSION, 1, [Define if strftime supports %E and %O modifiers.])
-            )
-
-dnl ===========================================================================
-
 #
 # Checks for Xft/XRender
 #
