--- enca-1.9/configure.ac.orig	2006-06-29 15:30:44.000000000 +0100
+++ enca-1.9/configure.ac	2006-06-29 15:31:41.000000000 +0100
@@ -92,14 +92,15 @@
 dnl And for gtk-doc
 gtk_CHECK_GTK_DOC
 
-dnl Dirty path hack.  Helps some people with badly set up search paths.
-if test "$prefix" = "NONE"; then
-  LDFLAGS="$LDFLAGS -L$ac_default_prefix/lib"
-  CPPFLAGS="$CPPFLAGS -I$ac_default_prefix/include"
-else
-  LDFLAGS="$LDFLAGS -L$prefix/lib"
-  CPPFLAGS="$CPPFLAGS -I$prefix/include"
-fi
+# This is really dumb code
+#dnl Dirty path hack.  Helps some people with badly set up search paths.
+#if test "$prefix" = "NONE"; then
+#  LDFLAGS="$LDFLAGS -L$ac_default_prefix/lib"
+#  CPPFLAGS="$CPPFLAGS -I$ac_default_prefix/include"
+#else
+#  LDFLAGS="$LDFLAGS -L$prefix/lib"
+#  CPPFLAGS="$CPPFLAGS -I$prefix/include"
+#fi
 
 dnl Checks for libraries.
 ye_CHECK_LIBM
