From ec82a194f3454ac203a134c9a99bac6d63095d25 Mon Sep 17 00:00:00 2001
From: "Meyer, Wolfgang" <Wolfgang.Meyer@gossenmetrawatt.com>
Date: Fri, 16 Sep 2022 18:29:03 +0200
Subject: [PATCH] meson.build: do not require xkbcommon

Upstream-Status: Pending
Signed-off-by: "Meyer, Wolfgang" <Wolfgang.Meyer@gossenmetrawatt.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 341939c..b5dab14 100644
--- a/meson.build
+++ b/meson.build
@@ -26,7 +26,7 @@ tiff_dep     = dependency('libtiff-4', required: get_option('tiff'))
 webp_dep     = dependency('libwebp', required : get_option('webp'))
 udev_dep     = dependency('libudev')
 input_dep    = dependency('libinput')
-xkb_dep      = dependency('xkbcommon')
+xkb_dep      = dependency('xkbcommon', required : false)
 tsm_dep      = dependency('libtsm', required : false)
 
 # other library deps
-- 
2.37.3

