From 36654a4484117e5f4d63a810c0d94bc9c7ee3a83 Mon Sep 17 00:00:00 2001
From: Fathi Boudra <fabo@debian.org>
Date: Tue, 2 Oct 2018 16:36:54 +0000
Subject: [PATCH] Remove bionic specific calls

Upstream-Status: Inappropriate

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/procattr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/procattr.c b/src/procattr.c
index f350808..761cf8e 100644
--- a/src/procattr.c
+++ b/src/procattr.c
@@ -8,7 +8,7 @@
 #include "selinux_internal.h"
 #include "policy.h"
 
-#ifdef HOST
+#ifndef __BIONIC__
 static pid_t gettid(void)
 {
 	return syscall(__NR_gettid);
