From e0d3e0bb1e38ff851696a7d8826e651d364ad8ce Mon Sep 17 00:00:00 2001
From: Li xin <lixin.fnst@cn.fujitsu.com>
Date: Fri, 5 Dec 2014 02:00:57 +0900
Subject: [PATCH 1/2] etc/openct.udev.in: disablePROGRAM

Bug fix: https://bugzilla.redhat.com/show_bug.cgi?id=287871

Upstream-Status: Pending

Signed-off-by: Li Xin <lixin.fnst@cn.fujitsu.com>
---
 etc/openct.udev.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/etc/openct.udev.in b/etc/openct.udev.in
index d11d0e1..48083c9 100644
--- a/etc/openct.udev.in
+++ b/etc/openct.udev.in
@@ -22,7 +22,8 @@ ACTION!="add", GOTO="openct_usb_rules_end"
 # 2010-01-06 removed, as latest udev doesn't know WAIT_FOR_ATTR any more.
 
 # sleep for 100ms - the wait_for_sysfs might not be enough
-PROGRAM="/bin/sleep 0.1"
+# Disabled in this package - see https://bugzilla.redhat.com/287871
+# PROGRAM="/bin/sleep 0.1"
 
 # ccid
 ATTR{bInterfaceClass}=="0b", ATTR{bInterfaceSubClass}=="00", ATTR{bInterfaceProtocol}=="00", ATTRS{idVendor}=="?*"  RUN+="@udevdir@/openct_usb /dev/$parent"
-- 
1.8.4.2

