Modernise configure to work with modern automake.

RP 2014/7/14

Upstream-Status: Pending

Index: re2c-0.13.5/configure.in
===================================================================
--- re2c-0.13.5.orig/configure.in	2008-05-25 14:42:34.000000000 +0000
+++ re2c-0.13.5/configure.in	2014-07-17 14:10:40.595821292 +0000
@@ -1,6 +1,6 @@
 AC_PREREQ([2.57])
 AC_INIT(re2c, 0.13.5, re2c-general@lists.sourceforge.net)
-AM_INIT_AUTOMAKE(re2c, $PACKAGE_VERSION)
+AM_INIT_AUTOMAKE([foreign])
 AC_CONFIG_SRCDIR(actions.cc)
 AM_CONFIG_HEADER(config.h)
 
