From 0ecaa5bab162abf0cb2057d77beeb7b89d5873b4 Mon Sep 17 00:00:00 2001
From: Jussi Kukkonen <jussi.kukkonen@intel.com>
Date: Tue, 21 Jun 2016 14:53:56 +0300
Subject: [PATCH 1/4] Hardcoded libtool

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
Signed-off-by: Jussi Kukkonen <jussi.kukkonen@intel.com>
---
 configure.ac | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 6628e21..f43ac09 100644
--- a/configure.ac
+++ b/configure.ac
@@ -617,7 +617,7 @@ AC_MSG_CHECKING([whether to write dependencies into .pc files])
 case $enable_explicit_deps in
   auto)
     export SED
-    deplibs_check_method=`(./libtool --config; echo 'eval echo \"$deplibs_check_method\"') | sh`
+    deplibs_check_method=`(./$host_alias-libtool --config; echo 'eval echo \"$deplibs_check_method\"') | sh`
     if test "x$deplibs_check_method" != xpass_all || test "x$enable_static" = xyes ; then
       enable_explicit_deps=yes
     else
@@ -895,7 +895,7 @@ else
     dnl Now we check to see if our libtool supports shared lib deps
     dnl (in a rather ugly way even)
     if $dynworks; then
-        module_libtool_config="${CONFIG_SHELL-/bin/sh} ./libtool --config"
+        module_libtool_config="${CONFIG_SHELL-/bin/sh} ./$host_alias-libtool --config"
         module_deplibs_check=`$module_libtool_config | \
             grep '^[[a-z_]]*check[[a-z_]]*_method=[['\''"]]' | \
             sed 's/.*[['\''"]]\(.*\)[['\''"]]$/\1/'`
@@ -1649,7 +1649,7 @@ AC_SUBST(GTK_PRINT_BACKENDS)
 # We are using gmodule-no-export now, but I'm leaving the stripping
 # code in place for now, since pango and atk still require gmodule.
 export SED
-export_dynamic=`(./libtool --config; echo eval echo \\$export_dynamic_flag_spec) | sh`
+export_dynamic=`(./$host_alias-libtool --config; echo eval echo \\$export_dynamic_flag_spec) | sh`
 if test -n "$export_dynamic"; then
   GDK_DEP_LIBS=`echo $GDK_DEP_LIBS | sed -e "s/$export_dynamic//"`
   GTK_DEP_LIBS=`echo $GTK_DEP_LIBS | sed -e "s/$export_dynamic//"`
-- 
2.12.0

