From e21d90ae8c131eae9d78e3d06f29cab55c952884 Mon Sep 17 00:00:00 2001
From: Matt Spinler <spinler@us.ibm.com>
Date: Thu, 19 Jan 2017 09:06:48 -0600
Subject: [PATCH] CFAM reg 0x2918 access hack

---
 libpdbg/bmcfsi.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/libpdbg/bmcfsi.c b/libpdbg/bmcfsi.c
index 64dbd69..f763f2a 100644
--- a/libpdbg/bmcfsi.c
+++ b/libpdbg/bmcfsi.c
@@ -247,7 +247,16 @@ static uint64_t fsi_abs_ar(uint32_t addr, int read)
 	/* Reformat the address. I'm not sure I fully understand this
 	 * yet but we basically shift the bottom byte and add 0b01
 	 * (for the write word?) */
-       	addr = ((addr & 0x1fff00) | ((addr & 0xff) << 2)) << 1;
+
+    /* Hack: this shifting doesn't work for all addresses, but it does
+     * work for 2918 and (and 2nd chip 2918) so use it just for that */
+    if ((addr == 0x2918) || (addr == 0x102918)) {
+        addr = ((addr & 0x1ffe00) | ((addr & 0x1ff) << 2)) << 1;
+    }
+    else {
+        addr = ((addr & 0x1fff00) | ((addr & 0xff) << 2)) << 1;
+    }
+
 	addr |= 0x3;
 	addr |= slave_id << 26;
 	addr |= (0x8ULL | !!(read)) << 22;
-- 
2.11.0

