From ed0b5a48c7ce59ea971a10c7e522933cf613841a Mon Sep 17 00:00:00 2001
From: Josep Puigdemont <josep.puigdemont@enea.com>
Date: Sun, 4 May 2014 16:02:07 +0200
Subject: [PATCH] Add install-ptest rules.

Signed-off-by: Josep Puigdemont <josep.puigdemont@enea.com>
Signed-off-by: Maxin B. John <maxin.john@enea.com>
Upstream-Status: Pending
---
 Makefile.am       |  5 +++++
 tests/Makefile.am | 12 ++++++++++++
 2 files changed, 17 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index d0b46ea..6c57b11 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -64,3 +64,8 @@ ChangeLog:
             mv ChangeLog.tmp ChangeLog;                                     \
           rm -f ChangeLog.tmp;                                              \
         fi
+
+install-ptest:
+	@for subdir in $(SUBDIRS); do \
+		$(MAKE) -C $$subdir DESTDIR=$(DESTDIR)/$$subdir $@; \
+	done
diff --git a/tests/Makefile.am b/tests/Makefile.am
index eb071ac..9d14fe8 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -145,3 +145,15 @@ d_creat_ce_LDADD = ../src/libgdbm.la ../compat/libgdbm_compat.la
 t_wordwrap_LDADD = ../tools/libgdbmapp.a
 
 SUBDIRS = gdbmtool
+
+buildtests: $(check_PROGRAMS) $(TESTSUITE)
+
+install-ptest: $(check_PROGRAMS) $(TESTSUITE)
+	@$(INSTALL) -d $(DESTDIR)
+	@for file in $^; do \
+		if [ -x .libs/$$file ]; then \
+			$(INSTALL_PROGRAM) .libs/$$file $(DESTDIR)/$$file; \
+		else \
+			$(INSTALL_PROGRAM) $$file $(DESTDIR) ; \
+		fi \
+	done
