Index: llvm-2.9/include/llvm/Support/FEnv.h
===================================================================
--- llvm-2.9.orig/include/llvm/Support/FEnv.h	2010-11-29 20:44:50.000000000 +0100
+++ llvm-2.9/include/llvm/Support/FEnv.h	2011-11-18 18:42:22.580161297 +0100
@@ -17,6 +17,9 @@
 
 #include "llvm/Config/config.h"
 #include <cerrno>
+
+#undef HAVE_FENV_H
+
 #ifdef HAVE_FENV_H
 #include <fenv.h>
 #endif
