From a9e57d4048c94f6bc2533ccc41e6c35d4416571a Mon Sep 17 00:00:00 2001
From: Liu Yiding <liuyd.fnst@fujitsu.com>
Date: Mon, 9 Jun 2025 06:44:06 +0000
Subject: [PATCH] fix uname declaration

Upstream-Status: Submitted [https://github.com/BuGlessRB/procmail/pull/11]
Signed-off-by: Liu Yiding <liuyd.fnst@fujitsu.com>
---
 src/includes.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/includes.h b/src/includes.h
index 4466907..7d6b41f 100644
--- a/src/includes.h
+++ b/src/includes.h
@@ -331,7 +331,7 @@ extern int errno;
 
 #ifndef NOuname
 #ifndef P		  /* SINIX V5.23 has the wrong prototype for uname() */
-extern int uname();					 /* so we fix it :-) */
+extern int uname(struct utsname *);
 #define Uname(name)		((int(*)(struct utsname*))uname)(name)
 #else
 #define Uname(name)		uname(name)		    /* no fix needed */
-- 
2.43.0

