Upstream-Status: Inappropriate [configuration]

--- lrzsz-0.12.20/acinclude.m4~autotools	1998-12-27 17:08:59.000000000 -0500
+++ lrzsz-0.12.20/acinclude.m4	2004-05-09 22:26:19.000000000 -0400
@@ -1,12 +1,11 @@
 dnl AC_REPLACE_GNU_GETOPT
-AC_DEFUN(AC_REPLACE_GNU_GETOPT,
-[AC_CHECK_FUNC(getopt_long, , [LIBOBJS="$LIBOBJS getopt1.o getopt.o"])
-AC_SUBST(LIBOBJS)dnl
-])
+AC_DEFUN([AC_REPLACE_GNU_GETOPT],
+[AC_CHECK_FUNC(getopt_long, , [AC_LIBOBJ([getopt1])
+AC_LIBOBJ([getopt])dnl])])
 
 dnl
 dnl taken from taylor uucp
-AC_DEFUN(LRZSZ_ERRNO_DECL,[
+AC_DEFUN([LRZSZ_ERRNO_DECL],[
 AC_MSG_CHECKING(for errno declaration)
 AC_CACHE_VAL(lrzsz_cv_decl_errno,
 [AC_TRY_COMPILE([#include <errno.h>], [int i = errno; errno = 1;],
@@ -23,7 +21,7 @@
 ])
 
 dnl LRZSZ_TYPE_SPEED_T
-AC_DEFUN(LRZSZ_TYPE_SPEED_T,[
+AC_DEFUN([LRZSZ_TYPE_SPEED_T],[
 AC_REQUIRE([AC_HEADER_STDC])dnl
 AC_REQUIRE([LRZSZ_HEADERS_TERM_IO])dnl
 AC_MSG_CHECKING(for speed_t)
@@ -61,7 +59,7 @@
 fi
 ])
 
-AC_DEFUN(lrzsz_HEADER_SYS_SELECT,
+AC_DEFUN([lrzsz_HEADER_SYS_SELECT],
 [AC_CACHE_CHECK([whether sys/time.h and sys/select.h may both be included],
   lrzsz_cv_header_sys_select,
 [AC_TRY_COMPILE([#include <sys/types.h>
