Index: SDL_image-1.2.12/configure.in
===================================================================
--- SDL_image-1.2.12.orig/configure.in	2012-01-21 01:51:33.000000000 +0000
+++ SDL_image-1.2.12/configure.in	2014-07-18 06:56:56.853466678 +0000
@@ -1,5 +1,4 @@
 dnl Process this file with autoconf to produce a configure script.
-AC_INIT(README)
 
 dnl Set various version strings - taken gratefully from the GTk sources
 
@@ -11,12 +10,19 @@
 # if backwards compatibility has been broken,
 # set BINARY_AGE and INTERFACE_AGE to 0.
 
-MAJOR_VERSION=1
-MINOR_VERSION=2
-MICRO_VERSION=12
+m4_define([sdlimage_major_version],[1])
+m4_define([sdlimage_minor_version],[2])
+m4_define([sdlimage_micro_version],[12])
+m4_define([sdlimage_version], [sdlimage_major_version.sdlimage_minor_version.sdlimage_micro_version])
+
+AC_INIT([SDL_image], [sdlimage_version])
+
+MAJOR_VERSION=sdlimage_major_version
+MINOR_VERSION=sdlimage_minor_version
+MICRO_VERSION=sdlimage_micro_version
 INTERFACE_AGE=4
 BINARY_AGE=12
-VERSION=$MAJOR_VERSION.$MINOR_VERSION.$MICRO_VERSION
+VERSION=sdlimage_version
 
 AC_SUBST(MAJOR_VERSION)
 AC_SUBST(MINOR_VERSION)
@@ -42,7 +48,7 @@
 AC_CANONICAL_HOST
 
 dnl Setup for automake
-AM_INIT_AUTOMAKE(SDL_image, $VERSION)
+AM_INIT_AUTOMAKE([foreign])
 
 dnl Check for tools
 AC_PROG_LIBTOOL
