From 9231a2d5b9c5406f33aef3ab4de2dbec9b08ed13 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20M=C3=BCller?= <schnitzeltony@googlemail.com>
Date: Fri, 18 May 2012 21:13:00 +0200
Subject: [PATCH] Do not run tests at build time
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Andreas Müller <schnitzeltony@googlemail.com>

---
 Makefile.am | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 226e79d..62ad18d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -9,8 +9,7 @@ SUBDIRS =								\
 	docs								\
 	icons								\
 	pixmaps								\
-	po								\
-	tests
+	po
 
 distclean-local:
 	rm -rf *.cache *~
