From d02a8e202d2fe5caf3449683e0ccdf84a482a325 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Wed, 5 Jun 2019 14:17:55 +0200
Subject: [PATCH] Build thumbnailer and tests also in cross builds.

Upstream-Status: Inappropriate [relies on oe-core specific hacks]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

---
 meson.build | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/meson.build b/meson.build
index 73034d1..d114016 100644
--- a/meson.build
+++ b/meson.build
@@ -378,10 +378,8 @@ subdir('gdk-pixbuf')
 # i18n
 subdir('po')
 
-if not meson.is_cross_build()
-  subdir('tests')
-  subdir('thumbnailer')
-endif
+subdir('tests')
+subdir('thumbnailer')
 
 # Documentation
 build_docs = get_option('gtk_doc') or get_option('docs')
