From 38afdacf526410f970afc58e147c7377c6c7112c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=BCmer=20Cip?= <sumer.cip@platform.sh>
Date: Fri, 25 Nov 2022 15:58:03 +0300
Subject: [PATCH 2/2] add 3.11 to the setup

---
 setup.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.py b/setup.py
index d006787..96e2a66 100644
--- a/setup.py
+++ b/setup.py
@@ -56,6 +56,7 @@ CLASSIFIERS = [
     'Programming Language :: Python :: 3.8',
     'Programming Language :: Python :: 3.9',
     'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
     'Programming Language :: Python :: Implementation :: CPython',
     'Operating System :: OS Independent',
     'Topic :: Software Development :: Libraries',
-- 
2.30.2

