From 63570b98ef63c91d8508478fcbe6b89c90c8398f Mon Sep 17 00:00:00 2001
From: Jens Rehsack <rehsack@gmail.com>
Date: Thu, 19 Nov 2015 20:45:56 +0100

---
 wscript | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/wscript b/wscript
index e7c05d2..1633593 100644
--- a/wscript
+++ b/wscript
@@ -161,9 +161,7 @@ def configure(conf):
     if conf.env.standalone_ldb:
         conf.CHECK_XSLTPROC_MANPAGES()
 
-        # we need this for the ldap backend
-        if conf.CHECK_FUNCS_IN('ber_flush ldap_open ldap_initialize', 'lber ldap', headers='lber.h ldap.h'):
-            conf.env.ENABLE_LDAP_BACKEND = True
+        conf.env.ENABLE_LDAP_BACKEND = False
 
         # we don't want any libraries or modules to rely on runtime
         # resolution of symbols
