From f13b26c6587bc061cb92ecda5936ff1971ee7f5d Mon Sep 17 00:00:00 2001
From: kfting <kfting@nuvoton.com>
Date: Thu, 16 Jul 2020 14:22:53 +0800
Subject: [PATCH] Modify dbus call style for calling setTime

Signed-off-by: kfting <kfting@nuvoton.com>
Signed-off-by: Allen Kang <jhkang@nuvoton.com>
---
 bmc_epoch.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/bmc_epoch.cpp b/bmc_epoch.cpp
index db4051a..3425468 100644
--- a/bmc_epoch.cpp
+++ b/bmc_epoch.cpp
@@ -125,6 +125,7 @@ bool BmcEpoch::setTime(const microseconds& usec)
                   false,  // relative
                   false); // user_interaction
 
+#if 0
     try
     {
         bus.call_noreply(method);
@@ -135,6 +136,15 @@ bool BmcEpoch::setTime(const microseconds& usec)
         using namespace xyz::openbmc_project::Time;
         elog<FailedError>(Failed::REASON(ex.what()));
     }
+#endif
+    auto reply = bus.call(method);
+    if (reply.is_method_error())
+    {
+        log<level::ERR>("Error in setting system time");
+        using namespace xyz::openbmc_project::Time;
+        //elog<InternalFailure>();
+        elog<FailedError>(Failed::REASON("Error in setting system time"));
+    }
     return true;
 }
 
-- 
2.34.1

