Implicit declaration compile warning fixes from Debian

Signed-off-by: Adrian Bunk <bunk@stusta.de>
Upstream-Status: Inappropriate [upstream is dead]

--- lrzsz-0.12.21.orig/lib/long-options.c
+++ lrzsz-0.12.21/lib/long-options.c
@@ -22,6 +22,7 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <getopt.h>
 #include "long-options.h"
 
--- lrzsz-0.12.21.orig/src/lsyslog.c
+++ lrzsz-0.12.21/src/lsyslog.c
@@ -22,6 +22,7 @@
 #ifdef ENABLE_SYSLOG
 #include "zglobal.h"
 #include <pwd.h>
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #endif
