From cec508013706ef06dbac036905a90cbe075fa03d Mon Sep 17 00:00:00 2001
From: Richard Barnes <rbarnes@umn.edu>
Date: Wed, 2 Oct 2024 10:35:09 -0700
Subject: [PATCH] Match `malloc` signature to its use

Upstream-Status: Submitted [https://github.com/westes/flex/pull/674]

Signed-off-by: Martin Jansa <martin.jansa@gmail.com>
---
 lib/malloc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/malloc.c b/lib/malloc.c
index 75e8ef9..701b9b3 100755
--- a/lib/malloc.c
+++ b/lib/malloc.c
@@ -3,7 +3,7 @@
      
      #include <sys/types.h>
      
-     void *malloc ();
+     void *malloc (size_t n);
      
      /* Allocate an N-byte block of memory from the heap.
         If N is zero, allocate a 1-byte block.  */
