From 4ef35041ba5c02df48c31f2382e7c3c4316ad936 Mon Sep 17 00:00:00 2001
From: Etienne Cordonnier <ecordonnier@snap.com>
Date: Tue, 14 Mar 2023 13:53:51 +0100
Subject: [PATCH] adb: Allow adbd to be run as root

Signed-off-by: Etienne Cordonnier <ecordonnier@snap.com>

---
Upstream-Status: Pending

 system/core/adb/daemon/main.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/system/core/adb/daemon/main.cpp b/system/core/adb/daemon/main.cpp
index e807d13d..309663a2 100644
--- a/system/core/adb/daemon/main.cpp
+++ b/system/core/adb/daemon/main.cpp
@@ -75,6 +75,7 @@ static bool should_drop_capabilities_bounding_set() {
 }
 
 static bool should_drop_privileges() {
+    return true;
     // "adb root" not allowed, always drop privileges.
     if (!ALLOW_ADBD_ROOT && !is_device_unlocked()) return true;
 
