From 9dcded5e79bb163a95bbb0edb8f05318460de8f2 Mon Sep 17 00:00:00 2001
From: Joseph Liu <kwliu@nuvoton.com>
Date: Fri, 17 Apr 2020 09:32:24 +0800
Subject: [PATCH] Run after /xyz/openbmc_project/user path created

---
 xyz.openbmc_project.Network.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xyz.openbmc_project.Network.service.in b/xyz.openbmc_project.Network.service.in
index 5b8ae01..2000049 100644
--- a/xyz.openbmc_project.Network.service.in
+++ b/xyz.openbmc_project.Network.service.in
@@ -4,6 +4,7 @@ After=obmc-mapper.target
 After=systemd-networkd.service
 
 [Service]
+ExecStartPre=/bin/sh -c "mapper wait /xyz/openbmc_project/user"
 ExecStart=/usr/bin/env phosphor-network-manager
 SyslogIdentifier=phosphor-network-manager
 Restart=always
-- 
2.17.1

