From 2326f14019248c2d4b159df2be2a5171394c816b Mon Sep 17 00:00:00 2001
From: Tim Lee <timlee660101@gmail.com>
Date: Fri, 10 Jun 2022 14:18:52 +0800
Subject: [PATCH] Run after /xyz/openbmc_project/user path created

---
 xyz.openbmc_project.Network.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xyz.openbmc_project.Network.service.in b/xyz.openbmc_project.Network.service.in
index 044b78f..9954784 100644
--- a/xyz.openbmc_project.Network.service.in
+++ b/xyz.openbmc_project.Network.service.in
@@ -4,6 +4,7 @@ After=obmc-mapper.target
 After=systemd-networkd.service
 
 [Service]
+ExecStartPre=/bin/sh -c "mapper wait /xyz/openbmc_project/user"
 ExecStart=/usr/bin/phosphor-network-manager
 SyslogIdentifier=phosphor-network-manager
 Restart=always
-- 
2.17.1

