From df5d45b2d7ad7e7b5f0542a816d08b0409a529a1 Mon Sep 17 00:00:00 2001
From: Roy Li <rongqing.li@windriver.com>
Date: Tue, 19 Aug 2014 11:31:35 +0800
Subject: [PATCH] [PATCH] add the needed stdarg.h

Upstream-status: Pending

Signed-off-by: Roy Li <rongqing.li@windriver.com>

---
 lib/defaults.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/defaults.c b/lib/defaults.c
index 0e48a78..667f35d 100644
--- a/lib/defaults.c
+++ b/lib/defaults.c
@@ -16,6 +16,7 @@
 
 #include <stdlib.h>
 #include <stdio.h>
+#include <stdarg.h>
 #include <ctype.h>
 #include <string.h>
 #include <sys/utsname.h>
