Subject: [PATCH] CMakeLists: do not use vendored libcurl

Upstream-Status: Inappropriate [upstream ticket https://github.com/nxp-imx/bcu/issues/17]

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

Update for 1.1.128
Signed-off-by: Liu Yiding <liuyd.fnst@fujitsu.com>
---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0d6b915..f8c76ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,14 +16,15 @@ find_package(PkgConfig REQUIRED)
 pkg_check_modules(LIBFTDI REQUIRED libftdi1)
 pkg_check_modules(LIBYAML REQUIRED yaml-0.1)
 pkg_check_modules(LIBUSB REQUIRED libusb-1.0)
+pkg_check_modules(LIBCURL REQUIRED libcurl)
 project(bcu)
-include_directories(${LIBUSB_INCLUDE_DIRS} ${LIBFTDI_INCLUDE_DIRS}include ${PROJECT_SOURCE_DIR}/libftdi/src ${LIBYAML_INCLUDEDIR})
+include_directories(${LIBUSB_INCLUDE_DIRS} ${LIBFTDI_INCLUDE_DIRS}include ${LIBYAML_INCLUDEDIR})
 include_directories(${PROJECT_SOURCE_DIR})
 
 message("operation system is ${CMAKE_SYSTEM}")
 if (LINUX)
         add_executable(bcu bcu.c port.c chip.c board.c bcu_parser.c bcu_yaml.c bcu_ftdi_eeprom.c)
-        link_directories(${LIBUSB_LIBRARY_DIRS} ${LIBFTDI_LIBRARY_DIRS} ${LIBYAML_LIBRARY_DIRS})
+        link_directories(${LIBUSB_LIBRARY_DIRS} ${LIBFTDI_LIBRARY_DIRS} ${LIBYAML_LIBRARY_DIRS} ${LIBCURL_LIBRARY_DIRS})
         target_link_libraries (bcu ${LIBUSB_LIBRARIES} ${LIBFTDI_LIBRARIES} ${LIBYAML_LIBRARIES} -lpthread -lm -ldl)
         execute_process( COMMAND ${PROJECT_SOURCE_DIR}/create_version_h.sh ${PROJECT_SOURCE_DIR} )
         install(TARGETS bcu DESTINATION bin)
-- 
2.43.0

