Index: git/configure
===================================================================
--- git.orig/configure	2012-11-19 21:07:51.917429465 -0800
+++ git/configure	2012-11-19 21:13:19.337437278 -0800
@@ -439,14 +442,18 @@
 	GZIP_SUFFIX=".gz"
 fi
 
-printf "Finding suitable compiler........"
-CC=`pathsearch "${CC:-cc}"`
-if test -z "$CC" -o ! -x "$CC"; then
-	CC=`pathsearch "${CC:-gcc}"`
+if test -z "$CC"; then
+	printf "Finding suitable compiler........"
+	CC=`pathsearch "${CC:-cc}"`
+	if test -z "$CC" -o ! -x "$CC"; then
+		CC=`pathsearch "${CC:-gcc}"`
+	fi
 fi
 assert "$CC" "not found"
 
-cat << EOF > .1.c
+if test -z "$COMPILER"; then
+
+	cat << EOF > .1.c
 #include <stdio.h>
 int main(void) {
 #if defined(__GNUC__) && (__GNUC__ >= 4)
@@ -569,16 +576,16 @@
 #endif
 }
 EOF
-
-$CC -o .1 .1.c
-COMPILER=`./.1`
-r=$?
-rm -f .1.c .1
-
-if test "$r" -ne 0; then
-	assert "" "update compiler"
-else
-	echo "success [$CC]"
+	$CC -o .1 .1.c
+	COMPILER=`./.1`
+	r=$?
+	rm -f .1.c .1
+
+	if test "$r" -ne 0; then
+		assert "" "update compiler"
+	else
+		echo "success [$CC]"
+	fi
 fi
 
 if test "$COMPILER" = "suncc"; then
