From 21d8e7700fa0a9c4bf569dd366134060ae858832 Mon Sep 17 00:00:00 2001
From: Christopher Larson <chris_larson@mentor.com>
Date: Thu, 5 May 2016 10:59:07 -0700
Subject: [PATCH] Obey LDFLAGS for the link of libruby

Signed-off-by: Christopher Larson <chris_larson@mentor.com>
Upstream-Status: Pending

---
 template/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/template/Makefile.in b/template/Makefile.in
index 1456313..15b98a4 100644
--- a/template/Makefile.in
+++ b/template/Makefile.in
@@ -127,7 +127,7 @@ ENABLE_SHARED = @ENABLE_SHARED@
 LDSHARED = @LIBRUBY_LDSHARED@
 DLDSHARED = @DLDSHARED@
 XDLDFLAGS = @DLDFLAGS@
-DLDFLAGS = @LIBRUBY_DLDFLAGS@ $(XLDFLAGS) $(ARCH_FLAG)
+DLDFLAGS = @LIBRUBY_DLDFLAGS@ @LDFLAGS@ $(XLDFLAGS) $(ARCH_FLAG)
 SOLIBS = @SOLIBS@
 ENABLE_DEBUG_ENV = @ENABLE_DEBUG_ENV@
 MAINLIBS = $(YJIT_LIBS) @MAINLIBS@
