From 2fcf1e23ef4b2a5c93526f12212aa892595261f6 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 29 Mar 2013 08:37:11 +0400
Subject: [PATCH 01/47] gcc-4.3.1: ARCH_FLAGS_FOR_TARGET

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-Status: Inappropriate [embedded specific]
---
 configure    | 2 +-
 configure.ac | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 32a38633ad8..b4760952085 100755
--- a/configure
+++ b/configure
@@ -7472,7 +7472,7 @@ fi
 # for target_alias and gcc doesn't manage it consistently.
 target_configargs="--cache-file=./config.cache ${target_configargs}"
 
-FLAGS_FOR_TARGET=
+FLAGS_FOR_TARGET="$ARCH_FLAGS_FOR_TARGET"
 case " $target_configdirs " in
  *" newlib "*)
   case " $target_configargs " in
diff --git a/configure.ac b/configure.ac
index 12377499295..176ebb921ed 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3065,7 +3065,7 @@ fi
 # for target_alias and gcc doesn't manage it consistently.
 target_configargs="--cache-file=./config.cache ${target_configargs}"
 
-FLAGS_FOR_TARGET=
+FLAGS_FOR_TARGET="$ARCH_FLAGS_FOR_TARGET"
 case " $target_configdirs " in
  *" newlib "*)
   case " $target_configargs " in
-- 
2.12.2

