From b5084616a3bc2a0d485f43aeae69c6025f3f857e Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Fri, 10 Jun 2011 18:44:21 +0200
Subject: [PATCH] add support for the angstrom distribtion

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 Utils/Platform.pm |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/Utils/Platform.pm b/Utils/Platform.pm
index fc76360..8b8c770 100644
--- a/Utils/Platform.pm
+++ b/Utils/Platform.pm
@@ -92,6 +92,7 @@ my $PLATFORM_INFO = {
   "nexenta-1.0"     => [ "Nexenta GNU/Solaris", "1.0", "Ellate" ],
   "yellowdog-4.1"   => [ "Yellow Dog Linux", "4.1", "Sagitta" ],
   "guadalinex-v4"   => [ "Guadalinex", "v4", "Toro" ],
+  "angstrom"        => [ "Angstrom", "", "" ],
 };
 
 sub get_platform_info
@@ -143,6 +144,7 @@ sub ensure_distro_map
      "vine-3.1"         => "vine-3.0",
      "vlos-1.2"         => "gentoo",
      "nexenta-1.0"      => "solaris-2.11",
+     "angstrom"         => "debian",
      );
 
   return $metamap{$distro} if ($metamap{$distro});
@@ -375,6 +377,7 @@ sub guess
                   [ \&check_ark ],
                   [ \&check_yoper ],
                   [ \&check_distro_file, "/etc/yellowdog-release", "yellowdog", "^Yellow Dog Linux release (\\S+)" ],
+                  [ \&check_file_exists, "/etc/angstrom-version", "angstrom" ],
                  ],
     "FreeBSD" => [[ \&check_freebsd ]],
     "SunOS"   => [[ \&check_solaris ]]
-- 
1.6.6.1

