From 5ee73f91b7f13dd5c700c69e7fc43cecab5868da Mon Sep 17 00:00:00 2001
From: Tom Geelen <t.f.g.geelen@gmail.com>
Date: Sun, 7 Apr 2024 22:39:35 +0000
Subject: [PATCH] Allow newer version of wheel and setuptools

Upstream-Status: Pending
Signed-off-by: Tom Geelen <t.f.g.geelen@gmail.com>
---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 56df5bd..17ca62d 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools~=65.6", "wheel~=0.37.1"]
+requires = ["setuptools>=65.6", "wheel>=0.37.1"]
 build-backend = "setuptools.build_meta"
 
 [project]
