From d7fb0a6ef3f3cbdfc1f21f5a67d54fa9d2c1e183 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Wed, 2 Nov 2011 11:07:24 +0100
Subject: [PATCH] disable AI_ADDRCONFIG flag

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>

Upstream-Status: unknown
---
 tornado/netutil.py |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/tornado/netutil.py b/tornado/netutil.py
index cfff0ba..c181f24 100644
--- a/tornado/netutil.py
+++ b/tornado/netutil.py
@@ -238,12 +238,6 @@ def bind_sockets(port, address=None, family=socket.AF_UNSPEC, backlog=128):
     if address == "":
         address = None
     flags = socket.AI_PASSIVE
-    if hasattr(socket, "AI_ADDRCONFIG"):
-        # AI_ADDRCONFIG ensures that we only try to bind on ipv6
-        # if the system is configured for it, but the flag doesn't
-        # exist on some platforms (specifically WinXP, although
-        # newer versions of windows have it)
-        flags |= socket.AI_ADDRCONFIG
     for res in set(socket.getaddrinfo(address, port, family, socket.SOCK_STREAM,
                                   0, flags):
         af, socktype, proto, canonname, sockaddr = res
-- 
1.7.2.5

