From a92fd7da3a262e21c0970a3908eb0b4502819a01 Mon Sep 17 00:00:00 2001
From: Yang Lyu <yangl@axis.com>
Date: Fri, 23 Feb 2018 16:31:49 +0100
Subject: [PATCH] Comment rst2man related stuff

---
 meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 56d38e4..3c3b109 100644
--- a/meson.build
+++ b/meson.build
@@ -15,7 +15,7 @@ add_project_arguments('-DBINDIR="' + join_paths(get_option('prefix'), get_option
 cc = meson.get_compiler('c')
 conf = configuration_data()
 mod_pkgconfig = import('pkgconfig')
-prog_rst2man = find_program('rst2man', 'rst2man.py')
+#prog_rst2man = find_program('rst2man', 'rst2man.py')
 
 sub_cdvar = subproject('c-dvar', version: '>=1')
 sub_clist = subproject('c-list', version: '>=3')
@@ -56,7 +56,7 @@ endif
 
 conf.set('bindir', join_paths(get_option('prefix'), get_option('bindir')))
 
-subdir('docs')
+#subdir('docs')
 subdir('src')
 
 if use_launcher
-- 
2.1.4

