From 5203821a936b1927c401034fe323686ad9eafb04 Mon Sep 17 00:00:00 2001
From: Enguerrand de Ribaucourt <enguerrand.de-ribaucourt@savoirfairelinux.com>
Date: Thu, 6 Jun 2024 15:57:12 +0200
Subject: [PATCH 1/1] include stddef.h for ptrdiff_t

This fixes compilation with musl.

Upstream-Status: Pending

Signed-off-by: Enguerrand de Ribaucourt <enguerrand.de-ribaucourt@savoirfairelinux.com>
---
 receivedata.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/receivedata.c b/receivedata.c
index 9a68f67..0aa797f 100644
--- a/receivedata.c
+++ b/receivedata.c
@@ -1,4 +1,5 @@
 #include <assert.h>
+#include <stddef.h>
 #include <sys/types.h>
 #include <unistd.h>
 #include <stdlib.h>
-- 
2.34.1

