From a4433978bf324525b4c260b0e9615ae27271fe55 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Wed, 20 Sep 2017 12:05:40 -0400
Subject: [PATCH] Makefile: force symlinks

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/import/Makefile b/src/import/Makefile
index e3e5050bb7f6..4ad3fb7ff0a9 100644
--- a/src/import/Makefile
+++ b/src/import/Makefile
@@ -53,7 +53,7 @@ help:
 .gopathok:
 ifeq ("$(wildcard $(GOPKGDIR))","")
 	mkdir -p "$(GOPKGBASEDIR)"
-	ln -s "$(CURDIR)" "$(GOPKGBASEDIR)"
+	ln -sf "$(CURDIR)" "$(GOPKGBASEDIR)"
 endif
 	touch "$(GOPATH)/.gopathok"
 
-- 
2.4.0.53.g8440f74

