From 85822fcb87dd531d5e8ed1c8230c67bddb56f66d Mon Sep 17 00:00:00 2001
From: reuk <reuk@users.noreply.github.com>
Date: Wed, 6 Apr 2022 15:43:09 +0100
Subject: [PATCH] StandardHeader:  Include <utility> for GCC 12 compatibility

Upstream-Status: Backport [https://github.com/juce-framework/JUCE/commit/ce8aff22c31c4d932a8414bc6f6185381dc970db]
Signed-off-by: Yoann Congal <yoann.congal@smile.fr>
---
 modules/juce_core/system/juce_StandardHeader.h | 1 +
 1 file changed, 1 insertion(+)

--- a/modules/juce_core/system/juce_StandardHeader.h
+++ b/modules/juce_core/system/juce_StandardHeader.h
@@ -64,6 +64,7 @@
 #include <typeindex>
 #include <unordered_map>
 #include <unordered_set>
+#include <utility>
 #include <vector>
 #include <set>
 
