Upstream-Status: Inappropriate [configuration]

---
 neon.pc.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- neon-0.30.1.orig/neon.pc.in
+++ neon-0.30.1/neon.pc.in
@@ -7,5 +7,5 @@ Name: neon
 Description: HTTP/WebDAV client library
 Version: @NEON_VERSION@
 Libs: -L${libdir} -lneon @NEON_PC_LIBS@
-Libs.private: @NEON_LIBS@
+Libs.private: -L${libdir} -lz -lgcrypt -lgpg-error -lexpat -lgnutls
 Cflags: -I${includedir}/neon @NEON_CFLAGS@
