Actually pass the arguments to RANLIB.

Upstream-Status: Pending
Signed-off-by: Ross Burton <ross.burton@intel.com>

Index: agent/Makefile.inc
===================================================================
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -92,6 +92,7 @@ ifeq ($(OPSYS),GNU/Linux)
     OPTS += -DUSE_uuid_generate=0
   endif
   OPTS += -DENABLE_arch_$(shell uname -m)
+  RANLIB += $@
 endif

 ifneq ($(OPSYS),Windows)
