From f9c455e81ad4d870c8ae20b9af8598139231ad26 Mon Sep 17 00:00:00 2001
From: "Meyer, Wolfgang" <Wolfgang.Meyer@gossenmetrawatt.com>
Date: Fri, 9 Sep 2022 09:32:22 +0200
Subject: [PATCH] meson.build: install fbgs shell script

Upstream-Status: Submitted [https://github.com/kraxel/fbida/pull/4]
Signed-off-by: "Meyer, Wolfgang" <Wolfgang.Meyer@gossenmetrawatt.com>
---
 meson.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/meson.build b/meson.build
index cbed982..2129896 100644
--- a/meson.build
+++ b/meson.build
@@ -126,6 +126,9 @@ executable('fbpdf',
            sources             : fbpdf_srcs,
            dependencies        : fbpdf_deps,
            install             : true)
+install_data('fbgs',
+             install_dir        : get_option('bindir'),
+             install_mode       : 'rwxr--r--')
 
 # build fbcon
 fbcon_srcs   = [ 'fbcon.c', 'drmtools.c', 'fbtools.c', 'gfx.c', 'vt.c', 'kbd.c' ]
-- 
2.37.3

