Don't check for tscd deamon on host.

Upstream-Status: OE Specific

Signed-off-by: Armin Kuster <akuster808@gmail.com>

Index: git/configure.ac
===================================================================
--- git.orig/configure.ac
+++ git/configure.ac
@@ -179,15 +179,6 @@ AC_SUBST([LIBTPMS_LIBS])
 AC_CHECK_LIB(c, clock_gettime, LIBRT_LIBS="", LIBRT_LIBS="-lrt")
 AC_SUBST([LIBRT_LIBS])
 
-AC_PATH_PROG([TCSD], tcsd)
-if test "x$TCSD" = "x"; then
-    have_tcsd=no
-    AC_MSG_WARN([tcsd could not be found; typically need it for tss user account and tests])
-else
-    have_tcsd=yes
-fi
-AM_CONDITIONAL([HAVE_TCSD], test "$have_tcsd" != "no")
-
 dnl We either need netstat (more common across systems) or 'ss' for test cases
 AC_PATH_PROG([NETSTAT], [netstat])
 if test "x$NETSTAT" = "x"; then
@@ -440,23 +431,6 @@ AC_ARG_WITH([tss-group],
             [TSS_GROUP="tss"]
 )
 
-case $have_tcsd in
-yes)
-	AC_MSG_CHECKING([whether TSS_USER $TSS_USER is available])
-	if ! test $(id -u $TSS_USER); then
-		AC_MSG_ERROR(["$TSS_USER is not available"])
-	else
-		AC_MSG_RESULT([yes])
-	fi
-	AC_MSG_CHECKING([whether TSS_GROUP $TSS_GROUP is available])
-	if ! test $(id -g $TSS_GROUP); then
-		AC_MSG_ERROR(["$TSS_GROUP is not available"])
-	else
-		AC_MSG_RESULT([yes])
-	fi
-	;;
-esac
-
 AC_SUBST([TSS_USER])
 AC_SUBST([TSS_GROUP])
 
Index: git/tests/Makefile.am
===================================================================
--- git.orig/tests/Makefile.am
+++ git/tests/Makefile.am
@@ -83,10 +83,6 @@ TESTS += \
 	test_tpm2_swtpm_cert \
 	test_tpm2_swtpm_cert_ecc \
 	test_tpm2_swtpm_setup_create_cert
-if HAVE_TCSD
-TESTS += \
-	test_tpm2_samples_create_tpmca
-endif
 endif
 
 EXTRA_DIST=$(TESTS) \
