From f9c4ad3a171d676e3818b8f6897f325ccf0b2203 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 9 May 2017 00:57:10 -0700

---
 Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index 8166d91..8e05e49 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2,6 +2,8 @@
 
 lib_LTLIBRARIES = libSDL2_ttf.la
 
+AUTOMAKE_OPTIONS = foreign
+
 libSDL2_ttfincludedir = $(includedir)/SDL2
 libSDL2_ttfinclude_HEADERS =	\
 	SDL_ttf.h
